--- SConstruct.orig	2014-02-12 10:21:42.227701032 +1300
+++ SConstruct	2014-02-12 10:29:06.820516919 +1300
@@ -105,12 +105,16 @@
 ## The other two options control the way the linker creates a namespace
 ## for the dynamic library; check the man page for ld on a mac to see
 ## the details.
-if os.environ['SAGE64']=="yes":
-    # We want the debug and optimization flags, since debug symbols are so useful, etc.
-    print "64 bit mode"
-    env.Append( CFLAGS="-O2 -g -m64" )
-    env.Append( CXXFLAGS="-O2 -g -m64" )
-    env.Append( LINKFLAGS="-m64" )
+#if os.environ['SAGE64']=="yes":
+#    # We want the debug and optimization flags, since debug symbols are so useful, etc.
+#    print "64 bit mode"
+#    env.Append( CFLAGS="-O2 -g -m64" )
+#    env.Append( CXXFLAGS="-O2 -g -m64" )
+#    env.Append( LINKFLAGS="-m64" )
+
+env['CFLAGS'] = os.environ['CFLAGS']
+env['CXXFLAGS'] = os.environ['CXXFLAGS']
+env['LINKFLAGS'] = os.environ['LDFLAGS']+' -Wl,-soname,libcsage.so'
 
 if env['PLATFORM']=="darwin":
     env.Append( LINKFLAGS=" -single_module -flat_namespace -undefined dynamic_lookup" )
@@ -118,6 +122,7 @@
 # SCons doesn't automatically pull in system environment variables
 # However, we only need SAGE_LOCAL, so that's easy.
 env['SAGE_LOCAL'] = os.environ['SAGE_LOCAL']
+env['DESTDIR'] = os.environ['DESTDIR']
 
 # Get the python version number. We want this to be a bit more resistant to major
 # change of version.
@@ -138,9 +143,8 @@
 
 lib = env.SharedLibrary( "csage", [ "src/" + x for x in srcFiles ],
                          LIBS=['ntl', 'pari', 'gmp', 'python$PYV'],
-                         LIBPATH=['$SAGE_LOCAL/lib','$SAGE_LOCAL/lib/python$PYV/config/'],
                          CPPPATH=includes )
-env.Install("$SAGE_LOCAL/lib", lib)
-env.Install("$SAGE_LOCAL/include/csage", [ os.path.join('include',x) for x in incFiles ])
+env.Install("$DESTDIR/$SAGE_LOCAL/lib", lib)
+env.Install("$DESTDIR/$SAGE_LOCAL/include/csage", [ os.path.join('include',x) for x in incFiles ])
 
-env.Alias("install", "$SAGE_LOCAL")
+env.Alias("install", "$DESTDIR/$SAGE_LOCAL")
