--- SConstruct.orig	2010-06-05 17:46:29.522933877 +0000
+++ SConstruct	2010-06-07 16:43:27.058457532 +0000
@@ -132,7 +132,7 @@
 srcFiles = cFiles + cppFiles
 
 lib = env.SharedLibrary( "csage", [ "src/" + x for x in srcFiles ],
-                         LIBS=['ntl', 'gmp', 'pari', 'python2.6'], 
+                         LIBS=['ntl', 'mpir', 'pari', 'python2.6'],
                          LIBPATH=['$SAGE_LOCAL/lib','$SAGE_LOCAL/lib/python/config/'],
                          CPPPATH=includes )
 env.Install("$SAGE_LOCAL/lib", lib)
--- include/convert.h.orig	2010-06-07 16:47:57.614457393 +0000
+++ include/convert.h	2010-06-07 16:48:07.968456694 +0000
@@ -8,7 +8,7 @@
   
 */
 
-#include <gmp.h>
+#include <mpir.h>
 #include <pari/pari.h>
 #include <stdio.h>
 
--- include/gmp_globals.h.orig	2010-06-07 16:48:17.627457253 +0000
+++ include/gmp_globals.h	2010-06-07 16:48:27.765457253 +0000
@@ -1,4 +1,4 @@
-#include <gmp.h>
+#include <mpir.h>
 
 #ifdef __cplusplus
 #define EXTERN extern "C"
--- include/mpn_pylong.h.orig	2010-06-07 16:45:31.956456415 +0000
+++ include/mpn_pylong.h	2010-06-07 16:45:46.956457323 +0000
@@ -2,7 +2,7 @@
 #define MPN_PYLONG_H
 
 #include <Python.h>
-#include <gmp.h>
+#include <mpir.h>
 
 /************************************************************/
 
--- include/mpz_longlong.h.orig	2010-06-07 16:47:06.606457253 +0000
+++ include/mpz_longlong.h	2010-06-07 16:47:19.452457672 +0000
@@ -1,5 +1,5 @@
 
-#include <gmp.h>
+#include <mpir.h>
 
 void mpz_set_longlong(mpz_ptr z, long long val);
 
--- include/mpz_pylong.h.orig	2010-06-07 16:46:03.928457951 +0000
+++ include/mpz_pylong.h	2010-06-07 16:46:17.136457183 +0000
@@ -2,7 +2,7 @@
 #define MPZ_PYLONG_H
 
 #include <Python.h>
-#include <gmp.h>
+#include <mpir.h>
 
 /* mpz -> pylong conversion */
 PyObject * mpz_get_pylong(mpz_srcptr z);
--- include/ntl_wrap.h.orig	2010-06-07 16:48:37.105456555 +0000
+++ include/ntl_wrap.h	2010-06-07 16:48:58.921458091 +0000
@@ -20,7 +20,7 @@
 #include <NTL/mat_GF2E.h>
 #include <NTL/HNF.h>
 #include <NTL/LLL.h>
-#include <gmp.h>
+#include <mpir.h>
 using namespace NTL;
 #endif
 
--- include/ZZ_pylong.h.orig	2010-06-07 16:46:30.164458440 +0000
+++ include/ZZ_pylong.h	2010-06-07 16:46:44.896458231 +0000
@@ -25,7 +25,7 @@
 #include <Python.h>
 #include <NTL/ZZ.h>
 using namespace NTL;
-#include "gmp.h"
+#include "mpir.h"
 
 /* ZZ -> pylong conversion */
 PyObject * ZZ_get_pylong(ZZ &z);
