--- SConstruct.orig	2010-05-26 12:13:50.000000000 +1200
+++ SConstruct	2010-06-28 22:47:20.080615376 +1200
@@ -120,6 +120,10 @@
 env['SAGE_LOCAL'] = os.environ['SAGE_LOCAL']
 if os.environ.has_key('SAGE_DEBIAN'):
     env['SAGE_LOCAL'] = '/usr'
+env['CFLAGS'] = os.environ['CFLAGS']
+env['CXXFLAGS'] = os.environ['CXXFLAGS']
+env['LINKFLAGS'] = os.environ['LDFLAGS']
+env.Append(LINKFLAGS = ['-Wl,-soname,libcsage.so'])
 
 # The SCons convenience function Split is the only strange thing 
 # to python programmers. It just makes a list by splitting on 
@@ -132,7 +136,7 @@
 srcFiles = cFiles + cppFiles
 
 lib = env.SharedLibrary( "csage", [ "src/" + x for x in srcFiles ],
-                         LIBS=['ntl', 'gmp', 'pari', 'python2.6'], 
+                         LIBS=['ntl', 'pari', 'gmp', 'python2.6'],
                          LIBPATH=['$SAGE_LOCAL/lib','$SAGE_LOCAL/lib/python/config/'],
                          CPPPATH=includes )
 env.Install("$SAGE_LOCAL/lib", lib)
