--- SConstruct.orig	2010-09-19 21:44:05.757218360 +1200
+++ SConstruct	2010-09-19 21:45:54.040219491 +1200
@@ -112,6 +112,10 @@
     env.Append( CXXFLAGS="-O2 -g -m64" )
     env.Append( LINKFLAGS="-m64" )
 
+env['CFLAGS'] = os.environ['CFLAGS']
+env['CXXFLAGS'] = os.environ['CXXFLAGS']
+env['LINKFLAGS'] = os.environ['LDFLAGS']+' -Wl,-soname,libcsage.so'
+
 if env['PLATFORM']=="darwin":
     env.Append( LINKFLAGS=" -single_module -flat_namespace -undefined dynamic_lookup" )
 
@@ -132,7 +136,7 @@
 srcFiles = cFiles + cppFiles
 
 lib = env.SharedLibrary( "csage", [ "src/" + x for x in srcFiles ],
-                         LIBS=['ntl', 'gmp', 'pari', 'python2.6'], 
+                         LIBS=['ntl', 'pari24', 'gmp', 'python2.6'],
                          LIBPATH=['$SAGE_LOCAL/lib','$SAGE_LOCAL/lib/python/config/'],
                          CPPPATH=includes )
 env.Install("$SAGE_LOCAL/lib", lib)
