--- SConstruct.orig	2010-02-16 21:45:46.213585333 +1300
+++ SConstruct	2010-02-16 21:45:58.931587437 +1300
@@ -132,7 +132,7 @@
 srcFiles = cFiles + cppFiles
 
 lib = env.SharedLibrary( "csage", [ "src/" + x for x in srcFiles ],
-                         LIBS=['ntl', 'gmp', 'pari'], LIBPATH=['$SAGE_LOCAL/lib'],
+                         LIBS=['ntl', 'mpir', 'pari'], LIBPATH=['$SAGE_LOCAL/lib'],
                          CPPPATH=includes )
 
 env.Alias( "install", [ lib ] )
--- include/ZZ_pylong.h.orig	2010-02-16 22:02:13.429586905 +1300
+++ include/ZZ_pylong.h	2010-02-16 22:05:48.741585471 +1300
@@ -25,7 +25,7 @@
 #include <Python.h>
 #include <NTL/ZZ.h>
 using namespace NTL;
-#include "gmp.h"
+#include "mpir.h"
 
 /* ZZ -> pylong conversion */
 PyObject * ZZ_get_pylong(ZZ &z);
--- include/convert.h.orig	2010-02-16 22:02:24.944585646 +1300
+++ include/convert.h	2010-02-16 22:05:48.751584714 +1300
@@ -8,7 +8,7 @@
   
 */
 
-#include <gmp.h>
+#include <mpir.h>
 #include <pari/pari.h>
 #include <stdio.h>
 
--- include/gmp_globals.h.orig	2010-02-16 22:02:37.237585764 +1300
+++ include/gmp_globals.h	2010-02-16 22:05:48.756584661 +1300
@@ -1,4 +1,4 @@
-#include <gmp.h>
+#include <mpir.h>
 
 #ifdef __cplusplus
 #define EXTERN extern "C"
--- include/mpn_pylong.h.orig	2010-02-16 22:02:51.925586559 +1300
+++ include/mpn_pylong.h	2010-02-16 22:05:48.764584529 +1300
@@ -2,7 +2,7 @@
 #define MPN_PYLONG_H
 
 #include <Python.h>
-#include <gmp.h>
+#include <mpir.h>
 
 /************************************************************/
 
--- include/mpz_pylong.h.orig	2010-02-16 22:03:26.001585584 +1300
+++ include/mpz_pylong.h	2010-02-16 22:05:48.764584529 +1300
@@ -2,7 +2,7 @@
 #define MPZ_PYLONG_H
 
 #include <Python.h>
-#include <gmp.h>
+#include <mpir.h>
 
 /* mpz -> pylong conversion */
 PyObject * mpz_get_pylong(mpz_srcptr z);
--- include/mpz_longlong.h.orig	2010-02-16 22:03:08.684585536 +1300
+++ include/mpz_longlong.h	2010-02-16 22:05:48.764584529 +1300
@@ -1,5 +1,5 @@
 
-#include <gmp.h>
+#include <mpir.h>
 
 void mpz_set_longlong(mpz_ptr z, long long val);
 
--- include/ntl_wrap.h.orig	2010-02-16 22:03:33.392585642 +1300
+++ include/ntl_wrap.h	2010-02-16 22:05:48.772584488 +1300
@@ -19,7 +19,7 @@
 #include <NTL/mat_GF2E.h>
 #include <NTL/HNF.h>
 #include <NTL/LLL.h>
-#include <gmp.h>
+#include <mpir.h>
 using namespace NTL;
 #endif
 
