http://github.com/sympy/sympy/commit/717516b8ffae806cdfdea8141ceb839107d92431

--- sympy/printing/pretty/stringpict.py
+++ sympy/printing/pretty/stringpict.py
@@ -81,7 +81,7 @@
         return '\n'.join(result), newBaseline
 
     def right(self, *args):
-        """Put pictures next to this one.
+        r"""Put pictures next to this one.
         Returns string, baseline arguments for stringPict.
         (Multiline) strings are allowed, and are given a baseline of 0.
         >>> from sympy.printing.pretty.stringpict import stringPict
--- sympy/utilities/runtests.py
+++ sympy/utilities/runtests.py
@@ -778,7 +778,7 @@
     def start(self):
         self.write_center("test process starts")
         executable = sys.executable
-        v = sys.version_info
+        v = tuple(sys.version_info)
         python_version = "%s.%s.%s-%s-%s" % v
         self.write("executable:   %s  (%s)\n\n" % (executable, python_version))
         self._t_start = clock()
