--- setup.py.orig	2011-08-02 15:03:59.748288474 +1200
+++ setup.py	2011-08-02 15:05:17.825461742 +1200
@@ -59,12 +59,6 @@
     'sympy.logic.algorithms',
     'sympy.logic.utilities',
     'sympy.matrices',
-    'sympy.mpmath',
-    'sympy.mpmath.calculus',
-    'sympy.mpmath.functions',
-    'sympy.mpmath.libmp',
-    'sympy.mpmath.matrices',
-    'sympy.mpmath.tests',
     'sympy.ntheory',
     'sympy.parsing',
     'sympy.physics',
@@ -208,7 +202,6 @@
     'sympy.integrals.tests',
     'sympy.logic.tests',
     'sympy.matrices.tests',
-    'sympy.mpmath.tests',
     'sympy.ntheory.tests',
     'sympy.parsing.tests',
     'sympy.physics.quantum.tests',
