diff -r -U1 sympy-0.7.2.orig/setup.py sympy-0.7.2/setup.py
--- sympy-0.7.2.orig/setup.py	2012-10-17 09:30:33.000000000 +0700
+++ sympy-0.7.2/setup.py	2012-10-20 23:07:53.000000000 +0700
@@ -67,7 +67,2 @@
     'sympy.matrices.expressions',
-    'sympy.mpmath',
-    'sympy.mpmath.calculus',
-    'sympy.mpmath.functions',
-    'sympy.mpmath.libmp',
-    'sympy.mpmath.matrices',
     'sympy.ntheory',
@@ -222,3 +217,2 @@
     'sympy.matrices.tests',
-    'sympy.mpmath.tests',
     'sympy.ntheory.tests',
