--- sympy/mpmath/tests/extratest_gamma.py.orig	2013-09-24 11:16:09.754443514 +1200
+++ sympy/mpmath/tests/extratest_gamma.py	2013-09-24 11:17:27.857447954 +1200
@@ -47,7 +47,7 @@
         if raise_:
             raise SystemExit
     if not err:
-        print("%s ok;" % name, end=' ')
+        sys.stdout.write("%s ok; " % name)
 
 def testcase(case):
     z, result = case
--- sympy/mpmath/tests/torture.py.orig	2013-09-24 11:16:49.786445729 +1200
+++ sympy/mpmath/tests/torture.py	2013-09-24 11:18:31.769451954 +1200
@@ -77,7 +77,7 @@
         exponents += [-1000, -100, -50, 50, 100, 1000]
     for n in exponents:
         if verbose:
-            print(".", end=' ')
+            sys.stdout.write(". ")
         mp.dps = 25
         xpos = mpf(10)**n / 1.1287
         xneg = -xpos
@@ -209,7 +209,7 @@
     if filt in line:
         print(line)
         t1 = clock()
-        exec_(line)
+        exec_(line, globals(), locals())
         t2 = clock()
         elapsed = t2-t1
         print("Time:", elapsed, "for", line, "(OK)")
