--- src/qsieve-fwd.H.orig	2007-06-14 19:14:10.000000000 +0000
+++ src/qsieve-fwd.H	2010-08-14 16:52:04.650593986 +0000
@@ -92,12 +92,12 @@
  */
 
 protected:
- static int PrimeNumbers[MaxSize] __attribute__ ((aligned (64)));
  static int PrimeNumberReciprocals[MaxSize] __attribute__ ((aligned (64)));
  static float PrimeNumberFloatReciprocals[MaxSize] __attribute__ ((aligned (64)));
  static FBsizetype Size_StaticFactorbase;
  static int biggest_Prime_in_Factorbase;
 public:
+ static int PrimeNumbers[MaxSize] __attribute__ ((aligned (64)));
  static FBsizetype Size() { return Size_StaticFactorbase; }
  static int StaticPrime(const int i) { return PrimeNumbers[i]; }
  static int BiggestPrime() { return biggest_Prime_in_Factorbase; }
--- src/unix_buffer.H.orig	2007-11-07 01:17:33.000000000 +0000
+++ src/unix_buffer.H	2010-08-14 16:51:27.685594125 +0000
@@ -50,6 +50,7 @@
  #endif
 #endif 
 
+#include <cstdio>
 #include <cerrno>
 #include <cstring>
 #include <string>
