diff --git a/setup.cfg b/setup.cfg
index dac401c..1a80b1d 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -14,10 +14,3 @@ packages =
 
 py_modules =
     sage_conf
-
-scripts =
-    bin/sage-env-config
-
-[options.entry_points]
-console_scripts =
-    sage-config = sage_conf:_main
diff --git a/setup.py b/setup.py
index a689f4d..60f964e 100644
--- a/setup.py
+++ b/setup.py
@@ -71,11 +71,10 @@ class build_py(setuptools_build_py):
 class build_scripts(distutils_build_scripts):
 
     def run(self):
-        self.distribution.scripts.append(os.path.join('bin', 'sage-env-config'))
         if not self.distribution.entry_points:
             self.entry_points = self.distribution.entry_points = dict()
         distutils_build_scripts.run(self)
 
 setup(
-    cmdclass=dict(build_py=build_py, build_scripts=build_scripts)
+    cmdclass=dict(build_scripts=build_scripts)
 )
