diff --git a/ginac/basic.cpp b/ginac/basic.cpp
index 0bb610e..473699f 100644
--- a/ginac/basic.cpp
+++ b/ginac/basic.cpp
@@ -604,10 +605,6 @@ ex basic::subs_one_level(const exmap & m, unsigned options) const
  *  will already be evaluated. */
 ex basic::subs(const exmap & m, unsigned options) const
 {
-        if (std::all_of(m.cbegin(), m.cend(),
-                                [](std::pair<ex,ex> p) 
-                                { return not haswild(p.first); } ))
-                options |= subs_options::no_pattern;
 	size_t num = nops();
 	if (num != 0u) {
 
diff --git a/ginac/expairseq.cpp b/ginac/expairseq.cpp
index b8b1569..ff85ced 100644
--- a/ginac/expairseq.cpp
+++ b/ginac/expairseq.cpp
@@ -474,10 +474,6 @@ found:		;
 
 ex expairseq::subs(const exmap & m, unsigned options) const
 {
-        if (std::all_of(m.cbegin(), m.cend(),
-                                [](std::pair<ex,ex> p) 
-                                { return not haswild(p.first); } ))
-                options |= subs_options::no_pattern;
 	std::unique_ptr<epvector> vp = subschildren(m, options);
 	if (vp.get() != nullptr)
 		return ex_to<basic>(thisexpairseq(std::move(vp), overall_coeff, (options & subs_options::no_index_renaming) == 0));

