From 1d23aeb9733728772954bfb7d2beabdd5e92b80f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fran=C3=A7ois=20Bissey?= <francois.bissey@canterbury.ac.nz>
Date: Tue, 15 Dec 2015 14:35:28 +1300
Subject: [PATCH 1/2] Check for gmp and link to it by default

---
 configure.ac      | 3 +++
 ginac/Makefile.am | 2 +-
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 1209f2f..ee00d75 100644
--- a/configure.ac
+++ b/configure.ac
@@ -95,6 +95,9 @@ AC_PROG_INSTALL
 AM_PROG_LIBTOOL
 AC_LIBTOOL_WIN32_DLL
 
+AC_CHECK_HEADERS([gmp.h], , AC_MSG_ERROR([This package needs gmp headers]))
+AC_SEARCH_LIBS([__gmpz_get_str], [gmp], [], [AC_MSG_ERROR([This package needs libgmp])])
+
 dnl Check for data types which are needed by the hash function 
 dnl (golden_ratio_hash).
 AC_CHECK_SIZEOF(int)
diff --git a/ginac/Makefile.am b/ginac/Makefile.am
index a0a88be..989942e 100644
--- a/ginac/Makefile.am
+++ b/ginac/Makefile.am
@@ -19,7 +19,7 @@ libpynac_la_LDFLAGS = -version-info $(LT_VERSION_INFO)
 endif
 libpynac_la_CPPFLAGS = $(PYTHON_CPPFLAGS) -Wall -Wextra -Wshadow -Wnon-virtual-dtor -Wno-unused-parameter
 
-libpynac_la_LIBADD = $(PYTHON_LDFLAGS)
+libpynac_la_LIBADD = $(PYTHON_LDFLAGS) $(LIBS)
 ginacincludedir = $(includedir)/pynac
 ginacinclude_HEADERS = ginac.h py_funcs.h add.h archive.h assertion.h basic.h class_info.h \
   clifford.h constant.h infinity.h container.h ex.h expair.h expairseq.h \

From 9dd3f6761d6c2a7829210ae1e07eaae8e96f9de9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fran=C3=A7ois=20Bissey?= <francois.bissey@canterbury.ac.nz>
Date: Tue, 15 Dec 2015 14:43:24 +1300
Subject: [PATCH 2/2] Remove duplacte gmp header check

---
 configure.ac | 2 --
 1 file changed, 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index ee00d75..b1449cb 100644
--- a/configure.ac
+++ b/configure.ac
@@ -126,8 +126,6 @@ if test "x$CONFIG_RUSAGE" = "xno"; then
     AC_CHECK_HEADER(ctime, , GINAC_ERROR([The standard <ctime> header file could not be found.]))
 fi
 
-AC_CHECK_HEADERS([gmp.h], , AC_MSG_ERROR([This package needs libgmp]))
-
 dnl Check for utilities needed by the different kinds of documentation.
 dnl Documentation needs only be built when extending it, so never mind if it
 dnl cannot find those helpers:
