From 23f41c286fe0132ab071ea1dbb40386569998b5a Mon Sep 17 00:00:00 2001
From: Ralf Stephan <ralf@ark.in-berlin.de>
Date: Thu, 5 Jan 2017 15:28:49 +0100
Subject: [PATCH] resultant() was outcommented with giac support ON

---
 ginac/mpoly-ginac.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/ginac/mpoly-ginac.cpp b/ginac/mpoly-ginac.cpp
index 55e202f..6630a57 100644
--- a/ginac/mpoly-ginac.cpp
+++ b/ginac/mpoly-ginac.cpp
@@ -27,8 +27,6 @@
 #include "pynac-config.h"
 #endif
 
-#ifndef PYNAC_HAVE_LIBGIAC
-
 #include "basic.h"
 #include "ex.h"
 #include "mpoly.h"
@@ -55,6 +53,8 @@
 
 namespace GiNaC {
 
+#ifndef PYNAC_HAVE_LIBGIAC
+
 // If comparing expressions (ex::compare()) is fast, you can set this to 1.
 // Some routines like quo(), rem() and gcd() will then return a quick answer
 // when they are called with two identical arguments.
@@ -708,6 +708,7 @@ ex sqrfree_parfrac(const ex & a, const symbol & x)
 	return red_poly + sum;
 }
 
+#endif // HAVE_LIBGIAC
 
 /** Resultant of two expressions e1,e2 with respect to symbol s.
  *  Method: Compute determinant of Sylvester matrix of e1,e2,s.  */
@@ -748,4 +749,3 @@ ex resultant(const ex & e1, const ex & e2, const ex & s)
 
 } // namespace GiNaC
 
-#endif // HAVE_LIBGIAC
