diff --git a/Makefile.am b/Makefile.am
index ad15723..776ae3e 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -2,14 +2,6 @@
 SUBDIRS = ginac 
 DIST_SUBDIRS = ginac
 
-# pkg-config metadata
-pkgconfigdir = $(libdir)/pkgconfig
-pkgconfig_DATA = pynac.pc
-$(pkgconfig_DATA): config.status
-
-# All the rest of the distributed files
-EXTRA_DIST = pynac.pc pynac.spec CONTRIBUTING.md PACKAGING.md
-
 # Rule to build tar-bzipped distribution package
 $(PACKAGE)-$(VERSION).tar.gz: dist
 
diff --git a/configure.ac b/configure.ac
index 9c50701..1e84fe8 100644
--- a/configure.ac
+++ b/configure.ac
@@ -217,8 +217,6 @@ RM="$RM -f"
 dnl Output makefiles etc.
 AC_CONFIG_FILES([
 Makefile
-pynac.spec
-pynac.pc
 ginac/Makefile
 ginac/version.h
 ])
