diff --git a/Makefile.am b/Makefile.am
index 485a19f..55a16d8 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -2,19 +2,7 @@
 SUBDIRS = ginac 
 DIST_SUBDIRS = ginac
 
-# pkg-config metadata
-pkgconfigdir = $(libdir)/pkgconfig
-pkgconfig_DATA = pynac.pc
-$(pkgconfig_DATA): config.status
-
-# All the rest of the distributed files
-EXTRA_DIST = pynac.pc pynac.spec
-
 # Rule to build tar-bzipped distribution package
 $(PACKAGE)-$(VERSION).tar.gz: dist
 
-# Rule to build RPM distribution package
-rpm: $(PACKAGE)-$(VERSION).tar.gz
-	rpmbuild -ta --clean $(PACKAGE)-$(VERSION).tar.gz
-
 ACLOCAL_AMFLAGS=-I m4
diff --git a/configure.ac b/configure.ac
index 7ceec03..d89b647 100644
--- a/configure.ac
+++ b/configure.ac
@@ -179,8 +179,6 @@ RM="$RM -f"
 dnl Output makefiles etc.
 AC_CONFIG_FILES([
 Makefile
-pynac.spec
-pynac.pc
 ginac/Makefile
 ginac/version.h
 ])
