--- module_list.py.orig	2010-06-26 16:23:29.532595646 +1200
+++ module_list.py	2010-06-26 16:50:54.526596704 +1200
@@ -516,6 +515,7 @@
               sources = ['sage/libs/singular/singular.pyx'],
               libraries = ['m', 'readline', 'singular', 'givaro', 'ntl', 'gmpxx', 'gmp'],
               language="c++",
+              runtime_library_dirs = [''],
               include_dirs = [SAGE_ROOT +'/local/include/singular'],
               depends = [SAGE_ROOT + "/local/include/libsingular.h"]),
     
@@ -523,6 +523,7 @@
               sources = ['sage/libs/singular/polynomial.pyx'],
               libraries = ['m', 'readline', 'singular', 'givaro', 'gmpxx', 'gmp'],
               language="c++",
+              runtime_library_dirs = [''],
               include_dirs = [SAGE_ROOT +'/local/include/singular'],
               depends = [SAGE_ROOT + "/local/include/libsingular.h"]),
 
@@ -530,6 +531,7 @@
               sources = ['sage/libs/singular/ring.pyx'],
               libraries = ['m', 'readline', 'singular', 'givaro', 'gmpxx', 'gmp'],
               language="c++",
+              runtime_library_dirs = [''],
               include_dirs = [SAGE_ROOT +'/local/include/singular'],
               depends = [SAGE_ROOT + "/local/include/libsingular.h"]),
 
@@ -537,6 +539,7 @@
               sources = ['sage/libs/singular/groebner_strategy.pyx'],
               libraries = ['m', 'readline', 'singular', 'givaro', 'gmpxx', 'gmp'],
               language="c++",
+              runtime_library_dirs = [''],
               include_dirs = [SAGE_ROOT +'/local/include/singular'],
               depends = [SAGE_ROOT + "/local/include/libsingular.h"]),
 
@@ -544,6 +547,7 @@
               sources = ['sage/libs/singular/function.pyx'],
               libraries = ['m', 'readline', 'singular', 'givaro', 'gmpxx', 'gmp'],
               language="c++",
+              runtime_library_dirs = [''],
               include_dirs = [SAGE_ROOT +'/local/include/singular'],
               depends = [SAGE_ROOT + "/local/include/libsingular.h"]),
 
@@ -551,6 +555,7 @@
               sources = ['sage/libs/singular/option.pyx'],
               libraries = ['m', 'readline', 'singular', 'givaro', 'gmpxx', 'gmp'],
               language="c++",
+              runtime_library_dirs = [''],
               include_dirs = [SAGE_ROOT +'/local/include/singular'],
               depends = [SAGE_ROOT + "/local/include/libsingular.h"]),
 
@@ -797,6 +802,7 @@
               sources = ['sage/matrix/matrix_mpolynomial_dense.pyx'],
               libraries = ['m', 'readline', 'singular', 'givaro', 'gmpxx', 'gmp'],
               language="c++",
+              runtime_library_dirs = [''],
               include_dirs = [SAGE_ROOT +'/local/include/singular'],
               depends = [SAGE_ROOT + "/local/include/libsingular.h"]),
 
@@ -1299,6 +1305,7 @@
               sources = ['sage/rings/polynomial/multi_polynomial_ideal_libsingular.pyx'],
               libraries = ['m', 'readline', 'singular', 'givaro', 'gmpxx', 'gmp'],
               language="c++",
+              runtime_library_dirs = [''],
               include_dirs = [SAGE_ROOT +'/local/include/singular'],
               depends = [SAGE_ROOT + "/local/include/libsingular.h"]),
 
@@ -1306,6 +1313,7 @@
               sources = ['sage/rings/polynomial/multi_polynomial_libsingular.pyx'],
               libraries = ['m', 'readline', 'singular', 'givaro', 'gmpxx', 'gmp'],
               language="c++",
+              runtime_library_dirs = [''],
               include_dirs = [SAGE_ROOT +'/local/include/singular'],
               depends = [SAGE_ROOT + "/local/include/libsingular.h"]),
 
