--- module_list.py.orig	2010-02-18 22:18:05.495344282 +0000
+++ module_list.py	2010-02-19 16:59:18.271371188 +0000
@@ -1119,7 +1119,8 @@
     #          language='c++'),                 
     
     Extension('sage.rings.padics.padic_base_generic_element',
-              sources = ['sage/rings/padics/padic_base_generic_element.pyx']),
+              sources = ['sage/rings/padics/padic_base_generic_element.pyx'],
+              libraries=['gmp']),
     
     Extension('sage.rings.padics.padic_capped_absolute_element',
               sources = ['sage/rings/padics/padic_capped_absolute_element.pyx'],
