diff --git a/Makefile.rules b/Makefile.rules
index e0c46d25..df2f85d1 100644
--- a/Makefile.rules
+++ b/Makefile.rules
@@ -563,7 +563,7 @@ clean:
 
 LTINSTALL=$(LIBTOOL) --mode=install $(INSTALL)
 
-install: install-bin install-doc install-gaproot install-sysinfo install-headers install-libgap
+install: install-bin install-gaproot install-sysinfo install-headers install-libgap
 	@echo "+--------------------------------------------------------------------------+"
 	@echo "| WARNING, 'make install' support is still experimental, and may be buggy. |"
 	@echo "| It also does not take care of installing GAP packages (and may never).   |"
