diff --git a/Makefile.rules b/Makefile.rules
index 368814d..c7ecd4c 100644
--- a/Makefile.rules
+++ b/Makefile.rules
@@ -586,7 +586,7 @@ export gap_wrapper
 install-bin: gap
 	# install the real GAP executable as $(libdir)/gap/gap
 	$(INSTALL) -d -m 0755 $(DESTDIR)$(libdir)/gap
-	$(LTINSTALL) -s gap $(DESTDIR)$(libdir)/gap
+	$(LTINSTALL) gap $(DESTDIR)$(libdir)/gap
 	
 	# install a wrapper shell script invoking the real GAP executable as $(bindir)/gap
 	$(INSTALL) -d -m 0755 $(DESTDIR)$(bindir)
@@ -663,7 +663,7 @@ install-headers: $(FFDATA_H) build/version.h
 
 install-libgap: libgap.la
 	$(INSTALL) -d -m 0755 $(DESTDIR)$(libdir)
-	$(LTINSTALL) -s libgap.la $(DESTDIR)$(libdir)
+	$(LTINSTALL) libgap.la $(DESTDIR)$(libdir)
 
 
 .PHONY: install install-bin install-doc install-gaproot install-headers install-libgap
