diff --git a/Makefile.rules b/Makefile.rules
index 28d87e61..40c56871 100644
--- a/Makefile.rules
+++ b/Makefile.rules
@@ -479,6 +479,7 @@ install-headers:
 	$(QUIET_INSTALL)$(INSTALL) -d -m 0755 $(DESTDIR)$(includedir)/gap
 	$(QUIET_INSTALL)$(INSTALL) -d -m 0755 $(DESTDIR)$(includedir)/gap/hpc
 	$(QUIET_INSTALL)$(INSTALL) -m 0644 $(srcdir)/src/*.h $(DESTDIR)$(includedir)/gap
+	$(QUIET_INSTALL)$(INSTALL) -m 0644 $(srcdir)/gen/config.h $(DESTDIR)$(includedir)/gap
 	$(QUIET_INSTALL)$(INSTALL) -m 0644 $(srcdir)/src/hpc/*.h $(DESTDIR)$(includedir)/gap/hpc
 	# TODO: take care of config.h
 
