
--- ./profiler-main.c.orig	2010-04-25 12:33:50.686541349 +0000
+++ ./profiler-main.c	2010-04-25 12:33:50.813541002 +0000
@@ -32,7 +32,7 @@
 #include <string.h>
 #include <math.h>
 #include <time.h>
-#include <gmp.h>
+#include <mpir.h>
 
 
 prof_Driver_t prof_active_Driver = NULL;
--- ./zn_poly/demo/bernoulli/bernoulli.c.orig	2010-04-25 12:33:51.012542188 +0000
+++ ./zn_poly/demo/bernoulli/bernoulli.c	2010-04-25 12:33:51.015541488 +0000
@@ -21,7 +21,7 @@
 */
 
 #include <stdio.h>
-#include <gmp.h>
+#include <mpir.h>
 #include "zn_poly.h"
 
 
--- ./zn_poly/include/support.h.orig	2010-04-25 12:33:51.018542047 +0000
+++ ./zn_poly/include/support.h	2010-04-25 12:33:51.020541837 +0000
@@ -25,7 +25,7 @@
 
 
 #include <stdio.h>
-#include <gmp.h>
+#include <mpir.h>
 #include "zn_poly_internal.h"
 
 
--- ./zn_poly/include/zn_poly_internal.h.orig	2010-04-25 12:33:51.022542187 +0000
+++ ./zn_poly/include/zn_poly_internal.h	2010-04-25 12:33:51.028541139 +0000
@@ -35,7 +35,7 @@
 #ifndef ZN_POLY_INTERNAL_H
 #define ZN_POLY_INTERNAL_H
 
-#include <gmp.h>
+#include <mpir.h>
 #include <stddef.h>
 #include <stdio.h>
 #include "zn_poly.h"
--- ./zn_poly/src/support.h.orig	2010-04-25 12:33:51.031541907 +0000
+++ ./zn_poly/src/support.h	2010-04-25 12:33:51.034541907 +0000
@@ -25,7 +25,7 @@
 
 
 #include <stdio.h>
-#include <gmp.h>
+#include <mpir.h>
 #include "zn_poly_internal.h"
 
 
--- ./zn_poly/src/zn_poly_internal.h.orig	2010-04-25 12:33:51.036541698 +0000
+++ ./zn_poly/src/zn_poly_internal.h	2010-04-25 12:33:51.053542396 +0000
@@ -35,7 +35,7 @@
 #ifndef ZN_POLY_INTERNAL_H
 #define ZN_POLY_INTERNAL_H
 
-#include <gmp.h>
+#include <mpir.h>
 #include <stddef.h>
 #include <stdio.h>
 #include "zn_poly.h"
--- ./fmpz.c.orig	2010-04-25 12:33:51.070542955 +0000
+++ ./fmpz.c	2010-04-25 12:33:51.104541279 +0000
@@ -25,7 +25,7 @@
 
 *****************************************************************************/
 
-#include <gmp.h>
+#include <mpir.h>
 #include <stdio.h>
 #include <stdlib.h>
 
--- ./F_mpz.h.orig	2010-04-25 12:33:51.112541977 +0000
+++ ./F_mpz.h	2010-04-25 12:33:51.124541628 +0000
@@ -32,7 +32,7 @@
  
 #include <stdlib.h>
 #include <stdio.h>
-#include <gmp.h>
+#include <mpir.h>
 #include "flint.h"
 #include "mpn_extras.h"
 #include <zn_poly/zn_poly.h>
--- ./long_extras.c.orig	2010-04-25 12:33:51.170540790 +0000
+++ ./long_extras.c	2010-04-25 12:33:51.206540929 +0000
@@ -32,7 +32,7 @@
 #include <stdarg.h>
 #include <stdint.h>
 #include <math.h>
-#include <gmp.h>
+#include <mpir.h>
 #include "flint.h"
 #include "long_extras.h"
 #include "longlong_wrapper.h"
--- ./F_mpz.c.orig	2010-04-25 12:33:51.222541977 +0000
+++ ./F_mpz.c	2010-04-25 12:33:51.247541279 +0000
@@ -27,7 +27,7 @@
 #include <string.h>
 #include <math.h>
 #include <stdio.h>
-#include <gmp.h>
+#include <mpir.h>
 
 #include "flint.h"
 #include "mpn_extras.h"
--- ./F_mpz_mul-timing.c.orig	2010-04-25 12:33:51.251542466 +0000
+++ ./F_mpz_mul-timing.c	2010-04-25 12:33:51.252542396 +0000
@@ -25,7 +25,7 @@
 //#define TIMING
 
 
-#include <gmp.h>
+#include <mpir.h>
 #include <stdlib.h>
 #include <stdio.h>
 #include <math.h>
--- ./fmpz_poly-profile.c.orig	2010-04-25 12:33:51.254542256 +0000
+++ ./fmpz_poly-profile.c	2010-04-25 12:33:51.273541698 +0000
@@ -29,7 +29,7 @@
 
 #include <string.h>
 #include <math.h>
-#include <gmp.h>
+#include <mpir.h>
 #include "profiler-main.h"
 #include "flint.h"
 #include "memory-manager.h"
--- ./ZmodF_poly-test.c.orig	2010-04-25 12:33:51.281541349 +0000
+++ ./ZmodF_poly-test.c	2010-04-25 12:33:51.303542745 +0000
@@ -27,7 +27,7 @@
 
 #include <stdio.h>
 #include <string.h>
-#include <gmp.h>
+#include <mpir.h>
 #include "flint.h"
 #include "memory-manager.h"
 #include "ZmodF_poly.h"
--- ./test-support.h.orig	2010-04-25 12:33:51.316540860 +0000
+++ ./test-support.h	2010-04-25 12:33:51.318540929 +0000
@@ -32,7 +32,7 @@
  extern "C" {
 #endif
  
-#include <gmp.h>
+#include <mpir.h>
 #include "profiler.h"
 
 #define RUN_TEST(targetfunc) \
--- ./zmod_mat-test.c.orig	2010-04-25 12:33:51.327542256 +0000
+++ ./zmod_mat-test.c	2010-04-25 12:33:51.331542256 +0000
@@ -28,7 +28,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
-#include <gmp.h>
+#include <mpir.h>
 #include "flint.h"
 #include "memory-manager.h"
 #include "test-support.h"
--- ./F_mpz-test.c.orig	2010-04-25 12:33:51.333542117 +0000
+++ ./F_mpz-test.c	2010-04-25 12:33:51.366541977 +0000
@@ -29,7 +29,7 @@
 
 #include <stdio.h>
 #include <string.h>
-#include <gmp.h>
+#include <mpir.h>
 #include <time.h>
 #include "flint.h"
 #include "memory-manager.h"
--- ./expmod.c.orig	2010-04-25 12:33:51.382540860 +0000
+++ ./expmod.c	2010-04-25 12:33:51.385540720 +0000
@@ -21,7 +21,7 @@
 ===============================================================================*/
 #include <stdio.h>
 
-#include <gmp.h>
+#include <mpir.h>
 #include "mpz_extras.h"
 
 /*int main(void)
--- ./BPTJCubes.c.orig	2010-04-25 12:33:51.389541628 +0000
+++ ./BPTJCubes.c	2010-04-25 12:33:51.391541768 +0000
@@ -32,7 +32,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <gmp.h>
+#include <mpir.h>
 #include "long_extras.h"
 
 #define START 1000000 
--- ./bernoulli.c.orig	2010-04-25 12:33:51.393541768 +0000
+++ ./bernoulli.c	2010-04-25 12:33:51.394542396 +0000
@@ -31,7 +31,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <gmp.h>
+#include <mpir.h>
 
 #include "flint.h"
 #include "mpz_poly.h"
--- ./NTL-interface.cpp.orig	2010-04-25 12:33:51.396542047 +0000
+++ ./NTL-interface.cpp	2010-04-25 12:33:51.398541907 +0000
@@ -33,7 +33,7 @@
 #include <NTL/lip.h>
 #include <NTL/ctools.h>
 #include <NTL/g_lip.h>
-#include <gmp.h>
+#include <mpir.h>
 
 #include "flint.h"
 #include "fmpz.h"
--- ./fmpz.h.orig	2010-04-25 12:33:51.400542047 +0000
+++ ./fmpz.h	2010-04-25 12:33:51.402541349 +0000
@@ -32,7 +32,7 @@
  extern "C" {
 #endif
  
-#include <gmp.h>
+#include <mpir.h>
 #include "memory-manager.h"
 #include "flint.h"
 #include "long_extras.h"
--- ./zmod_poly.h.orig	2010-04-25 12:33:51.404542047 +0000
+++ ./zmod_poly.h	2010-04-25 12:33:51.407541279 +0000
@@ -27,7 +27,7 @@
 
 *****************************************************************************/
 
-#include <gmp.h>
+#include <mpir.h>
 #include <math.h>
 #include <stdio.h>
 #include <stdlib.h>
--- ./mpz_poly.h.orig	2010-04-25 12:33:51.410542117 +0000
+++ ./mpz_poly.h	2010-04-25 12:33:51.412541837 +0000
@@ -34,7 +34,7 @@
  
 #include <stdlib.h>
 #include <stdio.h>
-#include <gmp.h>
+#include <mpir.h>
 #include "memory-manager.h"
 #include "fmpz_poly.h"
 
--- ./QS/mp_poly.h.orig	2010-04-25 12:33:51.414542047 +0000
+++ ./QS/mp_poly.h	2010-04-25 12:33:51.415542256 +0000
@@ -29,7 +29,7 @@
 #ifndef MPPOLY_H
 #define MPPOLY_H
 
-#include <gmp.h>
+#include <mpir.h>
 
 #include "common.h"
 
--- ./QS/block_lanczos.c.orig	2010-04-25 12:33:51.417542047 +0000
+++ ./QS/block_lanczos.c	2010-04-25 12:33:51.424541907 +0000
@@ -36,7 +36,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <gmp.h>
+#include <mpir.h>
 #include "block_lanczos.h"
 
 #include "../flint.h"
--- ./QS/mpQS.h.orig	2010-04-25 12:33:51.427542187 +0000
+++ ./QS/mpQS.h	2010-04-25 12:33:51.428541837 +0000
@@ -29,7 +29,7 @@
 #ifndef MPQS_H
 #define MPQS_H
 
-#include <gmp.h>
+#include <mpir.h>
 
 #include "mp_factor_base.h"
 #include "common.h"
--- ./QS/tinyQS.h.orig	2010-04-25 12:33:51.429541907 +0000
+++ ./QS/tinyQS.h	2010-04-25 12:33:51.430541977 +0000
@@ -29,7 +29,7 @@
 #ifndef TINYQS_H
 #define TINYQS_H
 
-#include <gmp.h>
+#include <mpir.h>
 
 #include "common.h"
 
--- ./QS/linear_algebra.c.orig	2010-04-25 12:33:51.432542187 +0000
+++ ./QS/linear_algebra.c	2010-04-25 12:33:51.434541837 +0000
@@ -27,7 +27,7 @@
 
 ******************************************************************************/
 
-#include <gmp.h>
+#include <mpir.h>
 #include <stdio.h>
 #include <stdlib.h>
 
--- ./QS/mp_lprels.c.orig	2010-04-25 12:33:51.436542117 +0000
+++ ./QS/mp_lprels.c	2010-04-25 12:33:51.439541488 +0000
@@ -27,7 +27,7 @@
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
-#include <gmp.h>
+#include <mpir.h>
 #include <unistd.h>
 
 #include "mp_lprels.h"
--- ./QS/poly.h.orig	2010-04-25 12:33:51.455541069 +0000
+++ ./QS/poly.h	2010-04-25 12:33:51.459542047 +0000
@@ -29,7 +29,7 @@
 #ifndef POLY_H
 #define POLY_H
 
-#include <gmp.h>
+#include <mpir.h>
 
 #include "common.h"
 
--- ./QS/mp_sieve.c.orig	2010-04-25 12:33:51.462541698 +0000
+++ ./QS/mp_sieve.c	2010-04-25 12:33:51.466546587 +0000
@@ -27,7 +27,7 @@
 
 ******************************************************************************/
 
-#include <gmp.h>
+#include <mpir.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
--- ./QS/mp_sieve.h.orig	2010-04-25 12:33:51.468542047 +0000
+++ ./QS/mp_sieve.h	2010-04-25 12:33:51.469542396 +0000
@@ -29,7 +29,7 @@
 #ifndef MPSIEVE_H
 #define MPSIEVE_H
 
-#include <gmp.h>
+#include <mpir.h>
 
 #include "mp_linear_algebra.h"
 #include "common.h"
--- ./QS/poly.c.orig	2010-04-25 12:33:51.471542745 +0000
+++ ./QS/poly.c	2010-04-25 12:33:51.473542187 +0000
@@ -27,7 +27,7 @@
 
 ******************************************************************************/
 
-#include <gmp.h>
+#include <mpir.h>
 #include <stdio.h>
 #include <stdlib.h>
 
--- ./QS/mp_poly.c.orig	2010-04-25 12:33:51.476542326 +0000
+++ ./QS/mp_poly.c	2010-04-25 12:33:51.478541209 +0000
@@ -27,7 +27,7 @@
 
 ******************************************************************************/
 
-#include <gmp.h>
+#include <mpir.h>
 #include <stdio.h>
 #include <stdlib.h>
 
--- ./QS/linear_algebra.h.orig	2010-04-25 12:33:51.480541977 +0000
+++ ./QS/linear_algebra.h	2010-04-25 12:33:51.481542466 +0000
@@ -29,7 +29,7 @@
 #ifndef LINALG_H
 #define LINALG_H
 
-#include <gmp.h>
+#include <mpir.h>
 
 #include "common.h"
 #include "poly.h"
--- ./QS/mp_linear_algebra.c.orig	2010-04-25 12:33:51.483542256 +0000
+++ ./QS/mp_linear_algebra.c	2010-04-25 12:33:51.489542047 +0000
@@ -27,7 +27,7 @@
 
 ******************************************************************************/
 
-#include <gmp.h>
+#include <mpir.h>
 #include <stdio.h>
 #include <stdlib.h>
 
--- ./QS/factor_base.c.orig	2010-04-25 12:33:51.491542466 +0000
+++ ./QS/factor_base.c	2010-04-25 12:33:51.493542117 +0000
@@ -28,7 +28,7 @@
 
 ******************************************************************************/
 
-#include <gmp.h>
+#include <mpir.h>
 #include <stdio.h>
 #include <math.h>
 
--- ./QS/sieve.c.orig	2010-04-25 12:33:51.495542187 +0000
+++ ./QS/sieve.c	2010-04-25 12:33:51.496542256 +0000
@@ -27,7 +27,7 @@
 
 ******************************************************************************/
 
-#include <gmp.h>
+#include <mpir.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
--- ./QS/tinyQS.c.orig	2010-04-25 12:33:51.498542117 +0000
+++ ./QS/tinyQS.c	2010-04-25 12:33:51.514540999 +0000
@@ -30,7 +30,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <math.h>
-#include <gmp.h>
+#include <mpir.h>
 #include <string.h>
 
 #include "../fmpz.h"
--- ./QS/factor_base.h.orig	2010-04-25 12:33:51.518542745 +0000
+++ ./QS/factor_base.h	2010-04-25 12:33:51.520542955 +0000
@@ -29,7 +29,7 @@
 #ifndef FACTORBASE_H
 #define FACTORBASE_H
 
-#include <gmp.h>
+#include <mpir.h>
 
 #include "tinyQS.h"
 #include "common.h"
--- ./QS/sieve.h.orig	2010-04-25 12:33:51.521542117 +0000
+++ ./QS/sieve.h	2010-04-25 12:33:51.522542745 +0000
@@ -29,7 +29,7 @@
 #ifndef SIEVE_H
 #define SIEVE_H
 
-#include <gmp.h>
+#include <mpir.h>
 
 #include "linear_algebra.h"
 #include "common.h"
--- ./QS/mp_factor_base.c.orig	2010-04-25 12:33:51.524542256 +0000
+++ ./QS/mp_factor_base.c	2010-04-25 12:33:51.526542326 +0000
@@ -28,7 +28,7 @@
 
 ******************************************************************************/
 
-#include <gmp.h>
+#include <mpir.h>
 #include <stdio.h>
 #include <math.h>
 
--- ./QS/mpQS.c.orig	2010-04-25 12:33:51.530542256 +0000
+++ ./QS/mpQS.c	2010-04-25 12:33:51.533541558 +0000
@@ -30,7 +30,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <math.h>
-#include <gmp.h>
+#include <mpir.h>
 #include <string.h>
 
 #include "../fmpz.h"
--- ./QS/mp_factor_base.h.orig	2010-04-25 12:33:51.539540790 +0000
+++ ./QS/mp_factor_base.h	2010-04-25 12:33:51.541540999 +0000
@@ -29,7 +29,7 @@
 #ifndef MP_FACTORBASE_H
 #define MP_FACTORBASE_H
 
-#include <gmp.h>
+#include <mpir.h>
 
 #include "mpQS.h"
 #include "common.h"
--- ./QS/mp_linear_algebra.h.orig	2010-04-25 12:33:51.544542326 +0000
+++ ./QS/mp_linear_algebra.h	2010-04-25 12:33:51.545542396 +0000
@@ -29,7 +29,7 @@
 #ifndef MPLINALG_H
 #define MPLINALG_H
 
-#include <gmp.h>
+#include <mpir.h>
 #include <stdio.h>
 
 #include "common.h"
--- ./fmpz_poly-test.c.orig	2010-04-25 12:33:51.603775808 +0000
+++ ./fmpz_poly-test.c	2010-04-25 12:33:51.716542046 +0000
@@ -27,7 +27,7 @@
 
 #include <stdio.h>
 #include <string.h>
-#include <gmp.h>
+#include <mpir.h>
 #include <time.h>
 
 #include "flint.h"
--- ./ZmodF_poly.h.orig	2010-04-25 12:33:51.764543094 +0000
+++ ./ZmodF_poly.h	2010-04-25 12:33:51.766542325 +0000
@@ -38,7 +38,7 @@
  
 #include <stdlib.h>
 #include <stdio.h>
-#include <gmp.h>
+#include <mpir.h>
 #include "memory-manager.h"
 #include "ZmodF.h"
 
--- ./zmod_poly-test.c.orig	2010-04-25 12:33:51.863116082 +0000
+++ ./zmod_poly-test.c	2010-04-25 12:33:51.910540999 +0000
@@ -30,7 +30,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
-#include <gmp.h>
+#include <mpir.h>
 #include "flint.h"
 #include "memory-manager.h"
 #include "test-support.h"
--- ./mpz_extras.c.orig	2010-04-25 12:33:51.920542117 +0000
+++ ./mpz_extras.c	2010-04-25 12:33:51.924541907 +0000
@@ -22,7 +22,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <gmp.h>
+#include <mpir.h>
 #include "mpz_extras.h"
 #include "flint.h"
 #include "mpn_extras.h"
--- ./thetaproduct.c.orig	2010-04-25 12:33:51.931541418 +0000
+++ ./thetaproduct.c	2010-04-25 12:33:51.943542117 +0000
@@ -27,7 +27,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <math.h>
-#include <gmp.h>
+#include <mpir.h>
 #include <omp.h>
 #include <sys/types.h>
 #include <sys/stat.h>
--- ./ZmodF.h.orig	2010-04-25 12:33:51.957541139 +0000
+++ ./ZmodF.h	2010-04-25 12:33:51.974541768 +0000
@@ -39,7 +39,7 @@
 #endif
  
 #include <stdlib.h>
-#include <gmp.h>
+#include <mpir.h>
 #include "flint.h"
 
 /*
--- ./memory-manager.c.orig	2010-04-25 12:33:51.981542326 +0000
+++ ./memory-manager.c	2010-04-25 12:33:51.983541907 +0000
@@ -26,7 +26,7 @@
 *****************************************************************************/
 
 #include <stdlib.h>
-#include <gmp.h>
+#include <mpir.h>
 #include <string.h>
 #include <stdio.h>
 #include "flint.h"
--- ./zmod_mat.h.orig	2010-04-25 12:33:51.986542326 +0000
+++ ./zmod_mat.h	2010-04-25 12:33:51.987542536 +0000
@@ -26,7 +26,7 @@
 
 *****************************************************************************/
 
-#include <gmp.h>
+#include <mpir.h>
 #include <math.h>
 #include <stdio.h>
 #include <stdlib.h>
--- ./mpn_extras.c.orig	2010-04-25 12:33:51.992542117 +0000
+++ ./mpn_extras.c	2010-04-25 12:33:52.000541628 +0000
@@ -29,7 +29,7 @@
 
 ******************************************************************************/
 
-#include <gmp.h>
+#include <mpir.h>
 #include <stdlib.h>
 #include <string.h>
 #include <stdio.h>
--- ./bernoulli_zmod.c.orig	2010-04-25 12:33:52.012542047 +0000
+++ ./bernoulli_zmod.c	2010-04-25 12:33:52.019541209 +0000
@@ -31,7 +31,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <gmp.h>
+#include <mpir.h>
 
 #include "flint.h"
 #include "long_extras.h"
--- ./bernoulli_fmpz.c.orig	2010-04-25 12:33:52.038542815 +0000
+++ ./bernoulli_fmpz.c	2010-04-25 12:33:52.047541977 +0000
@@ -31,7 +31,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <gmp.h>
+#include <mpir.h>
 
 #include "flint.h"
 #include "long_extras.h"
--- ./long_extras-test.c.orig	2010-04-25 12:33:52.067540999 +0000
+++ ./long_extras-test.c	2010-04-25 12:33:52.100541349 +0000
@@ -27,7 +27,7 @@
 
 #include <stdio.h>
 #include <string.h>
-#include <gmp.h>
+#include <mpir.h>
 #include "flint.h"
 #include "long_extras.h"
 #include "test-support.h"
--- ./fmpz_poly.h.orig	2010-04-25 12:33:52.107542117 +0000
+++ ./fmpz_poly.h	2010-04-25 12:33:52.113541349 +0000
@@ -34,7 +34,7 @@
  
 #include <stdlib.h>
 #include <stdio.h>
-#include <gmp.h>
+#include <mpir.h>
 #include "mpn_extras.h"
 #include "fmpz.h"
 #include "zmod_poly.h"
--- ./flint.h.orig	2010-04-25 12:33:52.116541907 +0000
+++ ./flint.h	2010-04-25 12:33:52.118542117 +0000
@@ -31,7 +31,7 @@
 
 #include <limits.h>
 #include <assert.h>
-#include <gmp.h>
+#include <mpir.h>
 #include <math.h>
 #include <stdint.h>
 #include "longlong_wrapper.h"
--- ./ZmodF_mul.h.orig	2010-04-25 12:33:52.120542117 +0000
+++ ./ZmodF_mul.h	2010-04-25 12:33:52.121542536 +0000
@@ -36,7 +36,7 @@
 #endif
  
 #include <stdlib.h>
-#include <gmp.h>
+#include <mpir.h>
 #include "mpn_extras.h"
 #include "ZmodF_poly.h"
 
--- ./delta_qexp.c.orig	2010-04-25 12:33:52.123542187 +0000
+++ ./delta_qexp.c	2010-04-25 12:33:52.124542466 +0000
@@ -24,7 +24,7 @@
 */
 
 #include <stdio.h>
-#include <gmp.h>
+#include <mpir.h>
 #include <math.h>
 #include "flint.h"
 #include "fmpz_poly.h"
--- ./mpn_extras-test.c.orig	2010-04-25 12:33:52.126542047 +0000
+++ ./mpn_extras-test.c	2010-04-25 12:33:52.128541977 +0000
@@ -27,7 +27,7 @@
 
 #include <stdio.h>
 #include <string.h>
-#include <gmp.h>
+#include <mpir.h>
 #include "flint.h"
 #include "memory-manager.h"
 #include "mpn_extras.h"
--- ./NTL-profile.c.orig	2010-04-25 12:33:52.130542187 +0000
+++ ./NTL-profile.c	2010-04-25 12:33:52.132542466 +0000
@@ -29,7 +29,7 @@
 
 #include <string.h>
 #include <math.h>
-#include <gmp.h>
+#include <mpir.h>
 #include "profiler-main.h"
 #include "flint.h"
 #include "memory-manager.h"
--- ./NTL-interface-test.cpp.orig	2010-04-25 12:33:52.140541768 +0000
+++ ./NTL-interface-test.cpp	2010-04-25 12:33:52.145541139 +0000
@@ -31,7 +31,7 @@
 #include <NTL/ZZ.h>
 #include <NTL/ZZX.h>
 #include <NTL/mat_ZZ.h>
-#include <gmp.h>
+#include <mpir.h>
 
 #include "NTL-interface.h"
 #include "fmpz.h"
--- ./fmpz-test.c.orig	2010-04-25 12:33:52.152542256 +0000
+++ ./fmpz-test.c	2010-04-25 12:33:52.161541418 +0000
@@ -28,7 +28,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
-#include <gmp.h>
+#include <mpir.h>
 #include "flint.h"
 #include "test-support.h"
 #include "long_extras.h"
--- ./mpz_extras.h.orig	2010-04-25 12:33:52.176541139 +0000
+++ ./mpz_extras.h	2010-04-25 12:33:52.181540999 +0000
@@ -23,7 +23,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdint.h>
-#include <gmp.h>
+#include <mpir.h>
 
 #ifndef FLINT_MPZ_EXTRAS_H
 #define FLINT_MPZ_EXTRAS_H
--- makefile.orig	2010-04-25 12:35:02.731541349 +0000
+++ makefile	2010-04-25 12:36:05.375541348 +0000
@@ -22,9 +22,9 @@
 
 CPP = $(FLINT_CPP) -fPIC
 
-LIBS = -L$(FLINT_GMP_LIB_DIR) $(FLINT_LINK_OPTIONS) -lgmp -lpthread -lm -lzn_poly
+LIBS = -L$(FLINT_GMP_LIB_DIR) $(FLINT_LINK_OPTIONS) -lmpir -lpthread -lm -lzn_poly
 
-LIBS2 = -L$(FLINT_GMP_LIB_DIR) -L$(FLINT_NTL_LIB_DIR) $(FLINT_LINK_OPTIONS) -lgmp -lpthread -lntl -lm -lzn_poly
+LIBS2 = -L$(FLINT_GMP_LIB_DIR) -L$(FLINT_NTL_LIB_DIR) $(FLINT_LINK_OPTIONS) -lmpir -lpthread -lntl -lm -lzn_poly
 
 ifndef FLINT_NTL_INCLUDE_DIR
 	INCS = -I$(FLINT_GMP_INCLUDE_DIR) 
