--- makefile.orig	2011-07-02 10:42:08.076386261 +1200
+++ makefile	2011-07-02 10:44:25.884385998 +1200
@@ -137,17 +137,17 @@
 
 library: $(FLINT_LIB)
 
-libflint.dylib: $(FLINTOBJ)
-	$(CC) -single_module -fPIC -dynamiclib -o libflint.dylib $(FLINTOBJ) $(LIBS)
+libflint.dylib: $(FLINTOBJ) NTL-interface.o
+	$(CXX) -single_module -fPIC -dynamiclib -o libflint.dylib $(FLINTOBJ) NTL-interface.o $(LIBS2)
 
-libflint.dylib64: $(FLINTOBJ)
-	$(CC) -m64 -single_module -fPIC -dynamiclib -o libflint.dylib $(FLINTOBJ) $(LIBS)
+libflint.dylib64: $(FLINTOBJ) NTL-interface.o
+	$(CXX) -m64 -single_module -fPIC -dynamiclib -o libflint.dylib $(FLINTOBJ) NTL-interface.o $(LIBS2)
 
-libflint.dll: $(FLINTOBJ)
-	$(CC) -fPIC -shared -o libflint.dll $(FLINTOBJ) $(LIBS)
+libflint.dll: $(FLINTOBJ) NTL-interface.o
+	$(CXX) -fPIC -shared -o libflint.dll $(FLINTOBJ) NTL-interface.o $(LIBS2)
 
-libflint.so: $(FLINTOBJ)
-	$(CC) -fPIC -shared -Wl,-soname,libflint.so -o libflint.so $(FLINTOBJ) $(LIBS)
+libflint.so: $(FLINTOBJ) NTL-interface.o
+	$(CXX) -fPIC -shared -Wl,-soname,libflint.so -o libflint.so $(FLINTOBJ) NTL-interface.o $(LIBS2)
 
 ##### Object files
 
