diff --git a/setup.py b/setup.py
index 41037e9..87f3c04 100644
--- a/setup.py
+++ b/setup.py
@@ -188,9 +188,9 @@ class develop_with_jquery(DevelopCommand):
         _download_jquery_to("lib/matplotlib/backends/web_backend/")
 
 
-cmdclass['sdist'] = sdist_with_jquery
-cmdclass['install_lib'] = install_lib_with_jquery
-cmdclass['develop'] = develop_with_jquery
+#cmdclass['sdist'] = sdist_with_jquery
+#cmdclass['install_lib'] = install_lib_with_jquery
+#cmdclass['develop'] = develop_with_jquery
 
 
 # One doesn't normally see `if __name__ == '__main__'` blocks in a setup.py,
diff --git a/setupext.py b/setupext.py
index c75725b..15442e3 100644
--- a/setupext.py
+++ b/setupext.py
@@ -871,6 +871,9 @@ class Matplotlib(SetupPackage):
                 'backends/web_backend/jquery/js/*.min.js',
                 'backends/web_backend/jquery/css/themes/base/*.min.css',
                 'backends/web_backend/jquery/css/themes/base/images/*',
+                'backends/web_backend/jquery-ui-1.12.1/*',
+                'backends/web_backend/jquery-ui-1.12.1/external/jquery/*',
+                'backends/web_backend/jquery-ui-1.12.1/images/*',
                 'backends/web_backend/css/*.*',
                 'backends/Matplotlib.nib/*',
                 'mpl-data/stylelib/*.mplstyle',
