Index: /trunk/scipy/io/dumbdbm_patched.py
===================================================================
--- /trunk/scipy/io/dumbdbm_patched.py (revision 3521)
+++ /trunk/scipy/io/dumbdbm_patched.py (revision 5967)
@@ -78,4 +78,7 @@
         return dat
 
+    def __contains__(self, key):
+        return key in self._index
+
     def _addval(self, val):
         f = _open(self._datfile, 'rb+')
