--- src/mpfi.c.orig	2010-06-23 22:53:15.181852072 +1200
+++ src/mpfi.c	2010-06-23 22:55:20.846851274 +1200
@@ -692,6 +692,7 @@
   mp_prec_t prec;
   mpfr_t diam, fact;
   int dummy;
+  gmp_randstate_t seed;
   prec=mpfr_get_prec(m);
 
   if ( MPFI_NAN_P(y) ) {
@@ -702,7 +703,8 @@
   mpfr_init2(fact, prec);
 
   dummy = mpfi_diam_abs(diam, y);
-  mpfr_random(fact);
+  gmp_randinit_default(seed);
+  mpfr_urandomb(fact,seed);
 
   /* fact lies between 0 and 1, the picked point lies at a relative
      distance "fact" of the left endpoint:  m = inf + (sup - inf)*fact  */
