--- a/C/cvxopt.h  Mon Jul 26 18:45:42 2010 +0300
+++ b/C/cvxopt.h  Thu Jul 29 09:12:21 2010 -0700
@@ -61,7 +61,7 @@
 typedef struct {
   PyObject_HEAD
   void *buffer;          /* in column-major-mode array of type 'id' */
-  int_t nrows, ncols;    /* number of rows and columns */
+  int nrows, ncols;    /* number of rows and columns -- was int_t */
   int   id;              /* DOUBLE, INT, COMPLEX */
 } matrix;