--- setup.py.orig	2013-10-03 14:43:21.608722052 +1300
+++ setup.py	2013-10-03 14:43:49.456771410 +1300
@@ -3,9 +3,9 @@
 $Date: 2005/11/13 01:31:46 $
 '''
 from distutils.core import setup
-setup (name='pexpect',
+setup (name='sage_pexpect',
     version='2.0',
-    py_modules=['pexpect'],
+    py_modules=['sage_pexpect'],
     description='Pexpect is a pure Python Expect. It allows easy control of other applications.',
     author='Noah Spurrier',
     author_email='noah@noah.org',
