--- g0n/Makefile.orig	2010-08-08 10:12:39.613699404 +1200
+++ g0n/Makefile	2010-08-08 10:20:17.526698392 +1200
@@ -12,9 +12,13 @@
 
 DOT_H = periods.h cusp.h homspace.h newforms.h nflist.h moddata.h oldforms.h symb.h fixc6.h pcprocs.h
 
-DOT_CC = h1bsd.cc newforms.cc symb.cc checkap.cc h1bsdc.cc homspace.cc nfcount.cc h1clist.cc homtest.cc nflist.cc th1nf.cc cnf.cc h1curisog.cc hpcurve.cc nftest.cc tmanin.cc curvesort.cc h1curve.cc oftest.cc cusp.cc h1degphi.cc lf1per.cc oldforms.cc fixc6.cc pcurve.cc tnf2.cc h1first.cc periods.cc vlift.cc mhcount.cc h1test.cc moddata.cc hecketest.cc modtest.cc putoldeigs.cc h1.cc nfhpcurve.cc   
+DOT_CC = h1bsd.cc newforms.cc symb.cc checkap.cc h1bsdc.cc homspace.cc nfcount.cc h1clist.cc \
+	homtest.cc nflist.cc th1nf.cc cnf.cc h1curisog.cc hpcurve.cc nftest.cc tmanin.cc curvesort.cc \
+	h1curve.cc oftest.cc cusp.cc h1degphi.cc lf1per.cc oldforms.cc fixc6.cc pcurve.cc tnf2.cc \
+	h1first.cc periods.cc vlift.cc mhcount.cc h1test.cc moddata.cc hecketest.cc modtest.cc \
+	putoldeigs.cc h1.cc nfhpcurve.cc
 
-HEADERS = $(DOT_H)
+headers: $(DOT_H)
 sources: $(DOT_H) $(DOT_CC)
 
 TESTS = modtest homtest hecketest nftest mhcount oftest
@@ -44,7 +48,8 @@
 CURVELIBFLAG = -L../lib -lcurvesntl.
 RANKLIBNAME = librankntl.a
 RANKLIBFLAG = -L../lib -lrankntl
-OBJ_SUF = _n
+OBJ_SUF = _n.o
+SOBJ_SUF = _n.lo
 EXE_SUFFIX = _n
 G0NLIBBASE = libg0nntl
 G0NLIBFLAG = -L../lib -lg0nntl
@@ -65,7 +70,11 @@
 
 G0NLIBNAME = $(G0NLIBBASE).a
 
-OBJS = cusp$(OBJ_SUF).o homspace$(OBJ_SUF).o symb$(OBJ_SUF).o moddata$(OBJ_SUF).o oldforms$(OBJ_SUF).o newforms$(OBJ_SUF).o periods$(OBJ_SUF).o fixc6$(OBJ_SUF).o
+OBJS = cusp$(OBJ_SUF) homspace$(OBJ_SUF) symb$(OBJ_SUF) moddata$(OBJ_SUF) oldforms$(OBJ_SUF) \
+	newforms$(OBJ_SUF) periods$(OBJ_SUF) fixc6$(OBJ_SUF)
+
+SOBJS = cusp$(SOBJ_SUF) homspace$(SOBJ_SUF) symb$(SOBJ_SUF) moddata$(SOBJ_SUF) oldforms$(SOBJ_SUF) \
+	newforms$(SOBJ_SUF) periods$(SOBJ_SUF) fixc6$(SOBJ_SUF)
 
 objs: $(OBJS)
 
@@ -76,15 +85,15 @@
 	$(CP) $(DOT_CC) $(HOME)/curves/g0ndist
 	chmod 644 $(HOME)/curves/g0ndist/*
 
-$(G0NLIBNAME): $(OBJS)
+$(G0NLIBNAME): objs
 	ar r $(G0NLIBNAME) $(OBJS)
 	$(RANLIB) $(G0NLIBNAME)
 
 lib: $(G0NLIBNAME)
 
-install_lib: $(HEADERS) $(G0NLIBNAME)
-	(cd $(INCDIR); rm -f $(HEADERS))
-	for f in $(HEADERS); \
+install_lib: headers lib
+	(cd $(INCDIR); rm -f $(DOT_H))
+	for f in $(DOT_H); \
 	 do cp $${f} $(INCDIR); \
 	done
 	chmod a+r $(INCDIR)/*
@@ -100,9 +109,14 @@
 
 LIBFLAGS = $(LFLAGS)
 
-%$(OBJ_SUF).o:   %.cc
+#template for compiling *.cc -> *$(OBJ_SUF) :
+.SUFFIXES: .cc $(OBJ_SUF) $(SOBJ_SUF)
+
+ .cc$(OBJ_SUF):
 	$(CXX) $(CFLAGS) $< -o $@
 
+ .cc$(SOBJ_SUF):
+	$(CXX) $(PICFLAG) $(CFLAGS) $< -o $@
 
 # TARGETS
 
@@ -112,12 +126,12 @@
 tests: $(TESTS)
 
 clean:
-	/bin/rm -f *~ *$(OBJ_SUF).o *.a *.so *.dylib *.dll* *.def *.exe
+	/bin/rm -f *~ *$(OBJ_SUF) *$(SOBJ_SUF) *.a *.so *.dylib *.dll* *.def *.exe
 
 veryclean: clean
 	/bin/rm -f $(PROGS) $(TESTS)
 
-check:  $(TESTS) $(PROGS)
+check:  tests progs
 	rm -f PRIMES t
 	./modtest < modtest.in > t 2>/dev/null && diff t modtest.out 
 	./homtest < homtest.in > t && diff t homtest.out
@@ -137,288 +151,288 @@
 input: input.cc
 	$(CXX) input.cc -o inp
 
-mhcount: mhcount$(OBJ_SUF).o
-	$(CXX) -o mhcount mhcount$(OBJ_SUF).o   $(LFLAGS)
+mhcount: mhcount$(OBJ_SUF)
+	$(CXX) -o mhcount mhcount$(OBJ_SUF)   $(LFLAGS)
 
-g0ngens: g0ngens$(OBJ_SUF).o symb$(OBJ_SUF).o moddata$(OBJ_SUF).o
-	$(CXX) -o g0ngens g0ngens$(OBJ_SUF).o symb$(OBJ_SUF).o moddata$(OBJ_SUF).o   $(LFLAGS)
+g0ngens: g0ngens$(OBJ_SUF) symb$(OBJ_SUF) moddata$(OBJ_SUF)
+	$(CXX) -o g0ngens g0ngens$(OBJ_SUF) symb$(OBJ_SUF) moddata$(OBJ_SUF)   $(LFLAGS)
 
-modtest: modtest$(OBJ_SUF).o symb$(OBJ_SUF).o moddata$(OBJ_SUF).o
-	$(CXX) -o modtest modtest$(OBJ_SUF).o symb$(OBJ_SUF).o moddata$(OBJ_SUF).o   $(LFLAGS)
+modtest: modtest$(OBJ_SUF) symb$(OBJ_SUF) moddata$(OBJ_SUF)
+	$(CXX) -o modtest modtest$(OBJ_SUF) symb$(OBJ_SUF) moddata$(OBJ_SUF)   $(LFLAGS)
 
-homtest: homtest$(OBJ_SUF).o symb$(OBJ_SUF).o moddata$(OBJ_SUF).o cusp$(OBJ_SUF).o homspace$(OBJ_SUF).o
-	$(CXX) -o homtest homtest$(OBJ_SUF).o symb$(OBJ_SUF).o moddata$(OBJ_SUF).o cusp$(OBJ_SUF).o homspace$(OBJ_SUF).o  $(LFLAGS)
+homtest: homtest$(OBJ_SUF) symb$(OBJ_SUF) moddata$(OBJ_SUF) cusp$(OBJ_SUF) homspace$(OBJ_SUF)
+	$(CXX) -o homtest homtest$(OBJ_SUF) symb$(OBJ_SUF) moddata$(OBJ_SUF) cusp$(OBJ_SUF) homspace$(OBJ_SUF)  $(LFLAGS)
 
-oftest: oftest$(OBJ_SUF).o symb$(OBJ_SUF).o moddata$(OBJ_SUF).o oldforms$(OBJ_SUF).o homspace$(OBJ_SUF).o newforms$(OBJ_SUF).o  cusp$(OBJ_SUF).o
-	$(CXX) -o oftest oftest$(OBJ_SUF).o symb$(OBJ_SUF).o moddata$(OBJ_SUF).o oldforms$(OBJ_SUF).o homspace$(OBJ_SUF).o newforms$(OBJ_SUF).o  cusp$(OBJ_SUF).o $(LFLAGS)
+oftest: oftest$(OBJ_SUF) symb$(OBJ_SUF) moddata$(OBJ_SUF) oldforms$(OBJ_SUF) homspace$(OBJ_SUF) newforms$(OBJ_SUF)  cusp$(OBJ_SUF)
+	$(CXX) -o oftest oftest$(OBJ_SUF) symb$(OBJ_SUF) moddata$(OBJ_SUF) oldforms$(OBJ_SUF) homspace$(OBJ_SUF) newforms$(OBJ_SUF)  cusp$(OBJ_SUF) $(LFLAGS)
 
-hecketest: hecketest$(OBJ_SUF).o cusp$(OBJ_SUF).o homspace$(OBJ_SUF).o symb$(OBJ_SUF).o moddata$(OBJ_SUF).o
-	$(CXX) -o hecketest hecketest$(OBJ_SUF).o cusp$(OBJ_SUF).o homspace$(OBJ_SUF).o symb$(OBJ_SUF).o moddata$(OBJ_SUF).o \
+hecketest: hecketest$(OBJ_SUF) cusp$(OBJ_SUF) homspace$(OBJ_SUF) symb$(OBJ_SUF) moddata$(OBJ_SUF)
+	$(CXX) -o hecketest hecketest$(OBJ_SUF) cusp$(OBJ_SUF) homspace$(OBJ_SUF) symb$(OBJ_SUF) moddata$(OBJ_SUF) \
 		 $(LFLAGS)
 
 
 
-gd: gd$(OBJ_SUF).o cusp$(OBJ_SUF).o homspace$(OBJ_SUF).o symb$(OBJ_SUF).o moddata$(OBJ_SUF).o
-	$(CXX) -o gd gd$(OBJ_SUF).o cusp$(OBJ_SUF).o homspace$(OBJ_SUF).o symb$(OBJ_SUF).o moddata$(OBJ_SUF).o \
+gd: gd$(OBJ_SUF) cusp$(OBJ_SUF) homspace$(OBJ_SUF) symb$(OBJ_SUF) moddata$(OBJ_SUF)
+	$(CXX) -o gd gd$(OBJ_SUF) cusp$(OBJ_SUF) homspace$(OBJ_SUF) symb$(OBJ_SUF) moddata$(OBJ_SUF) \
 		 $(LFLAGS)
 
-gd2: gd2$(OBJ_SUF).o cusp$(OBJ_SUF).o homspace$(OBJ_SUF).o symb$(OBJ_SUF).o moddata$(OBJ_SUF).o
-	$(CXX) -o gd2 gd2$(OBJ_SUF).o cusp$(OBJ_SUF).o homspace$(OBJ_SUF).o symb$(OBJ_SUF).o moddata$(OBJ_SUF).o \
+gd2: gd2$(OBJ_SUF) cusp$(OBJ_SUF) homspace$(OBJ_SUF) symb$(OBJ_SUF) moddata$(OBJ_SUF)
+	$(CXX) -o gd2 gd2$(OBJ_SUF) cusp$(OBJ_SUF) homspace$(OBJ_SUF) symb$(OBJ_SUF) moddata$(OBJ_SUF) \
 		 $(LFLAGS)
 
-tnfd: tnfd$(OBJ_SUF).o nfd$(OBJ_SUF).o cusp$(OBJ_SUF).o homspace$(OBJ_SUF).o symb$(OBJ_SUF).o moddata$(OBJ_SUF).o
-	$(CXX) -o tnfd tnfd$(OBJ_SUF).o nfd$(OBJ_SUF).o cusp$(OBJ_SUF).o homspace$(OBJ_SUF).o symb$(OBJ_SUF).o moddata$(OBJ_SUF).o \
+tnfd: tnfd$(OBJ_SUF) nfd$(OBJ_SUF) cusp$(OBJ_SUF) homspace$(OBJ_SUF) symb$(OBJ_SUF) moddata$(OBJ_SUF)
+	$(CXX) -o tnfd tnfd$(OBJ_SUF) nfd$(OBJ_SUF) cusp$(OBJ_SUF) homspace$(OBJ_SUF) symb$(OBJ_SUF) moddata$(OBJ_SUF) \
 		 $(LFLAGS)
 
-nftest: nftest$(OBJ_SUF).o cusp$(OBJ_SUF).o homspace$(OBJ_SUF).o symb$(OBJ_SUF).o moddata$(OBJ_SUF).o oldforms$(OBJ_SUF).o \
-		newforms$(OBJ_SUF).o
-	$(CXX) -o nftest nftest$(OBJ_SUF).o cusp$(OBJ_SUF).o homspace$(OBJ_SUF).o symb$(OBJ_SUF).o moddata$(OBJ_SUF).o \
-		oldforms$(OBJ_SUF).o newforms$(OBJ_SUF).o  $(LFLAGS)
+nftest: nftest$(OBJ_SUF) cusp$(OBJ_SUF) homspace$(OBJ_SUF) symb$(OBJ_SUF) moddata$(OBJ_SUF) oldforms$(OBJ_SUF) \
+		newforms$(OBJ_SUF)
+	$(CXX) -o nftest nftest$(OBJ_SUF) cusp$(OBJ_SUF) homspace$(OBJ_SUF) symb$(OBJ_SUF) moddata$(OBJ_SUF) \
+		oldforms$(OBJ_SUF) newforms$(OBJ_SUF)  $(LFLAGS)
 
-ecnf: ecnf$(OBJ_SUF).o cusp$(OBJ_SUF).o homspace$(OBJ_SUF).o symb$(OBJ_SUF).o moddata$(OBJ_SUF).o oldforms$(OBJ_SUF).o \
-		newforms$(OBJ_SUF).o
-	$(CXX) -o ecnf ecnf$(OBJ_SUF).o cusp$(OBJ_SUF).o homspace$(OBJ_SUF).o symb$(OBJ_SUF).o moddata$(OBJ_SUF).o \
-		oldforms$(OBJ_SUF).o newforms$(OBJ_SUF).o  $(LFLAGS)
+ecnf: ecnf$(OBJ_SUF) cusp$(OBJ_SUF) homspace$(OBJ_SUF) symb$(OBJ_SUF) moddata$(OBJ_SUF) oldforms$(OBJ_SUF) \
+		newforms$(OBJ_SUF)
+	$(CXX) -o ecnf ecnf$(OBJ_SUF) cusp$(OBJ_SUF) homspace$(OBJ_SUF) symb$(OBJ_SUF) moddata$(OBJ_SUF) \
+		oldforms$(OBJ_SUF) newforms$(OBJ_SUF)  $(LFLAGS)
 
-nfcount: nfcount$(OBJ_SUF).o
-	$(CXX) -o nfcount nfcount$(OBJ_SUF).o $(LFLAGS)
+nfcount: nfcount$(OBJ_SUF)
+	$(CXX) -o nfcount nfcount$(OBJ_SUF) $(LFLAGS)
 
-aplist: aplist$(OBJ_SUF).o
-	$(CXX) -o aplist aplist$(OBJ_SUF).o $(LFLAGS)
+aplist: aplist$(OBJ_SUF)
+	$(CXX) -o aplist aplist$(OBJ_SUF) $(LFLAGS)
 
-aplist2: aplist2$(OBJ_SUF).o
-	$(CXX) -o aplist2 aplist2$(OBJ_SUF).o $(LFLAGS)
+aplist2: aplist2$(OBJ_SUF)
+	$(CXX) -o aplist2 aplist2$(OBJ_SUF) $(LFLAGS)
 
-findcong: findcong$(OBJ_SUF).o nflist$(OBJ_SUF).o
-	$(CXX) -o findcong findcong$(OBJ_SUF).o nflist$(OBJ_SUF).o $(LFLAGS)
+findcong: findcong$(OBJ_SUF) nflist$(OBJ_SUF)
+	$(CXX) -o findcong findcong$(OBJ_SUF) nflist$(OBJ_SUF) $(LFLAGS)
 
-find1cong: find1cong$(OBJ_SUF).o nflist$(OBJ_SUF).o
-	$(CXX) -o find1cong find1cong$(OBJ_SUF).o nflist$(OBJ_SUF).o $(LFLAGS)
+find1cong: find1cong$(OBJ_SUF) nflist$(OBJ_SUF)
+	$(CXX) -o find1cong find1cong$(OBJ_SUF) nflist$(OBJ_SUF) $(LFLAGS)
 
-checkap: checkap$(OBJ_SUF).o
-	$(CXX) -o checkap checkap$(OBJ_SUF).o -I$(INCDIR) -L$(LIBDIR) $(LFLAGS)
+checkap: checkap$(OBJ_SUF)
+	$(CXX) -o checkap checkap$(OBJ_SUF) -I$(INCDIR) -L$(LIBDIR) $(LFLAGS)
 
-tmanin: tmanin$(OBJ_SUF).o $(G0NLIBNAME)
-	$(CXX) -o tmanin tmanin$(OBJ_SUF).o $(G0NLIBFLAG) $(LFLAGS)
+tmanin: tmanin$(OBJ_SUF) $(G0NLIBNAME)
+	$(CXX) -o tmanin tmanin$(OBJ_SUF) $(G0NLIBFLAG) $(LFLAGS)
 
-getoldeigs: getoldeigs$(OBJ_SUF).o
-	$(CXX) -o getoldeigs getoldeigs$(OBJ_SUF).o $(LFLAGS)
+getoldeigs: getoldeigs$(OBJ_SUF)
+	$(CXX) -o getoldeigs getoldeigs$(OBJ_SUF) $(LFLAGS)
 
-putoldeigs: putoldeigs$(OBJ_SUF).o
-	$(CXX) -o putoldeigs putoldeigs$(OBJ_SUF).o $(LFLAGS)
+putoldeigs: putoldeigs$(OBJ_SUF)
+	$(CXX) -o putoldeigs putoldeigs$(OBJ_SUF) $(LFLAGS)
 
 
-fix1290: fix1290$(OBJ_SUF).o $(G0NLIBNAME)
-	$(CXX) -o fix1290 fix1290$(OBJ_SUF).o $(G0NLIBFLAG) $(LFLAGS)
+fix1290: fix1290$(OBJ_SUF) $(G0NLIBNAME)
+	$(CXX) -o fix1290 fix1290$(OBJ_SUF) $(G0NLIBFLAG) $(LFLAGS)
 
-h1test: h1test$(OBJ_SUF).o $(G0NLIBNAME)
-	$(CXX) -o h1test h1test$(OBJ_SUF).o $(G0NLIBFLAG) $(LFLAGS)
+h1test: h1test$(OBJ_SUF) $(G0NLIBNAME)
+	$(CXX) -o h1test h1test$(OBJ_SUF) $(G0NLIBFLAG) $(LFLAGS)
 
-h1first: h1first$(OBJ_SUF).o $(G0NLIBNAME)
-	$(CXX) -o h1first h1first$(OBJ_SUF).o $(G0NLIBFLAG) $(LFLAGS)
+h1first: h1first$(OBJ_SUF) $(G0NLIBNAME)
+	$(CXX) -o h1first h1first$(OBJ_SUF) $(G0NLIBFLAG) $(LFLAGS)
 
-mh1first: h1first$(OBJ_SUF).o $(G0NLIBNAME)
-	$(CXX) -o mh1first h1first$(OBJ_SUF).o $(G0NLIBFLAG) $(LFLAGS)
+mh1first: h1first$(OBJ_SUF) $(G0NLIBNAME)
+	$(CXX) -o mh1first h1first$(OBJ_SUF) $(G0NLIBFLAG) $(LFLAGS)
 
-h1: h1$(OBJ_SUF).o $(G0NLIBNAME)
-	$(CXX) -o h1 h1$(OBJ_SUF).o $(G0NLIBFLAG) $(LFLAGS)
+h1: h1$(OBJ_SUF) $(G0NLIBNAME)
+	$(CXX) -o h1 h1$(OBJ_SUF) $(G0NLIBFLAG) $(LFLAGS)
 
-h1bsd: h1bsd$(OBJ_SUF).o $(G0NLIBNAME)
-	$(CXX) -o h1bsd h1bsd$(OBJ_SUF).o $(G0NLIBFLAG) $(LFLAGS)
+h1bsd: h1bsd$(OBJ_SUF) $(G0NLIBNAME)
+	$(CXX) -o h1bsd h1bsd$(OBJ_SUF) $(G0NLIBFLAG) $(LFLAGS)
 
-cong_vec: cong_vec$(OBJ_SUF).o $(G0NLIBNAME)
-	$(CXX) -o cong_vec cong_vec$(OBJ_SUF).o $(G0NLIBFLAG) $(LFLAGS)
+cong_vec: cong_vec$(OBJ_SUF) $(G0NLIBNAME)
+	$(CXX) -o cong_vec cong_vec$(OBJ_SUF) $(G0NLIBFLAG) $(LFLAGS)
 
-cong_vec_h1: cong_vec_h1$(OBJ_SUF).o cong_procs$(OBJ_SUF).o $(G0NLIBNAME)
-	$(CXX) -o cong_vec_h1 cong_vec_h1$(OBJ_SUF).o cong_procs$(OBJ_SUF).o $(G0NLIBFLAG) $(LFLAGS)
+cong_vec_h1: cong_vec_h1$(OBJ_SUF) cong_procs$(OBJ_SUF) $(G0NLIBNAME)
+	$(CXX) -o cong_vec_h1 cong_vec_h1$(OBJ_SUF) cong_procs$(OBJ_SUF) $(G0NLIBFLAG) $(LFLAGS)
 
-cong_vec_x: cong_vec_x$(OBJ_SUF).o cong_procs$(OBJ_SUF).o  $(G0NLIBNAME)
-	$(CXX) -o cong_vec_x cong_vec_x$(OBJ_SUF).o cong_procs$(OBJ_SUF).o $(G0NLIBFLAG) $(LFLAGS)
+cong_vec_x: cong_vec_x$(OBJ_SUF) cong_procs$(OBJ_SUF)  $(G0NLIBNAME)
+	$(CXX) -o cong_vec_x cong_vec_x$(OBJ_SUF) cong_procs$(OBJ_SUF) $(G0NLIBFLAG) $(LFLAGS)
 
-lf1per: lf1per$(OBJ_SUF).o $(G0NLIBNAME)
-	$(CXX) -o lf1per lf1per$(OBJ_SUF).o $(G0NLIBFLAG) $(LFLAGS)
+lf1per: lf1per$(OBJ_SUF) $(G0NLIBNAME)
+	$(CXX) -o lf1per lf1per$(OBJ_SUF) $(G0NLIBFLAG) $(LFLAGS)
 
-nfhpcurve: nfhpcurve$(OBJ_SUF).o pcprocs$(OBJ_SUF).o $(G0NLIBNAME)
-	$(CXX) -o nfhpcurve nfhpcurve$(OBJ_SUF).o pcprocs$(OBJ_SUF).o $(G0NLIBFLAG) $(LFLAGS)
+nfhpcurve: nfhpcurve$(OBJ_SUF) pcprocs$(OBJ_SUF) $(G0NLIBNAME)
+	$(CXX) -o nfhpcurve nfhpcurve$(OBJ_SUF) pcprocs$(OBJ_SUF) $(G0NLIBFLAG) $(LFLAGS)
 
-hpcurve: hpcurve$(OBJ_SUF).o $(G0NLIBNAME)
-	$(CXX) -o hpcurve hpcurve$(OBJ_SUF).o $(G0NLIBFLAG) $(LFLAGS)
+hpcurve: hpcurve$(OBJ_SUF) $(G0NLIBNAME)
+	$(CXX) -o hpcurve hpcurve$(OBJ_SUF) $(G0NLIBFLAG) $(LFLAGS)
 
-hpcurve1$(OBJ_SUF).o:  moddata.h oldforms.h newforms.h periods.h \
+hpcurve1$(OBJ_SUF):  moddata.h oldforms.h newforms.h periods.h \
 homspace.h symb.h hpcurve.cc
-	$(CXX) $(CFLAGS) -DSINGLE hpcurve.cc -o hpcurve1$(OBJ_SUF).o
+	$(CXX) $(CFLAGS) -DSINGLE hpcurve.cc -o hpcurve1$(OBJ_SUF)
 
-hpcurve1: hpcurve1$(OBJ_SUF).o $(G0NLIBNAME)
-	$(CXX) -o hpcurve1 hpcurve1$(OBJ_SUF).o $(G0NLIBFLAG) $(LFLAGS)
+hpcurve1: hpcurve1$(OBJ_SUF) $(G0NLIBNAME)
+	$(CXX) -o hpcurve1 hpcurve1$(OBJ_SUF) $(G0NLIBFLAG) $(LFLAGS)
 
-hppcurve: hppcurve$(OBJ_SUF).o $(G0NLIBNAME)
-	$(CXX) -o hppcurve hppcurve$(OBJ_SUF).o $(G0NLIBFLAG) $(LFLAGS)
+hppcurve: hppcurve$(OBJ_SUF) $(G0NLIBNAME)
+	$(CXX) -o hppcurve hppcurve$(OBJ_SUF) $(G0NLIBFLAG) $(LFLAGS)
 
-hpper: hpper$(OBJ_SUF).o $(G0NLIBNAME)
-	$(CXX) -o hpper hpper$(OBJ_SUF).o $(G0NLIBFLAG) $(LFLAGS)
+hpper: hpper$(OBJ_SUF) $(G0NLIBNAME)
+	$(CXX) -o hpper hpper$(OBJ_SUF) $(G0NLIBFLAG) $(LFLAGS)
 
-pcurve: pcurve$(OBJ_SUF).o pcprocs$(OBJ_SUF).o $(G0NLIBNAME)
-	$(CXX) -o pcurve pcurve$(OBJ_SUF).o pcprocs$(OBJ_SUF).o $(G0NLIBFLAG) $(LFLAGS)
+pcurve: pcurve$(OBJ_SUF) pcprocs$(OBJ_SUF) $(G0NLIBNAME)
+	$(CXX) -o pcurve pcurve$(OBJ_SUF) pcprocs$(OBJ_SUF) $(G0NLIBFLAG) $(LFLAGS)
 
-pcurve1$(OBJ_SUF).o:  moddata.h oldforms.h newforms.h periods.h \
+pcurve1$(OBJ_SUF):  moddata.h oldforms.h newforms.h periods.h \
 homspace.h symb.h pcurve.cc pcprocs.h
-	$(CXX) $(CFLAGS) -DSINGLE pcurve.cc -o pcurve1$(OBJ_SUF).o
+	$(CXX) $(CFLAGS) -DSINGLE pcurve.cc -o pcurve1$(OBJ_SUF)
 
-pcurve1: pcurve1$(OBJ_SUF).o pcprocs$(OBJ_SUF).o $(G0NLIBNAME)
-	$(CXX) -o pcurve1 pcurve1$(OBJ_SUF).o pcprocs$(OBJ_SUF).o $(G0NLIBFLAG) $(LFLAGS)
+pcurve1: pcurve1$(OBJ_SUF) pcprocs$(OBJ_SUF) $(G0NLIBNAME)
+	$(CXX) -o pcurve1 pcurve1$(OBJ_SUF) pcprocs$(OBJ_SUF) $(G0NLIBFLAG) $(LFLAGS)
 
-allpcurve: allpcurve$(OBJ_SUF).o pcprocs$(OBJ_SUF).o $(G0NLIBNAME)
-	$(CXX) -o allpcurve allpcurve$(OBJ_SUF).o pcprocs$(OBJ_SUF).o $(G0NLIBFLAG) $(LFLAGS)
+allpcurve: allpcurve$(OBJ_SUF) pcprocs$(OBJ_SUF) $(G0NLIBNAME)
+	$(CXX) -o allpcurve allpcurve$(OBJ_SUF) pcprocs$(OBJ_SUF) $(G0NLIBFLAG) $(LFLAGS)
 
-h1bsdc: h1bsdc$(OBJ_SUF).o $(G0NLIBNAME)
-	$(CXX) -o h1bsdc h1bsdc$(OBJ_SUF).o $(G0NLIBFLAG) $(LFLAGS)
+h1bsdc: h1bsdc$(OBJ_SUF) $(G0NLIBNAME)
+	$(CXX) -o h1bsdc h1bsdc$(OBJ_SUF) $(G0NLIBFLAG) $(LFLAGS)
 
-bsdtex2: bsdtex2$(OBJ_SUF).o $(G0NLIBNAME)
-	$(CXX) -o bsdtex2 bsdtex2$(OBJ_SUF).o $(G0NLIBFLAG) $(LFLAGS)
+bsdtex2: bsdtex2$(OBJ_SUF) $(G0NLIBNAME)
+	$(CXX) -o bsdtex2 bsdtex2$(OBJ_SUF) $(G0NLIBFLAG) $(LFLAGS)
 
-h1clist: h1clist$(OBJ_SUF).o $(G0NLIBNAME)
-	$(CXX) -o h1clist h1clist$(OBJ_SUF).o $(G0NLIBFLAG) $(LFLAGS)
+h1clist: h1clist$(OBJ_SUF) $(G0NLIBNAME)
+	$(CXX) -o h1clist h1clist$(OBJ_SUF) $(G0NLIBFLAG) $(LFLAGS)
 
-h1curve: h1curve$(OBJ_SUF).o $(G0NLIBNAME)
-	$(CXX) -o h1curve h1curve$(OBJ_SUF).o $(G0NLIBFLAG) $(LFLAGS)
+h1curve: h1curve$(OBJ_SUF) $(G0NLIBNAME)
+	$(CXX) -o h1curve h1curve$(OBJ_SUF) $(G0NLIBFLAG) $(LFLAGS)
 
-h1periods: h1periods$(OBJ_SUF).o $(G0NLIBNAME)
-	$(CXX) -o h1periods h1periods$(OBJ_SUF).o $(G0NLIBFLAG) $(LFLAGS)
+h1periods: h1periods$(OBJ_SUF) $(G0NLIBNAME)
+	$(CXX) -o h1periods h1periods$(OBJ_SUF) $(G0NLIBFLAG) $(LFLAGS)
 
-h1curve1$(OBJ_SUF).o: moddata.h symb.h cusp.h homspace.h oldforms.h newforms.h \
+h1curve1$(OBJ_SUF): moddata.h symb.h cusp.h homspace.h oldforms.h newforms.h \
 	    periods.h h1curve.cc curvesort.cc
-	$(CXX) $(CFLAGS) -DSINGLE h1curve.cc -o h1curve1$(OBJ_SUF).o
+	$(CXX) $(CFLAGS) -DSINGLE h1curve.cc -o h1curve1$(OBJ_SUF)
 
-h1curve1: h1curve1$(OBJ_SUF).o $(G0NLIBNAME)
-	$(CXX) -o h1curve1 h1curve1$(OBJ_SUF).o $(G0NLIBFLAG) $(LFLAGS)
+h1curve1: h1curve1$(OBJ_SUF) $(G0NLIBNAME)
+	$(CXX) -o h1curve1 h1curve1$(OBJ_SUF) $(G0NLIBFLAG) $(LFLAGS)
 
-h1curisog: h1curisog$(OBJ_SUF).o $(G0NLIBNAME)
-	$(CXX) -o h1curisog h1curisog$(OBJ_SUF).o $(G0NLIBFLAG) $(LFLAGS)
+h1curisog: h1curisog$(OBJ_SUF) $(G0NLIBNAME)
+	$(CXX) -o h1curisog h1curisog$(OBJ_SUF) $(G0NLIBFLAG) $(LFLAGS)
 
-h1isogbsd: h1isogbsd$(OBJ_SUF).o $(G0NLIBNAME)
-	$(CXX) -o h1isogbsd h1isogbsd$(OBJ_SUF).o $(G0NLIBFLAG) $(LFLAGS)
+h1isogbsd: h1isogbsd$(OBJ_SUF) $(G0NLIBNAME)
+	$(CXX) -o h1isogbsd h1isogbsd$(OBJ_SUF) $(G0NLIBFLAG) $(LFLAGS)
 
-nfconvert: nfconvert$(OBJ_SUF).o $(G0NLIBNAME)
-	$(CXX) -o nfconvert nfconvert$(OBJ_SUF).o $(G0NLIBFLAG) $(LFLAGS)
+nfconvert: nfconvert$(OBJ_SUF) $(G0NLIBNAME)
+	$(CXX) -o nfconvert nfconvert$(OBJ_SUF) $(G0NLIBFLAG) $(LFLAGS)
 
 
-h1bsdcurisog: h1bsdcurisog$(OBJ_SUF).o $(G0NLIBNAME)
-	$(CXX) -o h1bsdcurisog h1bsdcurisog$(OBJ_SUF).o $(G0NLIBFLAG) $(RANKLIBFLAG) $(LFLAGS)
+h1bsdcurisog: h1bsdcurisog$(OBJ_SUF) $(G0NLIBNAME)
+	$(CXX) -o h1bsdcurisog h1bsdcurisog$(OBJ_SUF) $(G0NLIBFLAG) $(RANKLIBFLAG) $(LFLAGS)
 
-th1nf: th1nf$(OBJ_SUF).o $(G0NLIBNAME)
-	$(CXX) -o th1nf th1nf$(OBJ_SUF).o $(G0NLIBFLAG) $(LFLAGS)
+th1nf: th1nf$(OBJ_SUF) $(G0NLIBNAME)
+	$(CXX) -o th1nf th1nf$(OBJ_SUF) $(G0NLIBFLAG) $(LFLAGS)
 
-mazurcheck$(OBJ_SUF).o: moddata.h symb.h cusp.h homspace.h oldforms.h newforms.h \
+mazurcheck$(OBJ_SUF): moddata.h symb.h cusp.h homspace.h oldforms.h newforms.h \
 	    periods.h mazurcheck.cc curvesort.cc
-	$(CXX) $(CFLAGS) mazurcheck.cc -o mazurcheck$(OBJ_SUF).o
-mazurcheck: mazurcheck$(OBJ_SUF).o $(G0NLIBNAME)
-	$(CXX) -o mazurcheck mazurcheck$(OBJ_SUF).o $(G0NLIBFLAG) $(LFLAGS)
-mazurcheck2$(OBJ_SUF).o: moddata.h symb.h cusp.h homspace.h oldforms.h newforms.h \
+	$(CXX) $(CFLAGS) mazurcheck.cc -o mazurcheck$(OBJ_SUF)
+mazurcheck: mazurcheck$(OBJ_SUF) $(G0NLIBNAME)
+	$(CXX) -o mazurcheck mazurcheck$(OBJ_SUF) $(G0NLIBFLAG) $(LFLAGS)
+mazurcheck2$(OBJ_SUF): moddata.h symb.h cusp.h homspace.h oldforms.h newforms.h \
 	    periods.h mazurcheck2.cc curvesort.cc
-	$(CXX) $(CFLAGS) mazurcheck2.cc -o mazurcheck2$(OBJ_SUF).o
-mazurcheck2: mazurcheck2$(OBJ_SUF).o $(G0NLIBNAME)
-	$(CXX) -o mazurcheck2 mazurcheck2$(OBJ_SUF).o $(G0NLIBFLAG) $(LFLAGS)
+	$(CXX) $(CFLAGS) mazurcheck2.cc -o mazurcheck2$(OBJ_SUF)
+mazurcheck2: mazurcheck2$(OBJ_SUF) $(G0NLIBNAME)
+	$(CXX) -o mazurcheck2 mazurcheck2$(OBJ_SUF) $(G0NLIBFLAG) $(LFLAGS)
 
 
-tnf2: tnf2$(OBJ_SUF).o $(G0NLIBNAME)
-	$(CXX) -o tnf2 tnf2$(OBJ_SUF).o $(G0NLIBFLAG) $(LFLAGS)
+tnf2: tnf2$(OBJ_SUF) $(G0NLIBNAME)
+	$(CXX) -o tnf2 tnf2$(OBJ_SUF) $(G0NLIBFLAG) $(LFLAGS)
 
 
-h1degphi: h1degphi$(OBJ_SUF).o $(G0NLIBNAME)
-	$(CXX) -o h1degphi h1degphi$(OBJ_SUF).o $(G0NLIBFLAG) $(LFLAGS)
+h1degphi: h1degphi$(OBJ_SUF) $(G0NLIBNAME)
+	$(CXX) -o h1degphi h1degphi$(OBJ_SUF) $(G0NLIBFLAG) $(LFLAGS)
 
-h1degphi_mw: h1degphi_mw$(OBJ_SUF).o $(G0NLIBNAME)
-	$(CXX) -o h1degphi_mw h1degphi_mw$(OBJ_SUF).o $(G0NLIBFLAG) $(LFLAGS)
+h1degphi_mw: h1degphi_mw$(OBJ_SUF) $(G0NLIBNAME)
+	$(CXX) -o h1degphi_mw h1degphi_mw$(OBJ_SUF) $(G0NLIBFLAG) $(LFLAGS)
 
-mazurdeg: mazurdeg$(OBJ_SUF).o $(G0NLIBNAME)
-	$(CXX) -o mazurdeg mazurdeg$(OBJ_SUF).o $(G0NLIBFLAG) $(LFLAGS)
+mazurdeg: mazurdeg$(OBJ_SUF) $(G0NLIBNAME)
+	$(CXX) -o mazurdeg mazurdeg$(OBJ_SUF) $(G0NLIBFLAG) $(LFLAGS)
 
-degphitab: degphitab$(OBJ_SUF).o $(G0NLIBNAME)
-	$(CXX) -o degphitab degphitab$(OBJ_SUF).o $(G0NLIBFLAG) $(LFLAGS)
+degphitab: degphitab$(OBJ_SUF) $(G0NLIBNAME)
+	$(CXX) -o degphitab degphitab$(OBJ_SUF) $(G0NLIBFLAG) $(LFLAGS)
 
-degphitex: degphitex$(OBJ_SUF).o
-	$(CXX) -o degphitex degphitex$(OBJ_SUF).o $(LFLAGS)
+degphitex: degphitex$(OBJ_SUF)
+	$(CXX) -o degphitex degphitex$(OBJ_SUF) $(LFLAGS)
 
-test: test$(OBJ_SUF).o
-	$(CXX) -o test test$(OBJ_SUF).o $(LFLAGS)
+test: test$(OBJ_SUF)
+	$(CXX) -o test test$(OBJ_SUF) $(LFLAGS)
 
-checkgens$(OBJ_SUF).o: checkgens.cc curvesort.cc
-checkgens: checkgens$(OBJ_SUF).o
-	   $(CXX) -o checkgens checkgens$(OBJ_SUF).o  $(LFLAGS)
+checkgens$(OBJ_SUF): checkgens.cc curvesort.cc
+checkgens: checkgens$(OBJ_SUF)
+	   $(CXX) -o checkgens checkgens$(OBJ_SUF)  $(LFLAGS)
 
 # DEPENDENCIES
 
-cusp$(OBJ_SUF).o: moddata.h symb.h cusp.h cusp.cc
-homspace$(OBJ_SUF).o: moddata.h symb.h cusp.h homspace.h homspace.cc
-moddata$(OBJ_SUF).o: moddata.h moddata.cc
-oldforms$(OBJ_SUF).o: moddata.h oldforms.h oldforms.cc
-newforms$(OBJ_SUF).o: newforms.cc newforms.h moddata.h homspace.h oldforms.h 
-
-modtest$(OBJ_SUF).o: modtest.cc moddata.h symb.h
-homtest$(OBJ_SUF).o: homtest.cc moddata.h symb.h cusp.h homspace.h
-hecketest$(OBJ_SUF).o: hecketest.cc moddata.h symb.h cusp.h homspace.h
-wsplit$(OBJ_SUF).o: wsplit.cc moddata.h symb.h cusp.h homspace.h
-gd2$(OBJ_SUF).o: gd2.cc moddata.h symb.h cusp.h homspace.h
-oftest$(OBJ_SUF).o: oftest.cc moddata.h symb.h cusp.h oldforms.h
-nftest$(OBJ_SUF).o: nftest.cc moddata.h symb.h cusp.h homspace.h oldforms.h newforms.h
-ecnf$(OBJ_SUF).o: ecnf.cc moddata.h symb.h cusp.h homspace.h oldforms.h newforms.h
+cusp$(OBJ_SUF): moddata.h symb.h cusp.h cusp.cc
+homspace$(OBJ_SUF): moddata.h symb.h cusp.h homspace.h homspace.cc
+moddata$(OBJ_SUF): moddata.h moddata.cc
+oldforms$(OBJ_SUF): moddata.h oldforms.h oldforms.cc
+newforms$(OBJ_SUF): newforms.cc newforms.h moddata.h homspace.h oldforms.h
+
+modtest$(OBJ_SUF): modtest.cc moddata.h symb.h
+homtest$(OBJ_SUF): homtest.cc moddata.h symb.h cusp.h homspace.h
+hecketest$(OBJ_SUF): hecketest.cc moddata.h symb.h cusp.h homspace.h
+wsplit$(OBJ_SUF): wsplit.cc moddata.h symb.h cusp.h homspace.h
+gd2$(OBJ_SUF): gd2.cc moddata.h symb.h cusp.h homspace.h
+oftest$(OBJ_SUF): oftest.cc moddata.h symb.h cusp.h oldforms.h
+nftest$(OBJ_SUF): nftest.cc moddata.h symb.h cusp.h homspace.h oldforms.h newforms.h
+ecnf$(OBJ_SUF): ecnf.cc moddata.h symb.h cusp.h homspace.h oldforms.h newforms.h
 
-tmanin$(OBJ_SUF).o: tmanin.cc moddata.h symb.h cusp.h homspace.h oldforms.h \
+tmanin$(OBJ_SUF): tmanin.cc moddata.h symb.h cusp.h homspace.h oldforms.h \
 	newforms.h 
 
-th1nf$(OBJ_SUF).o: th1nf.cc moddata.h symb.h cusp.h homspace.h  oldforms.h 
-tnf2$(OBJ_SUF).o: tnf2.cc moddata.h symb.h cusp.h homspace.h  oldforms.h \
+th1nf$(OBJ_SUF): th1nf.cc moddata.h symb.h cusp.h homspace.h  oldforms.h
+tnf2$(OBJ_SUF): tnf2.cc moddata.h symb.h cusp.h homspace.h  oldforms.h \
          newforms.h
-h1curve$(OBJ_SUF).o: moddata.h symb.h cusp.h homspace.h oldforms.h \
+h1curve$(OBJ_SUF): moddata.h symb.h cusp.h homspace.h oldforms.h \
 	newforms.h periods.h h1curve.cc curvesort.cc newforms.h
-h1curisog$(OBJ_SUF).o: h1curisog.cc moddata.h symb.h cusp.h homspace.h oldforms.h \
+h1curisog$(OBJ_SUF): h1curisog.cc moddata.h symb.h cusp.h homspace.h oldforms.h \
 	periods.h h1curve.cc curvesort.cc newforms.h
-h1isogbsd$(OBJ_SUF).o: h1isogbsd.cc moddata.h symb.h cusp.h homspace.h oldforms.h \
+h1isogbsd$(OBJ_SUF): h1isogbsd.cc moddata.h symb.h cusp.h homspace.h oldforms.h \
 	periods.h h1curve.cc curvesort.cc newforms.h
-h1clist$(OBJ_SUF).o: moddata.h symb.h cusp.h homspace.h oldforms.h \
+h1clist$(OBJ_SUF): moddata.h symb.h cusp.h homspace.h oldforms.h \
 	periods.h h1clist.cc curvesort.cc
-h1degphi$(OBJ_SUF).o: moddata.h symb.h cusp.h homspace.h oldforms.h \
+h1degphi$(OBJ_SUF): moddata.h symb.h cusp.h homspace.h oldforms.h \
 	periods.h h1degphi.cc curvesort.cc
-h1degphi_mw$(OBJ_SUF).o: moddata.h symb.h cusp.h homspace.h oldforms.h \
+h1degphi_mw$(OBJ_SUF): moddata.h symb.h cusp.h homspace.h oldforms.h \
 	periods.h h1degphi.cc curvesort.cc
-mazurdeg$(OBJ_SUF).o: moddata.h symb.h cusp.h homspace.h oldforms.h \
+mazurdeg$(OBJ_SUF): moddata.h symb.h cusp.h homspace.h oldforms.h \
 	periods.h mazurdeg.cc curvesort.cc
-degphitab$(OBJ_SUF).o: moddata.h symb.h cusp.h homspace.h oldforms.h \
+degphitab$(OBJ_SUF): moddata.h symb.h cusp.h homspace.h oldforms.h \
 	periods.h degphitab.cc curvesort.cc
-degphitex$(OBJ_SUF).o: degphitab.cc curvesort.cc
-h1bsd$(OBJ_SUF).o:  moddata.h oldforms.h newforms.h periods.h homspace.h symb.h h1bsd.cc
-h1bsdc$(OBJ_SUF).o: h1bsdc.cc moddata.h oldforms.h newforms.h symb.h cusp.h  periods.h
-h1bsdc$(OBJ_SUF).o: h1bsdc.cc moddata.h oldforms.h newforms.h symb.h cusp.h periods.h curvesort.cc
-lf1per$(OBJ_SUF).o:  moddata.h oldforms.h newforms.h periods.h homspace.h symb.h lf1per.cc
-nfhpcurve$(OBJ_SUF).o:  moddata.h oldforms.h newforms.h periods.h homspace.h symb.h cusp.h pcprocs.h nfhpcurve.cc
-hpcurve$(OBJ_SUF).o:  moddata.h oldforms.h newforms.h periods.h homspace.h symb.h hpcurve.cc
-pcurve$(OBJ_SUF).o:  moddata.h oldforms.h newforms.h periods.h homspace.h symb.h pcurve.cc
-allpcurve$(OBJ_SUF).o:  moddata.h oldforms.h newforms.h periods.h homspace.h symb.h allpcurve.cc
+degphitex$(OBJ_SUF): degphitab.cc curvesort.cc
+h1bsd$(OBJ_SUF):  moddata.h oldforms.h newforms.h periods.h homspace.h symb.h h1bsd.cc
+h1bsdc$(OBJ_SUF): h1bsdc.cc moddata.h oldforms.h newforms.h symb.h cusp.h  periods.h
+h1bsdc$(OBJ_SUF): h1bsdc.cc moddata.h oldforms.h newforms.h symb.h cusp.h periods.h curvesort.cc
+lf1per$(OBJ_SUF):  moddata.h oldforms.h newforms.h periods.h homspace.h symb.h lf1per.cc
+nfhpcurve$(OBJ_SUF):  moddata.h oldforms.h newforms.h periods.h homspace.h symb.h cusp.h pcprocs.h nfhpcurve.cc
+hpcurve$(OBJ_SUF):  moddata.h oldforms.h newforms.h periods.h homspace.h symb.h hpcurve.cc
+pcurve$(OBJ_SUF):  moddata.h oldforms.h newforms.h periods.h homspace.h symb.h pcurve.cc
+allpcurve$(OBJ_SUF):  moddata.h oldforms.h newforms.h periods.h homspace.h symb.h allpcurve.cc
 
-periods$(OBJ_SUF).o: periods.cc moddata.h symb.h cusp.h homspace.h oldforms.h \
+periods$(OBJ_SUF): periods.cc moddata.h symb.h cusp.h homspace.h oldforms.h \
 	newforms.h periods.h fixc6.h
-fixc6$(OBJ_SUF).o: fixc6.h fixc6.cc
-symb$(OBJ_SUF).o: symb.h moddata.h symb.cc
-getoldeigs$(OBJ_SUF).o: getoldeigs.cc
-putoldeigs$(OBJ_SUF).o: putoldeigs.cc
-
-nfd$(OBJ_SUF).o: nfd.cc nfd.h moddata.h symb.h homspace.h
-tnfd$(OBJ_SUF).o: tnfd.cc  nfd.h moddata.h symb.h homspace.h
-
-nflist$(OBJ_SUF).o: nflist.h nflist.cc
-findcong$(OBJ_SUF).o: findcong.cc nflist.h
-find1cong$(OBJ_SUF).o: find1cong.cc nflist.h
-cong_vec$(OBJ_SUF).o: cong_vec.cc moddata.h symb.h oldforms.h homspace.h newforms.h  periods.h
-cong_vec_h1$(OBJ_SUF).o: cong_vec_h1.cc cong_procs.h moddata.h symb.h oldforms.h homspace.h newforms.h  periods.h
-cong_vec_x$(OBJ_SUF).o: cong_vec_x.cc cong_procs.h
-cong_procs$(OBJ_SUF).o: cong_procs.h cong_procs.cc
-pcurve$(OBJ_SUF).o: pcurve.cc pcprocs.h moddata.h symb.h oldforms.h homspace.h newforms.h periods.h pcprocs.h
-pcprocs$(OBJ_SUF).o: pcprocs.cc moddata.h symb.h oldforms.h homspace.h newforms.h periods.h pcprocs.h
+fixc6$(OBJ_SUF): fixc6.h fixc6.cc
+symb$(OBJ_SUF): symb.h moddata.h symb.cc
+getoldeigs$(OBJ_SUF): getoldeigs.cc
+putoldeigs$(OBJ_SUF): putoldeigs.cc
+
+nfd$(OBJ_SUF): nfd.cc nfd.h moddata.h symb.h homspace.h
+tnfd$(OBJ_SUF): tnfd.cc  nfd.h moddata.h symb.h homspace.h
+
+nflist$(OBJ_SUF): nflist.h nflist.cc
+findcong$(OBJ_SUF): findcong.cc nflist.h
+find1cong$(OBJ_SUF): find1cong.cc nflist.h
+cong_vec$(OBJ_SUF): cong_vec.cc moddata.h symb.h oldforms.h homspace.h newforms.h  periods.h
+cong_vec_h1$(OBJ_SUF): cong_vec_h1.cc cong_procs.h moddata.h symb.h oldforms.h homspace.h newforms.h  periods.h
+cong_vec_x$(OBJ_SUF): cong_vec_x.cc cong_procs.h
+cong_procs$(OBJ_SUF): cong_procs.h cong_procs.cc
+pcurve$(OBJ_SUF): pcurve.cc pcprocs.h moddata.h symb.h oldforms.h homspace.h newforms.h periods.h pcprocs.h
+pcprocs$(OBJ_SUF): pcprocs.cc moddata.h symb.h oldforms.h homspace.h newforms.h periods.h pcprocs.h
 
-vlift$(OBJ_SUF).o: 
+vlift$(OBJ_SUF):
 
-vlift: vlift$(OBJ_SUF).o 
-	$(CXX) -o vlift vlift$(OBJ_SUF).o $(LFLAGS)
+vlift: vlift$(OBJ_SUF)
+	$(CXX) -o vlift vlift$(OBJ_SUF) $(LFLAGS)
 
 e2x: e2x.cc
 	$(CXX) $(OPTFLAG) e2x.cc -o e2x $(IOLIB)
@@ -426,22 +440,22 @@
 x2e: x2e.cc
 	$(CXX) $(OPTFLAG) x2e.cc -o x2e $(IOLIB)
 
-pair$(OBJ_SUF).o: pair.cc
-pair: pair$(OBJ_SUF).o
-	$(CXX) -static -o pair pair$(OBJ_SUF).o $(LFLAGS)
-pair2$(OBJ_SUF).o: pair2.cc
-pair2: pair2$(OBJ_SUF).o
-	$(CXX) -static -o pair2 pair2$(OBJ_SUF).o $(LFLAGS)
-pair1$(OBJ_SUF).o: pair1.cc
-pair1: pair1$(OBJ_SUF).o
-	$(CXX) -static -o pair1 pair1$(OBJ_SUF).o $(LFLAGS)
-
-aphomspace$(OBJ_SUF).o: moddata.h symb.h cusp.h homspace.h aphomspace.cc
-aphomtest$(OBJ_SUF).o: aphomtest.cc moddata.h symb.h cusp.h homspace.h
-aphomtest: homtest$(OBJ_SUF).o symb$(OBJ_SUF).o moddata$(OBJ_SUF).o cusp$(OBJ_SUF).o aphomspace$(OBJ_SUF).o
-	$(CXX) -o aphomtest homtest$(OBJ_SUF).o symb$(OBJ_SUF).o moddata$(OBJ_SUF).o cusp$(OBJ_SUF).o aphomspace$(OBJ_SUF).o  $(LFLAGS)
-aphecketest: hecketest$(OBJ_SUF).o cusp$(OBJ_SUF).o aphomspace$(OBJ_SUF).o symb$(OBJ_SUF).o moddata$(OBJ_SUF).o
-	$(CXX) -o aphecketest hecketest$(OBJ_SUF).o cusp$(OBJ_SUF).o aphomspace$(OBJ_SUF).o symb$(OBJ_SUF).o moddata$(OBJ_SUF).o \
+pair$(OBJ_SUF): pair.cc
+pair: pair$(OBJ_SUF)
+	$(CXX) -static -o pair pair$(OBJ_SUF) $(LFLAGS)
+pair2$(OBJ_SUF): pair2.cc
+pair2: pair2$(OBJ_SUF)
+	$(CXX) -static -o pair2 pair2$(OBJ_SUF) $(LFLAGS)
+pair1$(OBJ_SUF): pair1.cc
+pair1: pair1$(OBJ_SUF)
+	$(CXX) -static -o pair1 pair1$(OBJ_SUF) $(LFLAGS)
+
+aphomspace$(OBJ_SUF): moddata.h symb.h cusp.h homspace.h aphomspace.cc
+aphomtest$(OBJ_SUF): aphomtest.cc moddata.h symb.h cusp.h homspace.h
+aphomtest: homtest$(OBJ_SUF) symb$(OBJ_SUF) moddata$(OBJ_SUF) cusp$(OBJ_SUF) aphomspace$(OBJ_SUF)
+	$(CXX) -o aphomtest homtest$(OBJ_SUF) symb$(OBJ_SUF) moddata$(OBJ_SUF) cusp$(OBJ_SUF) aphomspace$(OBJ_SUF)  $(LFLAGS)
+aphecketest: hecketest$(OBJ_SUF) cusp$(OBJ_SUF) aphomspace$(OBJ_SUF) symb$(OBJ_SUF) moddata$(OBJ_SUF)
+	$(CXX) -o aphecketest hecketest$(OBJ_SUF) cusp$(OBJ_SUF) aphomspace$(OBJ_SUF) symb$(OBJ_SUF) moddata$(OBJ_SUF) \
 		 $(LFLAGS)
 
 
