--- qrank/Makefile.orig	2010-03-01 13:08:27.000000000 +1300
+++ qrank/Makefile	2010-08-17 23:16:43.199935599 +1200
@@ -32,7 +32,8 @@
 CURVELIBFLAG = -L../libs -lcurvesntl
 RANKLIBBASE = librankntl
 RANKLIBFLAG = -L../libs -lrankntl
-OBJ_SUF = _n
+OBJ_SUF = _n.o
+SOBJ_SUF = _n.lo
 EXE_SUFFIX = _n
 else
 NTLCFLAGS = 
@@ -55,38 +56,55 @@
 
 LIBFLAGS = $(LFLAGS)
 
-%$(OBJ_SUF).o:   %.cc
+#template for compiling *.cc -> *$(OBJ_SUF) :
+.SUFFIXES: .cc $(OBJ_SUF) $(SOBJ_SUF)
+
+ .cc$(OBJ_SUF):
 	$(CXX) $(CFLAGS) $< -o $@
 
+ .cc$(SOBJ_SUF):
+	$(CXX) $(PICFLAG) $(CFLAGS) $< -o $@
+
 # TARGETS
 
-HEADERS = mrank2.h mequiv.h mlocsol.h mglobsol.h msoluble.h mquartic.h mrank1.h qc.h sqfdiv.h version.h minim.h reduce.h transform.h desc2.h bitspace.h options.h GetOpt.h twoadic.h descent.h
+HEADERS = mrank2.h mequiv.h mlocsol.h mglobsol.h msoluble.h mquartic.h mrank1.h qc.h sqfdiv.h version.h \
+	minim.h reduce.h transform.h desc2.h bitspace.h options.h GetOpt.h twoadic.h descent.h
 headers: $(HEADERS)
 
-DOTCCS = mequiv.cc  mrank1.cc mlocsol.cc mglobsol.cc mquartic.cc mrank2.cc tmrank.cc ratpoint.cc qc.cc mwrank.cc sqfdiv.cc version.cc minim.cc reduce.cc transform.cc desc2.cc tred.cc bitspace.cc GetOpt.cc twoadic.cc descent.cc # d2.cc tequiv.cc tred.cc
+DOTCCS = mequiv.cc  mrank1.cc mlocsol.cc mglobsol.cc mquartic.cc mrank2.cc tmrank.cc ratpoint.cc qc.cc \
+	mwrank.cc sqfdiv.cc version.cc minim.cc reduce.cc transform.cc desc2.cc tred.cc bitspace.cc \
+	GetOpt.cc twoadic.cc descent.cc # d2.cc tequiv.cc tred.cc
 
 sources: $(HEADERS) $(DOTCCS)
 
 DOCS = mwrank.info mwrank.options mwrank.changes mwrank.readme
 docs: $(DOCS)
 
-DISTFILES = $(HEADERS) mequiv.cc mrank1.cc mlocsol.cc mglobsol.cc mquartic.cc mrank2.cc tmrank.cc ratpoint.cc d2.cc qc.cc mwrank.cc sqfdiv.cc version.cc minim.cc reduce.cc transform.cc desc2.cc bitspace.cc twoadic.h twoadic.cc descent.cc $(DOCS) tmrank.in
+DISTFILES = $(HEADERS) mequiv.cc mrank1.cc mlocsol.cc mglobsol.cc mquartic.cc mrank2.cc tmrank.cc \
+	ratpoint.cc d2.cc qc.cc mwrank.cc sqfdiv.cc version.cc minim.cc reduce.cc transform.cc desc2.cc \
+	bitspace.cc twoadic.h twoadic.cc descent.cc $(DOCS) tmrank.in
 
 distfiles: $(DISTFILES)
 
 clean:
-	/bin/rm -f *$(OBJ_SUF).o *~ $(RANKLIBNAME) *.a *.so *.dylib
+	/bin/rm -f *$(OBJ_SUF) *$(SOBJ_SUF)*~ $(RANKLIBNAME) *.a *.so *.dylib
 
-OBJS = mrank1$(OBJ_SUF).o mrank2$(OBJ_SUF).o mequiv$(OBJ_SUF).o mquartic$(OBJ_SUF).o mlocsol$(OBJ_SUF).o mglobsol$(OBJ_SUF).o  qc$(OBJ_SUF).o sqfdiv$(OBJ_SUF).o desc2$(OBJ_SUF).o transform$(OBJ_SUF).o minim$(OBJ_SUF).o reduce$(OBJ_SUF).o bitspace$(OBJ_SUF).o GetOpt$(OBJ_SUF).o twoadic$(OBJ_SUF).o descent$(OBJ_SUF).o
+OBJS = mrank1$(OBJ_SUF) mrank2$(OBJ_SUF) mequiv$(OBJ_SUF) mquartic$(OBJ_SUF) mlocsol$(OBJ_SUF) \
+	mglobsol$(OBJ_SUF)  qc$(OBJ_SUF) sqfdiv$(OBJ_SUF) desc2$(OBJ_SUF) transform$(OBJ_SUF) minim$(OBJ_SUF) \
+	reduce$(OBJ_SUF) bitspace$(OBJ_SUF) GetOpt$(OBJ_SUF) twoadic$(OBJ_SUF) descent$(OBJ_SUF)
+
+SOBJS = mrank1$(SOBJ_SUF) mrank2$(SOBJ_SUF) mequiv$(SOBJ_SUF) mquartic$(SOBJ_SUF) mlocsol$(SOBJ_SUF) \
+	mglobsol$(SOBJ_SUF)  qc$(SOBJ_SUF) sqfdiv$(SOBJ_SUF) desc2$(SOBJ_SUF) transform$(SOBJ_SUF) minim$(SOBJ_SUF) \
+	reduce$(SOBJ_SUF) bitspace$(SOBJ_SUF) GetOpt$(SOBJ_SUF) twoadic$(SOBJ_SUF) descent$(SOBJ_SUF)
 
 objs: $(OBJS)
 
-lib: $(OBJS)
+lib: objs
 	rm -f $(RANKLIBNAME)
 	ar r $(RANKLIBNAME) $(OBJS)
 	$(RANLIB) $(RANKLIBNAME)
 
-install_lib: $(HEADERS) lib
+install_lib: headers lib
 	(cd $(INCDIR); rm -f $(HEADERS))
 	for f in $(HEADERS); \
 	 do cp $${f} $(INCDIR); \
@@ -98,7 +116,7 @@
 
 install: lib install_lib
 
-install_docs: $(DOCS) 
+install_docs: docs
 	for f in $(DOCS); \
 	do cp $${f} $(FTP_DIR)/$${f}; chmod 644 $(FTP_DIR)/$${f}; done
 
@@ -114,7 +132,7 @@
 	time ./tmrank < tmrank.in
 
 test2: mwrank
-	time awk 'NF==6{print $$1,$$2,$$3,$$4,$$5;}' tmrank.in | ./mwrank -v 0 -q -o -S10
+	time awk 'NF==6{print $$1,$$2,$$3,$$4,$$5;}' tmrank.in | $(BINDIR)/mwrank -v 0 -q -o -S10
 
 alltests: tests test1 test2
 
@@ -142,88 +160,88 @@
 	chmod 755 $(HOME)/curves/rankdist/configure
 	chmod 755 $(HOME)/curves/rankdist/configure.ac
 
-check: $(PROGS) $(TESTS)
+check: progs tests
 	rm -f PRIMES t
 	./tequiv < tequiv.in > t && diff t tequiv.out
 	./tred < tred.in > t 2>/dev/null && diff t tred.out
-	./ratpoint < ratpoint.in > t 2>/dev/null && diff t ratpoint.out
-	./mwrank < mwrank.in > t 2>/dev/null && diff t mwrank.out
+	$(BINDIR)/ratpoint < ratpoint.in > t 2>/dev/null && diff t ratpoint.out
+	$(BINDIR)/mwrank < mwrank.in > t 2>/dev/null && diff t mwrank.out
 	rm -f PRIMES t
 
-mwrank: version$(OBJ_SUF).o mwrank$(OBJ_SUF).o descent$(OBJ_SUF).o mrank1$(OBJ_SUF).o mrank2$(OBJ_SUF).o mequiv$(OBJ_SUF).o mquartic$(OBJ_SUF).o \
-	mlocsol$(OBJ_SUF).o mglobsol$(OBJ_SUF).o  qc$(OBJ_SUF).o sqfdiv$(OBJ_SUF).o \
-	desc2$(OBJ_SUF).o transform$(OBJ_SUF).o minim$(OBJ_SUF).o reduce$(OBJ_SUF).o bitspace$(OBJ_SUF).o GetOpt$(OBJ_SUF).o \
-	twoadic$(OBJ_SUF).o
+mwrank: version$(OBJ_SUF) mwrank$(OBJ_SUF) descent$(OBJ_SUF) mrank1$(OBJ_SUF) mrank2$(OBJ_SUF) mequiv$(OBJ_SUF) mquartic$(OBJ_SUF) \
+	mlocsol$(OBJ_SUF) mglobsol$(OBJ_SUF)  qc$(OBJ_SUF) sqfdiv$(OBJ_SUF) \
+	desc2$(OBJ_SUF) transform$(OBJ_SUF) minim$(OBJ_SUF) reduce$(OBJ_SUF) bitspace$(OBJ_SUF) GetOpt$(OBJ_SUF) \
+	twoadic$(OBJ_SUF)
 	chmod og+r *
-	$(CXX) -o mwrank version$(OBJ_SUF).o mwrank$(OBJ_SUF).o descent$(OBJ_SUF).o mrank1$(OBJ_SUF).o mrank2$(OBJ_SUF).o mequiv$(OBJ_SUF).o mquartic$(OBJ_SUF).o \
-	mlocsol$(OBJ_SUF).o mglobsol$(OBJ_SUF).o  qc$(OBJ_SUF).o sqfdiv$(OBJ_SUF).o desc2$(OBJ_SUF).o \
-	transform$(OBJ_SUF).o minim$(OBJ_SUF).o reduce$(OBJ_SUF).o bitspace$(OBJ_SUF).o GetOpt$(OBJ_SUF).o twoadic$(OBJ_SUF).o \
+	$(CXX) $(LDFLAGS) -o $(BINDIR)/mwrank version$(OBJ_SUF) mwrank$(OBJ_SUF) descent$(OBJ_SUF) mrank1$(OBJ_SUF) mrank2$(OBJ_SUF) mequiv$(OBJ_SUF) mquartic$(OBJ_SUF) \
+	mlocsol$(OBJ_SUF) mglobsol$(OBJ_SUF)  qc$(OBJ_SUF) sqfdiv$(OBJ_SUF) desc2$(OBJ_SUF) \
+	transform$(OBJ_SUF) minim$(OBJ_SUF) reduce$(OBJ_SUF) bitspace$(OBJ_SUF) GetOpt$(OBJ_SUF) twoadic$(OBJ_SUF) \
 	$(LFLAGS)
 
-tmrank: version$(OBJ_SUF).o tmrank$(OBJ_SUF).o descent$(OBJ_SUF).o mrank1$(OBJ_SUF).o mrank2$(OBJ_SUF).o mequiv$(OBJ_SUF).o mquartic$(OBJ_SUF).o mlocsol$(OBJ_SUF).o mglobsol$(OBJ_SUF).o \
-	 qc$(OBJ_SUF).o sqfdiv$(OBJ_SUF).o desc2$(OBJ_SUF).o transform$(OBJ_SUF).o minim$(OBJ_SUF).o reduce$(OBJ_SUF).o bitspace$(OBJ_SUF).o twoadic$(OBJ_SUF).o
-	$(CXX) -o tmrank version$(OBJ_SUF).o tmrank$(OBJ_SUF).o descent$(OBJ_SUF).o mrank1$(OBJ_SUF).o mrank2$(OBJ_SUF).o mequiv$(OBJ_SUF).o mquartic$(OBJ_SUF).o \
-	mlocsol$(OBJ_SUF).o mglobsol$(OBJ_SUF).o  qc$(OBJ_SUF).o sqfdiv$(OBJ_SUF).o desc2$(OBJ_SUF).o transform$(OBJ_SUF).o minim$(OBJ_SUF).o \
-	reduce$(OBJ_SUF).o bitspace$(OBJ_SUF).o twoadic$(OBJ_SUF).o $(LFLAGS)
-
-ratpoint: version$(OBJ_SUF).o ratpoint$(OBJ_SUF).o mquartic$(OBJ_SUF).o mlocsol$(OBJ_SUF).o mglobsol$(OBJ_SUF).o  qc$(OBJ_SUF).o
-	$(CXX) -o ratpoint version$(OBJ_SUF).o ratpoint$(OBJ_SUF).o mquartic$(OBJ_SUF).o mlocsol$(OBJ_SUF).o mglobsol$(OBJ_SUF).o \
-	 qc$(OBJ_SUF).o $(LFLAGS)
-
-ratpoint_sta: version$(OBJ_SUF).o ratpoint$(OBJ_SUF).o mquartic$(OBJ_SUF).o mlocsol$(OBJ_SUF).o mglobsol$(OBJ_SUF).o  qc$(OBJ_SUF).o
-	$(CXX) -static -o ratpoint_sta version$(OBJ_SUF).o ratpoint$(OBJ_SUF).o mquartic$(OBJ_SUF).o mlocsol$(OBJ_SUF).o mglobsol$(OBJ_SUF).o \
-	 qc$(OBJ_SUF).o $(LFLAGS)
-
-d2: d2$(OBJ_SUF).o version$(OBJ_SUF).o desc2$(OBJ_SUF).o mquartic$(OBJ_SUF).o mlocsol$(OBJ_SUF).o mglobsol$(OBJ_SUF).o   qc$(OBJ_SUF).o reduce$(OBJ_SUF).o \
-	minim$(OBJ_SUF).o transform$(OBJ_SUF).o sqfdiv$(OBJ_SUF).o
-	$(CXX) -o d2 d2$(OBJ_SUF).o version$(OBJ_SUF).o desc2$(OBJ_SUF).o mquartic$(OBJ_SUF).o mlocsol$(OBJ_SUF).o mglobsol$(OBJ_SUF).o  \
-	qc$(OBJ_SUF).o reduce$(OBJ_SUF).o minim$(OBJ_SUF).o transform$(OBJ_SUF).o sqfdiv$(OBJ_SUF).o $(LFLAGS)
-
-tred: version$(OBJ_SUF).o tred$(OBJ_SUF).o reduce$(OBJ_SUF).o minim$(OBJ_SUF).o transform$(OBJ_SUF).o mquartic$(OBJ_SUF).o mlocsol$(OBJ_SUF).o mglobsol$(OBJ_SUF).o \
-	 qc$(OBJ_SUF).o
-	$(CXX) -o tred version$(OBJ_SUF).o tred$(OBJ_SUF).o reduce$(OBJ_SUF).o minim$(OBJ_SUF).o transform$(OBJ_SUF).o mquartic$(OBJ_SUF).o \
-	mlocsol$(OBJ_SUF).o mglobsol$(OBJ_SUF).o  qc$(OBJ_SUF).o $(LFLAGS)
+tmrank: version$(OBJ_SUF) tmrank$(OBJ_SUF) descent$(OBJ_SUF) mrank1$(OBJ_SUF) mrank2$(OBJ_SUF) mequiv$(OBJ_SUF) mquartic$(OBJ_SUF) mlocsol$(OBJ_SUF) mglobsol$(OBJ_SUF) \
+	 qc$(OBJ_SUF) sqfdiv$(OBJ_SUF) desc2$(OBJ_SUF) transform$(OBJ_SUF) minim$(OBJ_SUF) reduce$(OBJ_SUF) bitspace$(OBJ_SUF) twoadic$(OBJ_SUF)
+	$(CXX) $(LDFLAGS) -o tmrank version$(OBJ_SUF) tmrank$(OBJ_SUF) descent$(OBJ_SUF) mrank1$(OBJ_SUF) mrank2$(OBJ_SUF) mequiv$(OBJ_SUF) mquartic$(OBJ_SUF) \
+	mlocsol$(OBJ_SUF) mglobsol$(OBJ_SUF)  qc$(OBJ_SUF) sqfdiv$(OBJ_SUF) desc2$(OBJ_SUF) transform$(OBJ_SUF) minim$(OBJ_SUF) \
+	reduce$(OBJ_SUF) bitspace$(OBJ_SUF) twoadic$(OBJ_SUF) $(LFLAGS)
+
+ratpoint: version$(OBJ_SUF) ratpoint$(OBJ_SUF) mquartic$(OBJ_SUF) mlocsol$(OBJ_SUF) mglobsol$(OBJ_SUF)  qc$(OBJ_SUF)
+	$(CXX) $(LDFLAGS) -o $(BINDIR)/ratpoint version$(OBJ_SUF) ratpoint$(OBJ_SUF) mquartic$(OBJ_SUF) mlocsol$(OBJ_SUF) mglobsol$(OBJ_SUF) \
+	 qc$(OBJ_SUF) $(LFLAGS)
+
+ratpoint_sta: version$(OBJ_SUF) ratpoint$(OBJ_SUF) mquartic$(OBJ_SUF) mlocsol$(OBJ_SUF) mglobsol$(OBJ_SUF)  qc$(OBJ_SUF)
+	$(CXX) -static -o ratpoint_sta version$(OBJ_SUF) ratpoint$(OBJ_SUF) mquartic$(OBJ_SUF) mlocsol$(OBJ_SUF) mglobsol$(OBJ_SUF) \
+	 qc$(OBJ_SUF) $(LFLAGS)
+
+d2: d2$(OBJ_SUF) version$(OBJ_SUF) desc2$(OBJ_SUF) mquartic$(OBJ_SUF) mlocsol$(OBJ_SUF) mglobsol$(OBJ_SUF)   qc$(OBJ_SUF) reduce$(OBJ_SUF) \
+	minim$(OBJ_SUF) transform$(OBJ_SUF) sqfdiv$(OBJ_SUF)
+	$(CXX) $(LDFLAGS) -o d2 d2$(OBJ_SUF) version$(OBJ_SUF) desc2$(OBJ_SUF) mquartic$(OBJ_SUF) mlocsol$(OBJ_SUF) mglobsol$(OBJ_SUF)  \
+	qc$(OBJ_SUF) reduce$(OBJ_SUF) minim$(OBJ_SUF) transform$(OBJ_SUF) sqfdiv$(OBJ_SUF) $(LFLAGS)
+
+tred: version$(OBJ_SUF) tred$(OBJ_SUF) reduce$(OBJ_SUF) minim$(OBJ_SUF) transform$(OBJ_SUF) mquartic$(OBJ_SUF) mlocsol$(OBJ_SUF) mglobsol$(OBJ_SUF) \
+	 qc$(OBJ_SUF)
+	$(CXX) $(LDFLAGS) -o tred version$(OBJ_SUF) tred$(OBJ_SUF) reduce$(OBJ_SUF) minim$(OBJ_SUF) transform$(OBJ_SUF) mquartic$(OBJ_SUF) \
+	mlocsol$(OBJ_SUF) mglobsol$(OBJ_SUF)  qc$(OBJ_SUF) $(LFLAGS)
 
-tequiv: version$(OBJ_SUF).o tequiv$(OBJ_SUF).o mquartic$(OBJ_SUF).o mequiv$(OBJ_SUF).o
-	$(CXX) -o tequiv version$(OBJ_SUF).o tequiv$(OBJ_SUF).o mquartic$(OBJ_SUF).o mequiv$(OBJ_SUF).o $(LFLAGS)
+tequiv: version$(OBJ_SUF) tequiv$(OBJ_SUF) mquartic$(OBJ_SUF) mequiv$(OBJ_SUF)
+	$(CXX) $(LDFLAGS) -o tequiv version$(OBJ_SUF) tequiv$(OBJ_SUF) mquartic$(OBJ_SUF) mequiv$(OBJ_SUF) $(LFLAGS)
 
 
 # DEPENDENCIES
 
-version$(OBJ_SUF).o:  $(HEADERS) $(DOTCCS)
-mwrank$(OBJ_SUF).o:  mwrank.cc descent.h mrank1.h mrank2.h mquartic.h version.h options.h
-tmrank$(OBJ_SUF).o:  tmrank.cc descent.h mrank1.h mrank2.h mquartic.h version.h
-descent$(OBJ_SUF).o: descent.cc descent.h mrank1.h mrank2.h mquartic.h
-mrank1$(OBJ_SUF).o: mrank1.cc mrank1.h descent.h mquartic.h mequiv.h msoluble.h qc.h
-mrank2$(OBJ_SUF).o: mrank2.cc mrank2.h descent.h mquartic.h msoluble.h sqfdiv.h desc2.h bitspace.h
-mequiv$(OBJ_SUF).o: mequiv.cc mquartic.h mequiv.h
-mquartic$(OBJ_SUF).o: mquartic.cc mquartic.h
-
-mglobsol$(OBJ_SUF).o: mglobsol.cc mglobsol.h mquartic.h
-ratpoint$(OBJ_SUF).o: ratpoint.cc qc.h mlocsol.h mglobsol.h mquartic.h qc.h version.h
-desc2$(OBJ_SUF).o: desc2.cc desc2.h transform.h reduce.h mlocsol.h mglobsol.h mquartic.h
-d2$(OBJ_SUF).o: d2.cc desc2.h reduce.h qc.h mlocsol.h mglobsol.h mquartic.h qc.h
-reduce$(OBJ_SUF).o: reduce.cc reduce.h minim.h mquartic.h
-minim$(OBJ_SUF).o: minim.cc minim.h mquartic.h
-transform$(OBJ_SUF).o: transform.cc transform.h
-tred$(OBJ_SUF).o: tred.cc reduce.h mquartic.h
-qc$(OBJ_SUF).o: qc.cc qc.h mquartic.h
-tequiv$(OBJ_SUF).o: tequiv.cc mquartic.h mequiv.h
-
-bitspace$(OBJ_SUF).o: bitspace.h bitspace.cc
-
-sqfdiv$(OBJ_SUF).o: sqfdiv.cc sqfdiv.h
-sqftest$(OBJ_SUF).o: sqftest.cc sqfdiv.h
-
-testarg$(OBJ_SUF).o: testarg.cc
-cubic$(OBJ_SUF).o: cubic.h cubic.cc
-rcubic$(OBJ_SUF).o: cubic.h rcubic.cc
-lcubic$(OBJ_SUF).o: cubic.h lcubic.cc
+version$(OBJ_SUF):  $(HEADERS) $(DOTCCS)
+mwrank$(OBJ_SUF):  mwrank.cc descent.h mrank1.h mrank2.h mquartic.h version.h options.h
+tmrank$(OBJ_SUF):  tmrank.cc descent.h mrank1.h mrank2.h mquartic.h version.h
+descent$(OBJ_SUF): descent.cc descent.h mrank1.h mrank2.h mquartic.h
+mrank1$(OBJ_SUF): mrank1.cc mrank1.h descent.h mquartic.h mequiv.h msoluble.h qc.h
+mrank2$(OBJ_SUF): mrank2.cc mrank2.h descent.h mquartic.h msoluble.h sqfdiv.h desc2.h bitspace.h
+mequiv$(OBJ_SUF): mequiv.cc mquartic.h mequiv.h
+mquartic$(OBJ_SUF): mquartic.cc mquartic.h
+
+mglobsol$(OBJ_SUF): mglobsol.cc mglobsol.h mquartic.h
+ratpoint$(OBJ_SUF): ratpoint.cc qc.h mlocsol.h mglobsol.h mquartic.h qc.h version.h
+desc2$(OBJ_SUF): desc2.cc desc2.h transform.h reduce.h mlocsol.h mglobsol.h mquartic.h
+d2$(OBJ_SUF): d2.cc desc2.h reduce.h qc.h mlocsol.h mglobsol.h mquartic.h qc.h
+reduce$(OBJ_SUF): reduce.cc reduce.h minim.h mquartic.h
+minim$(OBJ_SUF): minim.cc minim.h mquartic.h
+transform$(OBJ_SUF): transform.cc transform.h
+tred$(OBJ_SUF): tred.cc reduce.h mquartic.h
+qc$(OBJ_SUF): qc.cc qc.h mquartic.h
+tequiv$(OBJ_SUF): tequiv.cc mquartic.h mequiv.h
+
+bitspace$(OBJ_SUF): bitspace.h bitspace.cc
+
+sqfdiv$(OBJ_SUF): sqfdiv.cc sqfdiv.h
+sqftest$(OBJ_SUF): sqftest.cc sqfdiv.h
+
+testarg$(OBJ_SUF): testarg.cc
+cubic$(OBJ_SUF): cubic.h cubic.cc
+rcubic$(OBJ_SUF): cubic.h rcubic.cc
+lcubic$(OBJ_SUF): cubic.h lcubic.cc
 
-twoadic$(OBJ_SUF).o: twoadic.cc twoadic.h mlocsol.h
+twoadic$(OBJ_SUF): twoadic.cc twoadic.h mlocsol.h
 
-GetOpt$(OBJ_SUF).o: GetOpt.cc
-new$(OBJ_SUF).o: new.C new.H
+GetOpt$(OBJ_SUF): GetOpt.cc
+new$(OBJ_SUF): new.C new.H
 	$(CXX) -g $(CFLAGS) -DHAVE_ATEXIT new.C
 
 LIBBASE=$(RANKLIBBASE)
