--- qcurves/Makefile.orig	2010-01-26 01:14:19.000000000 +1300
+++ qcurves/Makefile	2010-08-17 23:03:22.809936556 +1200
@@ -25,7 +25,8 @@
 LIBFLAG = -ljcntl -L$(LIBDIR)
 LIBBASE = libcurvesntl
 CURVELIBFLAG = -L../lib -lcurvesntl 
-OBJ_SUF = _n
+OBJ_SUF = _n.o
+SOBJ_SUF = _n.lo
 EXE_SUFFIX = _n
 else
 NTLCFLAGS = 
@@ -45,10 +46,22 @@
 
 LIBFLAGS = $(LFLAGS)
 
-%$(OBJ_SUF).o:   %.cc
+#template for compiling *.cc -> *$(OBJ_SUF) :
+.SUFFIXES: .cc $(OBJ_SUF) $(SOBJ_SUF)
+
+ .cc$(OBJ_SUF):
 	$(CXX) $(CFLAGS) $< -o $@
 
-OBJS = curve$(OBJ_SUF).o curvedata$(OBJ_SUF).o curvered$(OBJ_SUF).o points$(OBJ_SUF).o heights$(OBJ_SUF).o cperiods$(OBJ_SUF).o isogs$(OBJ_SUF).o mwprocs$(OBJ_SUF).o lambda$(OBJ_SUF).o sifter$(OBJ_SUF).o sieve_search$(OBJ_SUF).o htconst$(OBJ_SUF).o egr$(OBJ_SUF).o saturate$(OBJ_SUF).o divpol$(OBJ_SUF).o pointsmod$(OBJ_SUF).o curvemod$(OBJ_SUF).o ffmod$(OBJ_SUF).o tlss$(OBJ_SUF).o elog$(OBJ_SUF).o getcurve$(OBJ_SUF).o 
+ .cc$(SOBJ_SUF):
+	$(CXX) $(PICFLAG) $(CFLAGS) $< -o $@
+
+OBJS = curve$(OBJ_SUF) curvedata$(OBJ_SUF) curvered$(OBJ_SUF) points$(OBJ_SUF) heights$(OBJ_SUF) cperiods$(OBJ_SUF) isogs$(OBJ_SUF) mwprocs$(OBJ_SUF) lambda$(OBJ_SUF) sifter$(OBJ_SUF) sieve_search$(OBJ_SUF) htconst$(OBJ_SUF) egr$(OBJ_SUF) saturate$(OBJ_SUF) divpol$(OBJ_SUF) pointsmod$(OBJ_SUF) curvemod$(OBJ_SUF) ffmod$(OBJ_SUF) tlss$(OBJ_SUF) elog$(OBJ_SUF) getcurve$(OBJ_SUF)
+
+SOBJS = curve$(SOBJ_SUF) curvedata$(SOBJ_SUF) curvered$(SOBJ_SUF) points$(SOBJ_SUF) heights$(SOBJ_SUF) \
+	cperiods$(SOBJ_SUF) isogs$(SOBJ_SUF) mwprocs$(SOBJ_SUF) lambda$(SOBJ_SUF) sifter$(SOBJ_SUF) \
+	sieve_search$(SOBJ_SUF) htconst$(SOBJ_SUF) egr$(SOBJ_SUF) saturate$(SOBJ_SUF) divpol$(SOBJ_SUF) \
+	pointsmod$(SOBJ_SUF) curvemod$(SOBJ_SUF) ffmod$(SOBJ_SUF) tlss$(SOBJ_SUF) elog$(SOBJ_SUF) \
+	getcurve$(SOBJ_SUF)
 
 objs: $(OBJS)
 
@@ -71,12 +84,12 @@
 
 lib: $(CURVELIBNAME)
 
-$(CURVELIBNAME): $(OBJS)
+$(CURVELIBNAME): objs
 	rm -f $(CURVELIBNAME)
 	ar r $(CURVELIBNAME) $(OBJS)
 	$(RANLIB) $(CURVELIBNAME)
 
-install_lib: $(HEADERS) lib
+install_lib: headers lib
 	(cd $(INCDIR); rm -f $(HEADERS))
 	for f in $(HEADERS); \
 	 do cp $${f} $(INCDIR); \
@@ -120,7 +133,7 @@
 	chmod 644 $(HOME)/curves/g0ndist/*
 
 clean:
-	/bin/rm -f *~ *$(OBJ_SUF).o  $(TESTS) $(CURVELIBNAME) *_sta *.a *.so *.dylib *.dll *.def
+	/bin/rm -f *~ *$(OBJ_SUF)  $(TESTS) $(CURVELIBNAME) *_sta *.a *.so *.dylib *.dll *.def
 
 veryclean: clean
 	/bin/rm -f $(PROGS) $(TESTS) *.exe
@@ -130,10 +143,10 @@
 	shar -i files -o qcurves.shar
 	compress qcurves.shar
 
-check:  $(TESTS) $(PROGS)
+check:  tests progs
 	rm -f PRIMES t
-	./tcurve < tcurve.in > t && diff t tcurve.out 
-	./tpoints > t && diff t tpoints.out 
+	./tcurve < tcurve.in > t && diff t tcurve.out
+	./tpoints > t && diff t tpoints.out
 	./theight < theight.in > t && diff t theight.out
 	./thtconst < tcurve.in > t && diff t thtconst.out
 	./tegr < tegr.in > t && diff t tegr.out
@@ -143,216 +156,216 @@
 	./tsat < tsat.in > t 2>/dev/null && diff t tsat.out
 	./tsat2 < tsat2.in > t 2>/dev/null && diff t tsat2.out
 	./tsat3 < tsat2.in > t 2>/dev/null && diff t tsat3.out
-	./conductor < conductor.in > t 2>/dev/null && diff t conductor.out
-	./tate < tate.in > t 2>/dev/null && diff t tate.out
-	./torsion < torsion.in > t 2>/dev/null && diff t torsion.out
-	./twist < twist.in > t 2>/dev/null && diff t twist.out
-	./allisog < allisog.in > t 2>/dev/null && diff t allisog.out
-	./findinf < findinf.in > t 2>/dev/null && diff t findinf.out
-	./indep < indep.in > t 2>/dev/null && diff t indep.out
+	$(BINDIR)/conductor < conductor.in > t 2>/dev/null && diff t conductor.out
+	$(BINDIR)/tate < tate.in > t 2>/dev/null && diff t tate.out
+	$(BINDIR)/torsion < torsion.in > t 2>/dev/null && diff t torsion.out
+	$(BINDIR)/twist < twist.in > t 2>/dev/null && diff t twist.out
+	$(BINDIR)/allisog < allisog.in > t 2>/dev/null && diff t allisog.out
+	$(BINDIR)/findinf < findinf.in > t 2>/dev/null && diff t findinf.out
+	$(BINDIR)/indep < indep.in > t 2>/dev/null && diff t indep.out
 	rm -f PRIMES t
 
-tcurve: tcurve$(OBJ_SUF).o curve$(OBJ_SUF).o curvedata$(OBJ_SUF).o curvered$(OBJ_SUF).o curvemod$(OBJ_SUF).o pointsmod$(OBJ_SUF).o ffmod$(OBJ_SUF).o 
-	$(CXX)  -o $(BINDIR)/tcurve tcurve$(OBJ_SUF).o curve$(OBJ_SUF).o curvedata$(OBJ_SUF).o curvered$(OBJ_SUF).o curvemod$(OBJ_SUF).o pointsmod$(OBJ_SUF).o ffmod$(OBJ_SUF).o  $(LFLAGS)
+tcurve: tcurve$(OBJ_SUF) curve$(OBJ_SUF) curvedata$(OBJ_SUF) curvered$(OBJ_SUF) curvemod$(OBJ_SUF) pointsmod$(OBJ_SUF) ffmod$(OBJ_SUF)
+	$(CXX) $(LDFLAGS)  -o tcurve tcurve$(OBJ_SUF) curve$(OBJ_SUF) curvedata$(OBJ_SUF) curvered$(OBJ_SUF) curvemod$(OBJ_SUF) pointsmod$(OBJ_SUF) ffmod$(OBJ_SUF)  $(LFLAGS)
 
-tperiods: tperiods$(OBJ_SUF).o cperiods$(OBJ_SUF).o curve$(OBJ_SUF).o curvedata$(OBJ_SUF).o curvered$(OBJ_SUF).o curvemod$(OBJ_SUF).o pointsmod$(OBJ_SUF).o ffmod$(OBJ_SUF).o getcurve$(OBJ_SUF).o
-	$(CXX) -o $(BINDIR)/tperiods tperiods$(OBJ_SUF).o cperiods$(OBJ_SUF).o curve$(OBJ_SUF).o curvedata$(OBJ_SUF).o \
-		curvered$(OBJ_SUF).o curvemod$(OBJ_SUF).o pointsmod$(OBJ_SUF).o ffmod$(OBJ_SUF).o getcurve$(OBJ_SUF).o $(LFLAGS)
+tperiods: tperiods$(OBJ_SUF) cperiods$(OBJ_SUF) curve$(OBJ_SUF) curvedata$(OBJ_SUF) curvered$(OBJ_SUF) curvemod$(OBJ_SUF) pointsmod$(OBJ_SUF) ffmod$(OBJ_SUF) getcurve$(OBJ_SUF)
+	$(CXX) $(LDFLAGS) -o tperiods tperiods$(OBJ_SUF) cperiods$(OBJ_SUF) curve$(OBJ_SUF) curvedata$(OBJ_SUF) \
+		curvered$(OBJ_SUF) curvemod$(OBJ_SUF) pointsmod$(OBJ_SUF) ffmod$(OBJ_SUF) getcurve$(OBJ_SUF) $(LFLAGS)
 
-qperiods: qperiods$(OBJ_SUF).o
-	$(CXX)  -o $(BINDIR)/qperiods qperiods$(OBJ_SUF).o  $(LFLAGS)
+qperiods: qperiods$(OBJ_SUF)
+	$(CXX) $(LDFLAGS)  -o qperiods qperiods$(OBJ_SUF)  $(LFLAGS)
 
-tpoints: tpoints$(OBJ_SUF).o  $(CURVELIBNAME)
-	$(CXX)  -o $(BINDIR)/tpoints tpoints$(OBJ_SUF).o  $(CURVELIBNAME) $(LFLAGS)
+tpoints: tpoints$(OBJ_SUF)  $(CURVELIBNAME)
+	$(CXX) $(LDFLAGS)  -o tpoints tpoints$(OBJ_SUF)  $(CURVELIBNAME) $(LFLAGS)
 
-telog: telog$(OBJ_SUF).o $(CURVELIBNAME)
-	$(CXX)  -o $(BINDIR)/telog telog$(OBJ_SUF).o $(CURVELIBNAME) $(LFLAGS)
+telog: telog$(OBJ_SUF) $(CURVELIBNAME)
+	$(CXX) $(LDFLAGS)  -o telog telog$(OBJ_SUF) $(CURVELIBNAME) $(LFLAGS)
 
-theight: theight$(OBJ_SUF).o  $(CURVELIBNAME)
-	$(CXX)  -o $(BINDIR)/theight theight$(OBJ_SUF).o  $(CURVELIBNAME) $(LFLAGS)
+theight: theight$(OBJ_SUF)  $(CURVELIBNAME)
+	$(CXX) $(LDFLAGS)  -o theight theight$(OBJ_SUF)  $(CURVELIBNAME) $(LFLAGS)
 
-twist: twist$(OBJ_SUF).o $(CURVELIBNAME)
-	$(CXX)  -o $(BINDIR)/twist twist$(OBJ_SUF).o $(CURVELIBNAME) $(LFLAGS)
+twist: twist$(OBJ_SUF) $(CURVELIBNAME)
+	$(CXX) $(LDFLAGS)  -o $(BINDIR)/twist twist$(OBJ_SUF) $(CURVELIBNAME) $(LFLAGS)
 
-twist_sta: twist$(OBJ_SUF).o $(CURVELIBNAME)
-	$(CXX) -static -o $(BINDIR)/twist_sta twist$(OBJ_SUF).o $(CURVELIBNAME) $(LFLAGS)
+twist_sta: twist$(OBJ_SUF) $(CURVELIBNAME)
+	$(CXX) $(LDFLAGS) -static -o $(BINDIR)/twist_sta twist$(OBJ_SUF) $(CURVELIBNAME) $(LFLAGS)
 
-torsion: torsion$(OBJ_SUF).o $(CURVELIBNAME)
-	$(CXX)  -o $(BINDIR)/torsion torsion$(OBJ_SUF).o $(CURVELIBNAME) $(LFLAGS)
+torsion: torsion$(OBJ_SUF) $(CURVELIBNAME)
+	$(CXX) $(LDFLAGS)  -o $(BINDIR)/torsion torsion$(OBJ_SUF) $(CURVELIBNAME) $(LFLAGS)
 
-torsion_sta: torsion$(OBJ_SUF).o $(CURVELIBNAME)
-	$(CXX) -static -o $(BINDIR)/torsion_sta torsion$(OBJ_SUF).o $(CURVELIBNAME) $(LFLAGS)
+torsion_sta: torsion$(OBJ_SUF) $(CURVELIBNAME)
+	$(CXX) $(LDFLAGS) -static -o $(BINDIR)/torsion_sta torsion$(OBJ_SUF) $(CURVELIBNAME) $(LFLAGS)
 
-tate: tate$(OBJ_SUF).o  $(CURVELIBNAME)
-	$(CXX)  -o $(BINDIR)/tate tate$(OBJ_SUF).o $(CURVELIBNAME) $(LFLAGS)
+tate: tate$(OBJ_SUF)  $(CURVELIBNAME)
+	$(CXX) $(LDFLAGS)  -o $(BINDIR)/tate tate$(OBJ_SUF) $(CURVELIBNAME) $(LFLAGS)
 
-tate_sta: tate$(OBJ_SUF).o  $(CURVELIBNAME)
-	$(CXX)  -static -o $(BINDIR)/tate_sta tate$(OBJ_SUF).o $(CURVELIBNAME) $(LFLAGS)
+tate_sta: tate$(OBJ_SUF)  $(CURVELIBNAME)
+	$(CXX) $(LDFLAGS)  -static -o $(BINDIR)/tate_sta tate$(OBJ_SUF) $(CURVELIBNAME) $(LFLAGS)
 
-conductor: conductor$(OBJ_SUF).o  $(CURVELIBNAME)
-	$(CXX)  -o $(BINDIR)/conductor conductor$(OBJ_SUF).o $(CURVELIBNAME)  $(LFLAGS)
+conductor: conductor$(OBJ_SUF)  $(CURVELIBNAME)
+	$(CXX) $(LDFLAGS)  -o $(BINDIR)/conductor conductor$(OBJ_SUF) $(CURVELIBNAME)  $(LFLAGS)
 
-conductor_sta: conductor$(OBJ_SUF).o  $(CURVELIBNAME)
-	$(CXX)  -static -o $(BINDIR)/conductor_sta conductor$(OBJ_SUF).o $(CURVELIBNAME) $(LFLAGS)
+conductor_sta: conductor$(OBJ_SUF)  $(CURVELIBNAME)
+	$(CXX) $(LDFLAGS)  -static -o $(BINDIR)/conductor_sta conductor$(OBJ_SUF) $(CURVELIBNAME) $(LFLAGS)
 
-csearch: csearch$(OBJ_SUF).o  $(CURVELIBNAME)
-	$(CXX)  -o $(BINDIR)/csearch csearch$(OBJ_SUF).o   $(CURVELIBNAME)  $(LFLAGS)
+csearch: csearch$(OBJ_SUF)  $(CURVELIBNAME)
+	$(CXX) $(LDFLAGS)  -o csearch csearch$(OBJ_SUF)   $(CURVELIBNAME)  $(LFLAGS)
 
-bdw2357: bdw2357$(OBJ_SUF).o   $(CURVELIBNAME)
-	$(CXX) -o $(BINDIR)/bdw2357 bdw2357$(OBJ_SUF).o  $(CURVELIBNAME) $(LFLAGS)
+bdw2357: bdw2357$(OBJ_SUF)   $(CURVELIBNAME)
+	$(CXX) $(LDFLAGS) -o bdw2357 bdw2357$(OBJ_SUF)  $(CURVELIBNAME) $(LFLAGS)
 
-bdw2357a: bdw2357a$(OBJ_SUF).o  $(CURVELIBNAME)
-	$(CXX) -o $(BINDIR)/bdw2357a bdw2357a$(OBJ_SUF).o $(CURVELIBNAME) $(LFLAGS)
+bdw2357a: bdw2357a$(OBJ_SUF)  $(CURVELIBNAME)
+	$(CXX) $(LDFLAGS) -o bdw2357a bdw2357a$(OBJ_SUF) $(CURVELIBNAME) $(LFLAGS)
 
-bdw2311: bdw2311$(OBJ_SUF).o  $(CURVELIBNAME)
-	$(CXX) -o $(BINDIR)/bdw2311 bdw2311$(OBJ_SUF).o  $(CURVELIBNAME) $(LFLAGS)
+bdw2311: bdw2311$(OBJ_SUF)  $(CURVELIBNAME)
+	$(CXX) $(LDFLAGS) -o bdw2311 bdw2311$(OBJ_SUF)  $(CURVELIBNAME) $(LFLAGS)
 
-getjay: getjay$(OBJ_SUF).o  $(CURVELIBNAME)
-	$(CXX) -o $(BINDIR)/getjay getjay$(OBJ_SUF).o  $(CURVELIBNAME) $(LFLAGS)
+getjay: getjay$(OBJ_SUF)  $(CURVELIBNAME)
+	$(CXX) $(LDFLAGS) -o getjay getjay$(OBJ_SUF)  $(CURVELIBNAME) $(LFLAGS)
 
-isogsort: isogsort$(OBJ_SUF).o   $(CURVELIBNAME)
-	$(CXX) -o $(BINDIR)/isogsort isogsort$(OBJ_SUF).o   $(CURVELIBNAME) $(LFLAGS)
+isogsort: isogsort$(OBJ_SUF)   $(CURVELIBNAME)
+	$(CXX) $(LDFLAGS) -o isogsort isogsort$(OBJ_SUF)   $(CURVELIBNAME) $(LFLAGS)
 
-search: search$(OBJ_SUF).o $(CURVELIBNAME)
-	$(CXX)  -o $(BINDIR)/search search$(OBJ_SUF).o $(CURVELIBNAME) $(LFLAGS)
+search: search$(OBJ_SUF) $(CURVELIBNAME)
+	$(CXX) $(LDFLAGS)  -o search search$(OBJ_SUF) $(CURVELIBNAME) $(LFLAGS)
 
-findinf: findinf$(OBJ_SUF).o $(CURVELIBNAME)
-	$(CXX)  -o $(BINDIR)/findinf findinf$(OBJ_SUF).o $(CURVELIBNAME) $(LFLAGS)
+findinf: findinf$(OBJ_SUF) $(CURVELIBNAME)
+	$(CXX) $(LDFLAGS)  -o $(BINDIR)/findinf findinf$(OBJ_SUF) $(CURVELIBNAME) $(LFLAGS)
 
-indep: indep$(OBJ_SUF).o  $(CURVELIBNAME)
-	$(CXX)  -o $(BINDIR)/indep indep$(OBJ_SUF).o $(CURVELIBNAME) $(LFLAGS)
+indep: indep$(OBJ_SUF)  $(CURVELIBNAME)
+	$(CXX) $(LDFLAGS)  -o $(BINDIR)/indep indep$(OBJ_SUF) $(CURVELIBNAME) $(LFLAGS)
 
-indep_sta: indep$(OBJ_SUF).o  $(CURVELIBNAME)
-	$(CXX) -static -o $(BINDIR)/indep_sta indep$(OBJ_SUF).o $(CURVELIBNAME) $(LFLAGS)
+indep_sta: indep$(OBJ_SUF)  $(CURVELIBNAME)
+	$(CXX) $(LDFLAGS) -static -o $(BINDIR)/indep_sta indep$(OBJ_SUF) $(CURVELIBNAME) $(LFLAGS)
 
-ellisog: ellisog$(OBJ_SUF).o   $(CURVELIBNAME)
-	$(CXX)  -o $(BINDIR)/ellisog ellisog$(OBJ_SUF).o  $(CURVELIBNAME) $(LFLAGS)
+ellisog: ellisog$(OBJ_SUF)   $(CURVELIBNAME)
+	$(CXX) $(LDFLAGS)  -o ellisog ellisog$(OBJ_SUF)  $(CURVELIBNAME) $(LFLAGS)
 
-allisog: allisog$(OBJ_SUF).o $(CURVELIBNAME)
-	$(CXX)  -o $(BINDIR)/allisog allisog$(OBJ_SUF).o $(CURVELIBNAME) $(LFLAGS)
+allisog: allisog$(OBJ_SUF) $(CURVELIBNAME)
+	$(CXX) $(LDFLAGS)  -o $(BINDIR)/allisog allisog$(OBJ_SUF) $(CURVELIBNAME) $(LFLAGS)
 
-allisog_sta: allisog$(OBJ_SUF).o $(CURVELIBNAME)
-	$(CXX) -static -o $(BINDIR)/allisog_sta allisog$(OBJ_SUF).o $(CURVELIBNAME) $(LFLAGS)
+allisog_sta: allisog$(OBJ_SUF) $(CURVELIBNAME)
+	$(CXX) $(LDFLAGS) -static -o $(BINDIR)/allisog_sta allisog$(OBJ_SUF) $(CURVELIBNAME) $(LFLAGS)
 
-t2isog$(OBJ_SUF).o: t2isog.cc isogs.h
-t2isog: t2isog$(OBJ_SUF).o $(CURVELIBNAME)
-	$(CXX)  -o $(BINDIR)/t2isog t2isog$(OBJ_SUF).o  $(CURVELIBNAME) $(LFLAGS)
+t2isog$(OBJ_SUF): t2isog.cc isogs.h
+t2isog: t2isog$(OBJ_SUF) $(CURVELIBNAME)
+	$(CXX) $(LDFLAGS)  -o t2isog t2isog$(OBJ_SUF)  $(CURVELIBNAME) $(LFLAGS)
 
-tsat: tsat$(OBJ_SUF).o $(CURVELIBNAME)
-	$(CXX)  -o $(BINDIR)/tsat tsat$(OBJ_SUF).o  $(CURVELIBNAME) $(LFLAGS)
-tsat2: tsat2$(OBJ_SUF).o $(CURVELIBNAME)
-	$(CXX)  -o $(BINDIR)/tsat2 tsat2$(OBJ_SUF).o $(CURVELIBNAME) $(LFLAGS)
-tsat3: tsat3$(OBJ_SUF).o $(CURVELIBNAME)
-	$(CXX)  -o $(BINDIR)/tsat3 tsat3$(OBJ_SUF).o  $(CURVELIBNAME) $(LFLAGS)
+tsat: tsat$(OBJ_SUF) $(CURVELIBNAME)
+	$(CXX) $(LDFLAGS)  -o tsat tsat$(OBJ_SUF)  $(CURVELIBNAME) $(LFLAGS)
+tsat2: tsat2$(OBJ_SUF) $(CURVELIBNAME)
+	$(CXX) $(LDFLAGS)  -o tsat2 tsat2$(OBJ_SUF) $(CURVELIBNAME) $(LFLAGS)
+tsat3: tsat3$(OBJ_SUF) $(CURVELIBNAME)
+	$(CXX) $(LDFLAGS)  -o tsat3 tsat3$(OBJ_SUF)  $(CURVELIBNAME) $(LFLAGS)
 
 
-thtconst: thtconst$(OBJ_SUF).o htconst$(OBJ_SUF).o points$(OBJ_SUF).o curve$(OBJ_SUF).o cperiods$(OBJ_SUF).o curvedata$(OBJ_SUF).o	curvered$(OBJ_SUF).o egr$(OBJ_SUF).o heights$(OBJ_SUF).o mwprocs$(OBJ_SUF).o sieve_search$(OBJ_SUF).o saturate$(OBJ_SUF).o tlss$(OBJ_SUF).o elog$(OBJ_SUF).o curvemod$(OBJ_SUF).o pointsmod$(OBJ_SUF).o ffmod$(OBJ_SUF).o divpol$(OBJ_SUF).o
-	$(CXX) -o thtconst  thtconst$(OBJ_SUF).o htconst$(OBJ_SUF).o points$(OBJ_SUF).o curve$(OBJ_SUF).o curvedata$(OBJ_SUF).o curvered$(OBJ_SUF).o cperiods$(OBJ_SUF).o egr$(OBJ_SUF).o heights$(OBJ_SUF).o mwprocs$(OBJ_SUF).o tlss$(OBJ_SUF).o elog$(OBJ_SUF).o curvemod$(OBJ_SUF).o pointsmod$(OBJ_SUF).o ffmod$(OBJ_SUF).o divpol$(OBJ_SUF).o sieve_search$(OBJ_SUF).o saturate$(OBJ_SUF).o $(LFLAGS)
+thtconst: thtconst$(OBJ_SUF) htconst$(OBJ_SUF) points$(OBJ_SUF) curve$(OBJ_SUF) cperiods$(OBJ_SUF) curvedata$(OBJ_SUF)	curvered$(OBJ_SUF) egr$(OBJ_SUF) heights$(OBJ_SUF) mwprocs$(OBJ_SUF) sieve_search$(OBJ_SUF) saturate$(OBJ_SUF) tlss$(OBJ_SUF) elog$(OBJ_SUF) curvemod$(OBJ_SUF) pointsmod$(OBJ_SUF) ffmod$(OBJ_SUF) divpol$(OBJ_SUF)
+	$(CXX) $(LDFLAGS) -o thtconst  thtconst$(OBJ_SUF) htconst$(OBJ_SUF) points$(OBJ_SUF) curve$(OBJ_SUF) curvedata$(OBJ_SUF) curvered$(OBJ_SUF) cperiods$(OBJ_SUF) egr$(OBJ_SUF) heights$(OBJ_SUF) mwprocs$(OBJ_SUF) tlss$(OBJ_SUF) elog$(OBJ_SUF) curvemod$(OBJ_SUF) pointsmod$(OBJ_SUF) ffmod$(OBJ_SUF) divpol$(OBJ_SUF) sieve_search$(OBJ_SUF) saturate$(OBJ_SUF) $(LFLAGS)
 
 
-tshift: tshift$(OBJ_SUF).o htconst$(OBJ_SUF).o points$(OBJ_SUF).o curve$(OBJ_SUF).o cperiods$(OBJ_SUF).o curvedata$(OBJ_SUF).o	curvered$(OBJ_SUF).o egr$(OBJ_SUF).o heights$(OBJ_SUF).o mwprocs$(OBJ_SUF).o sieve_search$(OBJ_SUF).o saturate$(OBJ_SUF).o tlss$(OBJ_SUF).o elog$(OBJ_SUF).o curvemod$(OBJ_SUF).o pointsmod$(OBJ_SUF).o ffmod$(OBJ_SUF).o divpol$(OBJ_SUF).o 
-	$(CXX) -o tshift  tshift$(OBJ_SUF).o htconst$(OBJ_SUF).o points$(OBJ_SUF).o curve$(OBJ_SUF).o curvedata$(OBJ_SUF).o curvered$(OBJ_SUF).o cperiods$(OBJ_SUF).o egr$(OBJ_SUF).o heights$(OBJ_SUF).o mwprocs$(OBJ_SUF).o tlss$(OBJ_SUF).o elog$(OBJ_SUF).o curvemod$(OBJ_SUF).o pointsmod$(OBJ_SUF).o ffmod$(OBJ_SUF).o divpol$(OBJ_SUF).o sieve_search$(OBJ_SUF).o saturate$(OBJ_SUF).o $(LFLAGS)
+tshift: tshift$(OBJ_SUF) htconst$(OBJ_SUF) points$(OBJ_SUF) curve$(OBJ_SUF) cperiods$(OBJ_SUF) curvedata$(OBJ_SUF)	curvered$(OBJ_SUF) egr$(OBJ_SUF) heights$(OBJ_SUF) mwprocs$(OBJ_SUF) sieve_search$(OBJ_SUF) saturate$(OBJ_SUF) tlss$(OBJ_SUF) elog$(OBJ_SUF) curvemod$(OBJ_SUF) pointsmod$(OBJ_SUF) ffmod$(OBJ_SUF) divpol$(OBJ_SUF)
+	$(CXX) $(LDFLAGS) -o tshift  tshift$(OBJ_SUF) htconst$(OBJ_SUF) points$(OBJ_SUF) curve$(OBJ_SUF) curvedata$(OBJ_SUF) curvered$(OBJ_SUF) cperiods$(OBJ_SUF) egr$(OBJ_SUF) heights$(OBJ_SUF) mwprocs$(OBJ_SUF) tlss$(OBJ_SUF) elog$(OBJ_SUF) curvemod$(OBJ_SUF) pointsmod$(OBJ_SUF) ffmod$(OBJ_SUF) divpol$(OBJ_SUF) sieve_search$(OBJ_SUF) saturate$(OBJ_SUF) $(LFLAGS)
 
-htbound: htbound$(OBJ_SUF).o htconst$(OBJ_SUF).o points$(OBJ_SUF).o curve$(OBJ_SUF).o cperiods$(OBJ_SUF).o curvedata$(OBJ_SUF).o	curvered$(OBJ_SUF).o egr$(OBJ_SUF).o heights$(OBJ_SUF).o mwprocs$(OBJ_SUF).o sieve_search$(OBJ_SUF).o saturate$(OBJ_SUF).o tlss$(OBJ_SUF).o elog$(OBJ_SUF).o curvemod$(OBJ_SUF).o pointsmod$(OBJ_SUF).o ffmod$(OBJ_SUF).o divpol$(OBJ_SUF).o 
-	$(CXX) -o htbound  htbound$(OBJ_SUF).o htconst$(OBJ_SUF).o points$(OBJ_SUF).o curve$(OBJ_SUF).o curvedata$(OBJ_SUF).o curvered$(OBJ_SUF).o cperiods$(OBJ_SUF).o egr$(OBJ_SUF).o heights$(OBJ_SUF).o mwprocs$(OBJ_SUF).o tlss$(OBJ_SUF).o elog$(OBJ_SUF).o curvemod$(OBJ_SUF).o pointsmod$(OBJ_SUF).o ffmod$(OBJ_SUF).o divpol$(OBJ_SUF).o sieve_search$(OBJ_SUF).o saturate$(OBJ_SUF).o $(LFLAGS)
+htbound: htbound$(OBJ_SUF) htconst$(OBJ_SUF) points$(OBJ_SUF) curve$(OBJ_SUF) cperiods$(OBJ_SUF) curvedata$(OBJ_SUF)	curvered$(OBJ_SUF) egr$(OBJ_SUF) heights$(OBJ_SUF) mwprocs$(OBJ_SUF) sieve_search$(OBJ_SUF) saturate$(OBJ_SUF) tlss$(OBJ_SUF) elog$(OBJ_SUF) curvemod$(OBJ_SUF) pointsmod$(OBJ_SUF) ffmod$(OBJ_SUF) divpol$(OBJ_SUF)
+	$(CXX) $(LDFLAGS) -o htbound  htbound$(OBJ_SUF) htconst$(OBJ_SUF) points$(OBJ_SUF) curve$(OBJ_SUF) curvedata$(OBJ_SUF) curvered$(OBJ_SUF) cperiods$(OBJ_SUF) egr$(OBJ_SUF) heights$(OBJ_SUF) mwprocs$(OBJ_SUF) tlss$(OBJ_SUF) elog$(OBJ_SUF) curvemod$(OBJ_SUF) pointsmod$(OBJ_SUF) ffmod$(OBJ_SUF) divpol$(OBJ_SUF) sieve_search$(OBJ_SUF) saturate$(OBJ_SUF) $(LFLAGS)
 
 ratpoints: ratpoints-1.4.c
 	gcc ratpoints-1.4.c -Wall -o ratpoints -O2 -march=i386 -Dx86 -fomit-frame-pointer\
                             -lgmp -lgcc -lc -lm
 
-eco_appl: eco_appl$(OBJ_SUF).o
-	$(CXX)  -o eco_appl eco_appl$(OBJ_SUF).o  $(LFLAGS)
+eco_appl: eco_appl$(OBJ_SUF)
+	$(CXX) $(LDFLAGS)  -o eco_appl eco_appl$(OBJ_SUF)  $(LFLAGS)
 
-iso: iso$(OBJ_SUF).o
-	$(CXX)  -o iso iso$(OBJ_SUF).o  $(LFLAGS)
+iso: iso$(OBJ_SUF)
+	$(CXX) $(LDFLAGS)  -o iso iso$(OBJ_SUF)  $(LFLAGS)
 
-test_iso: test_iso$(OBJ_SUF).o curvemod$(OBJ_SUF).o pointsmod$(OBJ_SUF).o $(CURVELIBNAME)
-	   $(CXX) -o test_iso test_iso$(OBJ_SUF).o  $(CURVELIBFLAG) $(LFLAGS)
+test_iso: test_iso$(OBJ_SUF) curvemod$(OBJ_SUF) pointsmod$(OBJ_SUF) $(CURVELIBNAME)
+	   $(CXX) $(LDFLAGS) -o test_iso test_iso$(OBJ_SUF)  $(CURVELIBFLAG) $(LFLAGS)
 
-tegr: tegr$(OBJ_SUF).o egr$(OBJ_SUF).o $(CURVELIBNAME)
-	$(CXX)  -o $(BINDIR)/tegr tegr$(OBJ_SUF).o egr$(OBJ_SUF).o $(CURVELIBNAME) $(LFLAGS)
-tdivpol: tdivpol$(OBJ_SUF).o divpol$(OBJ_SUF).o curve$(OBJ_SUF).o curvedata$(OBJ_SUF).o
-	$(CXX)  -o $(BINDIR)/tdivpol tdivpol$(OBJ_SUF).o divpol$(OBJ_SUF).o curve$(OBJ_SUF).o curvedata$(OBJ_SUF).o $(LFLAGS)
+tegr: tegr$(OBJ_SUF) egr$(OBJ_SUF) $(CURVELIBNAME)
+	$(CXX) $(LDFLAGS)  -o tegr tegr$(OBJ_SUF) egr$(OBJ_SUF) $(CURVELIBNAME) $(LFLAGS)
+tdivpol: tdivpol$(OBJ_SUF) divpol$(OBJ_SUF) curve$(OBJ_SUF) curvedata$(OBJ_SUF)
+	$(CXX) $(LDFLAGS)  -o tdivpol tdivpol$(OBJ_SUF) divpol$(OBJ_SUF) curve$(OBJ_SUF) curvedata$(OBJ_SUF) $(LFLAGS)
 
 
 # DEPENDENCIES
 
-curve$(OBJ_SUF).o: curve.h curve.cc
-curvedata$(OBJ_SUF).o: curve.h curvedata.cc
-curvered$(OBJ_SUF).o: curve.h curvered.cc
-tcurve$(OBJ_SUF).o: tcurve.cc curve.h
-twist$(OBJ_SUF).o: twist.cc curve.h
-tate$(OBJ_SUF).o: tate.cc curve.h reader.h
-conductor$(OBJ_SUF).o: tate.cc curve.h reader.h
-points$(OBJ_SUF).o: points.h points.cc cperiods.h curve.h
-heights$(OBJ_SUF).o: heights.cc points.h curve.h
-tpoints$(OBJ_SUF).o: tpoints.cc points.h curve.h
-theight$(OBJ_SUF).o: theight.cc points.h curve.h
-lambda$(OBJ_SUF).o:   lambda.cc points.h curve.h lambda.h
-torsion$(OBJ_SUF).o: torsion.cc points.h curve.h reader.h
-findinf$(OBJ_SUF).o: findinf.cc mwprocs.h points.h curve.h
-mwprocs$(OBJ_SUF).o: mwprocs.cc mwprocs.h points.h curve.h sieve_search.h
-cperiods$(OBJ_SUF).o: cperiods.h cperiods.cc curve.h
-elog$(OBJ_SUF).o: elog.h elog.cc cperiods.h curve.h
-egr$(OBJ_SUF).o: egr.cc egr.h points.h curve.h
-qpemu$(OBJ_SUF).o: qpemu.cc cperiods.h points.h curve.h
-crlists$(OBJ_SUF).o: crlists.h crlists.cc curve.h
-isogs$(OBJ_SUF).o: isogs.h isogs.cc curve.h points.h  cperiods.h
-ellisog$(OBJ_SUF).o: ellisog.cc isogs.h curve.h points.h  cperiods.h
-allisog$(OBJ_SUF).o: allisog.cc isogs.h curve.h points.h  cperiods.h reader.h
-tperiods$(OBJ_SUF).o: tperiods.cc cperiods.h curve.h reader.h
-tesign$(OBJ_SUF).o: curve.h tesign.cc
-tesign: tesign$(OBJ_SUF).o curve$(OBJ_SUF).o curvedata$(OBJ_SUF).o curvered$(OBJ_SUF).o 
-	$(CXX) -o tesign tesign$(OBJ_SUF).o curve$(OBJ_SUF).o curvedata$(OBJ_SUF).o curvered$(OBJ_SUF).o $(LFLAGS)
-telog$(OBJ_SUF).o: telog.cc elog.h points.h curve.h
-elog$(OBJ_SUF).o: elog.h elog.cc cperiods.h curve.h
-egr$(OBJ_SUF).o: egr.cc egr.h points.h curve.h
-getcurve$(OBJ_SUF).o: getcurve.cc getcurve.h
-
-bdw2357$(OBJ_SUF).o: bdw2357.cc curve.h
-bdw2357a$(OBJ_SUF).o: bdw2357a.cc curve.h
-bdw2311$(OBJ_SUF).o: bdw2311.cc curve.h
-getjay$(OBJ_SUF).o: getjay.cc curve.h
-isogsort$(OBJ_SUF).o: isogsort.cc curve.h
-
-indep$(OBJ_SUF).o: indep.cc sifter.h points.h curve.h
-sifter$(OBJ_SUF).o: sifter.h sifter.cc points.h curve.h
-
-htconst$(OBJ_SUF).o: htconst.h htconst.cc points.h curve.h cperiods.h elog.h egr.h
-thtconst$(OBJ_SUF).o: thtconst.cc htconst.h points.h curve.h cperiods.h
-thtbound$(OBJ_SUF).o: thtbound.cc htconst.h points.h curve.h cperiods.h
-
-sieve_search$(OBJ_SUF).o: sieve_search.cc sieve_search.h
-
-divpol$(OBJ_SUF).o: divpol.cc divpol.h points.h
-saturate$(OBJ_SUF).o: saturate.cc saturate.h divpol.h points.h pointsmod.h ffmod.h tlss.h
-tsat$(OBJ_SUF).o: tsat.cc saturate.h points.h pointsmod.h ffmod.h curvemod.h tlss.h
-
-
-tsat2$(OBJ_SUF).o: tsat2.cc saturate.h divpol.h points.h pointsmod.h ffmod.h curvemod.h
-tsat3$(OBJ_SUF).o: tsat3.cc saturate.h divpol.h points.h pointsmod.h ffmod.h curvemod.h htconst.h egr.h
-tegr$(OBJ_SUF).o: tegr.cc egr.h saturate.h divpol.h points.h pointsmod.h ffmod.h curvemod.h htconst.h
-
-
-
-test_iso$(OBJ_SUF).o: test_iso.cc curvemod.h pointsmod.h
-
-tdivpol$(OBJ_SUF).o: tdivpol.cc divpol.h points.h
-curvemod$(OBJ_SUF).o: curvemod.h curvemod.cc curve.h
-pointsmod$(OBJ_SUF).o: pointsmod.h pointsmod.cc curve.h curvemod.h
-ffmod$(OBJ_SUF).o: ffmod.h ffmod.cc pointsmod.h curve.h curvemod.h
-tlss$(OBJ_SUF).o: tlss.h tlss.cc pointsmod.h curve.h ffmod.h curvemod.h
+curve$(OBJ_SUF): curve.h curve.cc
+curvedata$(OBJ_SUF): curve.h curvedata.cc
+curvered$(OBJ_SUF): curve.h curvered.cc
+tcurve$(OBJ_SUF): tcurve.cc curve.h
+twist$(OBJ_SUF): twist.cc curve.h
+tate$(OBJ_SUF): tate.cc curve.h reader.h
+conductor$(OBJ_SUF): tate.cc curve.h reader.h
+points$(OBJ_SUF): points.h points.cc cperiods.h curve.h
+heights$(OBJ_SUF): heights.cc points.h curve.h
+tpoints$(OBJ_SUF): tpoints.cc points.h curve.h
+theight$(OBJ_SUF): theight.cc points.h curve.h
+lambda$(OBJ_SUF):   lambda.cc points.h curve.h lambda.h
+torsion$(OBJ_SUF): torsion.cc points.h curve.h reader.h
+findinf$(OBJ_SUF): findinf.cc mwprocs.h points.h curve.h
+mwprocs$(OBJ_SUF): mwprocs.cc mwprocs.h points.h curve.h sieve_search.h
+cperiods$(OBJ_SUF): cperiods.h cperiods.cc curve.h
+elog$(OBJ_SUF): elog.h elog.cc cperiods.h curve.h
+egr$(OBJ_SUF): egr.cc egr.h points.h curve.h
+qpemu$(OBJ_SUF): qpemu.cc cperiods.h points.h curve.h
+crlists$(OBJ_SUF): crlists.h crlists.cc curve.h
+isogs$(OBJ_SUF): isogs.h isogs.cc curve.h points.h  cperiods.h
+ellisog$(OBJ_SUF): ellisog.cc isogs.h curve.h points.h  cperiods.h
+allisog$(OBJ_SUF): allisog.cc isogs.h curve.h points.h  cperiods.h reader.h
+tperiods$(OBJ_SUF): tperiods.cc cperiods.h curve.h reader.h
+tesign$(OBJ_SUF): curve.h tesign.cc
+tesign: tesign$(OBJ_SUF) curve$(OBJ_SUF) curvedata$(OBJ_SUF) curvered$(OBJ_SUF)
+	$(CXX) $(LDFLAGS) -o tesign tesign$(OBJ_SUF) curve$(OBJ_SUF) curvedata$(OBJ_SUF) curvered$(OBJ_SUF) $(LFLAGS)
+telog$(OBJ_SUF): telog.cc elog.h points.h curve.h
+elog$(OBJ_SUF): elog.h elog.cc cperiods.h curve.h
+egr$(OBJ_SUF): egr.cc egr.h points.h curve.h
+getcurve$(OBJ_SUF): getcurve.cc getcurve.h
+
+bdw2357$(OBJ_SUF): bdw2357.cc curve.h
+bdw2357a$(OBJ_SUF): bdw2357a.cc curve.h
+bdw2311$(OBJ_SUF): bdw2311.cc curve.h
+getjay$(OBJ_SUF): getjay.cc curve.h
+isogsort$(OBJ_SUF): isogsort.cc curve.h
+
+indep$(OBJ_SUF): indep.cc sifter.h points.h curve.h
+sifter$(OBJ_SUF): sifter.h sifter.cc points.h curve.h
+
+htconst$(OBJ_SUF): htconst.h htconst.cc points.h curve.h cperiods.h elog.h egr.h
+thtconst$(OBJ_SUF): thtconst.cc htconst.h points.h curve.h cperiods.h
+thtbound$(OBJ_SUF): thtbound.cc htconst.h points.h curve.h cperiods.h
+
+sieve_search$(OBJ_SUF): sieve_search.cc sieve_search.h
+
+divpol$(OBJ_SUF): divpol.cc divpol.h points.h
+saturate$(OBJ_SUF): saturate.cc saturate.h divpol.h points.h pointsmod.h ffmod.h tlss.h
+tsat$(OBJ_SUF): tsat.cc saturate.h points.h pointsmod.h ffmod.h curvemod.h tlss.h
+
+
+tsat2$(OBJ_SUF): tsat2.cc saturate.h divpol.h points.h pointsmod.h ffmod.h curvemod.h
+tsat3$(OBJ_SUF): tsat3.cc saturate.h divpol.h points.h pointsmod.h ffmod.h curvemod.h htconst.h egr.h
+tegr$(OBJ_SUF): tegr.cc egr.h saturate.h divpol.h points.h pointsmod.h ffmod.h curvemod.h htconst.h
+
+
+
+test_iso$(OBJ_SUF): test_iso.cc curvemod.h pointsmod.h
+
+tdivpol$(OBJ_SUF): tdivpol.cc divpol.h points.h
+curvemod$(OBJ_SUF): curvemod.h curvemod.cc curve.h
+pointsmod$(OBJ_SUF): pointsmod.h pointsmod.cc curve.h curvemod.h
+ffmod$(OBJ_SUF): ffmod.h ffmod.cc pointsmod.h curve.h curvemod.h
+tlss$(OBJ_SUF): tlss.h tlss.cc pointsmod.h curve.h ffmod.h curvemod.h
 
 
-new$(OBJ_SUF).o: new.C new.H
+new$(OBJ_SUF): new.C new.H
 	$(CXX) -g $(CFLAGS) -DHAVE_ATEXIT new.C
 #	$(CXX) -g $(CFLAGS) new.C
 
-lllbasis$(OBJ_SUF).o: lllbasis.cc lllbasis.h points.h curve.h
+lllbasis$(OBJ_SUF): lllbasis.cc lllbasis.h points.h curve.h
 
-tlllbasis$(OBJ_SUF).o: tlllbasis.cc lllbasis.h
-tlllbasis: tlllbasis$(OBJ_SUF).o lllbasis$(OBJ_SUF).o
-	$(CXX)  -o $(BINDIR)/tlllbasis tlllbasis$(OBJ_SUF).o lllbasis$(OBJ_SUF).o  $(CURVELIBFLAG) $(LFLAGS)
+tlllbasis$(OBJ_SUF): tlllbasis.cc lllbasis.h
+tlllbasis: tlllbasis$(OBJ_SUF) lllbasis$(OBJ_SUF)
+	$(CXX) $(LDFLAGS)  -o tlllbasis tlllbasis$(OBJ_SUF) lllbasis$(OBJ_SUF)  $(CURVELIBFLAG) $(LFLAGS)
 
 iso3: iso3.cc
 	g++ -g -static -o iso3 iso3.cc -I/usr/local/LiDIA/include \
