diff --git a/Makefile.in b/Makefile.in
index 0de0f214..1f8d5a17 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -171,7 +171,7 @@ install: library
 		cp -a $(shell ls $(ARB_LIBNAME)*) "$(DESTDIR)$(PREFIX)/$(LIBDIR)"; \
 	fi
 	$(AT)if [ "$(ARB_STATIC)" -eq "1" ]; then \
-		cp libarb.a $(DESTDIR)$(PREFIX)/lib; \
+		cp libarb.a $(DESTDIR)$(PREFIX)/$(LIBDIR); \
 	fi
 	cp $(HEADERS) $(DESTDIR)$(PREFIX)/include
 	$(AT)if [ ! -z $(EXT_HEADERS) ]; then \
