diff --git a/Makefile.am b/Makefile.am
index 22f836c..85e6bd8 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,4 +1,7 @@
-SUBDIRS = 	micropython-1.12 src check intl po debianold debian doc examples
+if BUILD_MICROPY
+    MICROPY_DIR = micropython-1.12
+endif
+SUBDIRS =       $(MICROPY_DIR) src check intl po debianold debian doc examples
 DIST_SUBDIRS = 	micropython-1.12 src check intl po debianold debian doc examples
 EXTRA_DIST = path.h.in setup inst TROUBLES Init.scm compte INSTALACION LEAME.WIN LEAME.ipaq giac.specs giac.specs32 fltk.1.1.changes README.MACOSX pariinl.h changeversion mkosx mkosx10 mkosx3 mkdebian mkdebian32 mkdebian2 win2unix.cc addu.cc xcas*.nsi xcas.desktop xcas.xpm xcas.applications README.visualcpp java.README config.h.win config.h.win64 win win64 mkiphone build_for_iphoneos path.iphone simupath.iphone config.h.iphone config.h.64 config.h.mingw path.mingw configure.mingw README.mingw archive.mingw compile.android mkrpm LISEZMOI.numworks icons/* README.hevea2mml README.itex2MML
 
diff --git a/configure.ac b/configure.ac
index bfa767d..e8bd5ed 100644
--- a/configure.ac
+++ b/configure.ac
@@ -511,6 +511,7 @@ if test "$CONFIG_MICROPY" = "yes"; then
   LDFLAGS="$LDFLAGS -L. -L.. -lmicropython"
   CXXFLAGS="$CXXFLAGS -DHAVE_LIBMICROPYTHON -fpermissive"
 fi
+AM_CONDITIONAL([BUILD_MICROPY], [test "$CONFIG_MICROPY" = "yes"])
 
 dnl Output makefiles etc.
 AC_OUTPUT([
