diff --git a/configure.in b/configure.in
index 0f85137..aba25f5 100644
--- a/configure.in
+++ b/configure.in
@@ -38,6 +38,7 @@ AC_LANG([C++])
 AC_PROG_LIBTOOL
 AC_PROG_YACC
 AM_PROG_LEX
+PKG_PROG_PKG_CONFIG
 
 dnl Check for standard C+headers
 GINAC_STD_CXX_HEADERS
@@ -138,20 +139,9 @@ AC_SUBST(GSL_LIBS)
 AM_CONDITIONAL(CONFIG_GSL, [test "$CONFIG_GSL" = "yes"])
   
 CONFIG_LAPACK="yes"
-AC_ARG_ENABLE([lapack],
-	[AS_HELP_STRING([--enable-lapack], [Use LAPACK [[default=yes]]])],
-	[ if test "$enableval" = "no"; then CONFIG_LAPACK="no"; fi], [])
-  
-if test "$CONFIG_LAPACK" = "yes"; then
-#	AC_CHECK_LIB(f2c, main, [], [CONFIG_LAPACK="no"])
-#	AC_CHECK_LIB(blas, main, [], [CONFIG_LAPACK="no"])
-#	AC_CHECK_LIB(tmglib, main, [], [CONFIG_LAPACK="no"])
-	AC_CHECK_LIB(gfortran, main)
-	AC_CHECK_LIB(blas, main)
-	AC_CHECK_LIB(lapack, main, [], [CONFIG_LAPACK="no"])
-#        AX_BLAS([have_blas=yes],[have_blas=no])
-#        AX_LAPACK([have_lapack=yes],[have_lapack=no])
-fi
+PKG_CHECK_MODULES([LAPACK], [lapack],[
+    AC_DEFINE(HAVE_LAPACK,1,[Define if LAPACK is installed])
+    AC_SUBST(LAPACK_LIBS)])
   
 CONFIG_PARI="yes"
 AC_ARG_ENABLE([pari],
