--- setup.py.orig	2011-11-13 07:22:00.742695902 +1300
+++ setup.py	2011-11-13 07:26:00.979228729 +1300
@@ -10,22 +10,8 @@
       url='http://bitbucket.org/ddrake/sagetex/',
       license='GPLv2+',
       py_modules=['sagetex'],
-      data_files = [('share/texmf/tex/generic/sagetex',
-        ['example.tex',
-         'tkz-berge.sty',
-         'tkz-graph.sty',
-         'tkz-arith.sty',
-         'CONTRIBUTORS',
-         'extractsagecode.py',
-         'makestatic.py',
-         'scripts.dtx',
-         'remote-sagetex.dtx',
+      data_files = [('share/texmf/tex/latex/sage-latex',
+        ['extractsagecode.py',
+         'makestatic.py',
          'remote-sagetex.py',
-         'py-and-sty.dtx',
-         'sagetex.dtx',
-         'sagetex.ins',
-         'sagetexparse.py',
-         'sagetex.sty']),
-      ('share/doc/sagetex', [
-         'sagetex.pdf',
-         'example.pdf'])])
+         'sagetexparse.py'])])
