diff --git a/m4/ac_find_gap.m4 b/m4/ac_find_gap.m4
index ed95f71..f614c9f 100755
--- a/m4/ac_find_gap.m4
+++ b/m4/ac_find_gap.m4
@@ -106,46 +106,47 @@ AC_DEFUN([AC_FIND_GAP],
   #####################################
   # Now check for the GAP header files
 
-  bad=0
-  AC_MSG_CHECKING([for GAP include files])
-  if test -r $GAPROOT/src/compiled.h; then
-    AC_MSG_RESULT([$GAPROOT/src/compiled.h])
-  else
-    AC_MSG_RESULT([Not found])
-    bad=1
-  fi
-  AC_MSG_CHECKING([for GAP config.h])
-  if test -r $GAPROOT/bin/$GAPARCH/config.h; then
-    AC_MSG_RESULT([$GAPROOT/bin/$GAPARCH/config.h])
-  else
-    AC_MSG_RESULT([Not found])
-    bad=1
-  fi
+  #bad=0
+  #AC_MSG_CHECKING([for GAP include files])
+  #if test -r $GAPROOT/src/compiled.h; then
+    #AC_MSG_RESULT([$GAPROOT/src/compiled.h])
+  #else
+    #AC_MSG_RESULT([Not found])
+    #bad=1
+  #fi
+  #AC_MSG_CHECKING([for GAP config.h])
+  #if test -r $GAPROOT/bin/$GAPARCH/config.h; then
+    #AC_MSG_RESULT([$GAPROOT/bin/$GAPARCH/config.h])
+  #else
+    #AC_MSG_RESULT([Not found])
+    #bad=1
+  #fi
 
-  if test "x$bad" = "x1"; then
-    echo ""
-    echo "********************************************************************"
-    echo "  ERROR"
-    echo ""
-    echo "  Failed to find the GAP source header files in src/ and"
-    echo "  GAP's config.h in the architecture dependend directory"
-    echo ""
-    echo "  The kernel build process expects to find the normal GAP "
-    echo "  root directory structure as it is after building GAP itself, and"
-    echo "  in particular the files"
-    echo "      <gaproot>/sysinfo.gap"
-    echo "      <gaproot>/src/<includes>"
-    echo "  and <gaproot>/bin/<architecture>/bin/config.h." 
-    echo "  Please make sure that your GAP root directory structure"
-    echo "  conforms to this layout, or give the correct GAP root using"
-    echo "  --with-gaproot=<path>"
-    echo "********************************************************************"
-    echo ""
-    AC_MSG_ERROR([Unable to find GAP include files in /src subdirectory])
-  fi
+  #if test "x$bad" = "x1"; then
+    #echo ""
+    #echo "********************************************************************"
+    #echo "  ERROR"
+    #echo ""
+    #echo "  Failed to find the GAP source header files in src/ and"
+    #echo "  GAP's config.h in the architecture dependend directory"
+    #echo ""
+    #echo "  The kernel build process expects to find the normal GAP "
+    #echo "  root directory structure as it is after building GAP itself, and"
+    #echo "  in particular the files"
+    #echo "      <gaproot>/sysinfo.gap"
+    #echo "      <gaproot>/src/<includes>"
+    #echo "  and <gaproot>/bin/<architecture>/bin/config.h." 
+    #echo "  Please make sure that your GAP root directory structure"
+    #echo "  conforms to this layout, or give the correct GAP root using"
+    #echo "  --with-gaproot=<path>"
+    #echo "********************************************************************"
+    #echo ""
+    #AC_MSG_ERROR([Unable to find GAP include files in /src subdirectory])
+  #fi
   
-  ARCHPATH=$GAPROOT/bin/$GAPARCH
-  GAP_CPPFLAGS="-I$GAPROOT -I$ARCHPATH"
+  #ARCHPATH=$GAPROOT/bin/$GAPARCH
+  #GAP_CPPFLAGS="-I$GAPROOT -I$ARCHPATH"
+  GAP_CPPFLAGS=""
 
   AC_MSG_CHECKING([for GAP's gmp.h location])
   if test -r "$ARCHPATH/extern/gmp/include/gmp.h"; then
