diff --git a/SPQRGPU/CMakeLists.txt b/SPQRGPU/CMakeLists.txt
index 5624c41..3cc48dc 100644
--- a/SPQRGPU/CMakeLists.txt
+++ b/SPQRGPU/CMakeLists.txt
@@ -74,19 +74,25 @@ target_include_directories ( spqr_cuda PUBLIC
 set_target_properties ( spqr_cuda PROPERTIES POSITION_INDEPENDENT_CODE ON )
 set_target_properties ( spqr_cuda PROPERTIES CUDA_SEPARABLE_COMPILATION ON )
 
+if ( NOT NSTATIC )
 target_include_directories ( spqr_cuda_static PUBLIC
         ${CUDAToolkit_INCLUDE_DIRS}
         ${SPQR_CUDA_INCLUDES} )
 set_target_properties ( spqr_cuda_static PROPERTIES CUDA_SEPARABLE_COMPILATION on )
 set_target_properties ( spqr_cuda_static PROPERTIES POSITION_INDEPENDENT_CODE on )
+endif ()
 
 target_link_libraries ( spqr_cuda ${CHOLMOD_LIBRARIES} )
+if ( NOT STATIC )
 target_link_libraries ( spqr_cuda_static ${CHOLMOD_LIBRARIES} )
+endif ()
 if ( SUITESPARSE_CUDA )
     target_link_libraries ( spqr_cuda CUDA::nvrtc CUDA::cudart_static
         CUDA::nvToolsExt CUDA::cublas )
+    if ( NOT NSTATIC )
     target_link_libraries ( spqr_cuda_static CUDA::nvrtc CUDA::cudart_static
         CUDA::nvToolsExt CUDA::cublas )
+    endif ()
 endif ( )
 
 #-------------------------------------------------------------------------------
@@ -97,5 +103,7 @@ install ( TARGETS spqr_cuda
     LIBRARY DESTINATION ${SUITESPARSE_LIBDIR}
     ARCHIVE DESTINATION ${SUITESPARSE_LIBDIR}
     RUNTIME DESTINATION ${SUITESPARSE_BINDIR} )
+if ( NOT NSTATIC )
 install ( TARGETS spqr_cuda_static
     ARCHIVE DESTINATION ${SUITESPARSE_LIBDIR} )
+endif ()
