diff --git a/setup.py b/setup.py
index cc59019..39dc3c0 100644
--- a/setup.py
+++ b/setup.py
@@ -13,29 +13,31 @@ else:
 
 # The planarity extension
 
-try:
-    import sage.libs
-    ext_modules = []
-except ImportError:    
-    planarity_dir = 'planarity_src/c/'
-    planarity_ui_sources = glob(planarity_dir + 'planarity*.c')
-    planarity_sources = [file for file in glob('planarity_src/c/*.c')
-                         if not file in planarity_ui_sources]
-
-    if sys.platform.startswith('win'):
-        extra_compile_args = ['-D_CRT_SECURE_NO_WARNINGS']
-    else:
-        extra_compile_args = []
+#try:
+    #import sage.libs
+    #ext_modules = []
+#except ImportError:    
+    #planarity_dir = 'planarity_src/c/'
+    #planarity_ui_sources = glob(planarity_dir + 'planarity*.c')
+    #planarity_sources = [file for file in glob('planarity_src/c/*.c')
+                         #if not file in planarity_ui_sources]
+
+    #if sys.platform.startswith('win'):
+        #extra_compile_args = ['-D_CRT_SECURE_NO_WARNINGS']
+    #else:
+        #extra_compile_args = []
     
-    Planarity = Extension(
-        name = 'spherogram.planarity',
-        sources = ['planarity_src/planarity.c'] + planarity_sources, 
-        include_dirs = [planarity_dir],
-        extra_compile_args = extra_compile_args,
-        extra_link_args = extra_link_args
-        )
-
-    ext_modules = [Planarity]
+    #Planarity = Extension(
+        #name = 'spherogram.planarity',
+        #sources = ['planarity_src/planarity.c'] + planarity_sources, 
+        #include_dirs = [planarity_dir],
+        #extra_compile_args = extra_compile_args,
+        #extra_link_args = extra_link_args
+        #)
+
+    #ext_modules = 
+# we don't need to import sage to know we have planarity in sage-on-gentoo
+ext_modules = []
 
 # A real clean
 
