--- configure.orig	2012-11-10 09:42:47.000000000 +1300
+++ configure	2013-05-21 12:31:25.294115084 +1200
@@ -92,12 +92,12 @@
 
 #find dependencies
 
-if [ -d "${MPIR_DIR}/lib" ]; then
-   MPIR_LIB_DIR="${MPIR_DIR}/lib"
-   MPIR_INCLUDE_DIR="${MPIR_DIR}/include"
-elif [ -d "${MPIR_DIR}/lib64" ]; then
+if [ -d "${MPIR_DIR}/lib64" ]; then
    MPIR_LIB_DIR="${MPIR_DIR}/lib64"
    MPIR_INCLUDE_DIR="${MPIR_DIR}/include"
+elif [ -d "${MPIR_DIR}/lib" ]; then
+   MPIR_LIB_DIR="${MPIR_DIR}/lib"
+   MPIR_INCLUDE_DIR="${MPIR_DIR}/include"
 elif [ -d "${MPIR_DIR}/.libs" ]; then
    MPIR_LIB_DIR="${MPIR_DIR}/.libs"
    MPIR_INCLUDE_DIR="${MPIR_DIR}"
@@ -106,12 +106,12 @@
    exit 1
 fi
 
-if [ -d "${MPFR_DIR}/lib" ]; then
-   MPFR_LIB_DIR="${MPFR_DIR}/lib"
-   MPFR_INCLUDE_DIR="${MPFR_DIR}/include"
-elif [ -d "${MPFR_DIR}/lib64" ]; then
+if [ -d "${MPFR_DIR}/lib64" ]; then
    MPFR_LIB_DIR="${MPFR_DIR}/lib64"
    MPFR_INCLUDE_DIR="${MPFR_DIR}/include"
+elif [ -d "${MPFR_DIR}/lib" ]; then
+   MPFR_LIB_DIR="${MPFR_DIR}/lib"
+   MPFR_INCLUDE_DIR="${MPFR_DIR}/include"
 elif [ -d "${MPFR_DIR}/.libs" ]; then
    MPFR_LIB_DIR="${MPFR_DIR}/.libs"
    MPFR_INCLUDE_DIR="${MPFR_DIR}"
@@ -123,18 +123,18 @@
    exit 1
 fi
 
-if [ -d "${NTL_DIR}/lib" ]; then
-   NTL_LIB_DIR="${NTL_DIR}/lib"
-   NTL_INCLUDE_DIR="${NTL_DIR}/include"
-elif [ -d "${NTL_DIR}/lib64" ]; then
+if [ -d "${NTL_DIR}/lib64" ]; then
    NTL_LIB_DIR="${NTL_DIR}/lib64"
    NTL_INCLUDE_DIR="${NTL_DIR}/include"
+elif [ -d "${NTL_DIR}/lib" ]; then
+   NTL_LIB_DIR="${NTL_DIR}/lib"
+   NTL_INCLUDE_DIR="${NTL_DIR}/include"
 elif [ "$WANT_NTL" = "0" ]; then
    NTL_DIR="/usr"
-   if [ -d "${NTL_DIR}/lib" ]; then
-      NTL_LIB_DIR="${NTL_DIR}/lib"
-   else
+   if [ -d "${NTL_DIR}/lib64" ]; then
       NTL_LIB_DIR="${NTL_DIR}/lib64"
+   else
+      NTL_LIB_DIR="${NTL_DIR}/lib"
    fi
    NTL_INCLUDE_DIR="${NTL_DIR}/include"
 else
@@ -261,21 +261,7 @@
 
 #various tuning parameters
 
-if [ -z "$FLINT_TUNE" ]; then
-   if [ "$KERNEL" = "Linux" -a "$ARCH" = "x86_64" ]; then
-      FLINT_TUNE="-funroll-loops"
-   elif [ "$KERNEL" = "Darwin" -a "$ARCH" = "ppc" ]; then
-      FLINT_TUNE=" -funroll-loops"
-   elif [ "`uname -p`" = "powerpc" ]; then
-      FLINT_TUNE="-m64 -mcpu=970 -mtune=970 -mpowerpc64 -falign-loops=16 -falign-functions=16 -falign-labels=16 -falign-jumps=16"
-   elif [ "$ARCH" = "ia64" ]; then
-      # -funroll-loops crashes the build on itanium under GCC-4.2.1, as reported by
-      # Kate Minola.
-      FLINT_TUNE=""
-   else
-      FLINT_TUNE="-funroll-loops"
-   fi
-fi
+FLINT_TUNE=""
 
 # defaults for CC and CXX
 
@@ -363,6 +349,7 @@
 echo "FLINT_NTL_INCLUDE_DIR=$NTL_INCLUDE_DIR" >> Makefile
 echo "" >> Makefile
 echo "FLINT_LIB=$FLINT_LIB" >> Makefile
+echo "FLINT_LIB2=$FLINT_LIB2" >> Makefile
 echo "CC=$CC" >> Makefile
 echo "CXX=$CXX" >> Makefile
 echo "CFLAGS=$CFLAGS" >> Makefile
--- Makefile.in.orig	2012-11-10 09:42:47.000000000 +1300
+++ Makefile.in	2013-05-21 12:44:48.813791146 +1200
@@ -4,7 +4,7 @@
    fmpz_poly_mat nmod_poly_mat fmpz_mod_poly fmpz_factor fmpz_poly_factor \
    fft qsieve
 
-LIBS=-L$(CURDIR) -L$(FLINT_MPIR_LIB_DIR) -L$(FLINT_MPFR_LIB_DIR) -L$(FLINT_NTL_LIB_DIR) -lflint $(EXTRA_LIBS) -lmpfr -lmpir -lm -lpthread
+LIBS=-L$(CURDIR) -lflint -L$(FLINT_MPIR_LIB_DIR) -lmpir -L$(FLINT_MPFR_LIB_DIR) -lmpfr -L$(FLINT_NTL_LIB_DIR) -lntl $(EXTRA_LIBS) -lm -lpthread
 LIBS2=-L$(FLINT_MPIR_LIB_DIR) -L$(FLINT_MPFR_LIB_DIR) -L$(FLINT_NTL_LIB_DIR) $(EXTRA_LIBS) -lmpfr -lmpir -lm -lpthread
 INCS=-I$(CURDIR) -I$(FLINT_MPIR_INCLUDE_DIR) -I$(FLINT_MPFR_INCLUDE_DIR) -I$(FLINT_NTL_INCLUDE_DIR)
 
@@ -72,7 +72,8 @@
 	$(foreach dir, $(BUILD_DIRS), mkdir -p build/$(dir); export BUILD_DIR=../build/$(dir); export MOD_DIR=$(dir); $(MAKE) -C $(dir) shared || exit $$?;)
 	if [ "$(WANT_NTL)" -eq "1" ]; then \
 	  $(MAKE) build/interfaces/NTL-interface.lo; \
-	  $(CXX) $(ABI_FLAG) -shared build/interfaces/NTL-interface.lo $(LOBJS) $(MOD_LOBJS) $(LIBS2) -o $(FLINT_LIB); \
+	  $(CXX) $(ABI_FLAG) -shared build/interfaces/NTL-interface.lo $(LOBJS) $(MOD_LOBJS) $(LIBS2) $(SHARE_FLAGS) -o $(FLINT_LIB); \
+	  ln -s $(FLINT_LIB) $(FLINT_LIB2); \
 	fi
 	if [ "$(WANT_NTL)" -ne "1" ]; then \
 	  $(CC) $(ABI_FLAG) -shared $(LOBJS) $(MOD_LOBJS) $(LIBS2) -o $(FLINT_LIB); \
@@ -149,7 +150,7 @@
 	$(CXX) $(CFLAGS) $(INCS) -c $< -o $@;
 
 build/interfaces/test/t-NTL-interface: interfaces/test/t-NTL-interface.cpp
-	$(CXX) $(CFLAGS) $(INCS) $< build/interfaces/NTL-interface.o -o $@ $(LIBS);
+	$(CXX) $(CFLAGS) $(INCS) $< build/interfaces/NTL-interface.lo -o $@ $(LIBS);
 
 .PHONY: profile library clean examples tune check distclean dist install all
 
