diff --git a/sagenb/flask_version/base.py b/sagenb/flask_version/base.py
index da6a990..371c524 100755
--- a/sagenb/flask_version/base.py
+++ b/sagenb/flask_version/base.py
@@ -11,7 +11,7 @@ from .decorators import global_lock
 from flask_oldsessions import OldSecureCookieSessionInterface
 
 from flask.ext.autoindex import AutoIndex
-from sage.env import SAGE_SRC, SAGE_DOC
+from sage.env import SAGE_SRC, SAGE_DOC, SAGE_LOCAL
 SRC = os.path.join(SAGE_SRC, 'sage')
 from flask.ext.openid import OpenID
 from flask.ext.babel import Babel, gettext, ngettext, lazy_gettext, get_locale
@@ -39,13 +39,13 @@ class SageNBFlask(Flask):
         self.add_static_path('/javascript', DATA)
         self.add_static_path('/static', DATA)
         self.add_static_path('/java', DATA)
-        self.add_static_path('/java/jmol', os.path.join(os.environ["SAGE_SHARE"],"jmol"))
-        self.add_static_path('/jsmol', os.path.join(os.environ["SAGE_SHARE"],"jsmol"))
-        self.add_static_path('/jsmol/js', os.path.join(os.environ["SAGE_SHARE"],"jsmol","js"))
-        self.add_static_path('/j2s', os.path.join(os.environ["SAGE_SHARE"],"jsmol","j2s"))
-        self.add_static_path('/jsmol/j2s', os.path.join(os.environ["SAGE_SHARE"],"jsmol","j2s"))
-        self.add_static_path('/j2s/core', os.path.join(os.environ["SAGE_SHARE"],"jsmol","j2s","core"))
-        self.add_static_path('/threejs', os.path.join(os.environ["SAGE_SHARE"],"threejs"))
+        self.add_static_path('/java/jmol', os.path.join(SAGE_LOCAL,"share","jmol"))
+        self.add_static_path('/jsmol', os.path.join(SAGE_LOCAL,"share","jsmol"))
+        self.add_static_path('/jsmol/js', os.path.join(SAGE_LOCAL,"share","jsmol","js"))
+        self.add_static_path('/j2s', os.path.join(SAGE_LOCAL,"share","jsmol","j2s"))
+        self.add_static_path('/jsmol/j2s', os.path.join(SAGE_LOCAL,"share","jsmol","j2s"))
+        self.add_static_path('/j2s/core', os.path.join(SAGE_LOCAL,"share","jsmol","j2s","core"))
+        self.add_static_path('/threejs', os.path.join(SAGE_LOCAL,"share","threejs"))
         import mimetypes
         mimetypes.add_type('text/plain','.jmol')
 
