diff -Naur sage-notebook-0.11.6.1.orig/sagenb/flask_version/base.py sage-notebook-0.11.6.1/sagenb/flask_version/base.py
--- sage-notebook-0.11.6.1.orig/sagenb/flask_version/base.py	2015-12-21 11:51:45.098183538 +1300
+++ sage-notebook-0.11.6.1/sagenb/flask_version/base.py	2015-12-21 11:54:28.538508812 +1300
@@ -8,10 +8,7 @@
 from flask_oldsessions import OldSecureCookieSessionInterface
 
 from flask.ext.autoindex import AutoIndex
-try:
-    from sage.env import SAGE_SRC
-except ImportError:
-    SAGE_SRC = os.environ.get('SAGE_SRC', os.path.join(os.environ['SAGE_ROOT'], 'devel', 'sage'))
+from sage.env import SAGE_SRC, SAGE_LOCAL
 SRC = os.path.join(SAGE_SRC, 'sage')
 from flask.ext.openid import OpenID
 from flask.ext.babel import Babel, gettext, ngettext, lazy_gettext, get_locale
@@ -39,12 +36,12 @@
         self.add_static_path('/javascript', DATA)
         self.add_static_path('/static', DATA)
         self.add_static_path('/java', DATA)
-        self.add_static_path('/java/jmol', os.path.join(os.environ["SAGE_ROOT"],"local","share","jmol"))
-        self.add_static_path('/jsmol', os.path.join(os.environ["SAGE_ROOT"],"local","share","jsmol"))
-        self.add_static_path('/jsmol/js', os.path.join(os.environ["SAGE_ROOT"],"local","share","jsmol","js"))
-        self.add_static_path('/j2s', os.path.join(os.environ["SAGE_ROOT"],"local","share","jsmol","j2s"))
-        self.add_static_path('/jsmol/j2s', os.path.join(os.environ["SAGE_ROOT"],"local","share","jsmol","j2s"))
-        self.add_static_path('/j2s/core', os.path.join(os.environ["SAGE_ROOT"],"local","share","jsmol","j2s","core"))
+        self.add_static_path('/java/jmol', os.path.join(SAGE_LOCAL,"share","sage-jmol-bin","lib"))
+        self.add_static_path('/jsmol', os.path.join(SAGE_LOCAL,"share","jsmol"))
+        self.add_static_path('/jsmol/js', os.path.join(SAGE_LOCAL,"share","jsmol","js"))
+        self.add_static_path('/j2s', os.path.join(SAGE_LOCAL,"share","jsmol","j2s"))
+        self.add_static_path('/jsmol/j2s', os.path.join(SAGE_LOCAL,"share","jsmol","j2s"))
+        self.add_static_path('/j2s/core', os.path.join(SAGE_LOCAL,"share","jsmol","j2s","core"))
         import mimetypes
         mimetypes.add_type('text/plain','.jmol')
 
