diff --git a/sagenb/notebook/misc.py b/sagenb/notebook/misc.py
index a04b16c..f18e4f7 100644
--- a/sagenb/notebook/misc.py
+++ b/sagenb/notebook/misc.py
@@ -213,11 +213,11 @@ def encode_response(obj, separators=(',', ':'), **kwargs):
         '[3,2.0,{"foo":"bar"},null]'
         sage: d = {'AR': 'MA', int(11): 'foo', 'bar': float(1.0), None: 'blah'}
         sage: encode_response(d, sort_keys = True)
-        '{"null":"blah","11":"foo","AR":"MA","bar":1.0}'
+        '{"11":"foo","AR":"MA","bar":1.0,"null":"blah"}'
         sage: d['archies'] = ['an', 'mon', 'hier']
         sage: d['sub'] = {'shape': 'triangle', 'color': 'blue', 'sides': [int(3), int(4), int(5)]}
         sage: encode_response(d, sort_keys = True)
-        '{"null":"blah","11":"foo","AR":"MA","archies":["an","mon","hier"],"bar":1.0,"sub":{"color":"blue","shape":"triangle","sides":[3,4,5]}}'
+        '{"11":"foo","AR":"MA","archies":["an","mon","hier"],"bar":1.0,"null":"blah","sub":{"color":"blue","shape":"triangle","sides":[3,4,5]}}'
         sage: print(encode_response(d, separators = (', ', ': '), indent = 4))
         {
             "...": ...
