--- setup.py.orig	2012-10-07 23:11:47.689898038 +1300
+++ setup.py	2012-10-07 23:13:37.219009136 +1300
@@ -51,7 +51,8 @@
           url         = 'http://github.com/sagemath/sagenb',
           install_requires = install_requires,
           test_suite = 'sagenb.testing.run_tests.all_tests',
-          packages    = [ 'sagenb'
+          packages    = [ 'flask_version'
+                        , 'sagenb'
                         , 'sagenb.interfaces'
                         , 'sagenb.misc'
                         , 'sagenb.notebook'
@@ -62,8 +63,6 @@
                         , 'sagenb.testing.tests'
                         , 'sagenb.testing.selenium'
                         ],
-          scripts      = [ 'sagenb/data/sage3d/sage3d',
-                         ],
           package_data = {'sagenb':
                               all_files('sagenb/data', 'sagenb/') +
                               all_files('sagenb/translations', 'sagenb/')
