--- flask_version/base.py.orig	2012-08-04 23:42:26.486249930 +1200
+++ flask_version/base.py	2012-08-04 23:48:05.365464915 +1200
@@ -6,10 +6,10 @@
 from decorators import global_lock
 
 from flaskext.autoindex import AutoIndex
-SRC = os.path.join(os.environ['SAGE_ROOT'], 'devel', 'sage', 'sage')
 from flaskext.openid import OpenID
 from flaskext.babel import Babel, gettext, ngettext, lazy_gettext, get_locale
-from sagenb.misc.misc import SAGENB_ROOT, DATA, SAGE_DOC, translations_path
+from sagenb.misc.misc import SAGENB_ROOT, SAGE_ROOT, SAGE_LOCAL, DATA, SAGE_DOC, translations_path
+SRC = os.path.join(os.environ['SAGE_ROOT'], 'devel', 'sage', 'sage')
 
 oid = OpenID()
 
@@ -27,7 +27,7 @@
         self.add_static_path('/javascript', DATA)
         self.add_static_path('/static', DATA)
         self.add_static_path('/java', DATA)
-        self.add_static_path('/java/jmol', os.path.join(os.environ["SAGE_ROOT"],"local","share","jmol"))
+        self.add_static_path('/java/jmol', os.path.join(SAGE_LOCAL,"share","jmol-applet"))
         import mimetypes
         mimetypes.add_type('text/plain','.jmol')
 
