--- setup.py.orig	2013-06-25 11:11:55.857524942 +1200
+++ setup.py	2013-06-25 11:12:46.177277646 +1200
@@ -63,8 +63,6 @@
                         , 'sagenb.testing.tests'
                         , 'sagenb.testing.selenium'
                         ],
-          scripts      = [ 'sagenb/data/sage3d/sage3d',
-                         ],
           package_data = {'sagenb':
                               all_files('sagenb/data', 'sagenb/') +
                               all_files('sagenb/translations', 'sagenb/')
