--- sagenb/notebook/notebook.py.orig	2012-08-05 00:12:14.575425811 +1200
+++ sagenb/notebook/notebook.py	2012-08-05 00:12:50.515450372 +1200
@@ -625,7 +625,7 @@
         else:
             import random
             user_at_host = random.choice(server_pool)
-            python_command = os.path.join(os.environ['SAGE_ROOT'], 'sage -python')
+            python_command = python
             return WorksheetProcess_RemoteExpectImplementation(user_at_host=user_at_host,
                              process_limits=process_limits,
                              remote_python=python_command)
