--- sagenb/misc/sageinspect.py.orig	2012-08-04 23:50:06.789542855 +1200
+++ sagenb/misc/sageinspect.py	2012-08-04 23:52:39.585641746 +1200
@@ -148,7 +148,7 @@
             obj.__class__.__module__ not in ('__builtin__', 'exceptions'))
 
 
-SAGE_ROOT = os.environ["SAGE_ROOT"]
+from sage.misc.misc import SAGE_ROOT
     
 import re
 # Parse strings of form "File: sage/rings/rational.pyx (starting at line 1080)"
