--- sagenb/notebook/wiki2html.py.orig	2010-01-25 00:39:57.000000000 +0000
+++ sagenb/notebook/wiki2html.py	2010-02-23 18:26:38.219772557 +0000
@@ -759,7 +759,7 @@
 
     def _heading_repl(self, word):
         """Handle section headings."""
-        import sha
+        from hashlib import sha1
 
         h = word.strip()
         level = 1
@@ -779,7 +779,7 @@
         if self.titles[pntt] > 1:
             unique_id = '-%d' % self.titles[pntt]
         result = self._closeP()
-        result += self.formatter.heading(1, depth, id="head-"+sha.new(pntt.encode(config.charset)).hexdigest()+unique_id)
+        result += self.formatter.heading(1, depth, id="head-"+sha1.new(pntt.encode(config.charset)).hexdigest()+unique_id)
                                      
         return (result + self.formatter.text(title_text) +
                 self.formatter.heading(0, depth))
