--- setup.py.orig	2012-07-19 12:44:18.543568217 +1200
+++ setup.py	2012-07-19 12:45:44.270617317 +1200
@@ -44,7 +44,8 @@
                          , 'pyOpenSSL<=0.12'
                          ],
       test_suite = 'sagenb.testing.run_tests.all_tests',
-      packages    = [ 'sagenb'
+      packages    = [ 'flask_version'
+                    , 'sagenb'
                     , 'sagenb.interfaces'
                     , 'sagenb.misc'
                     , 'sagenb.notebook'
@@ -55,8 +56,6 @@
                     , 'sagenb.testing.tests'
                     , 'sagenb.testing.selenium'
                     ],
-      scripts      = [ 'sagenb/data/sage3d/sage3d',
-                     ],
       package_data = {'sagenb':
                           all_files('sagenb/data', 'sagenb/') +
                           all_files('sagenb/translations', 'sagenb/')
