--- config/get_Qt.orig	2012-08-30 09:59:45.993224271 +1200
+++ config/get_Qt	2012-08-30 10:22:53.313576255 +1200
@@ -4,7 +4,7 @@
 QTDIR=
 case "$with_qt" in
 yes)
-   pth="/usr/local/lib /usr/local/share /usr/lib /usr/share"
+   pth="@GENTOO_PORTAGE_EPREFIX@/usr/local/lib @GENTOO_PORTAGE_EPREFIX@/usr/local/share @GENTOO_PORTAGE_EPREFIX@/usr/lib @GENTOO_PORTAGE_EPREFIX@/usr/share"
    QTDIR=`locatedir qt4 $pth`
    QTLIB="-lQtCore -lQtGui"
    which_graphic_lib=Qt4
--- config/get_X11.orig	2012-08-30 09:59:57.857236030 +1200
+++ config/get_X11	2012-08-30 10:28:11.090877811 +1200
@@ -10,6 +10,7 @@
 # X11 -- Headers
 if test ! -f $Xincroot/X11/Xos.h; then
   x11pth="\
+    @GENTOO_PORTAGE_EPREFIX@/usr/lib
     /usr/openwin/share/lib\
     /usr/openwin/lib\
     /usr/X11R6/lib       /usr/X11R5/lib       /usr/X11R4/lib\
--- config/get_include_path.orig	2012-08-30 10:00:26.872264764 +1200
+++ config/get_include_path	2012-08-30 10:28:45.202909949 +1200
@@ -7,8 +7,8 @@
     $cygtop/H-${arch}-cygwin32/${arch}-cygwin32/include\
     ";;
   *) incpth="$incpth /usr/include/$multipath\
-    /usr/local/include\
-    /usr/include\
+    @GENTOO_PORTAGE_EPREFIX@/usr/local/include\
+    @GENTOO_PORTAGE_EPREFIX@/usr/include\
     /opt/include\
     /opt/local/include\
     /opt/gnu/include\
--- config/get_libpth.orig	2012-08-30 10:00:48.200285857 +1200
+++ config/get_libpth	2012-08-30 10:30:16.194995460 +1200
@@ -1,5 +1,5 @@
 case "$sizeof_long" in
-  8) addlib64="/usr/local/lib64 /lib64 /usr/lib64"
+  8) addlib64="@GENTOO_PORTAGE_EPREFIX@/usr/local/lib64 @GENTOO_PORTAGE_EPREFIX@/lib64 @GENTOO_PORTAGE_EPREFIX@/usr/lib64"
 esac
 
 libpth="`echo $LIBRARY_PATH | sed -e \"s%\([^$dir_sep]*\)$dir_sep*%\1 %g\"`"
@@ -35,9 +35,9 @@
     fi;;
   os2) libpth=`echo $libpth | sed 's,\\\\,/,g'`;;
   *) libpth="$libpth $addlib64 /usr/lib/$multiarch\
-      /usr/local/lib\
-      /lib\
-      /usr/lib\
+      @GENTOO_PORTAGE_EPREFIX@/usr/local/lib\
+      @GENTOO_PORTAGE_EPREFIX@/lib\
+      @GENTOO_PORTAGE_EPREFIX@/usr/lib\
       /opt/lib\
       /opt/local/lib\
       /opt/gnu/lib\
