--- src/language/init.c.orig	2010-09-21 22:33:02.593520620 +1200
+++ src/language/init.c	2010-09-21 22:34:35.778520446 +1200
@@ -19,6 +19,7 @@
 /*                                                                 */
 /*******************************************************************/
 #include <string.h>
+#include <stdlib.h>
 #include "pari.h"
 #include "paripriv.h"
 #include "anal.h"
@@ -684,6 +685,7 @@
 static void
 pari_exit(void)
 {
+  abort();
   fprintferr("  ***   Error in the PARI system. End of program.\n");
   exit(1);
 }
--- src/kernel/gmp/mp.c.orig	2010-09-21 22:42:12.400521024 +1200
+++ src/kernel/gmp/mp.c	2010-09-21 22:43:20.172520407 +1200
@@ -56,7 +56,7 @@
 int pari_kernel_init(void)
 {
   /* Use pari_malloc instead of malloc */
-  mp_set_memory_functions((void *(*)(size_t)) pari_malloc, gmp_realloc, gmp_free);
+  /* mp_set_memory_functions((void *(*)(size_t)) pari_malloc, gmp_realloc, gmp_free); */
   return 0;
 }
 
