diff --git a/src/gp/gp.c b/src/gp/gp.c
index 678f1ba..af4ecdb 100644
--- a/src/gp/gp.c
+++ b/src/gp/gp.c
@@ -559,6 +559,7 @@ main(int argc, char **argv)
 #endif
   stdin_isatty = pari_stdin_isatty();
   pari_init_defaults();
+  DEBUGMEM = 1;  /* Use \gm1 by default in gp */
   pari_library_path = DL_DFLT_NAME;
   pari_stack_init(&s_A,sizeof(*A),(void**)&A);
   pari_init_opts(1000000 * sizeof(long), 0, INIT_SIGm | INIT_noPRIMEm | INIT_noIMTm);
diff --git a/src/language/init.c b/src/language/init.c
index a079cdf..3d0fda1 100644
--- a/src/language/init.c
+++ b/src/language/init.c
@@ -740,7 +740,8 @@ parivstack_resize(ulong newsize)
   evalstate_reset();
   paristack_setsize(pari_mainstack->rsize, newsize);
   s = pari_mainstack->vsize ? pari_mainstack->vsize : pari_mainstack->rsize;
-  pari_warn(warner,"new maximum stack size = %lu (%.3f Mbytes)", s, s/1048576.);
+  if (DEBUGMEM > 0)
+    pari_warn(warner,"new maximum stack size = %lu (%.3f Mbytes)", s, s/1048576.);
   pari_init_errcatch();
   cb_pari_err_recover(-1);
 }
@@ -754,7 +755,8 @@ paristack_newrsize(ulong newsize)
     pari_mainstack_resize(pari_mainstack, newsize, vsize);
   evalstate_reset();
   s = pari_mainstack->rsize;
-  pari_warn(warner,"new stack size = %lu (%.3f Mbytes)", s, s/1048576.);
+  if (DEBUGMEM > 0)
+    pari_warn(warner,"new stack size = %lu (%.3f Mbytes)", s, s/1048576.);
   pari_init_errcatch();
   cb_pari_err_recover(-1);
 }
@@ -768,7 +770,8 @@ paristack_resize(ulong newsize)
   newsize = maxuu(minuu(newsize, vsize), pari_mainstack->size);
   pari_mainstack->size = newsize;
   pari_mainstack->bot = pari_mainstack->top - pari_mainstack->size;
-  pari_warn(warner,"increasing stack size to %lu",newsize);
+  if (DEBUGMEM > 0)
+    pari_warn(warner,"increasing stack size to %lu",newsize);
 }
 
 void
