--- config/Makefile.SH.orig	2010-07-04 23:32:52.832562767 +1200
+++ config/Makefile.SH	2010-07-04 23:33:51.675562667 +1200
@@ -199,7 +199,7 @@
 LIBPARI      = \$(LIB_PREFIX)pari
 DLLIBPARI    = \$(DLLIB_PREFIX)pari
 LIBPARI_STA  = \$(LIBPARI)\$(_A)
-LIBPARI_SO   = \$(DLLIBPARI).$DLSUFFIX
+LIBPARI_SO   = \$(DLLIBPARI)24.$DLSUFFIX
 SOLIBPARI    = \$(DLLIB_PREFIX)\$(LIBPARI_BASE).$DLSUFFIX
 LIBPARI_DYN   = \$(SOLIBPARI)$sodest
 LIBPARI_SONAME= \$(SOLIBPARI)$soname
@@ -342,29 +342,27 @@
 install: install-lib-$dft install-include install-bin install-man install-misc install-doc install-examples install-cfg install-data
 
 install-include:
-	-mkdir -p \$(INCLUDEDIR)/pari
+	-mkdir -p \$(INCLUDEDIR)/pari24
 	-for i in paricfg.h mpinl.h; do \\
-	  \$(INSTALL_DATA) \$\$i \$(INCLUDEDIR)/pari; done
+	  \$(INSTALL_DATA) \$\$i \$(INCLUDEDIR)/pari24; done
 	-for i in $headerlist; do \\
-	   \$(INSTALL_DATA) $src/headers/\$\$i.h  \$(INCLUDEDIR)/pari; done
-	-\$(RM) \$(INCLUDEDIR)/pari/genpari.h
-	-\$(LN) pari.h \$(INCLUDEDIR)/pari/genpari.h
+	   \$(INSTALL_DATA) $src/headers/\$\$i.h  \$(INCLUDEDIR)/pari24; done
+	-\$(RM) \$(INCLUDEDIR)/pari24/genpari.h
+	-\$(LN) pari.h \$(INCLUDEDIR)/pari24/genpari.h
 
 install-bin: install-bin-$dft
 
 install-bin-dyn: gp-dyn install-lib-dyn
 	-mkdir -p \$(BINDIR)
 	-\$(RM) \$(BINDIR)/gp-$version$exe_suff \$(BINDIR)/gp$exe_suff
-	\$(LD) -o \$(BINDIR)/gp-$version$exe_suff \$(LDFLAGS) \$(OBJSGP) \$(RUNPTH_FINAL) \$(LDDYN) \$(RLLIBS) \$(PLOTLIBS) \$(LIBS) \$(DL_LIBS)
+	\$(LD) -o \$(BINDIR)/gp-$version$exe_suff \$(LDFLAGS) \$(OBJSGP) \$(RUNPTH_FINAL) -L\$(TOPLDDYN) \$(LDDYN) \$(RLLIBS) \$(PLOTLIBS) \$(LIBS) \$(DL_LIBS)
 	-\$(STRIP) \$(STRIPFLAGS) \$(BINDIR)/gp-$version$exe_suff
 	$postconfig \$(BINDIR)/gp-$version$exe_suff
-	-\$(LN) gp-$version$exe_suff \$(BINDIR)/gp$exe_suff
 
 install-bin-sta: gp-sta
 	-mkdir -p \$(BINDIR)
 	-\$(RM) \$(BINDIR)/gp-$version$exe_suff \$(BINDIR)/gp$exe_suff
 	\$(INSTALL_PROGRAM) gp-sta \$(BINDIR)/gp-$version$exe_suff
-	-\$(LN) gp-$version$exe_suff \$(BINDIR)/gp$exe_suff
 # Can't strip it if we want install() to work on OSF.
 #	-\$(STRIP) \$(STRIPFLAGS)  \$(BINDIR)/gp-$version
 	$postconfig \$(BINDIR)/gp-$version$exe_suff
@@ -372,15 +370,13 @@
 install-man::
 	-mkdir -p \$(MANDIR)
 	-\$(RM) \$(MANDIR)/pari.1 \$(MANDIR)/gp.1 \$(MANDIR)/gp-$version.1
-	\$(INSTALL_DATA) $doc/gphelp.1 \$(MANDIR)
+	\$(INSTALL_DATA) $doc/gphelp.1 \$(MANDIR)/gphelp24.1
 	\$(INSTALL_DATA) $doc/gp.1 \$(MANDIR)/gp-$version.1
-	-\$(LN) gp.1 \$(MANDIR)/pari.1
-	-\$(LN) gp-$version.1 \$(MANDIR)/gp.1
+	-\$(LN) gp24.1 \$(MANDIR)/pari24.1
+	-\$(LN) gp-$version.1 \$(MANDIR)/gp24.1
 
 install-misc:
 	-mkdir -p \$(MISCDIR) \$(BINDIR)
-	\$(INSTALL_PROGRAM) $misc/tex2mail \$(BINDIR)
-	\$(INSTALL_DATA) $doc/tex2mail.1 \$(MANDIR)
 	\$(INSTALL_DATA) $misc/README    \$(MISCDIR)
 	\$(INSTALL_DATA) $misc/color.dft \$(MISCDIR)
 	\$(INSTALL_DATA) $misc/gpalias   \$(MISCDIR)
