diff --git a/config/DOC_Make.SH b/config/DOC_Make.SH
index 37b1ed5..3242036 100644
--- a/config/DOC_Make.SH
+++ b/config/DOC_Make.SH
@@ -55,38 +55,41 @@ paricfg.tex: paricfg.tex.in
 %.dvi: %.tex \$(MACROS)
 	\$(TEX) \$<
 
-libpari.pdf: \$(PARI_TEX) \$(MACROS)
+# libpari needs to be able to find users.aux for possible cross-referencing
+# with the user manual. We ensure it is available by adding users.{dvi,pdf}
+# to the libpari.{dvi,pdf} dependencies.
+libpari.pdf: \$(PARI_TEX) users.pdf \$(MACROS)
 	-rm -f libpari.std
 	\$(PDFTEX) libpari
 	\$(PDFTEX) libpari
-	make libpari.std; \$(PDFTEX) libpari; rm -f libpari.std
-libpari.dvi: \$(PARI_TEX) \$(MACROS)
+	\$(MAKE) libpari.std; \$(PDFTEX) libpari; rm -f libpari.std
+libpari.dvi: \$(PARI_TEX) users.dvi \$(MACROS)
 	-rm -f libpari.std
 	\$(TEX) libpari
 	\$(TEX) libpari
-	make libpari.std; \$(TEX) libpari; rm -f libpari.std
+	\$(MAKE) libpari.std; \$(TEX) libpari; rm -f libpari.std
 
 develop.pdf: develop.tex \$(MACROS)
 	-rm -f develop.std
 	\$(PDFTEX) develop
 	\$(PDFTEX) develop
-	make develop.std; \$(PDFTEX) develop; rm -f develop.std
+	\$(MAKE) develop.std; \$(PDFTEX) develop; rm -f develop.std
 develop.dvi: develop.tex \$(MACROS)
 	-rm -f develop.std
 	\$(TEX) develop
 	\$(TEX) develop
-	make develop.std; \$(TEX) develop; rm -f develop.std
+	\$(MAKE) develop.std; \$(TEX) develop; rm -f develop.std
 
 parallel.pdf: parallel.tex \$(MACROS)
 	-rm -f parallel.std
 	\$(PDFTEX) parallel
 	\$(PDFTEX) parallel
-	make parallel.std; \$(PDFTEX) parallel; rm -f parallel.std
+	\$(MAKE) parallel.std; \$(PDFTEX) parallel; rm -f parallel.std
 parallel.dvi: parallel.tex \$(MACROS)
 	-rm -f parallel.std
 	\$(TEX) parallel
 	\$(TEX) parallel
-	make parallel.std; \$(TEX) parallel; rm -f parallel.std
+	\$(MAKE) parallel.std; \$(TEX) parallel; rm -f parallel.std
 
 tutorial.pdf: tutorial.tex \$(MACROS)
 	-rm -f tutorial.std
@@ -108,13 +111,13 @@ users.dvi: \$(USERS_TEX)
 	-rm -f users.std
 	\$(TEX) users
 	\$(TEX) users
-	make users.std; \$(TEX) users; rm -f users.std
+	\$(MAKE) users.std; \$(TEX) users; rm -f users.std
 
 users.pdf: \$(USERS_TEX)
 	-rm -f users.std
 	\$(PDFTEX) users
 	\$(PDFTEX) users
-	make users.std; \$(PDFTEX) users; rm -f users.std
+	\$(MAKE) users.std; \$(PDFTEX) users; rm -f users.std
 
 gpman: gp.1
 	nroff -man gp.1 | unix2dos -ascii > gp.man
