--- Configure.orig	2010-02-19 22:53:00.038780634 +1300
+++ Configure	2010-02-16 23:48:33.084586389 +1300
@@ -68,13 +68,17 @@
 . ./get_libpth # initialize
 if test -n "$kernel"; then # kernel explicitly provided
   case "$kernlvl1" in
-    none) with_gmp=;;
-    gmp) if test -z "$with_gmp"; then with_gmp=yes; fi ;;
+    none) with_gmp=; with_mpir=;;
+    gmp) if test -z "$with_gmp"; then with_gmp=yes; with_mpir=; fi ;;
+    mpir) if test -z "$with_mpir"; then with_mpir=yes; with_gmp=; fi ;;
   esac
 fi
 if test -n "$with_gmp" -o "$fastread" != yes; then
   . ./get_gmp  # $_gmp_list
 fi
+if test -n "$with_mpir" -o "$fastread" != yes; then
+  . ./get_mpir  # $_mpir_list
+fi
 if test "$optimization" != profiling; then
   . ./get_X11  # X11, X11_INC, X11_LIBS
   . ./get_graphic_lib # which_graphic_lib
@@ -88,6 +92,7 @@
 #############################################################################
 case $kernlvl1 in
 gmp) libpari_base=pari-gmp;;
+mpir) libpari_base=pari-gmp;;
 none) libpari_base=pari;;
 esac
 
--- config/get_config_options.orig	2010-02-16 23:30:02.366586100 +1300
+++ config/get_config_options	2010-02-19 22:59:34.542779895 +1300
@@ -96,6 +96,15 @@
   --with-gmp-include=*)
       with_gmp_include=`echo "$1" | sed -e 's/[-a-z]*=//'` ;;
 
+  --without-mpir|--with-mpir=no) without_mpir=yes ;;
+  --with-mpir) with_mpir=yes ;;
+  --with-mpir=*)
+      with_mpir=`echo "$1" | sed -e 's/[-a-z]*=//'` ;;
+  --with-mpir-lib=*)
+      with_mpir_lib=`echo "$1" | sed -e 's/[-a-z]*=//'` ;;
+  --with-mpir-include=*)
+      with_mpir_include=`echo "$1" | sed -e 's/[-a-z]*=//'` ;;
+
   --with-ncurses-lib=*|--with-ncurses=*)
       with_ncurses_lib=`echo "$1" | sed -e 's/[-a-z]*=//'` ;;
 
@@ -115,6 +124,7 @@
 # see get_install.
 for i in with_readline with_readline_lib with_readline_include \
          with_gmp with_gmp_lib with_gmp_include \
+         with_mpir with_mpir_lib with_mpir_include \
          with_ncurses_lib with_qt with_fltk ; do
   eval "dflt=\"\$$i\""
   case $dflt in 
@@ -164,11 +174,16 @@
   --with-readline-lib=DIR     specify location of readline libs
   --with-ncurses-lib=DIR      specify location of ncurses lib (for readline)
 
-  --without-gmp               use the native kernel instead of GNU MP
+  --without-gmp               use the native kernel (or MPIR) instead of GNU MP
   --with-gmp[=DIR]            use the GMP kernel [prefix for gmp files]
   --with-gmp-include=DIR      specify location of gmp headers
   --with-gmp-lib=DIR          specify location of gmp libs
 
+  --without-mpir              use the native kernel (or GNU MP) instead of MPIR
+  --with-mpir[=DIR]           use the MPIR kernel [prefix for mpir files]
+  --with-mpir-include=DIR     specify location of mpir headers
+  --with-mpir-lib=DIR         specify location of mpir libs
+
   --with-qt[=DIR]        use the Qt graphical library [prefix for Qt dir.]
   --with-fltk[=DIR]      use the FLTK graphical library [prefix for FLTK dir.]
 
