--- numpy/core/include/numpy/ufuncobject.h.orig	2009-07-28 15:04:42 -0400
+++ numpy/core/include/numpy/ufuncobject.h	2009-07-28 15:05:58 -0400
@@ -318,8 +318,10 @@
 
 #elif defined(__GLIBC__) || defined(__APPLE__) || defined(__CYGWIN__) || defined(__MINGW32__) || (defined(__FreeBSD__) && (__FreeBSD_version >= 502114))
 
-#if defined(__GLIBC__) || defined(__APPLE__) || defined(__MINGW32__) || defined(__FreeBSD__)
+#if defined(__GLIBC__) || defined(__APPLE__) || defined(__MINGW32__) 
 #include <fenv.h>
+#elif defined(__FreeBSD__)
+#include "fenv/fenv.h"
 #elif defined(__CYGWIN__)
 #include "fenv/fenv.c"
 #endif
--- numpy/numarray/_capi.c.orig	2009-07-28 15:18:13 -0400
+++ numpy/numarray/_capi.c	2009-07-28 15:19:04 -0400
@@ -8,8 +8,10 @@
 #include <sys/param.h>
 #endif
 
-#if defined(__GLIBC__) || defined(__APPLE__) || defined(__MINGW32__) || (defined(__FreeBSD__) && (__FreeBSD_version >= 502114))
+#if defined(__GLIBC__) || defined(__APPLE__) || defined(__MINGW32__)
 #include <fenv.h>
+#elif (defined(__FreeBSD__) && (__FreeBSD_version >= 502114))
+#include "numpy/fenv/fenv.h"
 #elif defined(__CYGWIN__)
 #include "numpy/fenv/fenv.h"
 #include "numpy/fenv/fenv.c"
