diff -ur numpy-1.3.0.orig/numpy/core/include/numpy/npy_cpu.h numpy-1.3.0/numpy/core/include/numpy/npy_cpu.h
--- numpy-1.3.0.orig/numpy/core/include/numpy/npy_cpu.h	2009-07-22 16:22:51.000000000 +0200
+++ numpy-1.3.0/numpy/core/include/numpy/npy_cpu.h	2009-07-22 16:23:18.000000000 +0200
@@ -44,6 +44,8 @@
     #define NPY_CPU_IA64
 #elif defined(__hppa__)
     #define NPY_CPU_PARISC
+#elif defined(__alpha__)
+    #define NPY_CPU_ALPHA
 #else
     #error Unknown CPU, please report this to numpy maintainers with \
     information about your platform (OS, CPU and compiler)
diff -ur numpy-1.3.0.orig/numpy/core/include/numpy/npy_endian.h numpy-1.3.0/numpy/core/include/numpy/npy_endian.h
--- numpy-1.3.0.orig/numpy/core/include/numpy/npy_endian.h	2009-07-22 16:22:51.000000000 +0200
+++ numpy-1.3.0/numpy/core/include/numpy/npy_endian.h	2009-07-22 16:23:04.000000000 +0200
@@ -22,7 +22,7 @@
     #include "npy_cpu.h"
 
     #if defined(NPY_CPU_X86) || defined(NPY_CPU_AMD64)\
-            || defined(NPY_CPU_IA64)
+            || defined(NPY_CPU_IA64) || defined(NPY_CPU_ALPHA)
         #define NPY_LITTLE_ENDIAN
         #define NPY_BYTE_ORDER 1234
     #elif defined(NPY_CPU_PPC) || defined(NPY_CPU_SPARC)\