Reported [2010.01.18]: http://projects.scipy.org/numpy/ticket/1365

diff -ru numpy-1.4.0.orig/numpy/core/setup.py numpy-1.4.0/numpy/core/setup.py
--- numpy-1.4.0.orig/numpy/core/setup.py	2010-01-15 16:49:14 +0100
+++ numpy-1.4.0/numpy/core/setup.py	2010-01-15 17:02:27 +0100
@@ -169,6 +169,9 @@
     priv = []
     pub = []
 
+    if os.uname()[0] == "Interix":
+        return priv, pub
+
     # Check for complex support
     st = config.check_header('complex.h')
     if st:
