diff -Naur sagetex-3.0.orig/setup.py sagetex-3.0/setup.py
--- sagetex-3.0.orig/setup.py	2015-08-28 14:02:02.989854644 +1200
+++ sagetex-3.0/setup.py	2015-08-28 14:05:44.170380486 +1200
@@ -10,21 +10,9 @@
       url='https://github.com/dandrake/sagetex',
       license='GPLv2+',
       py_modules=['sagetex'],
-      data_files = [('share/texmf/tex/latex/sagetex',
-        ['example.tex',
-         'CONTRIBUTORS',
-         'extractsagecode.py',
+      data_files = [('share/texmf/tex/latex/sage-latex',
+        ['extractsagecode.py',
          'run-sagetex-if-necessary.py',
          'makestatic.py',
-         'scripts.dtx',
-         'remote-sagetex.dtx',
          'remote-sagetex.py',
-         'py-and-sty.dtx',
-         'sagetex.dtx',
-         'sagetex.ins',
-         'sagetexparse.py',
-         'sagetex.sty']),
-      ('share/doc/sagetex', [
-         'example.tex',
-         'sagetex.pdf',
-         'example.pdf'])])
+         'sagetexparse.py'])])
