diff -r 887323b1bb48 sage-maxima.lisp
--- a/sage-maxima.lisp	Tue May 01 08:20:34 2007 -0700
+++ b/sage-maxima.lisp	Sat Oct 30 22:59:03 2010 -0400
@@ -1,4 +1,8 @@
-;(setf *prompt-prefix* "<prompt>")
-;(setf *prompt-suffix* "</prompt>")
-(setf *general-display-prefix* "<sage-display>")
+; (setf *prompt-prefix* "<prompt>")
+; (setf *prompt-suffix* "</prompt>")
 
+; This broke in maxima-5.22.1 as discussed in http://trac.sagemath.org/sage_trac/ticket/10187
+; (setf *general-display-prefix* "<sage-display>")
+
+(setf *prompt-prefix* "<sage-display>")
+
