--- configure.in.orig	2010-04-25 17:01:05.119862326 +0000
+++ configure.in	2010-04-25 17:06:44.065861693 +0000
@@ -22,8 +22,6 @@
 AC_PROG_MAKE_SET
 
 # Checks for libraries.
-AC_CHECK_LIB(gmp, __gmpz_init, , [AC_MSG_ERROR(
-[GNU MP not found, see http://gmplib.org])])
 AC_CHECK_LIB(mpfr, mpfr_add, , [AC_MSG_ERROR(
 [MPFR not found, see http://www.mpfr.org])])
 
@@ -35,14 +33,6 @@
        mpfr_lib_path="$withval/lib"
    ])
 
-AC_ARG_WITH(gmp,
-   [  --with-gmp=DIR          GMP install directory ], [
-      CPPFLAGS="$CPPFLAGS -I$withval/include"
-       LDFLAGS="$LDFLAGS -L$withval/lib"
-       gmp_lib_path="$withval/lib"
-   ])
-
-
 # Checks for header files.
 AC_HEADER_STDC
 AC_CHECK_HEADERS([float.h limits.h stdlib.h sys/time.h unistd.h])
@@ -59,9 +49,6 @@
 AC_CHECK_FUNCS([floor pow rint sqrt strtol])
 
 
-AC_CHECK_LIB(gmp, __gmp_get_memory_functions, , [AC_MSG_ERROR(
-[GMP version too old, need >= 4.2.0, see http://gmplib.org])])
-
 AC_CHECK_LIB(mpfr, mpfr_fms, , [AC_MSG_ERROR(
 [MPFR version too old, need >= 2.3.0, see http://www.mpfr.org])])
 
--- Makefile.am.orig	2010-04-25 17:09:37.157861694 +0000
+++ Makefile.am	2010-04-25 17:09:46.973871062 +0000
@@ -5,7 +5,7 @@
 
 libfplll_la_SOURCES = dummy.cpp
 
-libfplll_la_LIBADD = -lgmp -lmpfr
+libfplll_la_LIBADD = -lmpir -lmpfr
 libfplll_la_LDFLAGS = -version-info 1:0:1
 
 
--- src/Makefile.am.orig	2010-04-25 17:13:49.689908441 +0000
+++ src/Makefile.am	2010-04-25 17:10:53.842870484 +0000
@@ -21,12 +21,12 @@
 
 
 generate_SOURCES=generate.cpp defs.h dpe.h lexer.h matrix.h nr.h util.h vect.h
-generate_LDFLAGS=-lgmp -lmpfr
+generate_LDFLAGS=-lmpir -lmpfr
 
 llldiff_SOURCES=llldiff.cpp
 
 
-CCLIBS=-lgmp -lmpfr
+CCLIBS=-lmpir -lmpfr
 FLAGSDBG=-g -DDEBUG -DVERBOSE
 KFPOBJ=enumerate.o evaluator.o solver.o tools.o topenum.o
 UTILSRC=defs.h dpe.h lexer.h matrix.cpp \
--- src/defs.h.orig	2010-04-25 17:35:24.686863089 +0000
+++ src/defs.h	2010-04-25 17:35:58.899861414 +0000
@@ -31,7 +31,7 @@
 #include <cstdlib>
 #include <cmath>
 
-#include <gmp.h>
+#include <mpir.h>
 #include <mpfr.h>
 #include "dpe.h"
 
--- src/test.cpp.orig	2010-04-25 17:35:35.230992437 +0000
+++ src/test.cpp	2010-04-25 17:35:58.900861344 +0000
@@ -1,4 +1,4 @@
-#include <gmp.h>
+#include <mpir.h>
 #include <mpfr.h>
 #include <fplll.h>
 
--- src/util.h.orig	2010-04-25 17:35:44.012862531 +0000
+++ src/util.h	2010-04-25 17:35:58.909861344 +0000
@@ -23,7 +23,7 @@
 #define UTIL_H
 
 #include <cstdio>
-#include <gmp.h>
+#include <mpir.h>
 #include <mpfr.h>
 
 #include "defs.h"
