diff -u -r Python-2.7.5-orig//configure.ac Python-2.7.5/configure.ac
--- Python-2.7.5-orig//configure.ac	2014-03-06 04:35:31.000000000 +0100
+++ Python-2.7.5/configure.ac	2014-03-06 04:58:10.000000000 +0100
@@ -2300,10 +2300,13 @@
 
 if test "$with_system_ffi" = "yes" && test -n "$PKG_CONFIG"; then
     LIBFFI_INCLUDEDIR="`"$PKG_CONFIG" libffi --cflags-only-I 2>/dev/null | sed -e 's/^-I//;s/ *$//'`"
+    LIBFFI_LIB="`"$PKG_CONFIG" libffi --libs-only-l 2>/dev/null | sed -e 's/^-l//;s/ *$//'`"
 else
     LIBFFI_INCLUDEDIR=""
+    LIBFFI_LIB=""
 fi
 AC_SUBST(LIBFFI_INCLUDEDIR)
+AC_SUBST(LIBFFI_LIB)
 
 AC_MSG_RESULT($with_system_ffi)
 
diff -u -r Python-2.7.5-orig//Makefile.pre.in Python-2.7.5/Makefile.pre.in
--- Python-2.7.5-orig//Makefile.pre.in	2014-03-06 04:35:31.000000000 +0100
+++ Python-2.7.5/Makefile.pre.in	2014-03-06 05:03:59.000000000 +0100
@@ -224,6 +224,7 @@
 
 
 LIBFFI_INCLUDEDIR=	@LIBFFI_INCLUDEDIR@
+LIBFFI_LIB=	@LIBFFI_LIB@
 
 ##########################################################################
 # Parser
diff -u -r Python-2.7.5-orig//setup.py Python-2.7.5/setup.py
--- Python-2.7.5-orig//setup.py	2014-03-06 04:35:31.000000000 +0100
+++ Python-2.7.5/setup.py	2014-03-06 04:56:15.000000000 +0100
@@ -1893,8 +1893,10 @@
                     break
                 if line.startswith('#define LIBFFI_H'):
                     break
-        ffi_lib = None
-        if ffi_inc is not None:
+        ffi_lib = sysconfig.get_config_var("LIBFFI_LIB")
+        if not ffi_lib or ffi_lib == '':
+            ffi_lib = None
+        if ffi_inc is not None and ffi_lib is None:
             for lib_name in ('ffi_convenience', 'ffi_pic', 'ffi'):
                 if (self.compiler.find_library_file(lib_dirs, lib_name)):
                     ffi_lib = lib_name
