diff --git a/setup.py b/setup.py
index a9ecf28..9c9b10b 100755
--- a/setup.py
+++ b/setup.py
@@ -58,11 +58,12 @@ else:
 # SAGE
 have_sage = False
 
-try:
-    import sage
-    have_sage = True
-except ImportError:
-    pass
+if os.getenv('WANT_SAGE',False):
+    try:
+        import sage
+        have_sage = True
+    except ImportError:
+        pass
 
 if have_sage:
     from sage.env import SAGE_SRC
