From edb89c4c9c830ca6c82b2cdbddb6a5f85f729895 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fran=C3=A7ois=20Bissey?= <francois.bissey@canterbury.ac.nz>
Date: Tue, 13 Sep 2016 12:41:23 +1200
Subject: [PATCH] get SAGE_SRV from sage/env.py rather than the environment

---
 setup.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 4e6dd62..7a06a4c 100755
--- a/setup.py
+++ b/setup.py
@@ -65,7 +65,8 @@
     pass
 
 if have_sage:
-    fplll["include_dirs"].append(os.getenv("SAGE_SRC"))
+    from sage.env import SAGE_SRC
+    fplll["include_dirs"].append(SAGE_SRC)
     config_pxi.append("DEF HAVE_SAGE=True")
 else:
     config_pxi.append("DEF HAVE_SAGE=False")
