diff --git a/sage/matrix/matrix_gfpn_dense.pyx b/sage/matrix/matrix_gfpn_dense.pyx
index abd0754db4..5262b323c3 100644
--- a/sage/matrix/matrix_gfpn_dense.pyx
+++ b/sage/matrix/matrix_gfpn_dense.pyx
@@ -55,7 +55,7 @@ from sage.misc.randstate import current_randstate
 from sage.misc.randstate cimport randstate
 from sage.structure.element cimport Element, Matrix
 from sage.structure.richcmp import rich_to_bool
-from .args cimport MatrixArgs_init
+from sage.matrix.args cimport MatrixArgs_init
 
 from libc.string cimport memset, memcpy
 
