diff -Naur m4rie-20150908.orig/tests/Makefile.am m4rie-20150908/tests/Makefile.am
--- m4rie-20150908.orig/tests/Makefile.am	2015-10-30 09:50:39.870397062 +1300
+++ m4rie-20150908/tests/Makefile.am	2015-10-30 09:51:07.310459539 +1300
@@ -1,7 +1,7 @@
 AUTOMAKE_OPTIONS = foreign subdir-objects
 
 AM_CFLAGS = ${SIMD_FLAGS} ${OPENMP_CFLAGS} ${DEBUG_FLAGS} ${M4RIE_M4RI_CFLAGS} ${M4RI_CFLAGS}
-CFLAGS = ${AM_CFLAGS} -I.. @CFLAGS@ ${M4RIE_M4RI_CFLAGS} ${M4RI_CFLAGS}
+CFLAGS = ${AM_CFLAGS} -I${top_srcdir} @CFLAGS@ ${M4RIE_M4RI_CFLAGS} ${M4RI_CFLAGS}
 LDADD = ${top_builddir}/libm4rie.la -lm4ri -lm
 LDFLAGS = ${M4RIE_M4RI_LDFLAGS} -no-install
 
