diff -Naur grape.orig/lib/grape.g grape/lib/grape.g
--- grape.orig/lib/grape.g	2016-04-01 09:39:26.247960002 +1300
+++ grape/lib/grape.g	2016-04-01 09:42:54.298409197 +1300
@@ -39,13 +39,11 @@
 GRAPE_NRANGENS := 18;  # The number of random generators taken for a subgroup
 		       # when  GRAPE_RANDOM=true.
 
-GRAPE_NAUTY := true;   # Use nauty when true, else use bliss.
+GRAPE_NAUTY := @nauty@;   # Use nauty when true, else use bliss.
 
-GRAPE_DREADNAUT_EXE := 
-   ExternalFilename(DirectoriesPackagePrograms("grape"),"dreadnautB"); 
-   # filename of dreadnaut or dreadnautB executable
+GRAPE_DREADNAUT_EXE := "@GENTOO_PORTAGE_EPREFIX@/usr/bin/dreadnaut";
 
-GRAPE_BLISS_EXE := "/home/leonard/bliss-0.73/bliss"; 
+GRAPE_BLISS_EXE := "@GENTOO_PORTAGE_EPREFIX@/usr/bin/bliss"; 
    # filename of bliss executable
 
 BindGlobal("GRAPE_OrbitRepresentatives",function(arg)
