diff --git a/src/io.c b/src/io.c
index 8d8d1f3..fea741d 100644
--- a/src/io.c
+++ b/src/io.c
@@ -11,10 +11,10 @@
 /* Try to use as much of the GNU C library as possible: */
 #define _GNU_SOURCE
 
-#include "src/compiled.h"          /* GAP headers                */
+#include "gap/compiled.h"          /* GAP headers                */
 
 #if GAP_KERNEL_MAJOR_VERSION >= 6
-#include "src/profile.h"
+#include "gap/profile.h"
 #endif
 
 #undef PACKAGE
