diff --git a/src/bitarray.c b/src/bitarray.c
index b0b8250..10307da 100644
--- a/src/bitarray.c
+++ b/src/bitarray.c
@@ -14,7 +14,7 @@
 #include <stdlib.h>  // for free, calloc, malloc
 
 // GAP headers
-#include "src/compiled.h"  // for Obj, ELM_LIST, ISB_LIST, Fail
+#include "gap/compiled.h"  // for Obj, ELM_LIST, ISB_LIST, Fail
 
 // Digraphs headers
 #include "digraphs-debug.h"  // for DIGRAPHS_ASSERT
diff --git a/src/bitarray.h b/src/bitarray.h
index 0cd7c0e..d897820 100644
--- a/src/bitarray.h
+++ b/src/bitarray.h
@@ -18,7 +18,7 @@
 #include <stdint.h>   // for uint16_t
 
 // GAP headers
-#include "src/compiled.h"  // for COUNT_TRUES_BLOCKS, Obj, . . .
+#include "gap/compiled.h"  // for COUNT_TRUES_BLOCKS, Obj, . . .
 
 // Digraphs headers
 #include "digraphs-debug.h"  // for DIGRAPHS_ASSERT
diff --git a/src/conditions.h b/src/conditions.h
index b4e1c4d..9166b48 100644
--- a/src/conditions.h
+++ b/src/conditions.h
@@ -17,7 +17,7 @@
 #include <string.h>   // for NULL, memcpy, size_t
 
 // GAP headers
-#include "src/system.h"  // for ALWAYS_INLINE
+#include "gap/system.h"  // for ALWAYS_INLINE
 
 // Digraphs headers
 #include "bitarray.h"        // for BitArray, intersect_bit_arrays, size_b...
diff --git a/src/digraphs.h b/src/digraphs.h
index 8f06336..2355d3f 100644
--- a/src/digraphs.h
+++ b/src/digraphs.h
@@ -16,7 +16,7 @@
 #define DIGRAPHS_SRC_DIGRAPHS_H_
 
 // GAP headers
-#include "src/compiled.h"  // for Obj, Int
+#include "gap/compiled.h"  // for Obj, Int
 
 Int DigraphNrVertices(Obj D);
 Obj FuncOutNeighbours(Obj self, Obj D);
diff --git a/src/homos-graphs.c b/src/homos-graphs.c
index b3662fb..fe5cc64 100644
--- a/src/homos-graphs.c
+++ b/src/homos-graphs.c
@@ -14,7 +14,7 @@
 #include <stdlib.h>  // for free, malloc, NULL
 
 // GAP headers
-#include "src/compiled.h"  // for Obj, Int
+#include "gap/compiled.h"  // for Obj, Int
 
 // Digraphs headers
 #include "digraphs-config.h"  // for DIGRAPHS_WITH_INCLUDED_BLISS
diff --git a/src/homos.c b/src/homos.c
index 52f6eba..3ed90e7 100644
--- a/src/homos.c
+++ b/src/homos.c
@@ -36,7 +36,7 @@
 #include <time.h>     // for time
 
 // GAP headers
-#include "src/compiled.h"
+#include "gap/compiled.h"
 
 // Digraphs package headers
 #include "bitarray.h"         // for BitArray
diff --git a/src/homos.h b/src/homos.h
index 603b921..e239051 100644
--- a/src/homos.h
+++ b/src/homos.h
@@ -16,7 +16,7 @@
 #define MIN(a, b) (a < b ? a : b)
 
 // GAP headers
-#include "src/compiled.h"
+#include "gap/compiled.h"
 
 Obj FuncHomomorphismDigraphsFinder(Obj self, Obj args);
 
diff --git a/src/perms.h b/src/perms.h
index e5c5f6c..1831acd 100644
--- a/src/perms.h
+++ b/src/perms.h
@@ -22,8 +22,8 @@
 #include "digraphs-debug.h"  // for DIGRAPHS_ASSERT
 
 // GAP headers
-#include "src/compiled.h"  // for Obj, Int
-#include "src/permutat.h"  // for ADDR_PERM, IS_PERM
+#include "gap/compiled.h"  // for Obj, Int
+#include "gap/permutat.h"  // for ADDR_PERM, IS_PERM
 
 #define MAXVERTS 512
 #define UNDEFINED MAXVERTS + 1
diff --git a/src/planar.h b/src/planar.h
index d64960c..48f1497 100644
--- a/src/planar.h
+++ b/src/planar.h
@@ -13,7 +13,7 @@
 #define DIGRAPHS_SRC_PLANAR_H_
 
 // GAP headers
-#include "src/compiled.h"
+#include "gap/compiled.h"
 
 Obj FuncIS_PLANAR(Obj self, Obj digraph);
 Obj FuncKURATOWSKI_PLANAR_SUBGRAPH(Obj self, Obj digraph);
