diff --git a/src/scripts/cysignals-CSI b/src/scripts/cysignals-CSI
index 28187b6..4a0348c 100755
--- a/src/scripts/cysignals-CSI
+++ b/src/scripts/cysignals-CSI
@@ -75,7 +75,7 @@ def gdb_commands(pid, color):
     cmds += b('import sys; sys.stdout.flush()\n')
     cmds += b('end\n')
     cmds += b('bt full\n')
-    script = os.path.join(os.path.dirname(sys.argv[0]), 'cysignals-CSI-helper.py')
+    script = os.path.join(sysconfig.get_config_var('BINDIR'), 'cysignals-CSI-helper.py')
     with open(script, 'r') as f:
         cmds += b('python\n')
         cmds += b('color = {0}\n'.format(color))
