diff --git a/src/scripts/cysignals-CSI b/src/scripts/cysignals-CSI
index dd8b4e6..d29bd93 100755
--- a/src/scripts/cysignals-CSI
+++ b/src/scripts/cysignals-CSI
@@ -65,8 +65,8 @@ def gdb_commands(pid, color):
     cmds += b'import sys; sys.stdout.flush()\n'
     cmds += b'end\n'
     cmds += b'bt full\n'
-    cysignals_share = os.path.join(os.path.dirname(sys.argv[0]), '..',
-                                   'share', 'cysignals')
+    cysignals_share = os.path.join(sysconfig.get_config_var('BINDIR'),
+                                   '..', 'share', 'cysignals')
     script = os.path.join(cysignals_share, 'cysignals-CSI-helper.py')
     with open(script, 'rb') as f:
         cmds += b'python\n'
