diff --git a/src/cysignals/signals.pxd.in b/src/cysignals/signals.pxd.in
index c86e085..a98c8d1 100644
--- a/src/cysignals/signals.pxd.in
+++ b/src/cysignals/signals.pxd.in
@@ -54,7 +54,7 @@ cdef extern from "macros.h" nogil:
 # can be used to make Cython check whether there is a pending exception
 # (PyErr_Occurred() is non-NULL). To Cython, it will look like
 # cython_check_exception() actually raised the exception.
-cdef inline void cython_check_exception() nogil except *:
+cdef inline void cython_check_exception() except * nogil:
     pass


diff --git a/setup.py b/setup.py
index 37acdfc..f68270b 100755
--- a/setup.py
+++ b/setup.py
@@ -157,13 +157,17 @@ class build_ext(_build_ext):
         # Run Cython with -Werror on continuous integration services
         # with Python 3.6 or later
         from Cython.Compiler import Options
-        Options.warning_errors = True
+        Options.warning_errors = False

         from Cython.Build.Dependencies import cythonize
         return cythonize(extensions,
                 build_dir=cythonize_dir,
                 include_path=["src", os.path.join(cythonize_dir, "src")],
-                compiler_directives=dict(binding=True, language_level=2))
+                compiler_directives=dict(
+                    binding=True,
+                    language_level=2,
+                    legacy_implicit_noexcept=True,
+                    ))


 class build_py(_build_py):
