From d5a26efca4050d79b4b4770fcdab386c60daf243 Mon Sep 17 00:00:00 2001
From: Jeroen Demeyer <jdemeyer@cage.ugent.be>
Date: Fri, 18 Nov 2016 12:00:44 +0100
Subject: [PATCH] Improve installation and installation testing

---
 .travis.yml |  5 ++--
 Makefile    | 94 ++++++++++++++++++++++++++++++++++++++++++++++++++++---------
 setup.py    | 29 +++++++++++++------
 3 files changed, 104 insertions(+), 24 deletions(-)

diff --git a/Makefile b/Makefile
index 7a8c251..a00a300 100644
--- a/Makefile
+++ b/Makefile
@@ -1,28 +1,33 @@
 # Optional Makefile for easier development
 
-PYTHON=python
+PYTHON = python
+PIP = $(PYTHON) -m pip -v
+
+
+#####################
+# Build
+#####################
 
 all: build doc
 
 build: configure
 	$(PYTHON) setup.py build
 
-install: build
-	$(PYTHON) setup.py install
+install: configure
+	$(PIP) install --no-index --ignore-installed .
 
 dist: configure
 	chmod go+rX-w -R .
 	umask 0022 && $(PYTHON) setup.py sdist --formats=bztar
 
-distcheck: dist
-	mkdir -p dist/check
-	VERSION=`cat VERSION`; cd dist/check && tar xjf ../cysignals-$$VERSION.tar.bz2
-	VERSION=`cat VERSION`; cd dist/check/cysignals-$$VERSION && ./configure && $(MAKE) all check
-	rm -rf dist/check
-
 doc:
 	cd docs && $(MAKE) html
 
+
+#####################
+# Clean
+#####################
+
 clean: clean-doc clean-build
 
 clean-build:
@@ -35,15 +40,75 @@ distclean: clean
 	rm -rf autom4te.cache
 	rm -f config.log config.status
 
-check: check-doctest check-example
+
+#####################
+# Check
+#####################
+
+test: check
+
+check: check-tmp
+
+check-all: check-tmp check-install
+
+# Install and check
+check-install: check-doctest check-example
 
 check-doctest: install
 	$(PYTHON) -B rundoctests.py src/cysignals/*.pyx
 
 check-example: install
-	cd example && $(PYTHON) setup.py build
+	cd example && $(PYTHON) setup.py clean build
 
-test: check
+
+#####################
+# Check installation
+#####################
+#
+# Test 2 installation scenarios without a real installation
+# - prefix (with --prefix and --root)
+# - user (with --user)
+
+check-tmp: check-prefix check-user
+	rm -rf tmp
+
+prefix-install: configure
+	rm -rf tmp/local tmp/build
+	$(PYTHON) setup.py install --prefix="`pwd`/tmp/local" --root=tmp/build
+	cd tmp && mv "build/`pwd`/local" local
+	cd tmp && ln -s local/lib*/python*/site-packages site-packages
+
+check-prefix: check-prefix-doctest check-prefix-example
+
+check-prefix-doctest: prefix-install
+	export PYTHONPATH="`pwd`/tmp/site-packages" && $(PYTHON) -B rundoctests.py src/cysignals/*.pyx
+
+check-prefix-example: prefix-install
+	rm -rf example/build
+	export PYTHONPATH="`pwd`/tmp/site-packages" && cd example && $(PYTHON) setup.py clean build
+
+check-user: check-user-doctest check-user-example
+
+user-install: configure
+	rm -rf tmp/user
+	export PYTHONUSERBASE="`pwd`/user" && $(PYTHON) setup.py install --user --single-version-externally-managed --record=record
+
+check-user-doctest: user-install
+	export PYTHONUSERBASE="`pwd`/user" && $(PYTHON) -B rundoctests.py src/cysignals/*.pyx
+
+check-user-example: user-install
+	export PYTHONUSERBASE="`pwd`/user" && cd example && $(PYTHON) setup.py clean build
+
+distcheck: dist
+	mkdir -p dist/check
+	VERSION=`cat VERSION`; cd dist/check && tar xjf ../cysignals-$$VERSION.tar.bz2
+	VERSION=`cat VERSION`; cd dist/check/cysignals-$$VERSION && $(MAKE) all check-tmp
+	rm -rf dist/check
+
+
+#####################
+# Maintain
+#####################
 
 configure: configure.ac
 	autoconf
@@ -51,4 +116,7 @@ configure: configure.ac
 	@rm -f src/config.h.in~
 
 .PHONY: all build doc install dist doc clean clean-build clean-doc \
-	distclean check check-doctest check-example test
+	distclean test check check-all check-tmp check-install \
+	check-doctest check-example \
+	check-prefix prefix-install check-prefix-doctest check-prefix-example \
+	check-user user-install check-user-doctest check-user-example
diff --git a/setup.py b/setup.py
index 3c19c38..586ccaf 100755
--- a/setup.py
+++ b/setup.py
@@ -1,9 +1,11 @@
 #!/usr/bin/env python
 # -*- coding: utf-8 -*-
-from distutils.core import setup, Distribution
+
+from setuptools import setup
 from distutils.command.build_py import build_py as _build_py
 from distutils.command.build_ext import build_ext as _build_ext
-from distutils.extension import Extension
+from setuptools.command.install import install as _install
+from setuptools.extension import Extension
 from Cython.Build.Dependencies import cythonize
 
 import warnings
@@ -109,12 +111,13 @@ def create_init_pxd(self):
         The ``__init__.pxd`` file sets the correct compiler options for
         packages using cysignals.
         """
-        dist = self.distribution
-
         # Determine installation directory
-        inst = dist.get_command_obj("install")
-        inst.ensure_finalized()
-        install_dir = opj(inst.install_platlib, "cysignals")
+        inst = self.get_finalized_command("install")
+        try:
+            platlib = inst.noroot_platlib
+        except AttributeError:
+            platlib = inst.install_platlib
+        install_dir = opj(platlib, "cysignals")
 
         # The variable "init_pxd" is the string which should be written to
         # __init__.pxd
@@ -142,6 +145,16 @@ def get_init_pxd(self):
             return f.read()
 
 
+class install(_install):
+    # When handling the --root option, keep the original paths prefixed
+    # with "noroot" instead of "install"
+    def change_roots(self, *names):
+        for name in names:
+            path = getattr(self, "install_" + name)
+            setattr(self, "noroot_" + name, path)
+        _install.change_roots(self, *names)
+
+
 setup(
     name="cysignals",
     author=u"Martin R. Albrecht, François Bissey, Volker Braun, Jeroen Demeyer",
@@ -160,5 +173,5 @@ def get_init_pxd(self):
     package_data={"cysignals": ["*.pxi", "*.pxd", "*.h"],
                   "cysignals-cython": ["*.h"]},
     scripts=glob(opj("src", "scripts", "*")),
-    cmdclass=dict(build_py=build_py, build_ext=build_ext),
+    cmdclass=dict(build_py=build_py, build_ext=build_ext, install=install),
 )
