diff --git a/sage_setup/cython_options.py b/sage_setup/cython_options.py
index 086aa070ca9..9725ce0e1af 100644
--- a/sage_setup/cython_options.py
+++ b/sage_setup/cython_options.py
@@ -10,13 +10,17 @@ def compiler_directives(profile: bool):
         auto_pickle=False,
         # Do not create __test__ dictionary automatically from docstrings
         autotestdict=False,
+        binding=False,
+        c_api_binop_methods=True,
         # Do not check for division by 0 (this is about 35% quicker than with check)
         cdivision=True,
+        cpow=True,
         # Embed a textual copy of the call signature in the docstring (to support tools like IPython)
         embedsignature=True,
         fast_getattr=True,
         # Use Python 3 (including source code semantics) for module compilation
         language_level="3",
+        legacy_implicit_noexcept=True,
         # Enable support for late includes (make declarations in Cython code available to C include files)
         preliminary_late_includes_cy28=True,
         # Add hooks for Python profilers into the compiled C code
