--- a/GNUmakefile	2012-03-06 13:59:51.705509979 +1300
+++ b/GNUmakefile	2012-03-06 14:02:13.383739263 +1300
@@ -15,9 +15,9 @@
 MORI_OBJ= $(MORI_SRC:.c=.o)
 
 
-CC=gcc
+CC?=gcc
 
-CFLAGS=-O3 -g -W -Wall -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE
+MYCFLAGS=$(CPPFLAGS) $(CFLAGS) -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE
 # CFLAGS=-O3 -g				      # add -g for GNU debugger gdb
 # CFLAGS=-Ofast -O3 -mips4 -n32		      # SGI / 32 bit
 # CFLAGS=-Ofast -O3 -mips4 -64                # SGI / 64 bit
@@ -35,19 +35,19 @@
 
 
 poly:	poly.o $(OBJECTS) LG.o Global.h LG.h
-	$(CC) $(CFLAGS) -o poly.x poly.o $(OBJECTS) LG.o
+	$(CC) $(MYCFLAGS) $(LDFLAGS) -o poly.x poly.o $(OBJECTS) LG.o
 
 class:	class.o $(OBJECTS) $(CLASS_OBJ) Global.h Subpoly.h
-	$(CC) $(CFLAGS) -o class.x class.o $(OBJECTS) $(CLASS_OBJ)
+	$(CC) $(MYCFLAGS) $(LDFLAGS) -o class.x class.o $(OBJECTS) $(CLASS_OBJ)
 
 cws:    cws.o $(OBJECTS) LG.o Global.h LG.h
-	$(CC) $(CFLAGS) -o cws.x cws.o $(OBJECTS) LG.o
+	$(CC) $(MYCFLAGS) $(LDFLAGS) -o cws.x cws.o $(OBJECTS) LG.o
 
 nef:    nef.o $(OBJECTS) $(NEF_OBJ) Global.h 
-	$(CC) $(CFLAGS) -o nef.x nef.o $(OBJECTS) $(NEF_OBJ)
+	$(CC) $(MYCFLAGS) $(LDFLAGS) -o nef.x nef.o $(OBJECTS) $(NEF_OBJ)
 	
 mori:   mori.o  $(OBJECTS) $(MORI_OBJ) LG.o Mori.h 
-	$(CC) $(CFLAGS) -o mori.x mori.o $(OBJECTS) $(MORI_OBJ) LG.o
+	$(CC) $(MYCFLAGS) $(LDFLAGS) -o mori.x mori.o $(OBJECTS) $(MORI_OBJ) LG.o
 	
 
 
