--- GNUmakefile.orig	2010-08-18 22:49:55.489909197 +1200
+++ GNUmakefile	2010-08-18 22:54:43.263908035 +1200
@@ -11,13 +11,13 @@
 NEF_SRC= E_Poly.c Nefpart.c LG.c
 NEF_OBJ= $(NEF_SRC:.c=.o)
 
-CC=gcc
+CC?=gcc
 
-CFLAGS=-O3 -g -W -Wall -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE
-# CFLAGS=-O3 -g				      # add -g for GNU debugger gdb
-# CFLAGS=-Ofast -O3 -mips4 -n32		      # SGI / 32 bit
-# CFLAGS=-Ofast -O3 -mips4 -64                # SGI / 64 bit
-# CFLAGS=-O3 -fast -non_shared -arch pca56    # statically linked for alpha_PC
+MYCFLAGS=$(CFLAGS) -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE
+# MYCFLAGS=-O3 -g				      # add -g for GNU debugger gdb
+# MYCFLAGS=-Ofast -O3 -mips4 -n32		      # SGI / 32 bit
+# MYCFLAGS=-Ofast -O3 -mips4 -64                # SGI / 64 bit
+# MYCFLAGS=-O3 -fast -non_shared -arch pca56    # statically linked for alpha_PC
 
 #   targets : dependencies ; command
 #             command
@@ -31,16 +31,16 @@
 
 
 poly:	poly.o  $(OBJECTS) LG.o  Global.h LG.h
-	$(CC)   $(CFLAGS) -o  poly.x  poly.o  $(OBJECTS) LG.o
+	$(CC)   $(MYCFLAGS) $(LDFLAGS) -o  poly.x  poly.o  $(OBJECTS) LG.o
 
 class:	class.o $(OBJECTS) $(CLASS_OBJ) Global.h Subpoly.h
-	$(CC)   $(CFLAGS) -o class.x  class.o $(OBJECTS) $(CLASS_OBJ)
+	$(CC)   $(MYCFLAGS) $(LDFLAGS) -o class.x  class.o $(OBJECTS) $(CLASS_OBJ)
 
 cws:    cws.o   $(OBJECTS) LG.o Global.h LG.h
-	$(CC)   $(CFLAGS) -o cws.x  cws.o $(OBJECTS) LG.o
+	$(CC)   $(MYCFLAGS) $(LDFLAGS) -o cws.x  cws.o $(OBJECTS) LG.o
 
 nef:    nef.o   $(OBJECTS) $(NEF_OBJ) Global.h 
-	$(CC)   $(CFLAGS) -o  nef.x  nef.o  $(OBJECTS) $(NEF_OBJ)
+	$(CC)   $(MYCFLAGS) $(LDFLAGS) -o  nef.x  nef.o  $(OBJECTS) $(NEF_OBJ)
 
 
 #			     D E P E N D E N C I E S
@@ -66,9 +66,9 @@
 #	experimental stuff ...
 
 mori:	mori.o  $(OBJECTS) Global.h
-	$(CC)   $(CFLAGS) -o  mori.x  mori.o  $(OBJECTS)
+	$(CC)   $(MYCFLAGS) $(LDFLAGS) -o  mori.x  mori.o  $(OBJECTS)
 
 mori.o: mori.c	Global.h
 
 lgo:	lgotwist.c
-	$(CC)   $(CFLAGS) -o lgo.x lgotwist.c
+	$(CC)   $(MYCFLAGS) $(LDFLAGS) -o lgo.x lgotwist.c
