--- Global.h.old	2012-03-06 12:30:48.638575575 +1300
+++ Global.h	2012-03-06 12:31:29.790611566 +1300
@@ -19,7 +19,7 @@
 necessary to set even Long to 64 bits.
 */
 
-#define   		POLY_Dmax	6	/* max dim of polytope	    */
+/* #define   		POLY_Dmax	6	 max dim of polytope	    */
 /*
 POLY_Dmax should be set to the dimension of the polytopes that are analysed.
 While the programs still work if POLY_Dmax is set to a higher value, they may
