diff --git a/sphinx/ext/graphviz.py b/sphinx/ext/graphviz.py
index c9b1541..8de03ab 100644
--- a/sphinx/ext/graphviz.py
+++ b/sphinx/ext/graphviz.py
@@ -410,7 +410,7 @@ def man_visit_graphviz(self, node):
 
 def on_build_finished(app, exc):
     # type: (Sphinx, Exception) -> None
-    if exc is None:
+    if exc is None and not path.sep + 'inventory' + path.sep in app.outdir:
         src = path.join(sphinx.package_dir, 'templates', 'graphviz', 'graphviz.css')
         dst = path.join(app.outdir, '_static')
         copy_asset(src, dst)
