From 952bd16f81a8201bef77089607be00955c2a12fb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fran=C3=A7ois=20Bissey?= <frp.bissey@gmail.com>
Date: Wed, 19 Feb 2020 12:32:54 +1300
Subject: [PATCH] sane detection for standard paths installation of flint

---
 m4/flint-check.m4 | 53 +++++++++++++++++++++++++++++++++--------------
 1 file changed, 37 insertions(+), 16 deletions(-)

diff --git a/m4/flint-check.m4 b/m4/flint-check.m4
index d1616e46d3..cc7663e034 100644
--- a/m4/flint-check.m4
+++ b/m4/flint-check.m4
@@ -13,7 +13,7 @@ dnl FLINT_CFLAGS and FLINT_LIBS
 
 AC_DEFUN([LB_CHECK_FLINT],
 [
-DEFAULT_CHECKING_PATH="/usr /usr/local /sw /opt/local"
+DEFAULT_CHECKING_PATH="/sw /opt/local"
 
 AC_ARG_WITH(flint,
 [  --with-flint=<path>|yes|no  Use FLINT library. If argument is no, you do not have
@@ -40,26 +40,47 @@ fi
 
 AC_LANG_PUSH([C])
 
-for FLINT_HOME in ${FLINT_HOME_PATH}
- do
- if test -r "$FLINT_HOME/include/flint/fmpz.h"; then
-
-	FLINT_CFLAGS="-I${FLINT_HOME}/include/"
-	FLINT_LIBS="-L${FLINT_HOME}/lib -lflint -lmpfr"
+flint_found="no"
+dnl check for system installed libraries if FLINT_HOME_PATH is the default
+if test "$FLINT_HOME_PATH" = "$DEFAULT_CHECKING_PATH" ; then
+	FLINT_CFLAGS=""
+	FLINT_LIBS="-lflint -lmpfr"
 
 	# we suppose that mpfr and mpir to be in the same place or available by default
-	CFLAGS="${BACKUP_CFLAGS} ${FLINT_CFLAGS} ${GMP_CPPFLAGS}"
+	CFLAGS="${BACKUP_CFLAGS} ${GMP_CPPFLAGS}"
 	LIBS="${FLINT_LIBS} ${GMP_LIBS} ${BACKUPLIBS}"
 
-	AC_CHECK_LIB(flint,fmpz_init,
-	[flint_found="yes"],
-	[flint_found="mo"],
-	[]
-	)
-else
-	flint_found="no"
+	AC_CHECK_HEADER([flint/fmpz.h],
+		[AC_CHECK_LIB(flint,fmpz_init,
+			[flint_found="yes"],
+			[],
+			[])],
+		[],
+		[])
 fi
-done
+
+dnl if flint was not previously found, search FLINT_HOME_PATH
+if test "x$flint_found" = "xno" ; then
+	for FLINT_HOME in ${FLINT_HOME_PATH}
+	do
+		if test -r "$FLINT_HOME/include/flint/fmpz.h"; then
+
+		FLINT_CFLAGS="-I${FLINT_HOME}/include/"
+		FLINT_LIBS="-L${FLINT_HOME}/lib -lflint -lmpfr"
+
+	# we suppose that mpfr and mpir to be in the same place or available by default
+		CFLAGS="${BACKUP_CFLAGS} ${FLINT_CFLAGS} ${GMP_CPPFLAGS}"
+		LIBS="${FLINT_LIBS} ${GMP_LIBS} ${BACKUPLIBS}"
+
+		AC_CHECK_LIB(flint,fmpz_init,
+		[flint_found="yes"],
+		[],
+		[]
+		)
+		fi
+	done
+fi
+
 AC_LANG_POP([C])
 
 CFLAGS=${BACKUP_CFLAGS}
