From e180f5613c396d33aa3316cd450d3a1e80d69129 Mon Sep 17 00:00:00 2001
From: Hans Schoenemann <hannes@mathematik.uni-kl.de>
Date: Wed, 31 Aug 2016 09:55:25 +0200
Subject: [PATCH] fix typo in xalloc (Jean-Pierre Flori)

---
 xalloc/omalloc.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xalloc/omalloc.h b/xalloc/omalloc.h
index d905d97..b907ecc 100644
--- a/xalloc/omalloc.h
+++ b/xalloc/omalloc.h
@@ -66,7 +66,7 @@ typedef struct omOpts_s omOpts_t;
 extern int om_sing_opt_show_mem;
 
 static inline void * omalloc(size_t s)
-{ if (d!=0) {long *d=(long*)malloc(s+sizeof(long)); *d=s;d++;return d; }
+{ if (s!=0) {long *d=(long*)malloc(s+sizeof(long)); *d=s;d++;return d; }
   else return NULL;
 }
 static inline void * omAlloc(size_t s)
