--- factory/GNUmakefile.in.orig	2014-11-17 23:41:27.061615492 +1300
+++ factory/GNUmakefile.in	2014-11-17 23:42:00.371690221 +1300
@@ -99,7 +99,7 @@
 		$(DEFS) $(CPPFLAGS) $(CFLAGS) $(FLINT_CFLAGS)
 
 # flags to translate library c++ files
-LIBCXXFLAGS =	$(WARNFLAGS) -fno-implicit-templates \
+LIBCXXFLAGS =	$(WARNFLAGS) \
 		-I. -I.. -I$(srcdir) -I${prefix} -I$(includedir) \
 		$(DEFS) $(CPPFLAGS) $(CXXFLAGS) $(FLINT_CFLAGS)
 
