From 10ccdf4ec3a17fc1e6b47d06f9abf4565ae63341 Mon Sep 17 00:00:00 2001
From: Hans Schoenemann <hannes@mathematik.uni-kl.de>
Date: Mon, 5 Aug 2019 16:20:59 +0200
Subject: [PATCH] fix: factory/globaldefs.h for ingular.jl

---
 factory/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/factory/Makefile.am b/factory/Makefile.am
index 63174b6bb7..34c2994709 100644
--- a/factory/Makefile.am
+++ b/factory/Makefile.am
@@ -100,7 +100,6 @@ nodist_libfactory_la_SOURCES = factory.h factoryconf.h
 
 # factory header files
 factory_headers = \
-		globaldefs.h \
 		cf_assert.h \
 		canonicalform.h \
 		cf_algorithm.h \
@@ -174,6 +173,7 @@ noinst_HEADERS = $(factory_headers)
 libfactory_includedir = ${includedir}/factory
 
 nodist_libfactory_include_HEADERS = factory.h factoryconf.h
+libfactory_include_HEADERS = globaldefs.h
 
 
 ####################################################
