diff --git a/m4/options.m4 b/m4/options.m4
index fa91f635a..6443a4267 100644
--- a/m4/options.m4
+++ b/m4/options.m4
@@ -172,9 +172,13 @@ AC_DEFUN([SING_USE_RESOURCES],
   AC_SUBST(RESOURCES_INCLUDES)
   AC_SUBST(RESOURCES_LIBS)
 
+  if test "x$htmldir" = "x\${docdir}"; then
+    DEFAULT_HTMLDIR="--htmldir=$datarootdir/doc/singular"
+  fi
+
   ENABLE_ARG="--with-Singular RESOURCES_LIBS='$RESOURCES_LIBS' RESOURCES_INCLUDES='$RESOURCES_INCLUDES'"
 
-  ac_configure_args="$ac_configure_args $ENABLE_ARG"
+  ac_configure_args="$ac_configure_args $DEFAULT_HTMLDIR $ENABLE_ARG"
 
   PKG_REQUIRE="$PKG_REQUIRE singular_resources"
   AC_SUBST(PKG_REQUIRE)
diff --git a/resources/configure.ac b/resources/configure.ac
index 7e95b25c4..834e84a16 100644
--- a/resources/configure.ac
+++ b/resources/configure.ac
@@ -72,5 +72,10 @@ AX_RECURSIVE_EVAL([[$]datadir], [config_datadir])
 AX_NORMALIZE_PATH([config_datadir],['/'])
 AC_DEFINE_UNQUOTED([DATA_DIR],"$config_datadir",[datadir])
 
+AX_RECURSIVE_EVAL([[$]htmldir], [config_htmldir])
+AX_NORMALIZE_PATH([config_htmldir],['/'])
+AX_COMPUTE_RELATIVE_PATHS([config_datadir:config_htmldir:data_to_html])
+AC_DEFINE_UNQUOTED([DATA_TO_HTML_DIR],"%D/$data_to_html",[htmldir])
+
 AC_CONFIG_FILES([singular_resources.pc Makefile])
 AC_OUTPUT
diff --git a/resources/feResource.cc b/resources/feResource.cc
index 0eaa6fa6a..6694041c3 100644
--- a/resources/feResource.cc
+++ b/resources/feResource.cc
@@ -78,7 +78,7 @@ VAR feResourceConfig_s feResourceConfigs[] =
   {"DefaultDir",'d',    feResDir,   "SINGULAR_DEFAULT_DIR",  SINGULAR_DEFAULT_DIR,  (char *)""},
   {"InfoFile",  'i',    feResFile,  "SINGULAR_INFO_FILE",   "%D/info/singular.info", (char *)""},
   {"IdxFile",   'x',    feResFile,  "SINGULAR_IDX_FILE",    "%D/singular/singular.idx",  (char *)""},
-  {"HtmlDir",   'h',    feResDir,   "SINGULAR_HTML_DIR",    "%D/singular/html",              (char *)""},
+  {"HtmlDir",   'h',    feResDir,   "SINGULAR_HTML_DIR",    DATA_TO_HTML_DIR,       (char *)""},
   {"ManualUrl", 'u',    feResUrl,   "SINGULAR_URL",         "https://www.singular.uni-kl.de/Manual/",    (char *)""},
   {"ExDir",     'm',    feResDir,   "SINGULAR_EXAMPLES_DIR","%r/examples",          (char *)""},
   {"Path",      'p',    feResPath,  NULL,                   "%b;%P;$PATH",             (char *)""},
diff --git a/Makefile.am b/Makefile.am
index dd8957d59..829e291b5 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -17,9 +17,11 @@ bin_SCRIPTS = libsingular-config
 install-data-local:
 	if test -e $(srcdir)/doc/doc.tbz2; then\
 	  (cat ${top_srcdir}/doc/doc.tbz2| (cd $(DESTDIR)$(datadir)/singular; tar -jx --no-same-owner -f -));\
-	   mkdir $(DESTDIR)$(datadir)/info;\
-	   mkdir $(DESTDIR)$(datadir)/doc;\
+	   $(INSTALL) -d -v $(DESTDIR)$(datadir)/info;\
+	   $(INSTALL) -d -v $(DESTDIR)$(datadir)/doc;\
+	   $(INSTALL) -d -v `dirname $(DESTDIR)$(htmldir)`;\
 	   mv  $(DESTDIR)$(datadir)/singular/singular.info  $(DESTDIR)$(datadir)/info/.;\
+	   mv  $(DESTDIR)$(datadir)/singular/html $(DESTDIR)$(htmldir);\
 	fi
 
 uninstall-local:
