Index: kernel/polys.cc
===================================================================
--- a/kernel/polys.cc	(revision 1710703a348a673f5f83775001fd2d284d452af0)
+++ b/kernel/polys.cc	(revision 55ec4f789df5836f21154a2d6e25c0e9cb8cf814)
@@ -181,6 +181,7 @@
   else         return f; // g==0 => gcd=f (but do a p_Cleardenom/pNorm)
   if (f==NULL) return g; // f==0 => gcd=g (but do a p_Cleardenom/pNorm)
-  if(p_IsConstant(f,currRing)
-  ||p_IsConstant(g,currRing))
+  if(!rField_is_Ring(currRing)
+  && (p_IsConstant(f,currRing)
+  ||p_IsConstant(g,currRing)))
   {
     res=p_One(currRing);
