From 38cff47157a8f073de2a7e81fd00557a8a150222 Mon Sep 17 00:00:00 2001
From: Jeroen Demeyer <jdemeyer@cage.ugent.be>
Date: Wed, 17 Aug 2016 13:44:43 +0200
Subject: [PATCH] Replace "int(val) == 0" by "not val" for gdb.Value instances

---
 Cython/Debugger/libpython.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Cython/Debugger/libpython.py b/Cython/Debugger/libpython.py
index b43b83c..2158c1f 100644
--- a/Cython/Debugger/libpython.py
+++ b/Cython/Debugger/libpython.py
@@ -286,7 +286,7 @@ def type(self):
         return PyTypeObjectPtr(self.field('ob_type'))
 
     def is_null(self):
-        return 0 == int(self._gdbval)
+        return not self._gdbval
 
     def is_optimized_out(self):
         '''
@@ -989,7 +989,7 @@ def current_line_num(self):
         if self.is_optimized_out():
             return None
         f_trace = self.field('f_trace')
-        if int(f_trace) != 0:
+        if f_trace:
             # we have a non-NULL f_trace:
             return self.f_lineno
         else:
