From 116faa4bd66f0db17e011a767701cc78f220792a Mon Sep 17 00:00:00 2001
From: Jeroen Demeyer <jdemeyer@cage.ugent.be>
Date: Mon, 22 Oct 2018 17:27:05 +0200
Subject: [PATCH 1/2] PYTHONPATH: do not add trailing pathsep

---
 runtests.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/runtests.py b/runtests.py
index e30c5982f8..69cc1db7ce 100755
--- a/runtests.py
+++ b/runtests.py
@@ -1734,7 +1734,10 @@ def runTest(self):
             .replace("PYTHON", sys.executable))
         old_path = os.environ.get('PYTHONPATH')
         env = dict(os.environ)
-        env['PYTHONPATH'] = self.cython_syspath + os.pathsep + (old_path or '')
+        new_path = self.cython_syspath
+        if old_path:
+            new_path = new_path + os.pathsep + old_path
+        env['PYTHONPATH'] = new_path
         cmd = []
         out = []
         err = []

From d46b06fc0f98ff23a4d5736505258c6c335bdcc7 Mon Sep 17 00:00:00 2001
From: Jeroen Demeyer <jdemeyer@cage.ugent.be>
Date: Mon, 22 Oct 2018 17:32:18 +0200
Subject: [PATCH 2/2] Add cwd to sys.path inside subinterpreter

---
 tests/run/reimport_from_subinterpreter.srctree | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/tests/run/reimport_from_subinterpreter.srctree b/tests/run/reimport_from_subinterpreter.srctree
index 98b1599814..22b62517ec 100644
--- a/tests/run/reimport_from_subinterpreter.srctree
+++ b/tests/run/reimport_from_subinterpreter.srctree
@@ -59,11 +59,14 @@ def run_sub():
     assert 0 == run_in_subinterpreter(b'1+1')
     assert 0 == run_in_subinterpreter(b'2+2')
 
-    assert 0 == run_in_subinterpreter(b'import package')
-    assert 0 == run_in_subinterpreter(b'import package')
+    # The subinterpreter does not add the current working directory to
+    # sys.path, so we need to add it manually.
+    pre = b'import sys; sys.path.insert(0, "."); '
+    assert 0 == run_in_subinterpreter(pre + b'import package')
+    assert 0 == run_in_subinterpreter(pre + b'import package')
 
     import sys
-    result = run_in_subinterpreter(b'import package.subtest')
+    result = run_in_subinterpreter(pre + b'import package.subtest')
     if not MAIN_HAS_IMPORTED:
         assert result == 0, result  # imports only in subinterpreters are ok
     elif sys.version_info >= (3, 5):
