Index: src/library/utils/R/packages2.R
===================================================================
--- src/library/utils/R/packages2.R	(revision 58532)
+++ src/library/utils/R/packages2.R	(working copy)
@@ -229,8 +229,8 @@
                 domain = NA, immediate. = TRUE)
         userdir <- unlist(strsplit(Sys.getenv("R_LIBS_USER"),
                                    .Platform$path.sep))[1L]
-        if(interactive() && !file.exists(userdir)) {
-            msg <- gettext("Would you like to create a personal library\n%s\nto install packages into?")
+        if(interactive()) {
+            msg <- gettext("Would you like to use a personal library instead?")
             if(.Platform$OS.type == "windows") {
                 ans <- winDialog("yesno", sprintf(msg, sQuote(userdir)))
                 if(ans != "YES") stop("unable to install packages")
@@ -239,10 +239,21 @@
                 if(substr(ans, 1L, 1L) == "n")
                     stop("unable to install packages")
             }
-            if(!dir.create(userdir, recursive = TRUE))
-                stop("unable to create ", sQuote(userdir))
             lib <- userdir
-            .libPaths(c(userdir, .libPaths()))
+            if(!file.exists(userdir)) {
+                msg <- gettext("Would you like to create a personal library\n%s\nto install packages into?")
+                if(.Platform$OS.type == "windows") {
+                    ans <- winDialog("yesno", sprintf(msg, sQuote(userdir)))
+                    if(ans != "YES") stop("unable to install packages")
+                } else {
+                    ans <- readline(paste(sprintf(msg, userdir), " (y/n) "))
+                    if(substr(ans, 1L, 1L) == "n")
+                        stop("unable to install packages")
+                }
+                if(!dir.create(userdir, recursive = TRUE))
+                    stop("unable to create ", sQuote(userdir))
+                .libPaths(c(userdir, .libPaths()))
+            }
         } else stop("unable to install packages")
     }
 
