Fix for pcre-8.30 and above

--- src/main/util.c.orig	2012-02-09 17:15:08.000000000 +0000
+++ src/main/util.c	2012-02-09 17:20:31.000000000 +0000
@@ -1257,8 +1257,18 @@
 }
 
 #include "pcre.h"
+
+/* This changed at 8.30 */
+#if  PCRE_MAJOR > 8 || PCRE_MINOR >= 30
+extern int _pcre_valid_utf(const char *string, int length, int *erroroffset);
+
+Rboolean utf8Valid(const char *str)
+{
+    int errp;
+    return  (_pcre_valid_utf(str, (int) strlen(str), &errp) == 0);
+}
 /* This changed at 8.13: we don't allow < 8.0 */
-#if  PCRE_MAJOR > 8 || PCRE_MINOR >= 13
+#elif  PCRE_MAJOR > 8 || PCRE_MINOR >= 13
 extern int _pcre_valid_utf8(const char *string, int length, int *erroroffset);
 
 Rboolean utf8Valid(const char *str)
@@ -1266,7 +1276,7 @@
     int errp;
     return  (_pcre_valid_utf8(str, (int) strlen(str), &errp) == 0);
 }
-#else
+#else 
 extern int _pcre_valid_utf8(const char *string, int length);
 
 Rboolean utf8Valid(const char *str)
