Index: gap-atlasrep-1.5.1/init.g
===================================================================
--- gap-atlasrep-1.5.1.orig/init.g	2016-04-10 21:53:27.251489956 +0200
+++ gap-atlasrep-1.5.1/init.g	2016-04-16 22:18:27.033865113 +0200
@@ -5,6 +5,22 @@
 #Y  Copyright (C)  2001,   Lehrstuhl D für Mathematik,  RWTH Aachen,  Germany
 ##
 
+if UserPreference( "AtlasRep", "AtlasRepDataDirectory") = fail then
+  Perform(["datagens", "dataword"], function(dirname)
+    local package, l, dir;
+    package := "AtlasRep";
+    dir := GAPInfo.UserGapRoot;
+    for l in ["","pkg",package,dirname] do
+      dir := Concatenation([dir,"/",l]);
+      if not IsDirectoryPath(dir) then
+        CreateDir(dir);
+      fi;
+    od;
+    return 0;
+  end);
+  SetUserPreference( "AtlasRep", "AtlasRepDataDirectory",Concatenation(GAPInfo.UserGapRoot,"/pkg/AtlasRep/"));
+fi;
+
 # Read the declaration part.
 ReadPackage( "atlasrep", "gap/userpref.g"  );
 ReadPackage( "atlasrep", "gap/bbox.gd"     );

