--- makemakefile.py.orig	2010-04-25 07:14:12.665540999 +0000
+++ makemakefile.py	2010-04-25 07:17:13.284542466 +0000
@@ -99,7 +99,7 @@
 prefix = options.prefix
 
 includes = "-I" + gmp_include_dir + " -I./include"
-libs = "-L" + gmp_lib_dir + " -lgmp -lm"
+libs = "-L" + gmp_lib_dir + " -lmpir -lm"
 
 if options.use_flint:
    includes = includes + " -I" + flint_include_dir
--- include/support.h.orig	2010-04-25 07:15:15.233540859 +0000
+++ include/support.h	2010-04-25 07:16:14.276542886 +0000
@@ -25,7 +25,7 @@
 
 
 #include <stdio.h>
-#include <gmp.h>
+#include <mpir.h>
 #include "zn_poly_internal.h"
 
 
--- include/zn_poly_internal.h.orig	2010-04-25 07:15:33.437540930 +0000
+++ include/zn_poly_internal.h	2010-04-25 07:15:59.023542606 +0000
@@ -35,7 +35,7 @@
 #ifndef ZN_POLY_INTERNAL_H
 #define ZN_POLY_INTERNAL_H
 
-#include <gmp.h>
+#include <mpir.h>
 #include <stddef.h>
 #include <stdio.h>
 #include "zn_poly.h"
