--- makemakefile.py.orig	2010-02-17 22:16:49.049098999 +1300
+++ makemakefile.py	2010-02-17 22:21:26.027098608 +1300
@@ -76,6 +76,8 @@
 parser.add_option("--cflags", dest="cflags", default="-O2")
 parser.add_option("--ldflags", dest="ldflags", default="")
 parser.add_option("--gmp-prefix", dest="gmp_prefix", default="/usr/local")
+parser.add_option("--use-mpir", dest="use_mpir", action="store_true",
+                  default=False)
 parser.add_option("--ntl-prefix", dest="ntl_prefix", default="/usr/local")
 parser.add_option("--use-flint", dest="use_flint", action="store_true",
                   default=False)
@@ -99,7 +101,11 @@
 prefix = options.prefix
 
 includes = "-I" + gmp_include_dir + " -I./include"
-libs = "-L" + gmp_lib_dir + " -lgmp -lm"
+if options.use_mpir:
+   libs = "-L" + gmp_lib_dir + " -lmpir -lm"
+   cflags = cflags + " -DUSE_MPIR"
+else:
+   libs = "-L" + gmp_lib_dir + " -lgmp -lm"
 
 if options.use_flint:
    includes = includes + " -I" + flint_include_dir
--- include/zn_poly_internal.h.orig	2010-02-17 22:23:09.574098889 +1300
+++ include/zn_poly_internal.h	2010-02-17 22:26:32.229100863 +1300
@@ -35,7 +35,11 @@
 #ifndef ZN_POLY_INTERNAL_H
 #define ZN_POLY_INTERNAL_H
 
+#ifdef USE_MPIR
+#include <mpir.h>
+#else
 #include <gmp.h>
+#endif
 #include <stddef.h>
 #include <stdio.h>
 #include "zn_poly.h"
--- include/support.h.orig	2010-02-17 22:24:55.181097909 +1300
+++ include/support.h	2010-02-17 22:27:09.966099595 +1300
@@ -25,7 +25,11 @@
 
 
 #include <stdio.h>
+#ifdef USE_MPIR
+#include <mpir.h>
+#else
 #include <gmp.h>
+#endif
 #include "zn_poly_internal.h"
 
 
