diff --git a/setup.py b/setup.py
index 28d9ccc..f25e1dd 100755
--- a/setup.py
+++ b/setup.py
@@ -14,6 +14,7 @@ except ImportError:
 
 import sys, os
 import numpy as np
+import cypari2
 
 from distutils.core import setup
 from distutils.extension import Extension
@@ -50,13 +51,6 @@ except ImportError:
 else:
     WITH_PPL = True
 
-try:
-    import sage.modular.multiple_zeta
-except ImportError:
-    sys.stderr.write('Warning: multiple_zeta not available in Sage\n')
-    WITH_MZV = False
-else:
-    WITH_MZV = True
 
 extensions_data = {
     'origamis': {
@@ -108,19 +102,15 @@ for name, data in extensions_data.items():
         source_files.extend(sources)
         source_files.extend(headers)
 
-class build_py(_build_py):
-    r"""
-    Custom build_py command to not install generalized_multiple_zeta_values when
-    corresponding sage module not present.
-    """
-    def find_package_modules(self, package, package_dir):
-        modules = _build_py.find_package_modules(self, package, package_dir)
-        if not WITH_MZV and package == 'surface_dynamics/misc':
-            modules = [
-                (pkg, mod, file)
-                for (pkg, mod, file) in modules
-                if mod != 'generalized_multiple_zeta_values']
-        return modules
+# Work around changes in SageMath 9.7, see https://trac.sagemath.org/wiki/ReleaseTours/sage-9.7#Packagessagesage.rings...arenownamespaces
+try:
+    from sage.misc.package_dir import cython_namespace_package_support
+except (ImportError, ModuleNotFoundError):
+    from contextlib import nullcontext as cython_namespace_package_support
+
+with cython_namespace_package_support():
+    extensions = cythonize(extensions)
+
 
 setup(name='surface_dynamics',
       version=version,
@@ -143,7 +133,7 @@ setup(name='surface_dynamics',
           'surface_dynamics/databases': ['cylinder_diagrams/cyl_diags*', 'generalized_permutation_twins/twins*'],
           'surface_dynamics/flat_surfaces/origamis': ['origamis.db'],
           },
-    ext_modules=cythonize(extensions),
+    ext_modules=extensions,
     classifiers=[
       'Development Status :: 4 - Beta',
       'Intended Audience :: Science/Research',
@@ -156,5 +146,4 @@ setup(name='surface_dynamics',
       'Topic :: Scientific/Engineering :: Mathematics',
     ],
     keywords='surfaces, dynamics, geometry, flat surfaces, Abelian differentials, quadratic differentials, Riemann surfaces',
-    cmdclass={'build_py':build_py}
 )
