diff --git a/sage_docbuild/ext/sage_autodoc.py b/sage_docbuild/ext/sage_autodoc.py
index 399ac1532e..7f2e39f927 100644
--- a/sage_docbuild/ext/sage_autodoc.py
+++ b/sage_docbuild/ext/sage_autodoc.py
@@ -35,7 +35,8 @@ from docutils.statemachine import ViewList
 
 import sphinx
 from sphinx.ext.autodoc import mock, ObjectMember
-from sphinx.ext.autodoc.importer import import_object, get_object_members, get_module_members
+from sphinx.ext.autodoc.ModuleDocumenter import get_module_members
+from sphinx.ext.autodoc.importer import import_object, get_object_members
 from sphinx.locale import _, __
 from sphinx.pycode import ModuleAnalyzer
 from sphinx.errors import PycodeError
