diff --git a/sage/libs/polybori/decl.pxd b/sage/libs/polybori/decl.pxd
index dd6a3aa..edb7bb7 100644
--- a/sage/libs/polybori/decl.pxd
+++ b/sage/libs/polybori/decl.pxd
@@ -1,5 +1,5 @@
 # distutils: language = c++
-# distutils: extra_compile_args = -std=c++11
+# distutils: extra_compile_args = -std=c++11 -fno-delete-null-pointer-checks
 
 from libcpp.string cimport string as std_string
 from libcpp.vector cimport vector
