diff --git a/sage/interfaces/giac.py b/sage/interfaces/giac.py
index 68c9527..3c5fd12 100644
--- a/sage/interfaces/giac.py
+++ b/sage/interfaces/giac.py
@@ -990,7 +990,7 @@ class GiacElement(ExpectElement):
         EXAMPLES::
 
             sage: print(latex(giac('(x^4 - y)/(y^2-3*x)')))
-            "\frac{(x^{4}-y)}{(y^{2}-3\cdot x)}"
+            "\frac{x^{4}-y}{y^{2}-3 x}"
 
         """
         return self.parent().eval('latex(%s)'%self.name())
