--- patches/Makefile.sage	2008-04-14 03:07:10.000000000 +0000
+++ patches/Makefile.sage-copy	2009-10-29 20:28:57.142049936 +0000
@@ -1,6 +1,6 @@
 lcalc: cmdline.c
 	$(CXX) $(CFLAGS) -DINCLUDE_PARI   $(DEFINES)\
-      -I$(SAGE_LOCAL)/include/pari -I$(SAGE_LOCAL)/include\
+      -I/usr/include/pari -I$(SAGE_LOCAL)/include\
       -I ../include/ -L$(SAGE_LOCAL)/lib \
       cmdline.c \
       Lcommandline.cc Lcommandline_elliptic.cc Lcommandline_globals.cc \
