--- spkg-install	2009-10-26 18:42:55.639683224 +0000
+++ spkg-install.new	2009-10-26 18:45:10.880683193 +0000
@@ -91,7 +91,7 @@
         exit 1
     fi
 
-    cp -f gp "$SAGE_LOCAL/bin/"    # avoid "make install"
+    $MAKE install-bin
 
     # Install documentation source (compressed), for use by gphelp.
     $MKDIR -p "$SAGE_LOCAL/share/pari/doc/"
@@ -146,7 +146,7 @@
         fi
 
         cd `config/objdir`
-        cp -f libpari* $SAGE_LOCAL/lib
+        $MAKE install-lib-dyn
         cd "$CUR"
 
         # Also another patch since paripriv.h won't compile right on OS X
