diff --git a/sage_setup/docbuild/__init__.py b/sage_setup/docbuild/__init__.py
index b8f79c2..5c15e8e 100644
--- a/sage_setup/docbuild/__init__.py
+++ b/sage_setup/docbuild/__init__.py
@@ -26,7 +26,7 @@ from sage.misc.cachefunc import cached_method
 from sage.misc.misc import sage_makedirs
 from sage.env import SAGE_DOC_SRC, SAGE_DOC, SAGE_SRC
 
-from .build_options import (LANGUAGES, SPHINXOPTS, PAPER, OMIT,
+from sage_setup.docbuild.build_options import (LANGUAGES, SPHINXOPTS, PAPER, OMIT,
      PAPEROPTS, ALLSPHINXOPTS, NUM_THREADS, WEBSITESPHINXOPTS,
      INCREMENTAL_BUILD, ABORT_ON_ERROR)
 
diff --git a/sage_setup/docbuild/__main__.py b/sage_setup/docbuild/__main__.py
index 031df43..e68b438 100644
--- a/sage_setup/docbuild/__main__.py
+++ b/sage_setup/docbuild/__main__.py
@@ -1,2 +1,2 @@
-from . import main
+from sage_setup.docbuild import main
 main()
diff --git a/setup.py b/setup.py
index 5abd6dc..91a64b0 100755
--- a/setup.py
+++ b/setup.py
@@ -865,7 +865,7 @@ print("Discovering Python/Cython source code....")
 t = time.time()
 from sage_setup.find import find_python_sources
 python_packages, python_modules = find_python_sources(
-    SAGE_SRC, ['sage', 'sage_setup'])
+    SAGE_SRC, ['sage'])
 
 log.debug('python_packages = {0}'.format(python_packages))
 
@@ -880,10 +880,6 @@ class sage_install(install):
     def run(self):
         install.run(self)
         self.install_kernel_spec()
-        log.info('Cleaning up stale installed files....')
-        t = time.time()
-        self.clean_stale_files()
-        log.info('Finished cleaning, time: %.2f seconds.' % (time.time() - t))
 
     def install_kernel_spec(self):
         """
