--- module_list.py.orig	2010-06-26 16:23:29.532595646 +1200
+++ module_list.py	2010-06-26 16:49:44.259596047 +1200
@@ -113,7 +113,6 @@
 
     Extension('sage.algebras.quatalg.quaternion_algebra_element',
                sources = ['sage/algebras/quatalg/quaternion_algebra_element.pyx'],
-               extra_compile_args=["-std=c99"],
                language='c++',
                libraries = ["csage", "flint", "gmp", "gmpxx", "m", "stdc++", "ntl"],
                include_dirs = [SAGE_ROOT+'/local/include/FLINT/'],
