diff --git a/sage/doctest/control.py b/sage/doctest/control.py
index 96143a6..870511e 100644
--- a/sage/doctest/control.py
+++ b/sage/doctest/control.py
@@ -693,8 +693,11 @@ class DocTestController(SageObject):
         """
         opj = os.path.join
         from sage.env import SAGE_SRC, SAGE_DOC_SRC, SAGE_ROOT
-        DOT_GIT = opj(SAGE_ROOT, '.git')
-        have_git = os.path.exists(DOT_GIT)
+        if SAGE_ROOT is None:
+            have_git = False
+        else:
+            DOT_GIT = opj(SAGE_ROOT, '.git')
+            have_git = os.path.exists(DOT_GIT)
 
         def all_files():
             self.files.append(opj(SAGE_SRC, 'sage'))
@@ -1200,7 +1203,10 @@ class DocTestController(SageObject):
             self.test_safe_directory()
             self.create_run_id()
             from sage.env import SAGE_ROOT
-            DOT_GIT = os.path.join(SAGE_ROOT, '.git')
+            if SAGE_ROOT is None:
+                DOT_GIT = ""
+            else:
+                DOT_GIT = os.path.join(SAGE_ROOT, '.git')
             if os.path.isdir(DOT_GIT):
                 import subprocess
                 try:
diff --git a/sage/homology/simplicial_set_examples.py b/sage/homology/simplicial_set_examples.py
index f7f1c6d..e59d668 100644
--- a/sage/homology/simplicial_set_examples.py
+++ b/sage/homology/simplicial_set_examples.py
@@ -635,7 +635,7 @@ def simplicial_data_from_kenzo_output(filename):
     OUTPUT: data to construct a simplicial set from the Kenzo output
 
     Several files with Kenzo output are in the directory
-    :file:`SAGE_ROOT/src/ext/kenzo/`.
+    :file:`SAGE_EXTCODE/kenzo/`.
 
     EXAMPLES::
 
diff --git a/sage/misc/citation.pyx b/sage/misc/citation.pyx
index 86b850a..15f2647 100644
--- a/sage/misc/citation.pyx
+++ b/sage/misc/citation.pyx
@@ -5,7 +5,7 @@ Dependency usage tracking for citations
 from __future__ import absolute_import
 
 from sage.misc.all import tmp_filename
-from sage.env import SAGE_ROOT
+from sage.env import SAGE_LOCAL
 
 systems = {}
 systems['PARI'] = ['cypari2', 'sage.interfaces.gp']
@@ -110,7 +110,7 @@ def get_systems(cmd):
     stats = pstats.Stats(filename)
 
     #Strings is a list of method names and modules which get run
-    strings = [a[0].replace(SAGE_ROOT, "") + " " + a[2] for a in stats.stats.keys()]
+    strings = [a[0].replace(SAGE_LOCAL, "") + " " + a[2] for a in stats.stats.keys()]
 
     #Remove trivial functions
     bad_res = [re.compile(r'is_.*Element'), re.compile("is_[a-z_]*_type")]
diff --git a/sage/misc/inline_fortran.py b/sage/misc/inline_fortran.py
index fe2565c..092b504 100644
--- a/sage/misc/inline_fortran.py
+++ b/sage/misc/inline_fortran.py
@@ -141,12 +141,12 @@ class InlineFortran:
 
         TESTS::
 
-            sage: os.chdir(SAGE_ROOT)
+            sage: os.chdir(DOT_SAGE)
             sage: fortran.eval("SYNTAX ERROR !@#$")
             Traceback (most recent call last):
             ...
             RuntimeError: failed to compile Fortran code:...
-            sage: os.getcwd() == SAGE_ROOT
+            sage: os.getcwd() == DOT_SAGE
             True
         """
         if globals is None:
diff --git a/sage/misc/misc.py b/sage/misc/misc.py
index 8860669..939c53b 100644
--- a/sage/misc/misc.py
+++ b/sage/misc/misc.py
@@ -76,7 +76,7 @@ def sage_makedirs(dir):
     The following fails because we are trying to create a directory in
     place of an ordinary file (the main Sage executable)::
 
-        sage: sage_executable = os.path.join(SAGE_ROOT, 'sage')
+        sage: sage_executable = os.path.join(SAGE_LOCAL, 'bin', 'sage')
         sage: sage_makedirs(sage_executable)
         Traceback (most recent call last):
         ...
diff --git a/sage/misc/persist.pyx b/sage/misc/persist.pyx
index c493b20..3104e41 100644
--- a/sage/misc/persist.pyx
+++ b/sage/misc/persist.pyx
@@ -980,7 +980,7 @@ def picklejar(obj, dir=None):
     environment variable ``SAGE_PICKLE_JAR``, which will make it so
     :func:`dumps` will by default call :func:`picklejar` with the
     default dir.  Once you do that and doctest Sage, you'll find that
-    the ``SAGE_ROOT/tmp/pickle_jar`` directory contains a bunch of
+    the ``DOT_SAGE/tmp/pickle_jar`` directory contains a bunch of
     pickled objects along with corresponding txt descriptions of them.
     Use the :func:`unpickle_all` to see if they unpickle later.
 
@@ -989,7 +989,7 @@ def picklejar(obj, dir=None):
     - ``obj`` -- a pickleable object
 
     - ``dir`` -- a string or None; if None then ``dir`` defaults to
-      ``SAGE_ROOT/tmp/pickle_jar``
+      ``DOT_SAGE/tmp/pickle_jar``
 
     EXAMPLES::
 
@@ -1020,8 +1020,9 @@ def picklejar(obj, dir=None):
         OSError: ...
         sage: os.chmod(dir, 0o755)
     """
+    from sage.env import DOT_SAGE
     if dir is None:
-        dir = os.path.join(os.environ['SAGE_ROOT'], '/tmp/pickle_jar/')
+        dir = os.path.join(DOT_SAGE, '/tmp/pickle_jar/')
     try:
         os.makedirs(dir)
     except OSError as err:
