--- a/sage/doctest/control.py	2014-07-29 13:42:35.869541651 +1200
+++ b/sage/doctest/control.py	2014-07-29 13:43:42.389644902 +1200
@@ -415,12 +415,12 @@
             'sagenb'
         """
         opj = os.path.join
-        from sage.env import SAGE_SRC, SAGE_ROOT
+        from sage.env import SAGE_SRC, SAGE_DOC, SAGE_ROOT
         def all_files():
             from glob import glob
             self.files.append(opj(SAGE_SRC, 'sage'))
-            self.files.append(opj(SAGE_SRC, 'doc', 'common'))
-            self.files.extend(glob(opj(SAGE_SRC, 'doc', '[a-z][a-z]')))
+            self.files.append(opj(SAGE_DOC, 'common'))
+            self.files.extend(glob(opj(SAGE_DOC, '[a-z][a-z]')))
             self.options.sagenb = True
         if self.options.all:
             self.log("Doctesting entire Sage library.")
