--- module_list.py.orig	2014-07-23 10:24:33.471985622 +1200
+++ module_list.py	2014-07-23 10:25:19.962020237 +1200
@@ -4,6 +4,7 @@
 from distutils.core import setup
 from distutils.extension import Extension
 from sage.env import *
+import pkgconfig
 
 SAGE_INC = os.path.join(SAGE_LOCAL, 'include')
 
@@ -11,26 +12,8 @@
 ### BLAS setup
 #########################################################
 
-## Choose cblas library -- note -- make sure to update sage/misc/cython.py
-## if you change this!!
-if os.environ.has_key('SAGE_BLAS'):
-    BLAS=os.environ['SAGE_BLAS']
-    BLAS2=os.environ['SAGE_BLAS']
-elif os.path.exists('%s/lib/libatlas.so'%os.environ['SAGE_LOCAL']):
-    BLAS='cblas'
-    BLAS2='atlas'
-elif os.path.exists('/usr/lib/libcblas.dylib') or \
-     os.path.exists('/usr/lib/libcblas.so'):
-    BLAS='cblas'
-    BLAS2='cblas'
-elif os.path.exists('/usr/lib/libblas.dll.a'):
-    BLAS='gslcblas'
-    BLAS2='gslcblas'
-else:
-    # This is very slow  (?), but *guaranteed* to be available.
-    BLAS='gslcblas'
-    BLAS2='gslcblas'
-
+blas_libs = list(pkgconfig.parse('cblas')['libraries'])
+blas_dir  = list(pkgconfig.parse('cblas')['library_dir'])
 
 #########################################################
 ### Commonly used definitions
@@ -607,42 +590,50 @@
 
     Extension('sage.gsl.callback',
               sources = ['sage/gsl/callback.pyx'],
-              libraries = ['gsl', BLAS, BLAS2],
+              library_dirs = blas_dir,
+              libraries = ['gsl'] + blas_libs,
               define_macros=[('GSL_DISABLE_DEPRECATED','1')]),
 
     Extension('sage.gsl.dwt',
               sources = ['sage/gsl/dwt.pyx'],
-              libraries=['gsl',BLAS],
+              library_dirs = blas_dir,
+              libraries=['gsl'] + blas_libs,
               define_macros=[('GSL_DISABLE_DEPRECATED','1')]),
 
     Extension('sage.gsl.fft',
               sources = ['sage/gsl/fft.pyx'],
-              libraries = ['gsl', BLAS, BLAS2],
+              library_dirs = blas_dir,
+              libraries = ['gsl'] + blas_libs,
               define_macros=[('GSL_DISABLE_DEPRECATED','1')]),
 
     Extension('sage.gsl.gsl_array',
               sources = ['sage/gsl/gsl_array.pyx'],
-              libraries=['gsl', BLAS, BLAS2],
+              library_dirs = blas_dir,
+              libraries=['gsl'] + blas_libs,
               define_macros=[('GSL_DISABLE_DEPRECATED','1')]),
 
     Extension('sage.gsl.integration',
               sources = ['sage/gsl/integration.pyx'],
               define_macros=[('GSL_DISABLE_DEPRECATED','1')],
-              libraries=['gsl',BLAS, BLAS2]),
+              library_dirs = blas_dir,
+              libraries=['gsl'] + blas_libs),
 
     Extension('sage.gsl.interpolation',
               sources = ['sage/gsl/interpolation.pyx'],
-              libraries = ['gsl', BLAS, BLAS2],
+              library_dirs = blas_dir,
+              libraries = ['gsl'] + blas_libs,
               define_macros=[('GSL_DISABLE_DEPRECATED','1')]),
 
     Extension('sage.gsl.ode',
               sources = ['sage/gsl/ode.pyx'],
-              libraries=['gsl',BLAS, BLAS2],
+              library_dirs = blas_dir,
+              libraries=['gsl'] + blas_libs,
               define_macros=[('GSL_DISABLE_DEPRECATED','1')]),
 
     Extension('sage.gsl.probability_distribution',
               sources = ['sage/gsl/probability_distribution.pyx'],
-              libraries=['gsl', BLAS, BLAS2],
+              library_dirs = blas_dir,
+              libraries=['gsl'] + blas_libs,
               define_macros=[('GSL_DISABLE_DEPRECATED','1')]),
 
     ################################
@@ -695,10 +686,11 @@
 
     Extension('sage.libs.linbox.linbox',
               sources = ['sage/libs/linbox/linbox.pyx'],
+              library_dirs = blas_dir,
               # For this to work on cygwin, linboxsage *must* be
               # before ntl.
               libraries = ['linboxsage', 'ntl', 'iml', 'linbox',
-                           'stdc++', 'givaro', 'mpfr', 'gmp', 'gmpxx', BLAS, BLAS2],
+                           'stdc++', 'givaro', 'mpfr', 'gmp', 'gmpxx'] + blas_libs,
               language = 'c++',
               extra_compile_args = givaro_extra_compile_args,
               depends = givaro_depends),
@@ -998,7 +990,8 @@
 
     Extension('sage.matrix.change_ring',
               sources = ['sage/matrix/change_ring.pyx'],
-              libraries=[BLAS, BLAS2, 'gmp'],
+              library_dirs = blas_dir,
+              libraries=['gmp'] + blas_libs,
               include_dirs = numpy_include_dirs),
 
     Extension('sage.matrix.matrix',
@@ -1016,7 +1009,8 @@
 
     Extension('sage.matrix.matrix_complex_double_dense',
               sources = ['sage/matrix/matrix_complex_double_dense.pyx'],
-              libraries=[BLAS, BLAS2],
+              library_dirs = blas_dir,
+              libraries = blas_libs,
               include_dirs = numpy_include_dirs,
               depends = numpy_depends),
 
@@ -1030,7 +1024,8 @@
 
     Extension('sage.matrix.matrix_double_dense',
               sources = ['sage/matrix/matrix_double_dense.pyx'],
-              libraries=[BLAS, BLAS2],
+              library_dirs = blas_dir,
+              libraries = blas_libs,
               include_dirs = numpy_include_dirs,
               depends = numpy_depends),
 
@@ -1049,8 +1044,9 @@
               sources = ['sage/matrix/matrix_integer_dense.pyx'],
               extra_compile_args = m4ri_extra_compile_args,
               depends = [SAGE_INC + '/m4ri/m4ri.h'],
+              library_dirs = blas_dir,
               # order matters for cygwin!!
-              libraries = ['iml', 'pari', 'gmp', 'm', BLAS, BLAS2]),
+              libraries = ['iml', 'pari', 'gmp', 'm'] + blas_libs),
 
     Extension('sage.matrix.matrix_integer_sparse',
               sources = ['sage/matrix/matrix_integer_sparse.pyx'],
@@ -1077,13 +1073,15 @@
     Extension('sage.matrix.matrix_modn_dense_float',
               sources = ['sage/matrix/matrix_modn_dense_float.pyx'],
               language="c++",
-              libraries = ['linbox', 'givaro', 'mpfr', 'gmpxx', 'gmp', BLAS, BLAS2],
+              library_dirs = blas_dir,
+              libraries = ['linbox', 'givaro', 'mpfr', 'gmpxx', 'gmp'] + blas_libs,
               extra_compile_args = ['-DDISABLE_COMMENTATOR'] + givaro_extra_compile_args),
 
     Extension('sage.matrix.matrix_modn_dense_double',
               sources = ['sage/matrix/matrix_modn_dense_double.pyx'],
               language="c++",
-              libraries = ['linbox', 'givaro', 'mpfr', 'gmpxx', 'gmp', BLAS, BLAS2],
+              library_dirs = blas_dir,
+              libraries = ['linbox', 'givaro', 'mpfr', 'gmpxx', 'gmp'] + blas_libs,
               extra_compile_args = ['-DDISABLE_COMMENTATOR'] + givaro_extra_compile_args),
 
     Extension('sage.matrix.matrix_modn_sparse',
@@ -1107,7 +1105,8 @@
 
     Extension('sage.matrix.matrix_real_double_dense',
               sources = ['sage/matrix/matrix_real_double_dense.pyx'],
-              libraries=[BLAS, BLAS2],
+              library_dirs = blas_dir,
+              libraries = blas_libs,
               include_dirs = numpy_include_dirs,
               depends = numpy_depends),
 
@@ -1333,13 +1332,15 @@
 
     Extension('sage.modules.vector_complex_double_dense',
               ['sage/modules/vector_complex_double_dense.pyx'],
-              libraries = [BLAS, BLAS2],
+              library_dirs = blas_dir,
+              libraries = blas_libs,
               include_dirs = numpy_include_dirs,
               depends = numpy_depends),
 
     Extension('sage.modules.vector_double_dense',
               ['sage/modules/vector_double_dense.pyx'],
-              libraries = [BLAS, BLAS2],
+              library_dirs = blas_dir,
+              libraries = blas_libs,
               include_dirs = numpy_include_dirs,
               depends = numpy_depends),
 
@@ -1363,7 +1364,8 @@
 
     Extension('sage.modules.vector_real_double_dense',
               ['sage/modules/vector_real_double_dense.pyx'],
-              libraries = [BLAS, BLAS2],
+              library_dirs = blas_dir,
+              libraries = blas_libs,
               include_dirs = numpy_include_dirs,
               depends = numpy_depends),
 
@@ -1506,7 +1508,8 @@
     Extension('sage.rings.complex_double',
               sources = ['sage/rings/complex_double.pyx'],
               extra_compile_args=["-std=c99",  "-D_XPG6"],
-              libraries = (['gsl', BLAS, BLAS2, 'pari', 'gmp', 'm'])),
+              library_dirs = blas_dir,
+              libraries = ['gsl', 'pari', 'gmp', 'm'] + blas_libs),
 
     Extension('sage.rings.complex_interval',
               sources = ['sage/rings/complex_interval.pyx'],
@@ -1572,7 +1575,8 @@
 
     Extension('sage.rings.real_double',
               sources = ['sage/rings/real_double.pyx'],
-              libraries = ['gsl', 'gmp', BLAS, BLAS2],
+              library_dirs = blas_dir,
+              libraries = ['gsl', 'gmp'] + blas_libs,
               depends = numpy_depends,
               define_macros=[('GSL_DISABLE_DEPRECATED','1')]),
 
