diff -Naur src.orig/module_list.py src/module_list.py
--- src.orig/module_list.py	2016-12-10 14:45:51.896952970 +1300
+++ src/module_list.py	2016-12-10 14:46:11.940005381 +1300
@@ -44,7 +44,6 @@
 # The above cblas should already be in the list thanks to #20646
 gsl_pc = pkgconfig.parse('gsl')
 gsl_libs = gsl_pc['libraries']
-gsl_libs.remove('gslcblas')
 gsl_library_dirs = gsl_pc['library_dirs']
 gsl_include_dirs = gsl_pc['include_dirs']
 
