diff --git a/doc/en/reference/misc/index.rst b/doc/en/reference/misc/index.rst
index 8a5e6db..b88bea6 100644
--- a/doc/en/reference/misc/index.rst
+++ b/doc/en/reference/misc/index.rst
@@ -231,8 +231,6 @@ Distribution
 .. toctree::
    :maxdepth: 1
 
-   sage/misc/package
-   sage/misc/dist
    sage/features
    sage/features/bliss
    sage/features/csdp
diff --git a/sage/doctest/control.py b/sage/doctest/control.py
index 9d972b6..f420cdc 100644
--- a/sage/doctest/control.py
+++ b/sage/doctest/control.py
@@ -335,22 +335,6 @@ class DocTestController(SageObject):
                 # Special case to run all optional tests
                 options.optional = True
             else:
-                # We replace the 'optional' tag by all optional
-                # packages for which the installed version matches the
-                # latest available version (this implies in particular
-                # that the package is actually installed).
-                if 'optional' in options.optional:
-                    options.optional.discard('optional')
-                    from sage.misc.package import list_packages
-                    for pkg in list_packages('optional', local=True).values():
-                        if pkg['installed_version'] == pkg['remote_version']:
-                            options.optional.add(pkg['name'])
-
-                # Check that all tags are valid
-                for o in options.optional:
-                    if not optionaltag_regex.search(o):
-                        raise ValueError('invalid optional tag {!r}'.format(o))
-
                 if "sage" in options.optional:
                     options.optional |= auto_optional_tags
 
diff --git a/sage/doctest/test.py b/sage/doctest/test.py
index d43c91f..73d248f 100644
--- a/sage/doctest/test.py
+++ b/sage/doctest/test.py
@@ -463,14 +463,6 @@ Optional tests are run correctly::
     ...
     0
 
-Test an invalid value for ``--optional``::
-
-    sage: subprocess.call(["sage", "-t", "--warn-long", "0", "--optional=bad-option", "show_skipped.rst"], **kwds)
-    Traceback (most recent call last):
-    ...
-    ValueError: invalid optional tag 'bad-option'
-    1
-
 Test ``atexit`` support in the doctesting framework::
 
     sage: F = tmp_filename()
diff --git a/sage/misc/all.py b/sage/misc/all.py
index 5cd23ca..614f0aa 100644
--- a/sage/misc/all.py
+++ b/sage/misc/all.py
@@ -46,12 +46,6 @@ from .mrange import xmrange, mrange, xmrange_iter, mrange_iter, cartesian_produc
 
 from .fpickle import pickle_function, unpickle_function
 
-from .dist import install_scripts
-
-from .package import (installed_packages, is_package_installed,
-        standard_packages, optional_packages, experimental_packages,
-        package_versions)
-
 from .pager import pager
 
 lazy_import('sage.misc.sagedoc', ['browse_sage_doc',
diff --git a/sage/tests/cmdline.py b/sage/tests/cmdline.py
index 9edf619..7ad6f84 100644
--- a/sage/tests/cmdline.py
+++ b/sage/tests/cmdline.py
@@ -13,23 +13,18 @@ test.spyx
 --advanced
 -c
 --cython
---dev
 --ecl
---experimental
---fixdoctests
 --gap
 --gdb
 --gp
 -h
 --help
---info
 --ipython
 --kash
 --lisp
 --maxima
 --min
 --mwrank
---optional
 --preparse
 --python
 --python3
@@ -43,7 +38,6 @@ test.spyx
 --sh
 --singular
 --sqlite3
---standard
 --startuptime
 -t
 -v
@@ -203,36 +197,6 @@ def test_executable(args, input="", timeout=100.0, **kwds):
         sage: ret
         0
 
-    Test ``sage --info [packages]`` and the equivalent
-    ``sage -p --info --info [packages]`` (the doubling of ``--info``
-    is intentional, that option should be idempotent)::
-
-        sage: out, err, ret = test_executable(["sage", "--info", "sqlite"])
-        sage: print(out)
-        Found local metadata for sqlite-...
-        = SQLite =
-        ...
-        SQLite is a software library that implements a self-contained,
-        serverless, zero-configuration, transactional SQL database engine.
-        ...
-        sage: err
-        ''
-        sage: ret
-        0
-
-        sage: out, err, ret = test_executable(["sage", "-p", "--info", "--info", "sqlite"])
-        sage: print(out)
-        Found local metadata for sqlite-...
-        = SQLite =
-        ...
-        SQLite is a software library that implements a self-contained,
-        serverless, zero-configuration, transactional SQL database engine.
-        ...
-        sage: err
-        ''
-        sage: ret
-        0
-
     Test ``sage-run`` on a Python file, both with an absolute and with a relative path::
 
         sage: dir = tmp_dir(); name = 'python_test_file.py'
@@ -419,50 +383,6 @@ def test_executable(args, input="", timeout=100.0, **kwds):
         ...
         RuntimeError: refusing to run doctests...
 
-    Now run a test for the fixdoctests script and, in particular, check that the
-    issues raised in :trac:`10589` are fixed. We have to go to slightly silly
-    lengths to doctest the output.::
-
-        sage: test='r\"\"\"Add a doc-test for the fixdoctest command line option and, in particular, check that\n:trac:`10589` is fixed.\n\nEXAMPLES::\n\n    sage: 1+1              # incorrect output\n    3\n    sage: m=matrix(ZZ,3)   # output when none is expected\n    [0 0 0]\n    [0 0 0]\n    [1 0 0]\n    sage: (2/3)*m          # no output when it is expected\n    sage: mu=PartitionTuple([[4,4],[3,3,2,1],[1,1]])   # output when none is expected\n    [4, 4, 3, 3, 2, 1, 1]\n    sage: mu.pp()          # uneven indentation\n    ****\n    ****\n    sage: PartitionTuples.options(convention="French")\n    sage: mu.pp()         # fix doctest with uneven indentation\n    sage: PartitionTuples.options._reset()\n\"\"\"\n'
-        sage: test_file = os.path.join(tmp_dir(), 'test_file.py')
-        sage: F = open(test_file, 'w')
-        sage: _ = F.write(test)
-        sage: F.close()
-        sage: (out, err, ret) = test_executable(["sage", "--fixdoctests", test_file])
-        sage: with open(test_file, 'r') as f:
-        ....:     fixed_test = f.read()
-        sage: import difflib
-        sage: list(difflib.unified_diff(test.splitlines(), fixed_test.splitlines()))[2:-1]
-        ['@@ -4,18 +4,23 @@\n',
-         ' EXAMPLES::',
-         ' ',
-         '     sage: 1+1              # incorrect output',
-         '-    3',
-         '+    2',
-         '     sage: m=matrix(ZZ,3)   # output when none is expected',
-         '+    sage: (2/3)*m          # no output when it is expected',
-         '     [0 0 0]',
-         '     [0 0 0]',
-         '-    [1 0 0]',
-         '-    sage: (2/3)*m          # no output when it is expected',
-         '+    [0 0 0]',
-         '     sage: mu=PartitionTuple([[4,4],[3,3,2,1],[1,1]])   # output when none is expected',
-         '-    [4, 4, 3, 3, 2, 1, 1]',
-         '     sage: mu.pp()          # uneven indentation',
-         '-    ****',
-         '-    ****',
-         '+       ****   ***   *',
-         '+       ****   ***   *',
-         '+              **',
-         '+              *',
-         '     sage: PartitionTuples.options(convention="French")',
-         '     sage: mu.pp()         # fix doctest with uneven indentation',
-         '+    *',
-         '+    **',
-         '+    ****   ***   *',
-         '+    ****   ***   *',
-         '     sage: PartitionTuples.options._reset()']
-
     Test external programs being called by Sage::
 
         sage: (out, err, ret) = test_executable(["sage", "--sh"], "echo Hello World\nexit 42\n")
@@ -612,32 +532,6 @@ def test_executable(args, input="", timeout=100.0, **kwds):
         sage: ret
         0
 
-    Check some things requiring an internet connection::
-
-        sage: (out, err, ret) = test_executable(["sage", "--standard"])  # optional - internet
-        sage: out.find("cython") >= 0  # optional - internet
-        True
-        sage: err  # optional - internet
-        ''
-        sage: ret  # optional - internet
-        0
-
-        sage: (out, err, ret) = test_executable(["sage", "--optional"])  # optional - internet
-        sage: out.find("database_cremona_ellcurve") >= 0  # optional - internet
-        True
-        sage: err  # optional - internet
-        ''
-        sage: ret  # optional - internet
-        0
-
-        sage: (out, err, ret) = test_executable(["sage", "--experimental"])  # optional - internet
-        sage: out.find("valgrind") >= 0  # optional - internet
-        True
-        sage: err  # optional - internet
-        ''
-        sage: ret  # optional - internet
-        0
-
     Check an illegal command line option.  This outputs an error to stdout,
     but we allow stderr in case this changes in the future::
 
diff --git a/sage_setup/optional_extension.py b/sage_setup/optional_extension.py
index ad9e10c..49072c5 100644
--- a/sage_setup/optional_extension.py
+++ b/sage_setup/optional_extension.py
@@ -19,10 +19,7 @@ package which must be installed.
 
 
 from distutils.extension import Extension
-from sage.misc.package import list_packages
-
-all_packages = list_packages(local=True)
-
+import os
 
 class CythonizeExtension(Extension):
     """
@@ -76,14 +73,7 @@ def OptionalExtension(*args, **kwds):
         condition = kwds.pop("condition")
     except KeyError:
         pkg = kwds.pop("package")
-        from sage.misc.package import is_package_installed
-        try:
-            pkginfo = all_packages[pkg]
-        except KeyError:
-            # Might be an installed old-style package
-            condition = is_package_installed(pkg)
-        else:
-            condition = (pkginfo["installed_version"] == pkginfo["remote_version"])
+        condition = os.getenv('WANT_%s'%pkg,False)
 
     if condition:
         return Extension(*args, **kwds)
