diff --git a/sage/libs/gap/saved_workspace.py b/sage/libs/gap/saved_workspace.py
index 3cb8fd3..2dd79fd 100644
--- a/sage/libs/gap/saved_workspace.py
+++ b/sage/libs/gap/saved_workspace.py
@@ -8,7 +8,6 @@ workspaces.
 
 import os
 import glob
-from sage.env import GAP_ROOT_DIR
 from sage.interfaces.gap import GAP_BINARY
 from sage.interfaces.gap_workspace import gap_workspace_file
 
@@ -32,8 +31,7 @@ def timestamp():
     """
     libgap_dir = os.path.dirname(__file__)
     libgap_files = glob.glob(os.path.join(libgap_dir, '*'))
-    gap_packages = glob.glob(os.path.join(GAP_ROOT_DIR, 'pkg', '*'))
-    files = libgap_files + [GAP_BINARY] + gap_packages
+    files = libgap_files + [GAP_BINARY]
     if len(files) == 0:
         print('Unable to find LibGAP files.')
         return float('inf')
diff --git a/sage/libs/gap/util.pyx b/sage/libs/gap/util.pyx
index 3dc2c2e..7f773e5 100644
--- a/sage/libs/gap/util.pyx
+++ b/sage/libs/gap/util.pyx
@@ -19,7 +19,7 @@ from cpython.object cimport Py_LT, Py_LE, Py_EQ, Py_NE, Py_GT, Py_GE
 from cysignals.signals cimport sig_on, sig_off, sig_error
 
 from sage.interfaces.gap_workspace import prepare_workspace_dir
-from sage.env import SAGE_LOCAL, GAP_ROOT_DIR
+from sage.env import SAGE_LOCAL
 from .element cimport *
 
 
@@ -163,9 +163,6 @@ def gap_root():
         '/home/vbraun/opt/sage-5.3.rc0/local/gap/latest'
     """
     import os.path
-    if os.path.exists(GAP_ROOT_DIR):
-        return GAP_ROOT_DIR
-    print('The gap-4.5.5.spkg (or later) seems to be not installed!')
     gap_sh = open(os.path.join(SAGE_LOCAL, 'bin', 'gap')).read().splitlines()
     gapdir = filter(lambda dir:dir.strip().startswith('GAP_DIR'), gap_sh)[0]
     gapdir = gapdir.split('"')[1]
