diff --git a/sage/env.py b/sage/env.py
index 3a64c2d..74ac9c6 100644
--- a/sage/env.py
+++ b/sage/env.py
@@ -21,6 +21,7 @@ import glob
 import os
 import socket
 import site
+import sysconfig
 from . import version
 
 opj = os.path.join
@@ -117,13 +118,11 @@ _add_variable_or_fallback('HOSTNAME',        socket.gethostname())
 _add_variable_or_fallback('LOCAL_IDENTIFIER','$HOSTNAME.%s'%os.getpid())
 
 # bunch of sage directories and files
-_add_variable_or_fallback('SAGE_ROOT',       None)
-_add_variable_or_fallback('SAGE_LOCAL',      None)
-_add_variable_or_fallback('SAGE_ETC',        opj('$SAGE_LOCAL', 'etc'))
+_add_variable_or_fallback('SAGE_LOCAL',      sysconfig.get_config_var("prefix"))
+_add_variable_or_fallback('SAGE_ROOT',       opj('$SAGE_LOCAL', 'share', 'sage'))
+_add_variable_or_fallback('SAGE_ETC',        opj('$SAGE_LOCAL', '..', 'etc'))
 _add_variable_or_fallback('SAGE_INC',        opj('$SAGE_LOCAL', 'include'))
-_add_variable_or_fallback('SAGE_SHARE',      opj('$SAGE_LOCAL', 'share'))
-
-_add_variable_or_fallback('SAGE_SRC',        opj('$SAGE_ROOT', 'src'))
+_add_variable_or_fallback('SAGE_SHARE',      opj('$SAGE_LOCAL', 'share', 'sage'))
 
 try:
     sitepackages_dirs = site.getsitepackages()
@@ -133,19 +132,12 @@ except AttributeError:  # in case of use inside virtualenv
 _add_variable_or_fallback('SITE_PACKAGES',   sitepackages_dirs)
 
 _add_variable_or_fallback('SAGE_LIB',        SITE_PACKAGES[0])
+_add_variable_or_fallback('SAGE_SRC',        opj('$SAGE_LIB'))
 
-# Used by sage/misc/package.py.  Should be SAGE_SRC_ROOT in VPATH.
-_add_variable_or_fallback('SAGE_PKGS', opj('$SAGE_ROOT', 'build', 'pkgs'))
-
-
-_add_variable_or_fallback('SAGE_EXTCODE',    opj('$SAGE_SHARE', 'sage', 'ext'))
-_add_variable_or_fallback('SAGE_LOGS',       opj('$SAGE_ROOT', 'logs', 'pkgs'))
-_add_variable_or_fallback('SAGE_SPKG_INST',  opj('$SAGE_LOCAL', 'var', 'lib', 'sage', 'installed'))
-_add_variable_or_fallback('SAGE_DOC_SRC',    opj('$SAGE_SRC', 'doc'))
-_add_variable_or_fallback('SAGE_DOC',        opj('$SAGE_SHARE', 'doc', 'sage'))
+_add_variable_or_fallback('SAGE_EXTCODE',    opj('$SAGE_SHARE', 'ext'))
+_add_variable_or_fallback('SAGE_DOC',        opj('$SAGE_LOCAL', 'share', 'doc', '@GENTOO_PORTAGE_PF@'))
+_add_variable_or_fallback('SAGE_DOC_SRC',    opj('$SAGE_DOC'))
 _add_variable_or_fallback('DOT_SAGE',        opj(os.environ.get('HOME','$SAGE_ROOT'), '.sage'))
-_add_variable_or_fallback('SAGE_DOT_GIT',    opj('$SAGE_ROOT', '.git'))
-_add_variable_or_fallback('SAGE_DISTFILES',  opj('$SAGE_ROOT', 'upstream'))
 
 # misc
 _add_variable_or_fallback('SAGE_URL',                'http://sage.math.washington.edu/sage/')
@@ -164,14 +156,13 @@ _add_variable_or_fallback('CONWAY_POLYNOMIALS_DATA_DIR',  opj('$SAGE_SHARE','con
 _add_variable_or_fallback('GRAPHS_DATA_DIR',  opj('$SAGE_SHARE','graphs'))
 _add_variable_or_fallback('ELLCURVE_DATA_DIR',opj('$SAGE_SHARE','ellcurves'))
 _add_variable_or_fallback('POLYTOPE_DATA_DIR',opj('$SAGE_SHARE','reflexive_polytopes'))
-_add_variable_or_fallback('GAP_ROOT_DIR',     opj('$SAGE_SHARE','gap'))
-_add_variable_or_fallback('THEBE_DIR',        opj('$SAGE_SHARE','thebe'))
+_add_variable_or_fallback('THEBE_DIR',        opj('$SAGE_LOCAL','share','thebe'))
 _add_variable_or_fallback('COMBINATORIAL_DESIGN_DATA_DIR', opj('$SAGE_SHARE', 'combinatorial_designs'))
 _add_variable_or_fallback('CREMONA_MINI_DATA_DIR', opj('$SAGE_SHARE', 'cremona'))
 _add_variable_or_fallback('CREMONA_LARGE_DATA_DIR', opj('$SAGE_SHARE', 'cremona'))
-_add_variable_or_fallback('JMOL_DIR', opj('$SAGE_SHARE', 'jmol'))
-_add_variable_or_fallback('JSMOL_DIR', opj('$SAGE_SHARE', 'jsmol'))
-_add_variable_or_fallback('MATHJAX_DIR', opj('$SAGE_SHARE', 'mathjax'))
+_add_variable_or_fallback('JMOL_DIR', opj('$SAGE_LOCAL', 'share', 'sage-jmol-bin'))
+_add_variable_or_fallback('JSMOL_DIR', opj('$SAGE_LOCAL', 'share', 'jsmol'))
+_add_variable_or_fallback('MATHJAX_DIR', opj('$SAGE_LOCAL', 'share', 'mathjax'))
 _add_variable_or_fallback('THREEJS_DIR', opj('$SAGE_SHARE', 'threejs'))
 _add_variable_or_fallback('MAXIMA_FAS', None)
 
@@ -186,7 +177,7 @@ else:
     else:
         extension = "so"
     # library name changed from libsingular to libSingular btw 3.x and 4.x
-    SINGULAR_SO = SAGE_LOCAL+"/lib/libSingular."+extension
+    SINGULAR_SO = sysconfig.get_config_var("LIBDIR")+"/libSingular."+extension
 
 _add_variable_or_fallback('SINGULAR_SO', SINGULAR_SO)
 
@@ -220,7 +211,6 @@ if UNAME[:6] == 'CYGWIN':
     del _uname, re
 
 # things that need DOT_SAGE
-_add_variable_or_fallback('PYTHON_EGG_CACHE',   opj('$DOT_SAGE', '.python-eggs'))
 _add_variable_or_fallback('SAGE_STARTUP_FILE',  opj('$DOT_SAGE', 'init.sage'))
 
 # delete temporary variables used for setting up sage.env
diff --git a/sage/interfaces/jmoldata.py b/sage/interfaces/jmoldata.py
index 2e88022..910be2a 100644
--- a/sage/interfaces/jmoldata.py
+++ b/sage/interfaces/jmoldata.py
@@ -149,7 +149,7 @@ class JmolData(SageObject):
             True
         """
         # Set up paths, file names and scripts
-        jmolpath = os.path.join(JMOL_DIR, "JmolData.jar")
+        jmolpath = os.path.join(JMOL_DIR, 'lib', 'JmolData.jar')
         target_native = targetfile
 
         if sys.platform == 'cygwin':
diff --git a/sage/interfaces/singular.py b/sage/interfaces/singular.py
index 972fb4b..a3da0e0 100644
--- a/sage/interfaces/singular.py
+++ b/sage/interfaces/singular.py
@@ -2263,8 +2263,8 @@ def generate_docstring_dictionary():
     nodes.clear()
     node_names.clear()
 
-    import os
-    singular_docdir = os.environ['SINGULARPATH']+"/../info/"
+    from sage.env import SAGE_LOCAL
+    singular_docdir = SAGE_LOCAL+"/share/info/"
 
     new_node = re.compile(r"File: singular\.hlp,  Node: ([^,]*),.*")
     new_lookup = re.compile(r"\* ([^:]*):*([^.]*)\..*")
