diff -Naur src.orig/module_list.py src/module_list.py
--- src.orig/module_list.py	2016-09-07 11:21:13.300496441 +1200
+++ src/module_list.py	2016-09-07 11:22:03.440623869 +1200
@@ -88,6 +88,7 @@
     LINBOX_CFLAGS=linbox_cflags,
     LINBOX_LIBRARIES=linbox_libs,
     LINBOX_LIBDIR=linbox_library_dirs,
+    SINGULAR_EXTRA=['-DNDEBUG']
 )
 
 #########################################################
diff -Naur src.orig/sage/libs/singular/decl.pxd src/sage/libs/singular/decl.pxd
--- src.orig/sage/libs/singular/decl.pxd	2016-09-07 11:21:13.350496566 +1200
+++ src/sage/libs/singular/decl.pxd	2016-09-07 11:23:25.200831530 +1200
@@ -1,3 +1,5 @@
+# distutils: extra_compile_args = SINGULAR_EXTRA
+
 """
 Declarations of Singular's C/C++ Functions
 
