diff -Naur src.orig/sage_setup/docbuild/build_options.py src/sage_setup/docbuild/build_options.py
--- src.orig/sage_setup/docbuild/build_options.py	2016-02-09 15:26:07.546660530 +1300
+++ src/sage_setup/docbuild/build_options.py	2016-02-09 15:27:59.876951476 +1300
@@ -6,7 +6,18 @@
 
 from sage.env import SAGE_DOC_SRC, SAGE_DOC
 
-LANGUAGES = [d for d in os.listdir(SAGE_DOC_SRC) if re.match('^[a-z][a-z]$', d)]
+LANGUAGELIST = [d for d in os.listdir(SAGE_DOC) if re.match('^[a-z][a-z]$', d)]
+try:
+    DESIRED_LANGUAGES = list(set((os.environ['LANGUAGES']).split()))
+except KeyError:
+    DESIRED_LANGUAGES = LANGUAGELIST
+LANGUAGES = []
+for d in DESIRED_LANGUAGES:
+    if d in LANGUAGELIST:
+        LANGUAGES.append(d)
+if not LANGUAGES:
+    LANGUAGES=LANGUAGELIST
+
 SPHINXOPTS = ""
 PAPER = ""
 OMIT = ["introspect"]  # docs/dirs to omit when listing and building 'all'
