diff --git a/sage/graphs/generic_graph.py b/sage/graphs/generic_graph.py
index a789f13..07d21b5 100644
--- a/sage/graphs/generic_graph.py
+++ b/sage/graphs/generic_graph.py
@@ -19620,7 +19620,7 @@ class GenericGraph(GenericGraph_pyx):
                      vertex_colors=None, vertex_size=0.06, vertex_labels=False,
                      edge_colors=None, edge_size=0.02, edge_size2=0.0325,
                      pos3d=None, color_by_label=False,
-                     engine='jmol', **kwds):
+                     engine='tachyon', **kwds):
         r"""
         Plot a graph in three dimensions.
 
@@ -19663,11 +19663,11 @@ class GenericGraph(GenericGraph_pyx):
 
         - ``layout``, ``iterations``, ... -- layout options; see :meth:`layout`
 
-        - ``engine`` -- string (default: ``'jmol'``); the renderer to use among:
+        - ``engine`` -- string (default: ``'tachyon'``); the renderer to use among:
 
-          -  ``'jmol'`` - default
+          -  ``'jmol'``
 
-          -  ``'tachyon'``
+          -  ``'tachyon'`` - default
 
         - ``xres`` -- resolution
 
@@ -19860,7 +19860,7 @@ class GenericGraph(GenericGraph_pyx):
     def show3d(self, bgcolor=(1,1,1), vertex_colors=None, vertex_size=0.06,
                      edge_colors=None, edge_size=0.02, edge_size2=0.0325,
                      pos3d=None, color_by_label=False,
-                     engine='jmol', **kwds):
+                     engine='tachyon', **kwds):
         """
         Plot the graph using :class:`~sage.plot.plot3d.tachyon.Tachyon`, and
         show the resulting plot.
@@ -19901,11 +19901,11 @@ class GenericGraph(GenericGraph_pyx):
 
         - ``layout``, ``iterations``, ... -- layout options; see :meth:`layout`
 
-        - ``engine`` -- string (default: ``'jmol'``); the renderer to use among:
+        - ``engine`` -- string (default: ``'tachyon'``); the renderer to use among:
 
-          -  ``'jmol'`` - default
+          -  ``'jmol'``
 
-          -  ``'tachyon'``
+          -  ``'tachyon'`` - default
 
         - ``xres`` -- resolution
 
diff --git a/sage/plot/plot3d/base.pyx b/sage/plot/plot3d/base.pyx
index eb0b437..f5b6bd7 100644
--- a/sage/plot/plot3d/base.pyx
+++ b/sage/plot/plot3d/base.pyx
@@ -140,7 +140,7 @@ cdef class Graphics3d(SageObject):
             viewer = None
         # select suitable default
         if viewer is None:
-            viewer = 'jmol'
+            viewer = 'tachyon'
         # fall back to 2d image if necessary
         if viewer == 'canvas3d' and not can_view_canvas3d:   viewer = 'jmol'
         if viewer == 'wavefront' and not can_view_wavefront: viewer = 'jmol'
@@ -1956,7 +1956,7 @@ end_scene""" % (render_params.antialiasing,
         return self
 
 # if you add any default parameters you must update some code below
-SHOW_DEFAULTS = {'viewer': 'jmol',
+SHOW_DEFAULTS = {'viewer': 'tachyon',
                  'verbosity': 0,
                  'figsize': 5,
                  'aspect_ratio': "automatic",
