--- sage/doctest/control.py.orig	2013-12-17 14:09:20.313617816 +1300
+++ sage/doctest/control.py	2013-12-17 14:12:19.908137248 +1300
@@ -457,12 +457,12 @@
             'sagenb'
         """
         opj = os.path.join
-        from sage.env import SAGE_SRC, SAGE_ROOT
+        from sage.env import SAGE_SRC, SAGE_DOC, SAGE_ROOT
         def all_files():
             from glob import glob
             self.files.append(opj(SAGE_SRC, 'sage'))
-            self.files.append(opj(SAGE_SRC, 'doc', 'common'))
-            self.files.extend(glob(opj(SAGE_SRC, 'doc', '[a-z][a-z]')))
+            self.files.append(opj(SAGE_DOC, 'common'))
+            self.files.extend(glob(opj(SAGE_DOC, '[a-z][a-z]')))
             self.options.sagenb = True
         DOT_GIT= opj(SAGE_ROOT, '.git')
         have_git = os.path.exists(DOT_GIT)
