--- sage/numerical/backends/generic_backend.pyx.orig	2014-02-21 13:37:41.000000000 -0600
+++ sage/numerical/backends/generic_backend.pyx	2014-02-21 13:38:15.000000000 -0600
@@ -989,7 +989,7 @@
 
     .. SEEALSO::
 
-    - :func:`default_mip_solver` -- Returns/Sets the default MIP solver.
+        - :func:`default_mip_solver` -- Returns/Sets the default MIP solver.
 
     EXAMPLE::

--- sage/graphs/generic_graph.py.orig	2014-02-21 13:04:53.000000000 -0600
+++ sage/graphs/generic_graph.py	2014-02-21 13:05:25.000000000 -0600
@@ -9037,7 +9037,7 @@
 
         .. SEEALSO::
 
-        - :meth:`subdivide_edges` -- subdivides multiples edges at a time
+            - :meth:`subdivide_edges` -- subdivides multiples edges at a time
 
         """
 
@@ -9124,7 +9124,7 @@
 
         .. SEEALSO::
 
-        - :meth:`subdivide_edge` -- subdivides one edge
+            - :meth:`subdivide_edge` -- subdivides one edge
         """
         for e in edges:
             self.subdivide_edge(e, k)
@@ -11405,10 +11405,10 @@
 
         .. SEEALSO::
 
-        - :mod:`Interval Graph Recognition <sage.graphs.pq_trees>`.
+            - :mod:`Interval Graph Recognition <sage.graphs.pq_trees>`.
 
-        - :meth:`PQ <sage.graphs.pq_trees.PQ>`
-          -- Implementation of PQ-Trees.
+            - :meth:`PQ <sage.graphs.pq_trees.PQ>`
+              -- Implementation of PQ-Trees.
 
         """
         self._scream_if_not_simple()
@@ -13270,13 +13270,13 @@
 
         .. SEEALSO::
 
-        - :meth:`breadth_first_search <sage.graphs.base.c_graph.CGraphBackend.breadth_first_search>`
-          -- breadth-first search for fast compiled graphs.
+            - :meth:`breadth_first_search <sage.graphs.base.c_graph.CGraphBackend.breadth_first_search>`
+              -- breadth-first search for fast compiled graphs.
 
-        - :meth:`depth_first_search <sage.graphs.base.c_graph.CGraphBackend.depth_first_search>`
-          -- depth-first search for fast compiled graphs.
+            - :meth:`depth_first_search <sage.graphs.base.c_graph.CGraphBackend.depth_first_search>`
+              -- depth-first search for fast compiled graphs.
 
-        - :meth:`depth_first_search` -- depth-first search for generic graphs.
+            - :meth:`depth_first_search` -- depth-first search for generic graphs.
 
         EXAMPLES::
 
@@ -13395,13 +13395,13 @@
 
         .. SEEALSO::
 
-        - :meth:`breadth_first_search`
+            - :meth:`breadth_first_search`
 
-        - :meth:`breadth_first_search <sage.graphs.base.c_graph.CGraphBackend.breadth_first_search>`
-          -- breadth-first search for fast compiled graphs.
+            - :meth:`breadth_first_search <sage.graphs.base.c_graph.CGraphBackend.breadth_first_search>`
+              -- breadth-first search for fast compiled graphs.
 
-        - :meth:`depth_first_search <sage.graphs.base.c_graph.CGraphBackend.depth_first_search>`
-          -- depth-first search for fast compiled graphs.
+            - :meth:`depth_first_search <sage.graphs.base.c_graph.CGraphBackend.depth_first_search>`
+              -- depth-first search for fast compiled graphs.
 
         EXAMPLES::

--- sage/graphs/graph.py.orig	2014-02-21 13:04:22.000000000 -0600
+++ sage/graphs/graph.py	2014-02-21 13:05:40.000000000 -0600
@@ -2697,9 +2697,9 @@
 
         .. SEEALSO::
 
-          - :meth:`~Graph.is_arc_transitive`
-          - :meth:`~Graph.is_half_transitive`
-          - :meth:`~Graph.is_semi_symmetric`
+            - :meth:`~Graph.is_arc_transitive`
+            - :meth:`~Graph.is_half_transitive`
+            - :meth:`~Graph.is_semi_symmetric`
 
         EXAMPLES::
 
@@ -2744,9 +2744,9 @@
 
         .. SEEALSO::
 
-          - :meth:`~Graph.is_edge_transitive`
-          - :meth:`~Graph.is_half_transitive`
-          - :meth:`~Graph.is_semi_symmetric`
+            - :meth:`~Graph.is_edge_transitive`
+            - :meth:`~Graph.is_half_transitive`
+            - :meth:`~Graph.is_semi_symmetric`
 
         EXAMPLES::
 
@@ -2781,9 +2781,9 @@
 
         .. SEEALSO::
 
-          - :meth:`~Graph.is_edge_transitive`
-          - :meth:`~Graph.is_arc_transitive`
-          - :meth:`~Graph.is_semi_symmetric`
+            - :meth:`~Graph.is_edge_transitive`
+            - :meth:`~Graph.is_arc_transitive`
+            - :meth:`~Graph.is_semi_symmetric`
 
         EXAMPLES:
 
@@ -2820,9 +2820,9 @@
 
         .. SEEALSO::
 
-          - :meth:`~Graph.is_edge_transitive`
-          - :meth:`~Graph.is_arc_transitive`
-          - :meth:`~Graph.is_half_transitive`
+            - :meth:`~Graph.is_edge_transitive`
+            - :meth:`~Graph.is_arc_transitive`
+            - :meth:`~Graph.is_half_transitive`
 
         EXAMPLES:
 
@@ -5906,7 +5906,7 @@
 
         .. SEEALSO::
 
-        - :meth:`is_prime` -- Tests whether a graph is prime.
+            - :meth:`is_prime` -- Tests whether a graph is prime.
 
         REFERENCE:

--- sage/libs/lrcalc/lrcalc.pyx.orig	2014-02-21 13:09:26.000000000 -0600
+++ sage/libs/lrcalc/lrcalc.pyx	2014-02-21 13:11:40.000000000 -0600
@@ -79,12 +79,12 @@
 
 .. seealso::
 
-- :func:`lrcoef`
-- :func:`mult`
-- :func:`coprod`
-- :func:`skew`
-- :func:`lrskew`
-- :func:`mult_schubert`
+    - :func:`lrcoef`
+    - :func:`mult`
+    - :func:`coprod`
+    - :func:`skew`
+    - :func:`lrskew`
+    - :func:`mult_schubert`
 
 .. rubric:: Underlying algorithmic in lrcalc

