diff --git a/doc/common/conf.py b/doc/common/conf.py
index 70dd17f..8bbe4e5 100644
--- a/doc/common/conf.py
+++ b/doc/common/conf.py
@@ -178,7 +178,7 @@ intersphinx_mapping = {
     'python': ('https://docs.python.org/',
                 os.path.join(SAGE_DOC_SRC, "common",
                              "python{}.inv".format(python_version))),
-    'pplpy': (os.path.join(SAGE_SHARE, "doc", "pplpy"), None)}
+    'pplpy': (os.path.join(SAGE_LOCAL, "share", "doc", "pplpy"), None)}
 
 def set_intersphinx_mappings(app):
     """
