diff -Naur doc.orig/common/build_options.py doc/common/build_options.py
--- a/doc/common/build_options.py	2015-03-17 23:35:58.116605817 +1300
+++ b/doc/common/build_options.py	2015-03-17 23:40:42.687210590 +1300
@@ -4,7 +4,18 @@
 
 import os, re
 SAGE_DOC = os.environ['SAGE_DOC']
-LANGUAGES = [d for d in os.listdir(SAGE_DOC) if re.match('^[a-z][a-z]$', d)]
+LANGUAGELIST = [d for d in os.listdir(SAGE_DOC) if re.match('^[a-z][a-z]$', d)]
+try:
+    DESIRED_LANGUAGES = list(set((os.environ['LANGUAGES']).split()))
+except KeyError:
+    DESIRED_LANGUAGES = LANGUAGELIST
+LANGUAGES = []
+for d in DESIRED_LANGUAGES:
+    if d in LANGUAGELIST:
+        LANGUAGES.append(d)
+if not LANGUAGES:
+    LANGUAGES=LANGUAGELIST
+
 SPHINXOPTS = ""
 PAPER = ""
 OMIT = ["introspect"]  # docs/dirs to omit when listing and building 'all'
