diff --git a/setup.py b/setup.py
index cad807e..0bf8ecf 100755
--- a/setup.py
+++ b/setup.py
@@ -72,7 +72,7 @@ log.warn('distributions = {0}'.format(distributions))
 
 from sage_setup.find import find_python_sources
 python_packages, python_modules, cython_modules = find_python_sources(
-    SAGE_SRC, ['sage', 'sage_setup'], distributions=distributions)
+    SAGE_SRC, ['sage'], distributions=distributions)
 
 log.debug('python_packages = {0}'.format(python_packages))
 
