diff --git a/doc/common/conf.py b/doc/common/conf.py
index 70dd17f..f4a4f42 100644
--- a/doc/common/conf.py
+++ b/doc/common/conf.py
@@ -1,5 +1,5 @@
 import sys, os, sphinx
-from sage.env import SAGE_DOC_SRC, SAGE_DOC, SAGE_SRC, THEBE_DIR, SAGE_SHARE
+from sage.env import SAGE_DOC_SRC, SAGE_DOC, SAGE_SRC, THEBE_DIR, SAGE_SHARE, SAGE_LOCAL
 import sage.version
 from sage.misc.sagedoc import extlinks
 import dateutil.parser
@@ -178,7 +178,7 @@ intersphinx_mapping = {
     'python': ('https://docs.python.org/',
                 os.path.join(SAGE_DOC_SRC, "common",
                              "python{}.inv".format(python_version))),
-    'pplpy': (os.path.join(SAGE_SHARE, "doc", "pplpy"), None)}
+    'pplpy': (os.path.join(SAGE_LOCAL, "share", "doc", "pplpy", "html"), None)}
 
 def set_intersphinx_mappings(app):
     """
