diff -Naur src.orig/module_list.py src/module_list.py
--- src.orig/module_list.py	2015-06-05 11:42:18.275205274 +1200
+++ src/module_list.py	2015-06-05 11:43:39.765383736 +1200
@@ -4,6 +4,7 @@
 from glob import glob
 from distutils.extension import Extension
 from sage.env import SAGE_LOCAL
+import pkgconfig
 
 SAGE_INC = os.path.join(SAGE_LOCAL, 'include')
 
@@ -11,26 +12,8 @@
 ### BLAS setup
 #########################################################
 
-## Choose cblas library -- note -- make sure to update sage/misc/cython.py
-## if you change this!!
-if 'SAGE_BLAS' in os.environ:
-    BLAS=os.environ['SAGE_BLAS']
-    BLAS2=os.environ['SAGE_BLAS']
-elif os.path.exists('%s/lib/libatlas.so'%os.environ['SAGE_LOCAL']):
-    BLAS='cblas'
-    BLAS2='atlas'
-elif os.path.exists('/usr/lib/libcblas.dylib') or \
-     os.path.exists('/usr/lib/libcblas.so'):
-    BLAS='cblas'
-    BLAS2='cblas'
-elif os.path.exists('/usr/lib/libblas.dll.a'):
-    BLAS='gslcblas'
-    BLAS2='gslcblas'
-else:
-    # This is very slow  (?), but *guaranteed* to be available.
-    BLAS='gslcblas'
-    BLAS2='gslcblas'
-
+blas_libs = list(pkgconfig.parse('cblas')['libraries'])
+blas_dir  = list(pkgconfig.parse('cblas')['library_dir'])
 
 #########################################################
 ### Commonly used definitions and aliases
@@ -95,7 +78,7 @@
     "zn_poly", "gap",
     "gd", "png12",
     "m", "readline", "Lfunction",
-    BLAS, BLAS2,
+    ] + blas_libs + [
     "cryptominisat", "fplll", "z"]
 
 # Make a dict with library:order pairs, where the order are negative
@@ -591,42 +574,50 @@
 
     Extension('sage.gsl.callback',
               sources = ['sage/gsl/callback.pyx'],
-              libraries = ['gsl', BLAS, BLAS2],
+              library_dirs = blas_dir,
+              libraries = ['gsl'] + blas_libs,
               define_macros=[('GSL_DISABLE_DEPRECATED','1')]),
 
     Extension('sage.gsl.dwt',
               sources = ['sage/gsl/dwt.pyx'],
-              libraries=['gsl',BLAS],
+              library_dirs = blas_dir,
+              libraries=['gsl'] + blas_libs,
               define_macros=[('GSL_DISABLE_DEPRECATED','1')]),
 
     Extension('sage.gsl.fft',
               sources = ['sage/gsl/fft.pyx'],
-              libraries = ['gsl', BLAS, BLAS2],
+              library_dirs = blas_dir,
+              libraries = ['gsl'] + blas_libs,
               define_macros=[('GSL_DISABLE_DEPRECATED','1')]),
 
     Extension('sage.gsl.gsl_array',
               sources = ['sage/gsl/gsl_array.pyx'],
-              libraries=['gsl', BLAS, BLAS2],
+              library_dirs = blas_dir,
+              libraries=['gsl'] + blas_libs,
               define_macros=[('GSL_DISABLE_DEPRECATED','1')]),
 
     Extension('sage.gsl.integration',
               sources = ['sage/gsl/integration.pyx'],
               define_macros=[('GSL_DISABLE_DEPRECATED','1')],
-              libraries=['gsl',BLAS, BLAS2]),
+              library_dirs = blas_dir,
+              libraries=['gsl'] + blas_libs),
 
     Extension('sage.gsl.interpolation',
               sources = ['sage/gsl/interpolation.pyx'],
-              libraries = ['gsl', BLAS, BLAS2],
+              library_dirs = blas_dir,
+              libraries = ['gsl'] + blas_libs,
               define_macros=[('GSL_DISABLE_DEPRECATED','1')]),
 
     Extension('sage.gsl.ode',
               sources = ['sage/gsl/ode.pyx'],
-              libraries=['gsl',BLAS, BLAS2],
+              library_dirs = blas_dir,
+              libraries=['gsl'] + blas_libs,
               define_macros=[('GSL_DISABLE_DEPRECATED','1')]),
 
     Extension('sage.gsl.probability_distribution',
               sources = ['sage/gsl/probability_distribution.pyx'],
-              libraries=['gsl', BLAS, BLAS2],
+              library_dirs = blas_dir,
+              libraries=['gsl'] + blas_libs,
               define_macros=[('GSL_DISABLE_DEPRECATED','1')]),
 
     ################################
@@ -698,10 +689,11 @@
 
     Extension('sage.libs.linbox.linbox',
               sources = ['sage/libs/linbox/linbox.pyx'],
+              library_dirs = blas_dir,
               # For this to work on cygwin, linboxsage *must* be
               # before ntl.
               libraries = ['linboxsage', 'ntl', 'iml', 'linbox',
-                           'givaro', 'mpfr', 'gmp', 'gmpxx', BLAS, BLAS2],
+                           'givaro', 'mpfr', 'gmp', 'gmpxx'] + blas_libs,
               language = 'c++',
               extra_compile_args = givaro_extra_compile_args,
               depends = givaro_depends),
@@ -1012,7 +1004,8 @@
 
     Extension('sage.matrix.change_ring',
               sources = ['sage/matrix/change_ring.pyx'],
-              libraries=[BLAS, BLAS2]),
+              library_dirs = blas_dir,
+              libraries=blas_libs),
 
     Extension('sage.matrix.matrix',
               sources = ['sage/matrix/matrix.pyx']),
@@ -1028,7 +1021,8 @@
 
     Extension('sage.matrix.matrix_complex_double_dense',
               sources = ['sage/matrix/matrix_complex_double_dense.pyx'],
-              libraries=[BLAS, BLAS2]),
+              library_dirs = blas_dir,
+              libraries= blas_libs),
 
     Extension('sage.matrix.matrix_cyclo_dense',
               sources = ['sage/matrix/matrix_cyclo_dense.pyx'],
@@ -1040,7 +1034,7 @@
 
     Extension('sage.matrix.matrix_double_dense',
               sources = ['sage/matrix/matrix_double_dense.pyx'],
-              libraries=[BLAS, BLAS2]),
+              libraries=blas_libs),
 
     Extension('sage.matrix.matrix_generic_dense',
               sources = ['sage/matrix/matrix_generic_dense.pyx']),
@@ -1050,8 +1044,9 @@
 
     Extension('sage.matrix.matrix_integer_dense',
               sources = ['sage/matrix/matrix_integer_dense.pyx'],
+              library_dirs = blas_dir,
               extra_compile_args = ['-std=c99'] + m4ri_extra_compile_args,
-              libraries = ['iml', 'ntl', 'gmp', 'm', 'flint', BLAS, BLAS2],
+              libraries = ['iml', 'ntl', 'gmp', 'm', 'flint'] + blas_libs,
               depends = [SAGE_INC + '/m4ri/m4ri.h'] + flint_depends),
 
     Extension('sage.matrix.matrix_integer_sparse',
@@ -1072,14 +1067,16 @@
 
     Extension('sage.matrix.matrix_modn_dense_float',
               sources = ['sage/matrix/matrix_modn_dense_float.pyx'],
+              library_dirs = blas_dir,
               language="c++",
-              libraries = ['ntl', 'linbox', 'givaro', 'mpfr', 'gmpxx', 'gmp', BLAS, BLAS2],
+              libraries = ['ntl', 'linbox', 'givaro', 'mpfr', 'gmpxx', 'gmp'] + blas_libs,
               extra_compile_args = ['-DDISABLE_COMMENTATOR'] + givaro_extra_compile_args),
 
     Extension('sage.matrix.matrix_modn_dense_double',
               sources = ['sage/matrix/matrix_modn_dense_double.pyx'],
+              library_dirs = blas_dir,
               language="c++",
-              libraries = ['ntl', 'linbox', 'givaro', 'mpfr', 'gmpxx', 'gmp', BLAS, BLAS2],
+              libraries = ['ntl', 'linbox', 'givaro', 'mpfr', 'gmpxx', 'gmp'] + blas_libs,
               extra_compile_args = ["-D_XPG6", "-DDISABLE_COMMENTATOR"]
                     + m4ri_extra_compile_args + givaro_extra_compile_args),
 
@@ -1095,8 +1092,9 @@
 
     Extension('sage.matrix.matrix_rational_dense',
               sources = ['sage/matrix/matrix_rational_dense.pyx'],
+              library_dirs = blas_dir,
               extra_compile_args = ["-std=c99", "-D_XPG6"] + m4ri_extra_compile_args,
-              libraries = ['iml', 'ntl', 'm', 'flint', BLAS, BLAS2],
+              libraries = ['iml', 'ntl', 'm', 'flint'] + blas_libs,
               depends = [SAGE_INC + '/m4ri/m4ri.h'] + flint_depends),
 
     Extension('sage.matrix.matrix_rational_sparse',
@@ -1104,7 +1102,8 @@
 
     Extension('sage.matrix.matrix_real_double_dense',
               sources = ['sage/matrix/matrix_real_double_dense.pyx'],
-              libraries=[BLAS, BLAS2]),
+              library_dirs = blas_dir,
+              libraries= blas_libs),
 
     Extension('sage.matrix.matrix_sparse',
               sources = ['sage/matrix/matrix_sparse.pyx']),
@@ -1242,11 +1241,13 @@
 
     Extension('sage.modules.vector_complex_double_dense',
               ['sage/modules/vector_complex_double_dense.pyx'],
-              libraries = [BLAS, BLAS2]),
+              library_dirs = blas_dir,
+              libraries = blas_libs),
 
     Extension('sage.modules.vector_double_dense',
               ['sage/modules/vector_double_dense.pyx'],
-              libraries = [BLAS, BLAS2]),
+              library_dirs = blas_dir,
+              libraries = blas_libs),
 
     Extension('sage.modules.vector_integer_dense',
               sources = ['sage/modules/vector_integer_dense.pyx']),
@@ -1266,7 +1267,8 @@
 
     Extension('sage.modules.vector_real_double_dense',
               ['sage/modules/vector_real_double_dense.pyx'],
-              libraries = [BLAS, BLAS2]),
+              library_dirs = blas_dir,
+              libraries = blas_libs),
 
     ################################
     ##
@@ -1434,8 +1436,9 @@
 
     Extension('sage.rings.complex_double',
               sources = ['sage/rings/complex_double.pyx'],
+              library_dirs = blas_dir,
               extra_compile_args=["-std=c99", "-D_XPG6"],
-              libraries = (['gsl', BLAS, BLAS2, 'm'])),
+              libraries = ['gsl', 'm'] + blas_libs),
 
     Extension('sage.rings.complex_interval',
               sources = ['sage/rings/complex_interval.pyx'],
@@ -1497,7 +1500,8 @@
 
     Extension('sage.rings.real_double',
               sources = ['sage/rings/real_double.pyx'],
-              libraries = ['gsl', BLAS, BLAS2],
+              library_dirs = blas_dir,
+              libraries = ['gsl'] + blas_libs,
               define_macros=[('GSL_DISABLE_DEPRECATED','1')]),
 
     Extension('sage.rings.real_interval_absolute',
diff -Naur src.orig/sage/misc/cython.py src/sage/misc/cython.py
--- src.orig/sage/misc/cython.py	2015-06-05 11:42:18.195205097 +1200
+++ src/sage/misc/cython.py	2015-06-05 11:42:44.645263029 +1200
@@ -24,50 +24,35 @@
 from temporary_file import tmp_filename
 
 
-def cblas():
+def cblas_libs():
     """
-    Return the name of the cblas library on this system. If the environment
-    variable :envvar:`$SAGE_CBLAS` is set, just return its value. If not,
-    return ``'cblas'`` if :file:`/usr/lib/libcblas.so` or
-    :file:`/usr/lib/libcblas.dylib` exists, return ``'blas'`` if
-    :file:`/usr/lib/libblas.dll.a` exists, and return ``'gslcblas'`` otherwise.
+    Return the name of the cblas library on this system. Uses pkgconfig to determine
+    the libray used.
 
     EXAMPLES::
 
-        sage: sage.misc.cython.cblas() # random -- depends on OS, etc.
+        sage: sage.misc.cython.cblas_libs() # random -- depends on OS, etc.
         'cblas'
     """
-    if 'SAGE_CBLAS' in os.environ:
-        return os.environ['SAGE_CBLAS']
-    elif os.path.exists('/usr/lib/libcblas.dylib') or \
-         os.path.exists('/usr/lib/libcblas.so'):
-        return 'cblas'
-    elif os.path.exists('/usr/lib/libblas.dll.a'):   # untested.
-        return 'blas'
-    else:
-        # This is very slow (?), but *guaranteed* to be available.
-        return 'gslcblas'
 
-# In case of ATLAS we need to link against cblas as well as atlas
-# In the other cases we just return the same library name as cblas()
-# which is fine for the linker
-#
-# We should be using the Accelerate FrameWork on OS X, but that requires
-# some magic due to distutils having ridden on the short bus :)
-def atlas():
+    import pkgconfig
+
+    return list(pkgconfig.parse('cblas')['libraries'])
+
+def cblas_dir():
     """
-    Returns the name of the ATLAS library to use. On Darwin or Cygwin, this is
-    ``'blas'``, and otherwise it is ``'atlas'``.
+    Return the path where the cblas library is located on this system if it is
+    different from the normal search path. Uses pkgconfig to determine the path.
 
     EXAMPLES::
 
-        sage: sage.misc.cython.atlas() # random -- depends on OS
-        'atlas'
+        sage: sage.misc.cython.cblas_dir() # random -- depends on OS, etc.
+        ''
     """
-    if UNAME == "Darwin" or "CYGWIN" in UNAME:
-        return 'blas'
-    else:
-        return 'atlas'
+
+    import pkgconfig
+
+    return list(pkgconfig.parse('cblas')['library_dir'])
 
 include_dirs = [os.path.join(SAGE_LOCAL,'include'),
                 os.path.join(SAGE_LOCAL,'include','python'+platform.python_version().rsplit('.', 1)[0]),
@@ -78,7 +63,7 @@
 
 
 standard_libs = ['mpfr', 'gmp', 'gmpxx', 'stdc++', 'pari', 'm', \
-                 'ec', 'gsl', cblas(), atlas(), 'ntl']
+                 'ec', 'gsl'] + cblas_libs() + ['ntl']
 
 offset = 0
 
@@ -206,7 +191,6 @@
         'm',
         'ec',
         'gsl',
-        '...blas',
         ...,
         'ntl'],
         ['.../include',
@@ -216,8 +200,8 @@
         '...',
         '.../sage/gsl'],
         'c',
-        [], ['-w', '-O2'])
-        sage: s, libs, inc, lang, f, args = pyx_preparse("# clang c++\n #clib foo\n # cinclude bar\n")
+        [], ['-w', '-O2'],...)
+        sage: s, libs, inc, lang, f, args, libdirs = pyx_preparse("# clang c++\n #clib foo\n # cinclude bar\n")
         sage: lang
         'c++'
 
@@ -228,7 +212,7 @@
         'pari',
         'm',
         'ec',
-        'gsl', '...blas', ...,
+        'gsl', ...,
         'ntl']
         sage: libs[1:] == sage.misc.cython.standard_libs
         True
@@ -242,7 +226,7 @@
         '...',
         '.../sage/gsl']
 
-        sage: s, libs, inc, lang, f, args = pyx_preparse("# cargs -O3 -ggdb\n")
+        sage: s, libs, inc, lang, f, args, libdirs = pyx_preparse("# cargs -O3 -ggdb\n")
         sage: args
         ['-w', '-O2', '-O3', '-ggdb']
 
@@ -269,9 +253,10 @@
     s = """\ninclude "cdefs.pxi"\n""" + s
     s = """\ninclude "interrupt.pxi"  # ctrl-c interrupt block support\ninclude "stdsage.pxi"\n""" + s
     args, s = parse_keywords('cargs', s)
+    libdirs = cblas_dir()
     args = ['-w','-O2'] + args
 
-    return s, libs, inc, lang, additional_source_files, args
+    return s, libs, inc, lang, additional_source_files, args, libdirs
 
 ################################################################
 # If the user attaches a .spyx file and changes it, we have
@@ -408,7 +393,7 @@
 
     F = open(filename).read()
 
-    F, libs, includes, language, additional_source_files, extra_args = pyx_preparse(F)
+    F, libs, includes, language, additional_source_files, extra_args, libdirs = pyx_preparse(F)
 
     # add the working directory to the includes so custom headers etc. work
     includes.append(os.path.split(os.path.splitext(filename)[0])[0])
@@ -448,19 +433,17 @@
 
 from sage.env import SAGE_LOCAL
 
-extra_link_args =  ['-L' + SAGE_LOCAL + '/lib']
 extra_compile_args = %s
 
 ext_modules = [Extension('%s', sources=['%s.%s', %s],
                      libraries=%s,
-                     library_dirs=[SAGE_LOCAL + '/lib/'],
+                     library_dirs=[SAGE_LOCAL + '/lib/', '%s'],
                      extra_compile_args = extra_compile_args,
-                     extra_link_args = extra_link_args,
                      language = '%s' )]
 
 setup(ext_modules = ext_modules,
       include_dirs = %s)
-    """%(extra_args, name, name, extension, additional_source_files, libs, language, includes)
+    """%(extra_args, name, name, extension, additional_source_files, libs, libdirs, language, includes)
     open('%s/setup.py'%build_dir,'w').write(setup)
 
     cython_include = ' '.join(["-I '%s'"%x for x in includes if len(x.strip()) > 0 ])
