diff --git a/doc/en/reference/misc/index.rst b/doc/en/reference/misc/index.rst
index 8b438ad..e5c2ea7 100644
--- a/doc/en/reference/misc/index.rst
+++ b/doc/en/reference/misc/index.rst
@@ -232,8 +232,6 @@ Distribution
 .. toctree::
    :maxdepth: 1
 
-   sage/misc/package
-   sage/misc/dist
    sage/features
    sage/features/bliss
    sage/features/csdp
diff --git a/sage/doctest/control.py b/sage/doctest/control.py
index d408b18..e65ad2f 100644
--- a/sage/doctest/control.py
+++ b/sage/doctest/control.py
@@ -348,17 +348,6 @@ class DocTestController(SageObject):
                 # Special case to run all optional tests
                 options.optional = True
             else:
-                # We replace the 'optional' tag by all optional
-                # packages for which the installed version matches the
-                # latest available version (this implies in particular
-                # that the package is actually installed).
-                if 'optional' in options.optional:
-                    options.optional.discard('optional')
-                    from sage.misc.package import list_packages
-                    for pkg in list_packages('optional', local=True).values():
-                        if pkg['installed_version'] == pkg['remote_version']:
-                            options.optional.add(pkg['name'])
-
                 # Check that all tags are valid
                 for o in options.optional:
                     if not optionaltag_regex.search(o):
diff --git a/sage/misc/all.py b/sage/misc/all.py
index d3ea945..6dc25d0 100644
--- a/sage/misc/all.py
+++ b/sage/misc/all.py
@@ -46,12 +46,6 @@ from .mrange import xmrange, mrange, xmrange_iter, mrange_iter, cartesian_produc
 
 from .fpickle import pickle_function, unpickle_function
 
-from .dist import install_scripts
-
-lazy_import('sage.misc.package', ('installed_packages', 'is_package_installed',
-                                  'standard_packages', 'optional_packages',
-                                  'experimental_packages', 'package_versions'))
-
 lazy_import('sage.misc.pager', 'pager')
 
 lazy_import('sage.misc.sagedoc', ['browse_sage_doc',
diff --git a/sage/tests/cmdline.py b/sage/tests/cmdline.py
index 3838866..efb0b94 100644
--- a/sage/tests/cmdline.py
+++ b/sage/tests/cmdline.py
@@ -10,13 +10,9 @@ test.sage
 /path/to/test.sage
 test.spyx
 /path/to/test.spyx
---advanced
 -c
 --cython
---dev
 --ecl
---experimental
---fixdoctests
 --gap
 --gdb
 --gp
@@ -29,7 +25,6 @@ test.spyx
 --maxima
 --min
 --mwrank
---optional
 --preparse
 --python
 --python3
@@ -40,10 +35,8 @@ test.spyx
 --ipynb2rst
 --rst2txt
 --rst2sws
---sh
 --singular
 --sqlite3
---standard
 --startuptime
 -t
 -v
@@ -185,14 +178,6 @@ def test_executable(args, input="", timeout=100.0, pydebug_ignore_warnings=False
         sage: ret
         0
 
-        sage: (out, err, ret) = test_executable(["sage", "--advanced"])
-        sage: out.find("search through the Sage documentation") >= 0
-        True
-        sage: err
-        ''
-        sage: ret
-        0
-
     Basic information about the Sage installation::
 
         sage: (out, err, ret) = test_executable(["sage", "-v"])
@@ -427,64 +412,8 @@ def test_executable(args, input="", timeout=100.0, pydebug_ignore_warnings=False
         ...
         RuntimeError: refusing to run doctests...
 
-    Now run a test for the fixdoctests script and, in particular, check that the
-    issues raised in :trac:`10589` are fixed. We have to go to slightly silly
-    lengths to doctest the output.::
-
-        sage: test='r\"\"\"Add a doc-test for the fixdoctest command line option and, in particular, check that\n:trac:`10589` is fixed.\n\nEXAMPLES::\n\n    sage: 1+1              # incorrect output\n    3\n    sage: m=matrix(ZZ,3)   # output when none is expected\n    [0 0 0]\n    [0 0 0]\n    [1 0 0]\n    sage: (2/3)*m          # no output when it is expected\n    sage: mu=PartitionTuple([[4,4],[3,3,2,1],[1,1]])   # output when none is expected\n    [4, 4, 3, 3, 2, 1, 1]\n    sage: mu.pp()          # uneven indentation\n    ****\n    ****\n    sage: PartitionTuples.options(convention="French")\n    sage: mu.pp()         # fix doctest with uneven indentation\n    sage: PartitionTuples.options._reset()\n\"\"\"\n'
-        sage: test_file = os.path.join(tmp_dir(), 'test_file.py')
-        sage: F = open(test_file, 'w')
-        sage: _ = F.write(test)
-        sage: F.close()
-        sage: (out, err, ret) = test_executable(["sage", "--fixdoctests", test_file])
-        sage: with open(test_file, 'r') as f:
-        ....:     fixed_test = f.read()
-        sage: import difflib
-        sage: list(difflib.unified_diff(test.splitlines(), fixed_test.splitlines()))[2:-1]
-        ['@@ -4,18 +4,23 @@\n',
-         ' EXAMPLES::',
-         ' ',
-         '     sage: 1+1              # incorrect output',
-         '-    3',
-         '+    2',
-         '     sage: m=matrix(ZZ,3)   # output when none is expected',
-         '+    sage: (2/3)*m          # no output when it is expected',
-         '     [0 0 0]',
-         '     [0 0 0]',
-         '-    [1 0 0]',
-         '-    sage: (2/3)*m          # no output when it is expected',
-         '+    [0 0 0]',
-         '     sage: mu=PartitionTuple([[4,4],[3,3,2,1],[1,1]])   # output when none is expected',
-         '-    [4, 4, 3, 3, 2, 1, 1]',
-         '     sage: mu.pp()          # uneven indentation',
-         '-    ****',
-         '-    ****',
-         '+       ****   ***   *',
-         '+       ****   ***   *',
-         '+              **',
-         '+              *',
-         '     sage: PartitionTuples.options(convention="French")',
-         '     sage: mu.pp()         # fix doctest with uneven indentation',
-         '+    *',
-         '+    **',
-         '+    ****   ***   *',
-         '+    ****   ***   *',
-         '     sage: PartitionTuples.options._reset()']
-
     Test external programs being called by Sage::
 
-        sage: (out, err, ret) = test_executable(["sage", "--sh"], "echo Hello World\nexit 42\n")
-        sage: out.find("Hello World\n") >= 0
-        True
-        sage: ret
-        42
-
-        sage: (out, err, ret) = test_executable(["sage", "--sh", "-c", "echo Hello World; exit 42"])
-        sage: out.find("Hello World\n") >= 0
-        True
-        sage: ret
-        42
-
         sage: (out, err, ret) = test_executable(["sage", "--ipython"], "\n3**33\n", pydebug_ignore_warnings=True)
         sage: out.find("5559060566555523") >= 0
         True
@@ -620,32 +549,6 @@ def test_executable(args, input="", timeout=100.0, pydebug_ignore_warnings=False
         sage: ret
         0
 
-    Check some things requiring an internet connection::
-
-        sage: (out, err, ret) = test_executable(["sage", "--standard"])  # optional - internet
-        sage: out.find("cython") >= 0  # optional - internet
-        True
-        sage: err  # optional - internet
-        ''
-        sage: ret  # optional - internet
-        0
-
-        sage: (out, err, ret) = test_executable(["sage", "--optional"])  # optional - internet
-        sage: out.find("database_cremona_ellcurve") >= 0  # optional - internet
-        True
-        sage: err  # optional - internet
-        ''
-        sage: ret  # optional - internet
-        0
-
-        sage: (out, err, ret) = test_executable(["sage", "--experimental"])  # optional - internet
-        sage: out.find("valgrind") >= 0  # optional - internet
-        True
-        sage: err  # optional - internet
-        ''
-        sage: ret  # optional - internet
-        0
-
     Check an illegal command line option.  This outputs an error to stdout,
     but we allow stderr in case this changes in the future::
 
diff --git a/sage_setup/optional_extension.py b/sage_setup/optional_extension.py
index 546191f..dc72b54 100644
--- a/sage_setup/optional_extension.py
+++ b/sage_setup/optional_extension.py
@@ -19,10 +19,7 @@ package which must be installed.
 
 
 from distutils.extension import Extension
-from sage.misc.package import list_packages
-
-all_packages = list_packages(local=True)
-
+import os
 
 class CythonizeExtension(Extension):
     """
@@ -76,14 +73,7 @@ def OptionalExtension(*args, **kwds):
         condition = kwds.pop("condition")
     except KeyError:
         pkg = kwds.pop("package")
-        from sage.misc.package import is_package_installed
-        try:
-            pkginfo = all_packages[pkg]
-        except KeyError:
-            # Might be an installed old-style package
-            condition = is_package_installed(pkg)
-        else:
-            condition = (pkginfo["installed_version"] == pkginfo["remote_version"])
+        condition = os.getenv('WANT_%s'%pkg,False)
 
     if condition:
         return Extension(*args, **kwds)
