--- setup.py.orig	2014-06-25 11:41:28.093445850 +1200
+++ setup.py	2014-06-25 11:41:49.953492119 +1200
@@ -556,20 +556,6 @@
 
 
 #########################################################
-### Clean
-#########################################################
-
-print('Cleaning up stale installed files....')
-t = time.time()
-from sage_setup.clean import clean_install_dir
-output_dirs = SITE_PACKAGES + glob.glob(os.path.join(SAGE_SRC, 'build', 'lib*'))
-for output_dir in output_dirs:
-    print('- cleaning {0}'.format(output_dir))
-    clean_install_dir(output_dir, python_packages, python_modules, ext_modules)
-print('Finished cleaning, time: %.2f seconds.' % (time.time() - t))
-
-
-#########################################################
 ### Distutils
 #########################################################
 
--- sage/doctest/control.py.orig	2014-07-29 13:38:28.599162834 +1200
+++ sage/doctest/control.py	2014-07-29 13:38:55.929204730 +1200
@@ -461,7 +461,6 @@
         def all_files():
             from glob import glob
             self.files.append(opj(SAGE_SRC, 'sage'))
-            self.files.append(opj(SAGE_SRC, 'sage_setup'))
             self.files.append(opj(SAGE_SRC, 'doc', 'common'))
             self.files.extend(glob(opj(SAGE_SRC, 'doc', '[a-z][a-z]')))
             self.options.sagenb = True
