diff -Naur src.orig/sage/doctest/control.py src/sage/doctest/control.py
--- src.orig/sage/doctest/control.py	2015-07-13 11:29:04.490770881 +1200
+++ src/sage/doctest/control.py	2015-07-13 11:30:28.920909215 +1200
@@ -545,7 +545,6 @@
         def all_files():
             from glob import glob
             self.files.append(opj(SAGE_SRC, 'sage'))
-            self.files.append(opj(SAGE_SRC, 'sage_setup'))
             self.files.append(opj(SAGE_SRC, 'doc', 'common'))
             self.files.extend(glob(opj(SAGE_SRC, 'doc', '[a-z][a-z]')))
             self.options.sagenb = True
diff -Naur src.orig/setup.py src/setup.py
--- src.orig/setup.py	2015-07-13 11:29:04.520770929 +1200
+++ src/setup.py	2015-07-13 11:32:03.141062990 +1200
@@ -620,21 +620,6 @@
 
 
 #########################################################
-### Clean
-#########################################################
-
-print('Cleaning up stale installed files....')
-t = time.time()
-from sage_setup.clean import clean_install_dir
-output_dirs = SITE_PACKAGES + glob.glob(os.path.join(SAGE_SRC, 'build', 'lib*'))
-for output_dir in output_dirs:
-    print('- cleaning {0}'.format(output_dir))
-    clean_install_dir(output_dir, python_packages, python_modules,
-            ext_modules, python_data_files)
-print('Finished cleaning, time: %.2f seconds.' % (time.time() - t))
-
-
-#########################################################
 ### Distutils
 #########################################################
 
