diff --git a/sage/repl/rich_output/display_manager.py b/sage/repl/rich_output/display_manager.py
index f76be0d..4073e76 100644
--- a/sage/repl/rich_output/display_manager.py
+++ b/sage/repl/rich_output/display_manager.py
@@ -745,12 +745,10 @@ class DisplayManager(SageObject):
             offline threejs graphics
         """
         if online:
-            from sage.misc.package import installed_packages
-            version = installed_packages()['threejs'].split('.')[0]
             return """
-<script src="https://cdn.jsdelivr.net/gh/mrdoob/three.js@{0}/build/three.min.js"></script>
-<script src="https://cdn.jsdelivr.net/gh/mrdoob/three.js@{0}/examples/js/controls/OrbitControls.js"></script>
-            """.format(version)
+<script src="https://cdn.jsdelivr.net/gh/mrdoob/three.js@r80/build/three.min.js"></script>
+<script src="https://cdn.jsdelivr.net/gh/mrdoob/three.js@r80/examples/js/controls/OrbitControls.js"></script>
+            """
         try:
             return self._backend.threejs_offline_scripts()
         except AttributeError:
