diff --git a/src/sage/misc/sphinxify.py b/src/sage/misc/sphinxify.py
index 4270294..fa62413 100644
--- a/sage/misc/sphinxify.py
+++ b/sage/misc/sphinxify.py
@@ -26,7 +26,6 @@ import os
 import re
 import shutil
 from tempfile import mkdtemp
-from sage.env import SAGE_DOC_SRC
 from sphinx.application import Sphinx
 
 
@@ -82,12 +81,35 @@ def sphinxify(docstring, format='html'):
     with open(rst_name, 'w') as filed:
         filed.write(docstring)
 
-    confdir = os.path.join(SAGE_DOC_SRC, 'en', 'introspect')
-
-    open(os.path.join(srcdir, 'docutils.conf'), 'w').write(r"""
+    confdir = os.path.join(srcdir, 'en' , 'introspect')
+    os.makedirs(confdir)
+    with open(os.path.join(confdir, 'conf.py'), 'w') as filed:
+        filed.write(r"""
+extensions = ['sphinx.ext.autodoc', 'sphinx.ext.mathjax', 'sphinx.ext.todo', 'sphinx.ext.extlinks']
+
+templates_path = ['templates']
+html_static_path = ['static']
+
+html_use_modindex = False
+html_use_index = False
+html_split_index = False
+html_copy_source = False
+
+todo_include_todos = True""")
+    templatesdir = os.path.join(confdir, 'templates')
+    os.makedirs(templatesdir)
+    with open(os.path.join(templatesdir, 'layout.html'), 'w') as filed:
+        filed.write(r"""<div class="docstring">
+    {% block body %} {% endblock %}
+</div>""")
+    staticdir = os.path.join(confdir, 'static')
+    os.makedirs(staticdir)
+    with open(os.path.join(staticdir, 'empty'), 'w') as filed: pass
+
+    with open(os.path.join(srcdir, 'docutils.conf'), 'w') as filed:
+        filed.write(r"""
 [parsers]
-smart_quotes = no
-""")
+smart_quotes = no""")
     doctreedir = os.path.join(srcdir, 'doctrees')
     confoverrides = {'html_context': {}, 'master_doc': 'docstring'}
