diff --git a/sage/interfaces/maxima.py b/sage/interfaces/maxima.py
index bc00756..9575d35 100644
--- a/sage/interfaces/maxima.py
+++ b/sage/interfaces/maxima.py
@@ -473,7 +473,7 @@ import pexpect
 from random import randrange
 
 from sage.cpython.string import bytes_to_str
-from sage.env import DOT_SAGE, SAGE_LOCAL
+from sage.env import DOT_SAGE, SAGE_ETC
 from sage.misc.misc import ECL_TMP
 
 from .expect import (Expect, ExpectElement, FunctionElement,
@@ -525,7 +525,7 @@ class Maxima(MaximaAbstract, Expect):
         # setting inchar and outchar..
         eval_using_file_cutoff = 256
         self.__eval_using_file_cutoff = eval_using_file_cutoff
-        STARTUP = os.path.join(SAGE_LOCAL,'bin','sage-maxima.lisp')
+        STARTUP = os.path.join(SAGE_ETC, 'sage-maxima.lisp')
 
         # We set maxima's configuration directory to $DOT_SAGE/maxima
         # This avoids that sage's maxima inadvertently loads
