diff --git a/sage/repl/ipython_kernel/install.py b/sage/repl/ipython_kernel/install.py
index dc3dd6e..3717aab 100644
--- a/sage/repl/ipython_kernel/install.py
+++ b/sage/repl/ipython_kernel/install.py
@@ -180,16 +180,14 @@ class SageKernelSpec(object):
             sage: from sage.repl.ipython_kernel.install import SageKernelSpec
             sage: spec = SageKernelSpec(prefix=tmp_dir())
             sage: spec._kernel_cmd()
-            ['/.../sage',
-             '--python',
+            ['python',
              '-m',
              'sage.repl.ipython_kernel',
              '-f',
              '{connection_file}']
         """
         return [
-            os.path.join(SAGE_LOCAL, 'bin', 'sage'),
-            '--python',
+            'python',
             '-m', 'sage.repl.ipython_kernel',
             '-f', '{connection_file}',
         ]
