--- module_list.py.orig	2014-08-20 11:59:49.402371259 +1200
+++ module_list.py	2014-08-20 12:04:25.582794533 +1200
@@ -45,7 +45,7 @@
 singular_depends = [SAGE_INC + '/libsingular.h']
 givaro_depends = [SAGE_INC + '/givaro/givconfig.h']
 
-singular_incs = [SAGE_INC + '/singular', SAGE_INC + '/factory']
+singular_incs = [SAGE_INC + '/singular']
 
 #########################################################
 ### M4RI flags
@@ -66,7 +66,8 @@
 ### Givaro flags
 #########################################################
 
-givaro_extra_compile_args =['-D__STDC_LIMIT_MACROS']
+givaro_extra_compile_args =['-D__STDC_LIMIT_MACROS', '-DNDEBUG']
+singular_extra_compile_args = ['-DNDEBUG']
 
 #########################################################
 ### PolyBoRi settings
@@ -110,21 +111,24 @@
               libraries = singular_libs,
               language="c++",
               include_dirs = singular_incs,
-              depends = singular_depends),
+              depends = singular_depends,
+              extra_compile_args = singular_extra_compile_args),
 
     Extension('sage.algebras.letterplace.free_algebra_element_letterplace',
               sources = ['sage/algebras/letterplace/free_algebra_element_letterplace.pyx'],
               libraries = singular_libs,
               language="c++",
               include_dirs = singular_incs,
-              depends = singular_depends),
+              depends = singular_depends,
+              extra_compile_args = singular_extra_compile_args),
 
     Extension('sage.algebras.letterplace.letterplace_ideal',
               sources = ['sage/algebras/letterplace/letterplace_ideal.pyx'],
               libraries = singular_libs,
               language="c++",
               include_dirs = singular_incs,
-              depends = singular_depends),
+              depends = singular_depends,
+              extra_compile_args = singular_extra_compile_args),
 
     Extension('sage.algebras.quatalg.quaternion_algebra_cython',
                sources = ['sage/algebras/quatalg/quaternion_algebra_cython.pyx'],
@@ -770,28 +774,31 @@
               language="c++",
               include_dirs = singular_incs,
               depends = singular_depends + givaro_depends,
-              extra_compile_args = givaro_extra_compile_args),
+              extra_compile_args = givaro_extra_compile_args + singular_extra_compile_args),
 
     Extension('sage.libs.singular.polynomial',
               sources = ['sage/libs/singular/polynomial.pyx'],
               libraries = singular_libs,
               language="c++",
               include_dirs = singular_incs,
-              depends = singular_depends),
+              depends = singular_depends,
+              extra_compile_args = singular_extra_compile_args),
 
     Extension('sage.libs.singular.ring',
               sources = ['sage/libs/singular/ring.pyx'],
               libraries = singular_libs,
               language="c++",
               include_dirs = singular_incs,
-              depends = singular_depends),
+              depends = singular_depends,
+              extra_compile_args = singular_extra_compile_args),
 
     Extension('sage.libs.singular.groebner_strategy',
               sources = ['sage/libs/singular/groebner_strategy.pyx'],
               libraries = singular_libs,
               language="c++",
               include_dirs = singular_incs,
-              depends = singular_depends),
+              depends = singular_depends,
+              extra_compile_args = singular_extra_compile_args),
 
     Extension('sage.libs.singular.function',
               sources = ['sage/libs/singular/function.pyx'],
@@ -799,14 +806,15 @@
               language="c++",
               include_dirs = singular_incs,
               depends = singular_depends,
-              extra_compile_args = givaro_extra_compile_args),
+              extra_compile_args = givaro_extra_compile_args + singular_extra_compile_args),
 
     Extension('sage.libs.singular.option',
               sources = ['sage/libs/singular/option.pyx'],
               libraries = singular_libs,
               language="c++",
               include_dirs = singular_incs,
-              depends = singular_depends),
+              depends = singular_depends,
+              extra_compile_args = singular_extra_compile_args),
 
     Extension('sage.libs.symmetrica.symmetrica',
               sources = ["sage/libs/symmetrica/%s"%s for s in ["symmetrica.pyx"]],
@@ -1103,7 +1111,8 @@
               libraries = singular_libs,
               language="c++",
               include_dirs = singular_incs,
-              depends = singular_depends),
+              depends = singular_depends,
+              extra_compile_args = singular_extra_compile_args),
 
     Extension('sage.matrix.matrix_rational_dense',
               sources = ['sage/matrix/matrix_rational_dense.pyx'],
@@ -1794,7 +1803,8 @@
               libraries = singular_libs,
               language="c++",
               include_dirs = singular_incs,
-              depends = singular_depends),
+              depends = singular_depends,
+              extra_compile_args = singular_extra_compile_args),
 
     Extension('sage.rings.polynomial.plural',
               sources = ['sage/rings/polynomial/plural.pyx'],
@@ -1809,7 +1819,8 @@
               libraries = singular_libs,
               language="c++",
               include_dirs = singular_incs,
-              depends = singular_depends),
+              depends = singular_depends,
+              extra_compile_args = singular_extra_compile_args),
 
     Extension('sage.rings.polynomial.multi_polynomial_ring_generic',
               sources = ['sage/rings/polynomial/multi_polynomial_ring_generic.pyx']),
