diff -Naur src.orig/sage/misc/lazy_import_cache.py src/sage/misc/lazy_import_cache.py
--- src.orig/sage/misc/lazy_import_cache.py	2015-07-29 10:58:51.217807168 +1200
+++ src/sage/misc/lazy_import_cache.py	2015-07-29 10:59:27.337873956 +1200
@@ -6,7 +6,7 @@
 
 import os
 
-from ..env import SAGE_SRC, DOT_SAGE
+from ..env import SAGE_VERSION, DOT_SAGE
 
 def get_cache_file():
     """
@@ -30,6 +30,6 @@
         True
         sage: sage.misc.lazy_import_cache.DOT_SAGE = OLD
     """
-    mangled = os.path.realpath(SAGE_SRC).replace(os.sep, '_')
+    mangled = os.path.realpath(SAGE_VERSION).replace(os.sep, '_')
     return os.path.join(DOT_SAGE, 'cache',
                         "%s-lazy_import_cache.pickle" % mangled)
