diff --git a/sage_setup/docbuild/__init__.py b/sage_setup/docbuild/__init__.py
index b07e9c1..0786c08 100644
--- a/sage_docbuild/__init__.py
+++ b/sage_docbuild/__init__.py
@@ -59,7 +59,7 @@ from sage.misc.cachefunc import cached_method
 from sage.misc.misc import sage_makedirs
 from sage.env import SAGE_DOC_SRC, SAGE_DOC, SAGE_SRC, DOT_SAGE
 
-from .build_options import (LANGUAGES, SPHINXOPTS, PAPER, OMIT,
+from sage_docbuild.build_options import (LANGUAGES, SPHINXOPTS, PAPER, OMIT,
      PAPEROPTS, ALLSPHINXOPTS, NUM_THREADS, WEBSITESPHINXOPTS,
      INCREMENTAL_BUILD, ABORT_ON_ERROR)
 
diff --git a/sage_setup/docbuild/__main__.py b/sage_setup/docbuild/__main__.py
index 031df43..e68b438 100644
--- a/sage_docbuild/__main__.py
+++ b/sage_docbuild/__main__.py
@@ -1,2 +1,2 @@
-from . import main
+from sage_docbuild import main
 main()
 
