--- module_list.py.orig	2010-08-30 22:48:44.781370340 +1200
+++ module_list.py	2010-08-30 22:54:58.297371782 +1200
@@ -459,6 +459,7 @@
               sources = ["sage/libs/ecl.pyx"],
               libraries = ["ecl"],
               include_dirs = [SAGE_ROOT+'/local/include/ecl/'],
+              extra_compile_args=["-mno-altivec", "-mabi=no-altivec"],
               depends = [SAGE_ROOT + '/local/include/ecl/ecl.h']),
 
     Extension('sage.libs.flint.flint',
