--- sage/numerical/test.py.orig	2010-02-04 18:36:20.845887211 +0000
+++ sage/numerical/test.py	2010-02-04 18:36:54.362885056 +0000
@@ -9,7 +9,7 @@
 sage: from scipy.sparse.linalg import dsolve   
 sage: from scipy import interpolate
 sage: from scipy import sparse    
-sage: import arpack
+sage: from scipy.sparse.linalg.eigen.arpack import speigs
 
 #Test arpack
 #This matrix is the finite difference approximation to
@@ -24,7 +24,7 @@
 sage: n[2,:]=-1
 sage: A=sparse.spdiags(n,[-1,0,1],int(500),int(500))
 sage: A=aslinearoperator(A)
-sage: e,v=arpack.speigs.ARPACK_eigs(A.matvec,500,6,which='SM')
+sage: e,v=speigs.ARPACK_eigs(A.matvec,500,6,which='SM')
 sage: e[0]*float(501/pi)**2
 0.999............
 """
