diff -Naur src.orig/sage/misc/cython.py src/sage/misc/cython.py
--- src.orig/sage/misc/cython.py	2017-02-13 15:53:47.732447432 +1300
+++ src/sage/misc/cython.py	2017-02-13 17:13:51.382955834 +1300
@@ -307,9 +307,12 @@
     Before :trac:`12975`, it would have been needed to write ``#clang c++``,
     but upper case ``C++`` has resulted in an error::
 
+        sage: import pkgconfig
+        sage: singular_pc = pkgconfig.parse('Singular')
         sage: code = [
         ....: "#clang C++",
-        ....: "#clib m readline Singular givaro ntl gmpxx gmp",
+        ....: "#clib " + " ".join(singular_pc['libraries']),
+        ....: "#cargs " + pkgconfig.cflags('Singular'),
         ....: "from sage.rings.polynomial.multi_polynomial_libsingular cimport MPolynomial_libsingular",
         ....: "from sage.libs.singular.polynomial cimport singular_polynomial_pow",
         ....: "def test(MPolynomial_libsingular p):",
