diff --git a/sage/env.py b/sage/env.py
index b09d6f6..524d936 100644
--- a/sage/env.py
+++ b/sage/env.py
@@ -117,13 +117,11 @@ _add_variable_or_fallback('HOSTNAME',        socket.gethostname())
 _add_variable_or_fallback('LOCAL_IDENTIFIER','$HOSTNAME.%s'%os.getpid())
 
 # bunch of sage directories and files
-_add_variable_or_fallback('SAGE_ROOT',       None)
-_add_variable_or_fallback('SAGE_LOCAL',      None)
-_add_variable_or_fallback('SAGE_ETC',        opj('$SAGE_LOCAL', 'etc'))
-_add_variable_or_fallback('SAGE_INC',        opj('$SAGE_LOCAL', 'include'))
-_add_variable_or_fallback('SAGE_SHARE',      opj('$SAGE_LOCAL', 'share'))
-
-_add_variable_or_fallback('SAGE_SRC',        opj('$SAGE_ROOT', 'src'))
+_add_variable_or_fallback('SAGE_ROOT',       opj('@GENTOO_PORTAGE_EPREFIX@/usr', 'share', 'sage'))
+_add_variable_or_fallback('SAGE_LOCAL',      opj('@GENTOO_PORTAGE_EPREFIX@/usr'))
+_add_variable_or_fallback('SAGE_ETC',        opj('@GENTOO_PORTAGE_EPREFIX@/etc'))
+_add_variable_or_fallback('SAGE_INC',        opj('@GENTOO_PORTAGE_EPREFIX@/usr', 'include'))
+_add_variable_or_fallback('SAGE_SHARE',      opj('$SAGE_LOCAL', 'share', 'sage'))
 
 try:
     sitepackages_dirs = site.getsitepackages()
@@ -133,19 +131,12 @@ except AttributeError:  # in case of use inside virtualenv
 _add_variable_or_fallback('SITE_PACKAGES',   sitepackages_dirs)
 
 _add_variable_or_fallback('SAGE_LIB',        SITE_PACKAGES[0])
+_add_variable_or_fallback('SAGE_SRC',        opj('$SAGE_LIB'))
 
-# Used by sage/misc/package.py.  Should be SAGE_SRC_ROOT in VPATH.
-_add_variable_or_fallback('SAGE_PKGS', opj('$SAGE_ROOT', 'build', 'pkgs'))
-
-
-_add_variable_or_fallback('SAGE_EXTCODE',    opj('$SAGE_SHARE', 'sage', 'ext'))
-_add_variable_or_fallback('SAGE_LOGS',       opj('$SAGE_ROOT', 'logs', 'pkgs'))
-_add_variable_or_fallback('SAGE_SPKG_INST',  opj('$SAGE_LOCAL', 'var', 'lib', 'sage', 'installed'))
-_add_variable_or_fallback('SAGE_DOC_SRC',    opj('$SAGE_SRC', 'doc'))
-_add_variable_or_fallback('SAGE_DOC',        opj('$SAGE_SHARE', 'doc', 'sage'))
+_add_variable_or_fallback('SAGE_EXTCODE',    opj('$SAGE_SHARE', 'ext'))
+_add_variable_or_fallback('SAGE_DOC',        opj('$SAGE_LOCAL', 'share', 'doc', 'sage'))
+_add_variable_or_fallback('SAGE_DOC_SRC',    opj('$SAGE_DOC'))
 _add_variable_or_fallback('DOT_SAGE',        opj(os.environ.get('HOME','$SAGE_ROOT'), '.sage'))
-_add_variable_or_fallback('SAGE_DOT_GIT',    opj('$SAGE_ROOT', '.git'))
-_add_variable_or_fallback('SAGE_DISTFILES',  opj('$SAGE_ROOT', 'upstream'))
 
 # misc
 _add_variable_or_fallback('SAGE_URL',                'http://sage.math.washington.edu/sage/')
@@ -164,16 +155,15 @@ _add_variable_or_fallback('CONWAY_POLYNOMIALS_DATA_DIR',  opj('$SAGE_SHARE','con
 _add_variable_or_fallback('GRAPHS_DATA_DIR',  opj('$SAGE_SHARE','graphs'))
 _add_variable_or_fallback('ELLCURVE_DATA_DIR',opj('$SAGE_SHARE','ellcurves'))
 _add_variable_or_fallback('POLYTOPE_DATA_DIR',opj('$SAGE_SHARE','reflexive_polytopes'))
-_add_variable_or_fallback('GAP_ROOT_DIR',     opj('$SAGE_LOCAL','gap','latest'))
-_add_variable_or_fallback('THEBE_DIR',        opj('$SAGE_SHARE','thebe'))
+_add_variable_or_fallback('THEBE_DIR',        opj('$SAGE_LOCAL','share','thebe'))
 _add_variable_or_fallback('COMBINATORIAL_DESIGN_DATA_DIR', opj('$SAGE_SHARE', 'combinatorial_designs'))
 _add_variable_or_fallback('CREMONA_MINI_DATA_DIR', opj('$SAGE_SHARE','cremona'))
 _add_variable_or_fallback('CREMONA_LARGE_DATA_DIR', opj('$SAGE_SHARE', 'cremona'))
-_add_variable_or_fallback('JMOL_DIR', opj('$SAGE_SHARE', 'jmol'))
-_add_variable_or_fallback('JSMOL_DIR', SAGE_ENV['JMOL_DIR'])
-_add_variable_or_fallback('MATHJAX_DIR', opj('$SAGE_SHARE', 'mathjax'))
+_add_variable_or_fallback('JMOL_DIR', opj('$SAGE_LOCAL','share', 'sage-jmol-bin'))
+_add_variable_or_fallback('JSMOL_DIR', opj('$SAGE_LOCAL','share', 'jsmol'))
+_add_variable_or_fallback('MATHJAX_DIR', opj('$SAGE_LOCAL','share', 'mathjax'))
 _add_variable_or_fallback('THREEJS_DIR', opj('$SAGE_SHARE', 'threejs'))
-_add_variable_or_fallback('ECLDIR', opj('$SAGE_LOCAL', 'lib', 'ecl'))
+_add_variable_or_fallback('ECLDIR', opj('$SAGE_LOCAL', 'lib', 'ecl-16.1.2'))
 _add_variable_or_fallback('MAXIMA_FAS', opj(SAGE_ENV['ECLDIR'], 'maxima.fas'))
 
 
@@ -221,7 +211,6 @@ if UNAME[:6] == 'CYGWIN':
     del _uname, re
 
 # things that need DOT_SAGE
-_add_variable_or_fallback('PYTHON_EGG_CACHE',   opj('$DOT_SAGE', '.python-eggs'))
 _add_variable_or_fallback('SAGE_STARTUP_FILE',  opj('$DOT_SAGE', 'init.sage'))
 
 # delete temporary variables used for setting up sage.env
diff --git a/sage/interfaces/gap.py b/sage/interfaces/gap.py
index ef378dd..2d0f889 100644
--- a/sage/interfaces/gap.py
+++ b/sage/interfaces/gap.py
@@ -197,8 +197,7 @@ import string
 
 WORKSPACE = gap_workspace_file()
 
-from sage.env import GAP_ROOT_DIR
-GAP_BINARY = os.path.join(GAP_ROOT_DIR, 'bin', 'gap.sh')
+GAP_BINARY = os.path.join(SAGE_LOCAL, 'bin', 'gap')
 
 first_try = True
 
diff --git a/sage/interfaces/jmoldata.py b/sage/interfaces/jmoldata.py
index f58b8f1..9bdb0d3 100644
--- a/sage/interfaces/jmoldata.py
+++ b/sage/interfaces/jmoldata.py
@@ -149,7 +149,7 @@ class JmolData(SageObject):
             True
         """
         # Set up paths, file names and scripts
-        jmolpath = os.path.join(JMOL_DIR, "JmolData.jar")
+        jmolpath = os.path.join(JMOL_DIR, 'lib', 'JmolData.jar')
         target_native = targetfile
 
         if sys.platform == 'cygwin':
diff --git a/sage/interfaces/singular.py b/sage/interfaces/singular.py
index f3c8162..7d2f8de 100644
--- a/sage/interfaces/singular.py
+++ b/sage/interfaces/singular.py
@@ -2263,8 +2263,8 @@ def generate_docstring_dictionary():
     nodes.clear()
     node_names.clear()
 
-    import os
-    singular_docdir = os.environ['SINGULARPATH']+"/../info/"
+    from sage.env import SAGE_LOCAL
+    singular_docdir = SAGE_LOCAL+"/share/info/"
 
     new_node = re.compile("File: singular\.hlp,  Node: ([^,]*),.*")
     new_lookup = re.compile("\* ([^:]*):*([^.]*)\..*")
