diff --git a/bin/sage-cython b/bin/sage-cython
index 2980858..f7843d9 100755
--- a/bin/sage-cython
+++ b/bin/sage-cython
@@ -4,9 +4,7 @@ import os
 import sys
 args = sys.argv[1:]
 
-if "SAGE_SRC" not in os.environ:
-    raise RuntimeError("The environment variable SAGE_SRC must be set.")
-SAGE_SRC = os.environ["SAGE_SRC"]
+from sage.env import SAGE_SRC
 
 # args can have length 0, in case we're printing a usage message (see trac 12207)
 pyx_file = os.path.abspath(args[-1]) if len(args) else None
