--- sage/interfaces/rubik.py.orig	2014-02-05 22:00:58.683127836 +1300
+++ sage/interfaces/rubik.py	2014-02-05 22:01:25.463166088 +1300
@@ -34,7 +34,8 @@
 #                  http://www.gnu.org/licenses/
 ########################################################################
 
-import pexpect, time
+import pexpect
+import time
 import cleaner
 
 from sage.groups.perm_gps.cubegroup import *
--- sage/interfaces/ecm.py.orig	2014-02-05 22:16:09.138304807 +1300
+++ sage/interfaces/ecm.py	2014-02-05 22:16:28.612324974 +1300
@@ -7,7 +7,8 @@
 for more about GMP-ECM.
 """
 
-import os, pexpect
+import os
+import pexpect
 from math import ceil, floor
 
 from sage.rings.integer import Integer
