diff -r 6e03cd2545db -r e41ff0c7ae27 sage/functions/hyperbolic.py
--- a/sage/functions/hyperbolic.py	Sat Oct 02 15:44:26 2010 +0200
+++ b/sage/functions/hyperbolic.py	Sat Oct 02 20:45:11 2010 +0200
@@ -474,7 +474,7 @@
             sage: import numpy
             sage: a = numpy.linspace(0,1,3)
             sage: arcsech(a)
-            array([       Inf,  1.3169579,  0.       ])
+            array([       inf,  1.3169579,  0.       ])
         """
         return arccosh(1.0 / x)
 
@@ -523,7 +523,7 @@
             sage: import numpy
             sage: a = numpy.linspace(0,1,3)
             sage: arccsch(a)
-            array([        Inf,  1.44363548,  0.88137359])
+            array([        inf,  1.44363548,  0.88137359])
         """
         return arcsinh(1.0 / x)
 
diff -r 6e03cd2545db -r e41ff0c7ae27 sage/matrix/matrix_double_dense.pyx
--- a/sage/matrix/matrix_double_dense.pyx	Sat Oct 02 15:44:26 2010 +0200
+++ b/sage/matrix/matrix_double_dense.pyx	Sat Oct 02 20:45:11 2010 +0200
@@ -1523,6 +1523,7 @@
             [51.9689561987  74.736564567]
             [112.104846851 164.073803049]
             sage: A.exp(method='eig')
+            doctest:94: ComplexWarning: Casting complex values to real discards the imaginary part
             [51.9689561987  74.736564567]
             [112.104846851 164.073803049]
             sage: A.exp(order=2)
diff -r 6e03cd2545db -r e41ff0c7ae27 sage/symbolic/function.pyx
--- a/sage/symbolic/function.pyx	Sat Oct 02 15:44:26 2010 +0200
+++ b/sage/symbolic/function.pyx	Sat Oct 02 20:45:11 2010 +0200
@@ -609,7 +609,7 @@
             sage: import numpy
             sage: a = numpy.arange(5)
             sage: csc(a)
-            array([        Inf,  1.18839511,  1.09975017,  7.0861674 , -1.32134871])
+            array([        inf,  1.18839511,  1.09975017,  7.0861674 , -1.32134871])
 
             sage: factorial(a)
             Traceback (most recent call last):
