--- sage/misc/cython.py.orig	2012-06-19 20:25:24.000000000 +1200
+++ sage/misc/cython.py	2012-06-21 10:45:50.600052215 +1200
@@ -334,10 +334,10 @@
     Before :trac:`12975`, it would have beeen needed to write ``#clang c++``,
     but upper case ``C++`` has resulted in an error::
 
-        sage: from sage.all import SAGE_ROOT
+        sage: from sage.misc.misc import SAGE_LOCAL
         sage: code = [                     
         ... "#clang C++",
-        ... "#cinclude %s/local/include/singular %s/local/include/factory"%(SAGE_ROOT,SAGE_ROOT),
+        ... "#cinclude %s/include/singular"%(SAGE_LOCAL),
         ... "#clib m readline singular givaro ntl gmpxx gmp",
         ... "from sage.rings.polynomial.multi_polynomial_libsingular cimport MPolynomial_libsingular",
         ... "from sage.libs.singular.polynomial cimport singular_polynomial_pow",
@@ -453,7 +453,7 @@
     sys.exit(1)
 else:
     SAGE_ROOT  = os.environ['SAGE_ROOT']
-    SAGE_LOCAL = SAGE_ROOT + '/local/'
+    SAGE_LOCAL = os.environ['SAGE_LOCAL']
 
 extra_link_args =  ['-L' + SAGE_LOCAL + '/lib']
 extra_compile_args = %s
--- sage/interfaces/maxima.py.orig	2010-09-30 20:46:50.000000000 -0500
+++ sage/interfaces/maxima.py	2010-09-30 20:48:17.000000000 -0500
@@ -461,7 +461,7 @@
 ##import sage.rings.all
 import sage.rings.complex_number
 
-from sage.misc.misc import DOT_SAGE, SAGE_ROOT
+from sage.misc.misc import DOT_SAGE, SAGE_LOCAL
 
 from sage.misc.multireplace import multiple_replace
 
@@ -497,7 +497,7 @@
         # setting inchar and outchar..
         eval_using_file_cutoff = 256
         self.__eval_using_file_cutoff = eval_using_file_cutoff
-        STARTUP = '%s/local/bin/sage-maxima.lisp'%SAGE_ROOT
+        STARTUP = '%s/bin/sage-maxima.lisp'%SAGE_LOCAL
         
         # We set maxima's configuration directory to $DOT_SAGE/maxima
         # This avoids that sage's maxima inadvertently loads
 
--- sage/server/trac/trac.py.orig	2010-10-01 17:54:16.000000000 -0500
+++ sage/server/trac/trac.py	2010-10-01 17:57:01.000000000 -0500
@@ -120,7 +120,7 @@
         print "    cd %s" % os.path.abspath(os.path.join(directory, 'conf'))
         print "    htdigest passwd %s <username>" % address
         print "\nTo grant full admin permissions to a user:"
-        print "    %s %s permission add <username> TRAC_ADMIN" % (os.path.join(SAGE_ROOT, 'local/bin/trac-admin'), os.path.abspath(directory))
+        print "    %s %s permission add <username> TRAC_ADMIN" % (os.path.join(SAGE_LOCAL, '/bin/trac-admin'), os.path.abspath(directory))
         print "\nThen restart the trac server."
         print "*" * 80
         open(passwd,'w').close()
--- sage/quadratic_forms/quadratic_form__automorphisms.py.orig	2010-09-30 18:22:43.000000000 -0500
+++ sage/quadratic_forms/quadratic_form__automorphisms.py	2010-09-30 18:24:03.000000000 -0500
@@ -10,7 +10,7 @@
 from sage.modules.free_module_element import vector
 from sage.rings.arith import GCD
 from sage.misc.sage_eval import sage_eval
-from sage.misc.misc import SAGE_ROOT
+from sage.misc.misc import SAGE_LOCAL
 
 import tempfile, os
 from random import random
@@ -452,7 +452,7 @@
     #os.system("less " + F.name)
 
     ## Call the Souvigner automorphism code
-    souvigner_auto_path = SAGE_ROOT + "/local/bin/Souvigner_AUTO"                 ## FIX THIS LATER!!!
+    souvigner_auto_path = SAGE_LOCAL + "/bin/Souvigner_AUTO"                 ## FIX THIS LATER!!!
     G1 = tempfile.NamedTemporaryFile(prefix='tmp_auto_ouput', suffix=".txt")
     #print "Output filename = ", G1.name
     os.system(souvigner_auto_path + " '" +  F.name + "' > '" + G1.name +"'")
--- sage/quadratic_forms/quadratic_form__equivalence_testing.py.orig	2010-09-30 18:22:54.000000000 -0500
+++ sage/quadratic_forms/quadratic_form__equivalence_testing.py	2010-09-30 18:24:39.000000000 -0500
@@ -11,7 +11,7 @@
 from quadratic_form__genus import CS_genus_symbol_list
 
 
-from sage.misc.misc import SAGE_ROOT
+from sage.misc.misc import SAGE_LOCAL
 
 import tempfile, os
 
@@ -78,7 +78,7 @@
     #os.system("less " + F.name)
 
     ## Call the Souvigner automorphism code
-    souvigner_isom_path = SAGE_ROOT + "/local/bin/Souvigner_ISOM" 
+    souvigner_isom_path = SAGE_LOCAL + "/bin/Souvigner_ISOM" 
     G1 = tempfile.NamedTemporaryFile(prefix='tmp_isom_ouput', suffix=".txt")
     #print "Output filename = ", G1.name
     #print  "Executing the shell command:   " + souvigner_isom_path + " '" +  F.name + "' > '" + G1.name + "'"

--- sage/misc/misc.py.orig	2012-03-07 13:29:58.053935723 +1300
+++ sage/misc/misc.py	2012-03-07 13:35:35.555188804 +1300
@@ -40,7 +40,7 @@
 from banner import version, banner
 
 SAGE_ROOT = os.environ["SAGE_ROOT"]
-SAGE_LOCAL = SAGE_ROOT + '/local/'
+SAGE_LOCAL = os.environ["SAGE_LOCAL"]
 SAGE_DOC = os.environ["SAGE_DOC"]
 HOSTNAME = socket.gethostname().replace('-','_').replace('/','_').replace('\\','_')
 
@@ -79,8 +79,6 @@
 except KeyError:
     SAGE_URL = "http://sage.math.washington.edu/sage/"     # default server
     
-LOGFILE = "%s/log/sage_log"%SAGE_ROOT
-
 
 try:
     DOT_SAGE = os.environ['DOT_SAGE']
@@ -90,6 +88,8 @@
     except KeyError:
         DOT_SAGE = '%s/.sage/'%SAGE_ROOT
 
+LOGFILE = "%s/log/sage_log"%DOT_SAGE
+
 UNAME=os.uname()[0]
 if ' ' in DOT_SAGE:
     if UNAME[:6] == 'CYGWIN':
@@ -2116,7 +2116,7 @@
     r"""
     Return a string describing the current branch and that the library
     is being loaded.  This function is called in the
-    ``SAGE_ROOT/local/bin/ipy_profile_sage.py`` script.
+    ``SAGE_LOCAL/bin/ipy_profile_sage.py`` script.
     
     INPUT:
     
