diff --git a/src/sage/libs/mpfi.pxd b/src/sage/libs/mpfi.pxd
index fe50da6..ef2da40 100644
--- a/sage/libs/mpfi.pxd
+++ b/sage/libs/mpfi.pxd
@@ -193,24 +193,6 @@ cdef extern from "mpfi.h":
     bint (*mpfi_is_strictly_pos) (mpfi_srcptr)
     bint (*mpfi_is_strictly_neg) (mpfi_srcptr)
 
-    # default comparison functions
-    bint mpfi_is_pos_default(mpfi_srcptr)
-    bint mpfi_is_nonneg_default(mpfi_srcptr)
-    bint mpfi_is_neg_default(mpfi_srcptr)
-    bint mpfi_is_nonpos_default(mpfi_srcptr)
-    bint mpfi_is_zero_default(mpfi_srcptr)
-    bint mpfi_is_strictly_neg_default(mpfi_srcptr a)
-    bint mpfi_is_strictly_pos_default(mpfi_srcptr a)
-
-    int mpfi_cmp_default(mpfi_srcptr, mpfi_srcptr)
-    int mpfi_cmp_d_default(mpfi_srcptr, double)
-    int mpfi_cmp_ui_default(mpfi_srcptr, unsigned long)
-    int mpfi_cmp_si_default(mpfi_srcptr, long)
-    int mpfi_cmp_z_default(mpfi_srcptr, mpz_srcptr)
-    int mpfi_cmp_q_default(mpfi_srcptr, mpq_srcptr)
-    int mpfi_cmp_fr_default(mpfi_srcptr, mpfr_srcptr)
-
-
     bint mpfi_has_zero(mpfi_srcptr)
 
     bint mpfi_nan_p(mpfi_srcptr)
diff --git a/src/sage/rings/real_mpfi.pyx b/src/sage/rings/real_mpfi.pyx
index 11c13f2..511bad5 100644
--- a/sage/rings/real_mpfi.pyx
+++ b/sage/rings/real_mpfi.pyx
@@ -3137,9 +3137,9 @@ cdef class RealIntervalFieldElement(RingElement):
         elif field.rnd == MPFR_RNDU:
             mpfi_get_right(x.value, self.value)
         elif field.rnd == MPFR_RNDZ:
-            if mpfi_is_strictly_pos_default(self.value):    # interval is > 0
+            if mpfi_is_strictly_pos(self.value):    # interval is > 0
                 mpfi_get_left(x.value, self.value)
-            elif mpfi_is_strictly_neg_default(self.value):  # interval is < 0
+            elif mpfi_is_strictly_neg(self.value):  # interval is < 0
                 mpfi_get_right(x.value, self.value)
             else:
                 mpfr_set_zero(x.value, 1)                   # interval contains 0
