diff --git a/sage/misc/misc.py b/sage/misc/misc.py
index fc9eaea..eb4c9dc 100644
--- a/sage/misc/misc.py
+++ b/sage/misc/misc.py
@@ -73,7 +73,7 @@ def sage_makedirs(dir):
     The following fails because we are trying to create a directory in
     place of an ordinary file (the main Sage executable)::
 
-        sage: sage_executable = os.path.join(SAGE_ROOT, 'sage')
+        sage: sage_executable = os.path.join(SAGE_LOCAL, 'bin', 'sage')
         sage: sage_makedirs(sage_executable)
         Traceback (most recent call last):
         ...
diff --git a/sage/misc/sage_ostools.pyx b/sage/misc/sage_ostools.pyx
index 8b55b93..9aa3747 100644
--- a/sage/misc/sage_ostools.pyx
+++ b/sage/misc/sage_ostools.pyx
@@ -30,7 +30,7 @@ def have_program(program, path=None):
         True
         sage: have_program('there_is_not_a_program_with_this_name')
         False
-        sage: have_program('sage', path=SAGE_ROOT)
+        sage: have_program('sage')
         True
         sage: have_program('ls', path=SAGE_ROOT)
         False
