diff --git a/src/sage/interfaces/maxima.py b/src/sage/interfaces/maxima.py
index 80369d52c8..1e31e8cf2d 100644
--- a/sage/interfaces/maxima.py
+++ b/sage/interfaces/maxima.py
@@ -468,6 +468,7 @@ from __future__ import print_function, absolute_import
 import os
 import re
 import pexpect
+import shlex
 
 from random import randrange
 
@@ -545,7 +546,7 @@ class Maxima(MaximaAbstract, Expect):
         Expect.__init__(self,
                         name = 'maxima',
                         prompt = r'\(\%i[0-9]+\) ',
-                        command = '"{0}" -p "{1}"'.format(MAXIMA, STARTUP),
+                        command = '{0} -p {1}'.format(MAXIMA, shlex.quote(STARTUP)),
                         env = {'TMPDIR': str(ECL_TMP)},
                         script_subdirectory = script_subdirectory,
                         restart_on_ctrlc = False,
