diff -Naur src.orig/sage/repl/ipython_kernel/install.py src/sage/repl/ipython_kernel/install.py
--- src.orig/sage/repl/ipython_kernel/install.py	2015-10-27 11:33:11.505520893 +1300
+++ src/sage/repl/ipython_kernel/install.py	2015-10-27 11:35:16.565814491 +1300
@@ -10,7 +10,7 @@
 import errno
 
 from sage.env import (
-    SAGE_ROOT, SAGE_DOC, SAGE_LOCAL, SAGE_EXTCODE,
+    SAGE_DOC, SAGE_LOCAL, SAGE_EXTCODE, SAGE_SRC,
     SAGE_VERSION
 )
 from jupyter_core.paths import ENV_JUPYTER_PATH
@@ -31,8 +31,7 @@
             'SageMath 6.9'
         """
         self._display_name = 'SageMath {0}'.format(SAGE_VERSION)
-        self.nbextensions_dir = os.path.join(JUPYTER_PATH, "nbextensions")
-        self.kernel_dir = os.path.join(JUPYTER_PATH, "kernels", self.identifier())
+        self.kernel_dir = os.path.join(SAGE_SRC, "build", "spec" , "kernels", self.identifier())
         self._mkdirs()
 
     def _mkdirs(self):
@@ -53,7 +52,6 @@
             except OSError:
                 if not os.path.isdir(path):
                     raise
-        mkdir_p(self.nbextensions_dir)
         mkdir_p(self.kernel_dir)
 
     @classmethod
@@ -143,16 +141,14 @@
             sage: from sage.repl.ipython_kernel.install import SageKernelSpec
             sage: spec = SageKernelSpec()
             sage: spec._kernel_cmd()
-            ['/.../sage',
-             '--python',
+            ['python2.7',
              '-m',
              'sage.repl.ipython_kernel',
              '-f',
              '{connection_file}']
         """
         return [
-            os.path.join(SAGE_ROOT, 'sage'),
-            '--python',
+            'python2.7',
             '-m', 'sage.repl.ipython_kernel',
             '-f', '{connection_file}',
         ]
@@ -177,7 +173,7 @@
             display_name=self._display_name,
         )
 
-    def _install_spec(self):
+    def _build_spec(self):
         """
         Install the SageMath Jupyter kernel
 
@@ -219,25 +215,28 @@
         )
 
     @classmethod
-    def update(cls):
+    def ipython_extra_files(cls):
         """
-        Configure the Jupyter notebook for the SageMath kernel
+        Return a data_files style list of files that will be
+        installed for the SageMath Jupyter kernel to work
 
-        This method does everything necessary to use the SageMath kernel,
-        you should never need to call any of the other methods
-        directly.
+        This method prepares the kernel and notebook extensions for installation.
+        There shouldn't be any needs to call any of the other methods directly.
 
         EXAMPLES::
 
             sage: from sage.repl.ipython_kernel.install import SageKernelSpec
             sage: spec = SageKernelSpec()
-            sage: spec.update()  # not tested
+            sage: spec.ipython_extra_files()  # not tested
         """
         instance = cls()
-        instance.use_local_mathjax()
-        instance.use_local_jsmol()
-        instance._install_spec()
-        instance._symlink_resources()
+        instance._build_spec()
+
+        kernel_installdir = os.path.join(JUPYTER_PATH, "kernels", instance.identifier())
+        kernelpath = os.path.join("build", "spec" , "kernels", instance.identifier(), "kernel.json")
+        data_files = [(kernel_installdir, [kernelpath])]
+
+        return data_files
 
 
 def have_prerequisites(debug=True):
diff -Naur src.orig/setup.py src/setup.py
--- src.orig/setup.py	2015-10-27 11:33:11.535520962 +1300
+++ src/setup.py	2015-10-27 11:33:36.495579559 +1300
@@ -635,19 +635,11 @@
 
 
 #########################################################
-### Install also Jupyter kernel spec
+### Add Jupyter kernel spec and notebook extensions
 #########################################################
 
-# We cannot just add the installation of the kernel spec to data_files
-# since the file is generated, not copied.
-class sage_install(install):
-    def run(self):
-        install.run(self)
-        self.install_kernel_spec()
-
-    def install_kernel_spec(self):
-        from sage.repl.ipython_kernel.install import SageKernelSpec
-        SageKernelSpec.update()
+from sage.repl.ipython_kernel.install import SageKernelSpec
+python_data_files.append(SageKernelSpec.ipython_extra_files()[0])
 
 
 #########################################################
@@ -664,5 +656,5 @@
       packages    = python_packages,
       data_files  = python_data_files,
       scripts = [],
-      cmdclass = dict(build_ext=sage_build_ext, install=sage_install),
+      cmdclass = { 'build_ext' : sage_build_ext},
       ext_modules = ext_modules)
