diff --git a/sage/graphs/generic_graph.py b/sage/graphs/generic_graph.py
index e5648e8..1bf308c 100644
--- a/sage/graphs/generic_graph.py
+++ b/sage/graphs/generic_graph.py
@@ -19947,7 +19947,7 @@ class GenericGraph(GenericGraph_pyx):
                      vertex_colors=None, vertex_size=0.06, vertex_labels=False,
                      edge_colors=None, edge_size=0.02, edge_size2=0.0325,
                      pos3d=None, color_by_label=False,
-                     engine='jmol', **kwds):
+                     engine='tachyon', **kwds):
         r"""
         Plot a graph in three dimensions.
 
@@ -19991,9 +19991,9 @@ class GenericGraph(GenericGraph_pyx):
 
         -  ``engine`` - which renderer to use. Options:
 
-           -  ``'jmol'`` - default
+           -  ``'jmol'``
 
-           -  ``'tachyon'``
+           -  ``'tachyon'`` - default
 
         -  ``xres`` - resolution
 
@@ -20178,7 +20178,7 @@ class GenericGraph(GenericGraph_pyx):
     def show3d(self, bgcolor=(1,1,1), vertex_colors=None, vertex_size=0.06,
                      edge_colors=None, edge_size=0.02, edge_size2=0.0325,
                      pos3d=None, color_by_label=False,
-                     engine='jmol', **kwds):
+                     engine='tachyon', **kwds):
         """
         Plots the graph using Tachyon, and shows the resulting plot.
 
@@ -20211,7 +20211,7 @@ class GenericGraph(GenericGraph_pyx):
 
         -  ``engine`` - which renderer to use. Options:
 
-        -  ``'jmol'`` - default 'tachyon'
+        -  ``'tachyon'`` - default or 'jmol'
 
         -  ``xres`` - resolution
 
diff --git a/sage/plot/plot3d/base.pyx b/sage/plot/plot3d/base.pyx
index 7401225..4dc0026 100644
--- a/sage/plot/plot3d/base.pyx
+++ b/sage/plot/plot3d/base.pyx
@@ -137,7 +137,7 @@ cdef class Graphics3d(SageObject):
             viewer = None
         # select suitable default
         if viewer is None:
-            viewer = 'jmol'
+            viewer = 'tachyon'
         # fall back to 2d image if necessary
         if viewer == 'canvas3d' and not can_view_canvas3d:   viewer = 'jmol'
         if viewer == 'wavefront' and not can_view_wavefront: viewer = 'jmol'
@@ -1949,7 +1949,7 @@ end_scene""" % (render_params.antialiasing,
         return self
 
 # if you add any default parameters you must update some code below
-SHOW_DEFAULTS = {'viewer': 'jmol',
+SHOW_DEFAULTS = {'viewer': 'tachyon',
                  'verbosity': 0,
                  'figsize': 5,
                  'aspect_ratio': "automatic",
