--- sage/misc/dist.py.orig	2011-11-25 10:25:58.915503321 +1300
+++ sage/misc/dist.py	2011-11-25 10:27:11.582425922 +1300
@@ -97,8 +97,8 @@
 
     from sage.misc.sage_ostools import have_program
     script_created = False
-    SAGE_ROOT = os.environ['SAGE_ROOT']
-    SAGE_BIN = os.path.join(SAGE_ROOT, 'local', 'bin')
+    SAGE_LOCAL = os.environ['SAGE_LOCAL']
+    SAGE_BIN = os.path.join(SAGE_LOCAL, 'bin')
     # See if 'directory' is already in PATH, and then remove
     # SAGE_ROOT/local/bin from PATH so that we can later check whether
     # cmd is available outside of Sage.
@@ -108,7 +108,7 @@
     PATH = os.pathsep.join([d for d in PATH if not
                             os.path.samefile(d, SAGE_BIN)])
     for cmd in ['gap', 'gp', 'hg', 'ipython', 'maxima',
-              'mwrank', 'R', 'singular', 'sqlite3', 'M2', 'kash']:
+              'mwrank', 'R', 'Singular', 'sqlite3', 'M2', 'kash']:
         print "Checking that Sage has the command '%s' installed" % cmd
         # Check to see if Sage includes cmd.
         cmd_inside_sage = have_program(cmd, path=SAGE_BIN)
