--- setup.py	2009-10-23 09:20:29.000000000 +0000
+++ setup.py-copy	2009-11-07 22:06:01.913721652 +0000
@@ -68,29 +68,28 @@
 ### Debian-related stuff
 #########################################################
 
-if os.environ.has_key('SAGE_DEBIAN'):
-    debian_include_dirs=["/usr/include",
-                         "/usr/include/cudd",
-                         "/usr/include/eclib",
-                         "/usr/include/FLINT",
-                         "/usr/include/fplll",
-                         "/usr/include/givaro",
-                         "/usr/include/gmp++",
-                         "/usr/include/gsl",
-                         "/usr/include/linbox",
-                         "/usr/include/NTL",
-                         "/usr/include/numpy",
-                         "/usr/include/pari",
-                         "/usr/include/polybori",
-                         "/usr/include/polybori/groebner",
-                         "/usr/include/qd",
-                         "/usr/include/singular",
-                         "/usr/include/singular/singular",
-                         "/usr/include/symmetrica",
-                         "/usr/share/python-support/cython/Cython/Includes/"
-                         "/usr/share/pyshared/Cython/Includes/"
-                         "/usr/include/zn_poly"]
-    include_dirs = include_dirs + debian_include_dirs
+debian_include_dirs=["/usr/include",
+                     "/usr/include/cudd",
+                     "/usr/include/eclib",
+                     "/usr/include/FLINT",
+                     "/usr/include/fplll",
+                     "/usr/include/givaro",
+                     "/usr/include/gmp++",
+                     "/usr/include/gsl",
+                     "/usr/include/linbox",
+                     "/usr/include/NTL",
+                     "/usr/include/numpy",
+                     "/usr/include/pari",
+                     "/usr/include/polybori",
+                     "/usr/include/polybori/groebner",
+                     "/usr/include/qd",
+                     "/usr/include/singular",
+                     "/usr/include/singular/singular",
+                     "/usr/include/symmetrica",
+                     "/usr/share/python-support/cython/Cython/Includes/"
+                     "/usr/share/pyshared/Cython/Includes/"
+                     "/usr/include/zn_poly"]
+include_dirs = include_dirs + debian_include_dirs
 
 extra_compile_args = [ ]
 
