diff --git a/doc/common/conf.py b/doc/common/conf.py
index c4dfbf5..897d41a 100644
--- a/sage/docs/conf.py
+++ b/sage/docs/conf.py
@@ -219,7 +219,7 @@ if (os.environ.get('SAGE_DOC_MATHJAX', 'no') != 'no'
     from sage.misc.latex_macros import sage_mathjax_macros
     html_theme_options['mathjax_macros'] = sage_mathjax_macros()
 
-    mathjax_relative = 'mathjax'
+    mathjax_relative = '../mathjax'
 
     # It would be really nice if sphinx would copy the entire mathjax directory,
     # (so we could have a _static/mathjax directory), rather than the contents of the directory
