--- patches/setup.py.orig	2009-11-27 13:40:36.350371201 +0000
+++ patches/setup.py	2009-11-27 13:44:10.015074775 +0000
@@ -136,9 +136,8 @@
             runtime_library_dirs = r_libs,
             #extra_compile_args=['-O0', '-g'],
             extra_link_args = get_rconfig(RHOME, '--ldflags') +\
-                              get_rconfig(RHOME, 'LAPACK_LIBS',
-                                          allow_empty = True) +\
-                              get_rconfig(RHOME, 'BLAS_LIBS', allow_empty=True)
+                              tuple(get_rconfig(RHOME, 'LAPACK_LIBS')[0].split()) +\
+			      tuple(get_rconfig(RHOME, 'BLAS_LIBS')[0].split())
             )
 
     print get_rconfig(RHOME, '--ldflags') +\
