diff -Naur src.orig/sage/doctest/sources.py src/sage/doctest/sources.py
--- src.orig/sage/doctest/sources.py	2015-07-13 11:34:07.741265459 +1200
+++ src/sage/doctest/sources.py	2015-07-13 11:35:45.871424272 +1200
@@ -27,7 +27,7 @@
 from parsing import SageDocTestParser
 from util import NestedName
 from sage.structure.dynamic_class import dynamic_class
-from sage.env import SAGE_SRC, SAGE_LOCAL
+from sage.env import SAGE_SRC, SAGE_LOCAL, SAGE_LIB
 
 # Python file parsing
 triple_quotes = re.compile("\s*[rRuU]*((''')|(\"\"\"))")
@@ -80,7 +80,7 @@
     # If the file is in the sage library, we can use our knowledge of
     # the directory structure
     dev = SAGE_SRC
-    sp = os.path.join(SAGE_LOCAL, 'lib', 'python', 'site-packages')
+    sp = SAGE_LIB
     if path.startswith(dev):
         # there will be a branch name
         i = path.find(os.path.sep, len(dev))
diff -Naur src.orig/setup.py src/setup.py
--- src.orig/setup.py	2015-07-13 11:34:07.781265524 +1200
+++ src/setup.py	2015-07-13 11:37:53.831630610 +1200
@@ -613,9 +613,9 @@
 t = time.time()
 from sage_setup.find import find_python_sources, find_extra_files
 python_packages, python_modules = find_python_sources(
-    SAGE_SRC, ['sage', 'sage_setup'])
+    SAGE_SRC, ['sage'])
 python_data_files = find_extra_files(python_packages,
-    ".", SAGE_CYTHONIZED, SAGE_LIB, ["ntlwrap.cpp"])
+    ".", SAGE_CYTHONIZED, SAGE_LIB, ["ntlwrap.cpp", "tests/*", "*.pyx"])
 print("Discovered Python/Cython sources, time: %.2f seconds." % (time.time() - t))
 
 
