--- sage/numerical/mip.pyx.orig	2013-04-11 09:36:55.000000000 +0000
+++ sage/numerical/mip.pyx	2013-04-14 09:45:49.342090177 +0000
@@ -2029,7 +2029,7 @@
             sage: b.solver_parameter("simplex_or_intopt", "simplex_only")
             sage: b.solver_parameter("verbosity_simplex", "GLP_MSG_ALL")
             sage: p.solve()  # tol 0.00001
-            GLPK Simplex Optimizer, v4.44
+            GLPK Simplex Optimizer, v...
             2 rows, 2 columns, 4 non-zeros
             *     0: obj =   7.000000000e+00  infeas =  0.000e+00 (0)
             *     2: obj =   9.400000000e+00  infeas =  0.000e+00 (0)
