diff -Naur src.orig/sage/misc/sagedoc.py src/sage/misc/sagedoc.py
--- src.orig/sage/misc/sagedoc.py	2016-02-11 11:17:57.853577546 +1300
+++ src/sage/misc/sagedoc.py	2016-02-11 11:24:17.836548184 +1300
@@ -18,8 +18,8 @@
 Check that argspecs of extension function/methods appear correctly,
 see :trac:`12849`::
 
-    sage: from sage.env import SAGE_DOC_OUTPUT
-    sage: docfilename = os.path.join(SAGE_DOC_OUTPUT, 'html', 'en', 'reference', 'calculus', 'sage', 'symbolic', 'expression.html')
+    sage: from sage.env import SAGE_DOC
+    sage: docfilename = os.path.join(SAGE_DOC, 'html', 'en', 'reference', 'calculus', 'sage', 'symbolic', 'expression.html')
     sage: for line in open(docfilename):
     ....:     if "#sage.symbolic.expression.Expression.N" in line:
     ....:         print line
@@ -40,7 +40,7 @@
 from sage.misc.viewer import browser
 from sage.misc.temporary_file import tmp_dir
 import sage.version
-from sage.env import SAGE_DOC, SAGE_DOC_OUTPUT, SAGE_SRC
+from sage.env import SAGE_DOC, SAGE_SRC
 
 # The detex function does two kinds of substitutions: math, which
 # should only be done on the command line -- in the notebook, these
@@ -800,58 +800,7 @@
         module = ''
         exts = ['html']
         title = 'Documentation'
-        base_path = SAGE_DOC_OUTPUT
-        doc_path = SAGE_DOC
-
-        from sage_setup.docbuild.build_options import LANGUAGES, OMIT
-        # List of languages
-        lang = LANGUAGES
-        # Documents in SAGE_DOC/LANG/ to omit
-        omit = OMIT
-
-        # List of documents, minus the omitted ones
-        documents = []
-        for L in lang:
-            documents += [os.path.join(L, dir) for dir
-                          in os.listdir(os.path.join(doc_path, L))
-                          if dir not in omit]
-
-        # Check to see if any documents are missing.  This just
-        # checks to see if the appropriate output directory exists,
-        # not that it contains a complete build of the docs.
-        missing = [os.path.join(SAGE_DOC_OUTPUT, 'html', doc)
-                   for doc in documents if not
-                   os.path.exists(os.path.join(SAGE_DOC_OUTPUT, 'html', doc))]
-        num_missing = len(missing)
-        if num_missing > 0:
-            print("""Warning, the following Sage documentation hasn't been built,
-so documentation search results may be incomplete:
-""")
-            for s in missing:
-                print(s)
-            if num_missing > 1:
-                print("""
-You can build these with 'sage -docbuild DOCUMENT html',
-where DOCUMENT is one of""", end=' ')
-                for s in missing:
-                    if s.find('en') != -1:
-                        print("'{}',".format(os.path.split(s)[-1]), end=' ')
-                    else:
-                        print("'{}',".format(os.path.join(
-                            os.path.split(os.path.split(s)[0])[-1],
-                            os.path.split(s)[-1])), end=' ')
-                print("""
-or you can use 'sage -docbuild all html' to build all of the missing documentation.""")
-            else:
-                s = missing[0]
-                if s.find('en') != -1:
-                    s = os.path.split(s)[-1]
-                else:
-                    s = os.path.join(
-                        os.path.split(os.path.split(s)[0])[-1],
-                        os.path.split(s)[-1])
-                print("""
-You can build this with 'sage -docbuild {} html'.""".format(s))
+        base_path = SAGE_DOC
 
     strip = len(base_path)
     results = ''
@@ -1102,7 +1051,7 @@
     search is case-insensitive by default.
 
     The file paths in the output are relative to
-    ``$SAGE_DOC_OUTPUT``.
+    ``$SAGE_DOC``.
 
     INPUT: same as for :func:`search_src`.
 
@@ -1309,7 +1258,7 @@
             'http://localhost:8000/doc/live/'
         """
         self._base_url = "http://localhost:8000/doc/live/"
-        self._base_path = os.path.join(SAGE_DOC_OUTPUT, "html", "en")
+        self._base_path = os.path.join(SAGE_DOC, "html", "en")
 
     def __call__(self, obj, output='html', view=True):
         r"""
@@ -1376,7 +1325,7 @@
                 path = os.path.join(tmp_dir(), "temp.html")
                 filed = open(path, 'w')
 
-                static_path = os.path.join(SAGE_DOC_OUTPUT, "html", "en", "_static")
+                static_path = os.path.join(SAGE_DOC, "html", "en", "_static")
                 if os.path.exists(static_path):
                     title = obj_name + ' - Sage ' + sage.version.version + ' Documentation'
                     template = """<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
