--- module_list.py.orig	2012-10-07 23:35:20.470331075 +1300
+++ module_list.py	2012-10-07 23:36:31.299402918 +1300
@@ -1609,8 +1609,8 @@
               sources = ['sage/rings/polynomial/plural.pyx'],
               libraries = ['m', 'readline', 'singular', 'givaro', 'gmpxx', 'gmp'],
               language="c++",
-              include_dirs = [SAGE_ROOT +'/local/include/singular'],
-              depends = [SAGE_ROOT + "/local/include/libsingular.h"],
+              include_dirs = [SAGE_INC +'singular'],
+              depends = singular_depends,
               extra_compile_args = givaro_extra_compile_args),
 
     Extension('sage.rings.polynomial.multi_polynomial_libsingular',
