--- sage/misc/sagedoc.py.orig	2013-12-30 22:24:50.497934009 +1300
+++ sage/misc/sagedoc.py	2013-12-30 22:27:59.468374318 +1300
@@ -1370,7 +1370,7 @@
             sage: browse_sage_doc._open("reference", testing=True)[0]
             'http://localhost:8000/doc/live/reference/index.html'
             sage: browse_sage_doc._open("tutorial", testing=True)[1]
-            '...doc/output/html/en/tutorial/index.html'
+            '.../output/html/en/tutorial/index.html'
         """
         url = self._base_url + os.path.join(name, "index.html")
         path = os.path.join(self._base_path, name, "index.html")
