diff --git a/sage/features/databases.py b/sage/features/databases.py
index 82d81be..c21aa40 100644
--- a/sage/features/databases.py
+++ b/sage/features/databases.py
@@ -87,7 +87,10 @@ class DatabaseKnotInfo(PythonModule):
             sage: isinstance(DatabaseKnotInfo(), DatabaseKnotInfo)
             True
         """
-        PythonModule.__init__(self, 'sage.knots.knotinfo', spkg='database_knotinfo',)
-        import os
         from sage.env import SAGE_SHARE
+        StaticFile.__init__(self, "Knot info database",
+                      filename='knotinfo/num_knots.sobj',
+                      search_path=SAGE_SHARE,
+                      spkg='database_knotinfo')
+        import os
         self._sobj_path = os.path.join(SAGE_SHARE, 'knotinfo')
