diff -Nuar src.orig/sage/doctest/control.py src/sage/doctest/control.py
--- src.orig/sage/doctest/control.py	2015-06-13 19:13:51.002817153 +1200
+++ src/sage/doctest/control.py	2015-06-13 19:16:45.993193698 +1200
@@ -545,7 +545,6 @@
         def all_files():
             from glob import glob
             self.files.append(opj(SAGE_SRC, 'sage'))
-            self.files.append(opj(SAGE_SRC, 'sage_setup'))
             self.files.append(opj(SAGE_SRC, 'doc', 'common'))
             self.files.extend(glob(opj(SAGE_SRC, 'doc', '[a-z][a-z]')))
             self.options.sagenb = True
diff -Nuar src.orig/setup.py src/setup.py
--- src.orig/setup.py	2015-06-13 19:13:51.062817283 +1200
+++ src/setup.py	2015-06-13 19:17:46.803324555 +1200
@@ -585,19 +585,6 @@
 
 
 #########################################################
-### Clean
-#########################################################
-
-print('Cleaning up stale installed files....')
-t = time.time()
-from sage_setup.clean import clean_install_dir
-output_dirs = SITE_PACKAGES + glob.glob(os.path.join(SAGE_SRC, 'build', 'lib*'))
-for output_dir in output_dirs:
-    print('- cleaning {0}'.format(output_dir))
-    clean_install_dir(output_dir, python_packages, python_modules, ext_modules)
-print('Finished cleaning, time: %.2f seconds.' % (time.time() - t))
-
-#########################################################
 ### Extra files to install
 #########################################################
 
