--- sage/matrix/matrix1.pyx.orig	2013-03-15 09:20:53.000000000 +0000
+++ sage/matrix/matrix1.pyx	2013-03-20 13:13:08.578095914 +0000
@@ -492,7 +492,7 @@
 
             sage: import numpy
             sage: sorted(numpy.typecodes.items())
-            [('All', '?bhilqpBHILQPfdgFDGSUVO'), ('AllFloat', 'fdgFDG'), ('AllInteger', 'bBhHiIlLqQpP'), ('Character', 'c'), ('Complex', 'FDG'), ('Float', 'fdg'), ('Integer', 'bhilqp'), ('UnsignedInteger', 'BHILQP')]
+            [('All', '?bhilqpBHILQPefdgFDGSUVOMm'), ('AllFloat', 'efdgFDG'), ('AllInteger', 'bBhHiIlLqQpP'), ('Character', 'c'), ('Complex', 'FDG'), ('Datetime', 'Mm'), ('Float', 'efdg'), ('Integer', 'bhilqp'), ('UnsignedInteger', 'BHILQP')]
 
         Alternatively, numpy automatically calls this function (via
         the magic :meth:`__array__` method) to convert Sage matrices
--- sage/rings/complex_number.pyx.orig	2013-03-15 09:20:53.000000000 +0000
+++ sage/rings/complex_number.pyx	2013-03-20 13:06:22.266101805 +0000
@@ -246,7 +246,7 @@
                 sage: numpy.array([1.0, 2.5j]).dtype
                 dtype('complex128')
                 sage: numpy.array([1.000000000000000000000000000000000000j]).dtype
-                dtype('object')
+                dtype('O')
             """
             if self._prec <= 57: # max size of repr(float)
                 return numpy_complex_interface
--- sage/rings/integer.pyx.orig	2013-03-15 09:20:53.000000000 +0000
+++ sage/rings/integer.pyx	2013-03-20 13:05:29.452102571 +0000
@@ -4924,7 +4924,7 @@
                 sage: numpy.array(2**40).dtype
                 dtype('int64')
                 sage: numpy.array(2**400).dtype
-                dtype('object')
+                dtype('O')
                 
                 sage: numpy.array([1,2,3,0.1]).dtype
                 dtype('float64')
--- sage/rings/rational.pyx.orig	2013-03-15 09:20:53.000000000 +0000
+++ sage/rings/rational.pyx	2013-03-20 13:04:47.662103177 +0000
@@ -715,7 +715,7 @@
                 sage: numpy.array(QQ(2**40)).dtype
                 dtype('int64')
                 sage: numpy.array(QQ(2**400)).dtype
-                dtype('object')
+                dtype('O')
 
                 sage: numpy.array([1, 1/2, 3/4])
                 array([ 1.  ,  0.5 ,  0.75])
--- sage/rings/real_mpfr.pyx.orig	2013-03-15 09:20:53.000000000 +0000
+++ sage/rings/real_mpfr.pyx	2013-03-20 13:07:09.437101122 +0000
@@ -1324,7 +1324,7 @@
                 sage: numpy.array([1.0, 1.1, 1.2]).dtype
                 dtype('float64')
                 sage: numpy.array([1.000000000000000000000000000000000000]).dtype
-                dtype('object')
+                dtype('O')
             """
             if (<RealField_class>self._parent).__prec <= 57: # max size of repr(float)
                 return numpy_double_interface
