diff --git a/sage/docs/conf.py b/sage/docs/conf.py
index c788623..527ed90 100644
--- a/sage/docs/conf.py
+++ b/sage/docs/conf.py
@@ -719,8 +719,8 @@ def call_intersphinx(app, env, node, contnode):
     Check that the link from the thematic tutorials to the reference
     manual is relative, see :trac:`20118`::
 
-        sage: from sage.env import SAGE_DOC  # optional - sagemath_doc_html
-        sage: thematic_index = os.path.join(SAGE_DOC, "html", "en", "thematic_tutorials", "index.html")  # optional - sagemath_doc_html
+        sage: from sage.env import SAGE_DOC
+        sage: thematic_index = os.path.join(SAGE_DOC, "html", "en", "thematic_tutorials", "index.html")
         sage: for line in open(thematic_index).readlines():  # optional - sagemath_doc_html
         ....:     if "padics" in line:
         ....:         _ = sys.stdout.write(line)
diff --git a/sage/misc/sagedoc.py b/sage/misc/sagedoc.py
index 4c56aea..583ca06 100644
--- a/sage/misc/sagedoc.py
+++ b/sage/misc/sagedoc.py
@@ -18,8 +18,8 @@ TESTS:
 Check that argspecs of extension function/methods appear correctly,
 see :trac:`12849`::
 
-    sage: from sage.env import SAGE_DOC  # optional - sagemath_doc_html
-    sage: docfilename = os.path.join(SAGE_DOC, 'html', 'en', 'reference', 'calculus', 'sage', 'symbolic', 'expression.html')  # optional - sagemath_doc_html
+    sage: from sage.env import SAGE_DOC
+    sage: docfilename = os.path.join(SAGE_DOC, 'html', 'en', 'reference', 'calculus', 'sage', 'symbolic', 'expression.html')
     sage: with open(docfilename) as fobj:  # optional - sagemath_doc_html
     ....:     for line in fobj:
     ....:         if "#sage.symbolic.expression.Expression.numerical_approx" in line:
@@ -845,12 +845,12 @@ def _search_src_or_doc(what, string, extra1='', extra2='', extra3='',
 
     ::
 
-        sage: from sage.misc.sagedoc import _search_src_or_doc  # optional - sagemath_doc_html
-        sage: len(_search_src_or_doc('src', r'matrix\(', 'incidence_structures', 'self', 'combinat', interact=False).splitlines()) > 1  # optional - sagemath_doc_html
+        sage: from sage.misc.sagedoc import _search_src_or_doc
+        sage: len(_search_src_or_doc('src', r'matrix\(', 'incidence_structures', 'self', 'combinat', interact=False).splitlines()) > 1
         True
         sage: 'abvar/homology' in _search_src_or_doc('doc', 'homology', 'variety', interact=False)  # optional - sagemath_doc_html, long time (4s on sage.math, 2012)
         True
-        sage: 'divisors' in _search_src_or_doc('src', '^ *def prime', interact=False)  # optional - sagemath_doc_html
+        sage: 'divisors' in _search_src_or_doc('src', '^ *def prime', interact=False)
         True
 
     When passing ``interactive=True``, in a terminal session this will pass the
@@ -1369,7 +1369,7 @@ class _sage_doc:
 
         sage: browse_sage_doc._open("reference", testing=True)[0]  # optional - sagemath_doc_html, indirect doctest
         'http://localhost:8000/doc/live/reference/index.html'
-        sage: browse_sage_doc(identity_matrix, 'rst')[-107:-47]  # optional - sagemath_doc_html
+        sage: browse_sage_doc(identity_matrix, 'rst')[-107:-47]
         'Full MatrixSpace of 3 by 3 sparse matrices over Integer Ring'
     """
     def __init__(self):

diff --git a/bin/sage-runtests b/bin/sage-runtests
index 15ab1ce..6d140aa 100755
--- a/bin/sage-runtests
+++ b/bin/sage-runtests
@@ -17,13 +17,6 @@ def _get_optional_defaults():
     """Return the default value for the --optional flag."""
     optional = ['sage', 'optional']
 
-    # This should be good-enough to determine if we are in a sage source
-    # repository working tree
-    if (SAGE_ROOT and os.path.isdir(SAGE_ROOT) and
-            os.path.exists(os.path.join(SAGE_ROOT, '.git')) and
-            os.path.isfile(os.path.join(SAGE_ROOT, 'sage'))):
-        optional.append('build')
-
     return ','.join(optional)
 
 
@@ -56,7 +49,6 @@ if __name__ == "__main__":
              'if "sagemath_doc_html" is listed, will also run the tests relying on the HTML documentation; '
              'if "optional" is listed, will also run tests for installed optional (new-style) packages; '
              'if "external" is listed, will also run tests for available external software; '
-             'if "build" is listed, will also run tests specific to Sage\'s build/packaging system; '
              'if set to "all", then all tests will be run')
     parser.add_argument("--randorder", type=int, metavar="SEED", help="randomize order of tests")
     parser.add_argument("--random-seed", dest="random_seed", type=int, metavar="SEED", help="random seed (integer) for fuzzing doctests",
