diff -Naur src.orig/sage/plot/contour_plot.py src/sage/plot/contour_plot.py
--- src.orig/sage/plot/contour_plot.py	2016-01-16 21:27:17.379311337 +1300
+++ src/sage/plot/contour_plot.py	2016-01-16 21:27:49.874377729 +1300
@@ -568,7 +568,7 @@
     g.add_primitive(ContourPlot(xy_data_array, xrange, yrange, options))
     return g
 
-@options(plot_points=150, contours=(0,0), fill=False, cmap=["blue"])
+@options(plot_points=150, fill=False, cmap=["blue"])
 def implicit_plot(f, xrange, yrange, **options):
     r"""
     ``implicit_plot`` takes a function of two variables, `f(x,y)`
