diff --git a/sage/doctest/sources.py b/sage/doctest/sources.py
index 21c5fb1..44c264d 100644
--- a/sage/doctest/sources.py
+++ b/sage/doctest/sources.py
@@ -33,7 +33,7 @@ from sage.misc.lazy_attribute import lazy_attribute
 from .parsing import SageDocTestParser
 from .util import NestedName
 from sage.structure.dynamic_class import dynamic_class
-from sage.env import SAGE_SRC, SAGE_LOCAL
+from sage.env import SAGE_SRC, SAGE_LOCAL, SAGE_LIB
 
 # Python file parsing
 triple_quotes = re.compile("\s*[rRuU]*((''')|(\"\"\"))")
@@ -86,7 +86,7 @@ def get_basename(path):
     # If the file is in the sage library, we can use our knowledge of
     # the directory structure
     dev = SAGE_SRC
-    sp = os.path.join(SAGE_LOCAL, 'lib', 'python', 'site-packages')
+    sp = SAGE_LIB
     if path.startswith(dev):
         # there will be a branch name
         i = path.find(os.path.sep, len(dev))
diff --git a/sage_setup/docbuild/__init__.py b/sage_setup/docbuild/__init__.py
index 4315663..e87483f 100644
--- a/sage_setup/docbuild/__init__.py
+++ b/sage_setup/docbuild/__init__.py
@@ -25,7 +25,7 @@ from sage.misc.cachefunc import cached_method
 from sage.misc.misc import sage_makedirs
 from sage.env import SAGE_DOC_SRC, SAGE_DOC, SAGE_SRC
 
-from .build_options import (LANGUAGES, SPHINXOPTS, PAPER, OMIT,
+from sage_setup.docbuild.build_options import (LANGUAGES, SPHINXOPTS, PAPER, OMIT,
      PAPEROPTS, ALLSPHINXOPTS, NUM_THREADS, WEBSITESPHINXOPTS,
      INCREMENTAL_BUILD, ABORT_ON_ERROR)
 
diff --git a/sage_setup/docbuild/__main__.py b/sage_setup/docbuild/__main__.py
index 031df43..e68b438 100644
--- a/sage_setup/docbuild/__main__.py
+++ b/sage_setup/docbuild/__main__.py
@@ -1,2 +1,2 @@
-from . import main
+from sage_setup.docbuild import main
 main()
diff --git a/sage_setup/find.py b/sage_setup/find.py
index cba29bf..36d5ab1 100644
--- a/sage_setup/find.py
+++ b/sage_setup/find.py
@@ -114,7 +114,7 @@ def find_extra_files(packages, src_dir, cythonized_dir, special_filenames=[]):
         cydir = os.path.join(cythonized_dir, dir)
 
         files = [os.path.join(sdir, f) for f in os.listdir(sdir)
-                if f.endswith((".pxd", ".pxi")) or f in special_filenames]
+                if f.endswith((".pxd", ".pxi", ".pyx", ".rst")) or f in special_filenames]
         if os.path.isdir(cydir):  # Not every directory contains Cython files
             files += [os.path.join(cydir, f) for f in os.listdir(cydir)
                     if f.endswith(".h")]
diff --git a/setup.py b/setup.py
index 2215379..16c2971 100755
--- a/setup.py
+++ b/setup.py
@@ -834,7 +834,7 @@ print("Discovering Python/Cython source code....")
 t = time.time()
 from sage_setup.find import find_python_sources
 python_packages, python_modules = find_python_sources(
-    SAGE_SRC, ['sage', 'sage_setup'])
+    SAGE_SRC, ['sage'])
 
 log.debug('python_packages = {0}'.format(python_packages))
 
@@ -849,10 +849,6 @@ class sage_install(install):
     def run(self):
         install.run(self)
         self.install_kernel_spec()
-        log.info('Cleaning up stale installed files....')
-        t = time.time()
-        self.clean_stale_files()
-        log.info('Finished cleaning, time: %.2f seconds.' % (time.time() - t))
 
     def install_kernel_spec(self):
         """
@@ -863,7 +859,7 @@ class sage_install(install):
             The files are generated, not copied. Therefore, we cannot
             use ``data_files`` for this.
         """
-        from sage.repl.ipython_kernel.install import SageKernelSpec
+        from sage_setup.jupyter.install import SageKernelSpec
         SageKernelSpec.update()
 
     def clean_stale_files(self):
