diff -Naur src.orig/sage/env.py src/sage/env.py
--- src.orig/sage/env.py	2016-02-08 21:03:30.802138665 +1300
+++ src/sage/env.py	2016-02-08 21:28:38.713809278 +1300
@@ -87,26 +87,22 @@
 _add_variable_or_fallback('LOCAL_IDENTIFIER','$HOSTNAME.%s'%os.getpid())
 
 # bunch of sage directories and files
-_add_variable_or_fallback('SAGE_ROOT',       None)
-_add_variable_or_fallback('SAGE_LOCAL',      opj('$SAGE_ROOT', 'local'))
-_add_variable_or_fallback('SAGE_ETC',        opj('$SAGE_LOCAL', 'etc'))
-_add_variable_or_fallback('SAGE_INC',        opj('$SAGE_LOCAL', 'include'))
-_add_variable_or_fallback('SAGE_SHARE',      opj('$SAGE_LOCAL', 'share'))
+_add_variable_or_fallback('SAGE_ROOT',       opj('@GENTOO_PORTAGE_EPREFIX@/usr', 'share', 'sage'))
+_add_variable_or_fallback('SAGE_LOCAL',      opj('@GENTOO_PORTAGE_EPREFIX@/usr')
+_add_variable_or_fallback('SAGE_ETC',        opj('@GENTOO_PORTAGE_EPREFIX@/etc'))
+_add_variable_or_fallback('SAGE_INC',        opj('@GENTOO_PORTAGE_EPREFIX@/usr/include'))
+_add_variable_or_fallback('SAGE_SHARE',      opj('$SAGE_LOCAL', 'share', 'sage'))
 
-_add_variable_or_fallback('SAGE_SRC',        opj('$SAGE_ROOT', 'src'))
 _add_variable_or_fallback('SITE_PACKAGES',   site.getsitepackages())
 _add_variable_or_fallback('SAGE_LIB',        SITE_PACKAGES[0])
+_add_variable_or_fallback('SAGE_SRC',        opj('$SAGE_LIB'))
 
 _add_variable_or_fallback('SAGE_CYTHONIZED', opj('$SAGE_SRC', 'build', 'cythonized'))
 
-_add_variable_or_fallback('SAGE_EXTCODE',    opj('$SAGE_SHARE', 'sage', 'ext'))
-_add_variable_or_fallback('SAGE_LOGS',       opj('$SAGE_ROOT', 'logs', 'pkgs'))
-_add_variable_or_fallback('SAGE_SPKG_INST',  opj('$SAGE_LOCAL', 'var', 'lib', 'sage', 'installed'))
-_add_variable_or_fallback('SAGE_DOC',        opj('$SAGE_SRC', 'doc'))
-_add_variable_or_fallback('SAGE_DOC_OUTPUT', opj('$SAGE_DOC', 'output'))
+_add_variable_or_fallback('SAGE_EXTCODE',    opj('$SAGE_SHARE', 'ext'))
+_add_variable_or_fallback('SAGE_DOC',        opj('$SAGE_LOCAL', 'share', 'doc', 'sage'))
+_add_variable_or_fallback('SAGE_DOC_OUTPUT', opj('$SAGE_DOC'))
 _add_variable_or_fallback('DOT_SAGE',        opj(os.environ.get('HOME','$SAGE_ROOT'), '.sage'))
-_add_variable_or_fallback('SAGE_DOT_GIT',    opj('$SAGE_ROOT', '.git'))
-_add_variable_or_fallback('SAGE_DISTFILES',  opj('$SAGE_ROOT', 'upstream'))
 
 # misc
 _add_variable_or_fallback('SAGE_URL',                'http://sage.math.washington.edu/sage/')
@@ -135,7 +131,6 @@
         print("permissions to before you start sage.")
 
 # things that need DOT_SAGE
-_add_variable_or_fallback('PYTHON_EGG_CACHE',   opj('$DOT_SAGE', '.python-eggs'))
 _add_variable_or_fallback('SAGE_STARTUP_FILE',  opj('$DOT_SAGE', 'init.sage'))
 
 # delete temporary variables used for setting up sage.env
