--- module_list.py.orig	2014-03-06 14:15:15.000000000 -0600
+++ module_list.py	2014-03-06 14:19:05.000000000 -0600
@@ -740,9 +740,11 @@
 
     Extension('sage.libs.ppl',
               sources = ['sage/libs/ppl.pyx', 'sage/libs/ppl_shim.cc'],
+              runtime_library_dirs = [SAGE_LOCAL + "/lib/ppl1"],
+              library_dirs = [SAGE_LOCAL + "/lib/ppl1"],
               libraries = ['ppl', 'gmpxx', 'gmp', 'm'],
               language="c++",
-              depends = [SAGE_INC + "/ppl.hh"]),
+              depends = [SAGE_INC + "/ppl1/ppl.hh"]),
 
     Extension('sage.libs.ratpoints',
               sources = ["sage/libs/ratpoints.pyx"],
@@ -1416,6 +1418,8 @@
     Extension("sage.numerical.backends.ppl_backend",
               ["sage/numerical/backends/ppl_backend.pyx"],
               include_dirs = [SAGE_INC, "sage/c_lib/include/"],
+              runtime_library_dirs = [SAGE_LOCAL + "/lib/ppl1"],
+              library_dirs = [SAGE_LOCAL + "/lib/ppl1"],
               libraries=["stdc++"]),
 
     Extension("sage.numerical.backends.glpk_graph_backend",
--- sage/libs/ppl.pyx.orig	2014-03-06 21:21:03.000000000 -0600
+++ sage/libs/ppl.pyx	2014-03-06 21:22:10.000000000 -0600
@@ -176,7 +176,7 @@
 
 ####################################################
 # PPL can use floating-point arithmetic to compute integers
-cdef extern from "ppl.hh" namespace "Parma_Polyhedra_Library":
+cdef extern from "ppl1/ppl.hh" namespace "Parma_Polyhedra_Library":
     cdef void set_rounding_for_PPL()
     cdef void restore_pre_PPL_rounding()
 
@@ -186,15 +186,15 @@
 
 ####################################################
 # Cython does not support ctypedef within cppclass; Hack around this restriction:
-cdef extern from "ppl.hh" namespace "Parma_Polyhedra_Library::Generator":
+cdef extern from "ppl1/ppl.hh" namespace "Parma_Polyhedra_Library::Generator":
     ctypedef enum PPL_GeneratorType "Parma_Polyhedra_Library::Generator::Type":
         LINE, RAY, POINT, CLOSURE_POINT
 
-cdef extern from "ppl.hh" namespace "Parma_Polyhedra_Library::Constraint":
+cdef extern from "ppl1/ppl.hh" namespace "Parma_Polyhedra_Library::Constraint":
     ctypedef enum PPL_ConstraintType "Parma_Polyhedra_Library::Constraint::Type":
         EQUALITY, NONSTRICT_INEQUALITY, STRICT_INEQUALITY
 
-cdef extern from "ppl.hh" namespace "Parma_Polyhedra_Library::MIP_Problem":
+cdef extern from "ppl1/ppl.hh" namespace "Parma_Polyhedra_Library::MIP_Problem":
     ctypedef enum PPL_MIP_Problem_Control_Parameter_Name:
         PRICING
     ctypedef enum PPL_MIP_Problem_Control_Parameter_Value:
@@ -202,7 +202,7 @@
 
 
 ####################################################
-cdef extern from "ppl.hh" namespace "Parma_Polyhedra_Library":
+cdef extern from "ppl1/ppl.hh" namespace "Parma_Polyhedra_Library":
 
     ctypedef size_t PPL_dimension_type  "Parma_Polyhedra_Library::dimension_type"
     ctypedef mpz_class PPL_Coefficient  "Parma_Polyhedra_Library::Coefficient"
@@ -435,7 +435,7 @@
         PPL_MIP_Problem_Control_Parameter_Value get_control_parameter(PPL_MIP_Problem_Control_Parameter_Name name)
         void set_control_parameter(PPL_MIP_Problem_Control_Parameter_Value value)
 
-cdef extern from "ppl.hh":
+cdef extern from "ppl1/ppl.hh":
     PPL_Generator PPL_line          "Parma_Polyhedra_Library::line"          (PPL_Linear_Expression &e) except +ValueError
     PPL_Generator PPL_ray           "Parma_Polyhedra_Library::ray"           (PPL_Linear_Expression &e) except +ValueError
     PPL_Generator PPL_point         "Parma_Polyhedra_Library::point"         (PPL_Linear_Expression &e, PPL_Coefficient &d) except +ValueError
@@ -444,7 +444,7 @@
 
 ####################################################
 # Cython does not support static methods; hack around
-cdef extern from "ppl.hh":
+cdef extern from "ppl1/ppl.hh":
 
     PPL_Poly_Gen_Relation PPL_Poly_Gen_Relation_nothing  "Parma_Polyhedra_Library::Poly_Gen_Relation::nothing"  ()
     PPL_Poly_Gen_Relation PPL_Poly_Gen_Relation_subsumes "Parma_Polyhedra_Library::Poly_Gen_Relation::subsumes" ()
--- sage/libs/ppl_shim.hh.orig	2014-03-06 14:22:57.000000000 -0600
+++ sage/libs/ppl_shim.hh	2014-03-06 14:23:30.000000000 -0600
@@ -3,7 +3,7 @@
 
 
 
-#include <ppl.hh>
+#include <ppl1/ppl.hh>
 
 using namespace Parma_Polyhedra_Library;
 
