diff -Naur src.orig/sage/repl/ipython_kernel/install.py src/sage/repl/ipython_kernel/install.py
--- src.orig/sage/repl/ipython_kernel/install.py	2015-04-04 14:26:50.139284265 +1300
+++ src/sage/repl/ipython_kernel/install.py	2015-04-04 14:28:53.689572657 +1300
@@ -15,7 +15,7 @@
 from IPython.utils.path import get_ipython_dir
 
 from sage.env import (
-    SAGE_ROOT, SAGE_DOC, SAGE_LOCAL, SAGE_EXTCODE,
+    SAGE_DOC, SAGE_LOCAL, SAGE_EXTCODE,
     SAGE_VERSION
 )
 from jupyter_core.paths import ENV_JUPYTER_PATH
@@ -143,16 +143,14 @@
             sage: from sage.repl.ipython_kernel.install import SageKernelSpec
             sage: spec = SageKernelSpec()
             sage: spec._kernel_cmd()
-            ['/.../sage',
-             '--python',
+            ['python2.7',
              '-m',
              'sage.repl.ipython_kernel',
              '-f',
              '{connection_file}']
         """
         return [
-            os.path.join(SAGE_ROOT, 'sage'),
-            '--python',
+            'python2.7',
             '-m', 'sage.repl.ipython_kernel',
             '-f', '{connection_file}',
         ]
