diff -Naur src.orig/sage/misc/sagedoc.py src/sage/misc/sagedoc.py
--- src.orig/sage/misc/sagedoc.py	2015-03-13 13:51:42.829117147 +1300
+++ src/sage/misc/sagedoc.py	2015-03-13 13:53:07.159355497 +1300
@@ -18,7 +18,7 @@
 Check that argspecs of extension function/methods appear correctly,
 see :trac:`12849`::
 
-    sage: docfilename = os.path.join(SAGE_DOC, 'output', 'html', 'en', 'reference', 'calculus', 'sage', 'symbolic', 'expression.html')
+    sage: docfilename = os.path.join(SAGE_DOC, 'html', 'en', 'reference', 'calculus', 'sage', 'symbolic', 'expression.html')
     sage: for line in open(docfilename):
     ...       if "#sage.symbolic.expression.Expression.N" in line:
     ...           print line
@@ -725,7 +725,7 @@
         module = ''
         exts = ['html']
         title = 'Documentation'
-        base_path = os.path.join(SAGE_DOC, 'output')
+        base_path = SAGE_DOC
         doc_path = SAGE_DOC
 
         # We need to import stuff from SAGE_DOC/common
@@ -749,9 +749,9 @@
         # Check to see if any documents are missing.  This just
         # checks to see if the appropriate output directory exists,
         # not that it contains a complete build of the docs.
-        missing = [os.path.join(doc_path, 'output', 'html', doc)
+        missing = [os.path.join(doc_path, 'html', doc)
                    for doc in documents if not
-                   os.path.exists(os.path.join(doc_path, 'output', 'html', doc))]
+                   os.path.exists(os.path.join(doc_path, 'html', doc))]
         num_missing = len(missing)
         if num_missing > 0:
             print("""Warning, the following Sage documentation hasn't been built,
@@ -1235,7 +1235,7 @@
             'http://localhost:8000/doc/live/'
         """
         self._base_url = "http://localhost:8000/doc/live/"
-        self._base_path = os.path.join(SAGE_DOC, "output/html/en/")
+        self._base_path = os.path.join(SAGE_DOC, "html/en/")
 
     def __call__(self, obj, output='html', view=True):
         r"""
@@ -1302,7 +1302,7 @@
                 path = os.path.join(tmp_dir(), "temp.html")
                 filed = open(path, 'w')
 
-                static_path = os.path.join(SAGE_DOC, 'output/html/en/_static')
+                static_path = os.path.join(SAGE_DOC, 'html/en/_static')
                 if os.path.exists(static_path):
                     title = obj_name + ' - Sage ' + sage.version.version + ' Documentation'
                     template = """<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
@@ -1389,7 +1389,7 @@
             sage: browse_sage_doc._open("reference", testing=True)[0]
             'http://localhost:8000/doc/live/reference/index.html'
             sage: browse_sage_doc._open("tutorial", testing=True)[1]
-            '...doc/output/html/en/tutorial/index.html'
+            '.../html/en/tutorial/index.html'
         """
         url = self._base_url + os.path.join(name, "index.html")
         path = os.path.join(self._base_path, name, "index.html")
