--- module_list.py.orig	2013-05-28 19:47:12.050971179 -0500
+++ module_list.py	2013-05-28 19:48:05.994977813 -0500
@@ -196,6 +196,7 @@
     Extension('sage.combinat.matrices.dancing_links',
               sources = ['sage/combinat/matrices/dancing_links.pyx'],
               libraries = ["stdc++"],
+              depends = ['sage/combinat/matrices/dancing_links_c.h'],
               language='c++'),
 
     Extension('sage.structure.list_clone',
@@ -279,6 +280,28 @@
 
     ################################
     ##
+    ## sage.ext.intepreters
+    ##
+    ################################
+
+    Extension('sage.ext.interpreters.wrapper_el',
+              sources = ['sage/ext/interpreters/wrapper_el.pyx'],
+              depends = ['sage/ext/interpreters/interp_rdf.h', 'sage/ext/interpreters/interp_el.h']),
+
+    Extension('sage.ext.interpreters.wrapper_py',
+              sources = ['sage/ext/interpreters/wrapper_py.pyx'],
+              depends = ['sage/ext/interpreters/interp_py.h']),
+
+    Extension('sage.ext.interpreters.wrapper_cdf',
+              sources = ['sage/ext/interpreters/wrapper_cdf.pyx'],
+              depends = ['sage/ext/interpreters/interp_cdf.h']),
+
+    Extension('sage.ext.interpreters.wrapper_rr',
+              sources = ['sage/ext/interpreters/wrapper_rr.pyx'],
+              depends = ['sage/ext/interpreters/interp_rr.h']),
+
+    ################################
+    ##
     ## sage.finance
     ##
     ################################
@@ -445,7 +468,8 @@
 
     Extension('sage.graphs.graph_decompositions.rankwidth',
               sources = ['sage/graphs/graph_decompositions/rankwidth.pyx',
-                         'sage/graphs/graph_decompositions/rankwidth_c/rw.c']),
+                         'sage/graphs/graph_decompositions/rankwidth_c/rw.c'],
+              depends = ['sage/graphs/graph_decompositions/rankwidth_c/rw.h']),
 
     Extension('sage.graphs.spanning_tree',
               sources = ['sage/graphs/spanning_tree.pyx']),
@@ -618,7 +642,7 @@
               sources = ["sage/libs/ecl.pyx"],
               libraries = ["ecl", "gmp"],
               include_dirs = [SAGE_INC + '/ecl'],
-              depends = [SAGE_INC + '/ecl/ecl.h']),
+              depends = [SAGE_INC + '/ecl/ecl.h', 'sage/libs/eclsig.h']),
 
     Extension('sage.libs.flint.flint',
               sources = ["sage/libs/flint/flint.pyx"],
@@ -655,6 +679,7 @@
                            'Lfunction', 'stdc++'],
               include_dirs = [SAGE_INC + "/libLfunction"],
               extra_compile_args=["-O3", "-ffast-math"],
+              depends = ['sage/libs/lcalc/lcalc_sage.h'],
               language = 'c++'),
      
 
@@ -680,13 +705,14 @@
     
     Extension('sage.libs.pari.gen',
               sources = ["sage/libs/pari/gen.pyx"],
+              depends = ['sage/libs/pari/pari_err.h', 'sage/libs/pari/misc.h'],
               libraries = ['pari', 'gmp']),
     
     Extension('sage.libs.ppl',
               sources = ['sage/libs/ppl.pyx', 'sage/libs/ppl_shim.cc'],
               libraries = ['ppl', 'gmpxx', 'gmp', 'm'], 
               language="c++",
-              depends = [SAGE_INC + "/ppl.hh"]),
+              depends = [SAGE_INC + "/ppl.hh", 'sage/libs/ppl_shim.hh']),
 
     Extension('sage.libs.ratpoints',
               sources = ["sage/libs/ratpoints.pyx"],
@@ -1200,7 +1226,8 @@
     ################################
 
     Extension('sage.modular.arithgroup.congroup_pyx',
-              sources = ['sage/modular/arithgroup/congroup_pyx.pyx']),
+              sources = ['sage/modular/arithgroup/congroup_pyx.pyx'],
+              depends = ['sage/modular/arithgroup/sl2z.hpp', 'sage/modular/arithgroup/farey.hpp']),
 
     Extension('sage.modular.arithgroup.farey_symbol',
               sources = ['sage/modular/arithgroup/farey_symbol.pyx',
@@ -1521,7 +1548,8 @@
 
     Extension('sage.rings.finite_rings.integer_mod',
               sources = ['sage/rings/finite_rings/integer_mod.pyx'],
-              libraries = ['gmp']),
+              libraries = ['gmp'],
+              depends = ['sage/rings/finite_rings/stdint.h']),
 
     Extension('sage.rings.finite_rings.element_givaro',
               sources = ["sage/rings/finite_rings/element_givaro.pyx"],
@@ -1922,7 +1950,7 @@
     Extension('sage.symbolic.pynac',
               sources = ['sage/symbolic/pynac.pyx'],
               language = 'c++',
-              depends = ginac_depends,
+              depends = ginac_depends + ['sage/symbolic/pynac_cc.h'],
               libraries = ["pynac", "gmp", "gsl"]),
     
     Extension('sage.symbolic.ring',