--- module_list.py.orig	2012-10-08 22:52:08.555265166 +1300
+++ module_list.py	2012-10-08 23:01:59.834936662 +1300
@@ -677,7 +677,7 @@
               language="c++",
               include_dirs = [SAGE_INC + 'singular'],
               depends = singular_depends,
-              extra_compile_args = givaro_extra_compile_args),
+              extra_compile_args = givaro_extra_compile_args + ['-DNDEBUG']),
     
     Extension('sage.libs.singular.polynomial',
               sources = ['sage/libs/singular/polynomial.pyx'],
@@ -685,7 +685,7 @@
               language="c++",
               include_dirs = [SAGE_INC + 'singular'],
               depends = singular_depends,
-              extra_compile_args = givaro_extra_compile_args),
+              extra_compile_args = givaro_extra_compile_args + ['-DNDEBUG']),
 
     Extension('sage.libs.singular.ring',
               sources = ['sage/libs/singular/ring.pyx'],
@@ -693,7 +693,7 @@
               language="c++",
               include_dirs = [SAGE_INC + 'singular'],
               depends = singular_depends,
-              extra_compile_args = givaro_extra_compile_args),
+              extra_compile_args = givaro_extra_compile_args + ['-DNDEBUG']),
 
     Extension('sage.libs.singular.groebner_strategy',
               sources = ['sage/libs/singular/groebner_strategy.pyx'],
@@ -701,7 +701,7 @@
               language="c++",
               include_dirs = [SAGE_INC + 'singular'],
               depends = singular_depends,
-              extra_compile_args = givaro_extra_compile_args),
+              extra_compile_args = givaro_extra_compile_args + ['-DNDEBUG']),
 
     Extension('sage.libs.singular.function',
               sources = ['sage/libs/singular/function.pyx'],
@@ -709,7 +709,7 @@
               language="c++",
               include_dirs = [SAGE_INC +'singular'],
               depends = singular_depends,
-              extra_compile_args = givaro_extra_compile_args),
+              extra_compile_args = givaro_extra_compile_args + ['-DNDEBUG']),
 
     Extension('sage.libs.singular.option',
               sources = ['sage/libs/singular/option.pyx'],
@@ -717,7 +717,7 @@
               language="c++",
               include_dirs = [SAGE_INC + 'singular'],
               depends = singular_depends,
-              extra_compile_args = givaro_extra_compile_args),
+              extra_compile_args = givaro_extra_compile_args + ['-DNDEBUG']),
 
     Extension('sage.libs.symmetrica.symmetrica',
               sources = ["sage/libs/symmetrica/%s"%s for s in ["symmetrica.pyx"]],
@@ -1009,7 +1009,7 @@
               language="c++",
               include_dirs = [SAGE_INC + 'singular'],
               depends = singular_depends,
-              extra_compile_args = givaro_extra_compile_args),
+              extra_compile_args = givaro_extra_compile_args + ['-DNDEBUG']),
 
     #Extension('sage.matrix.matrix_pid_dense',
     #          sources = ['sage/matrix/matrix_pid_dense.pyx']),
@@ -1603,7 +1603,7 @@
               language="c++",
               include_dirs = [SAGE_INC + 'singular'],
               depends = singular_depends,
-              extra_compile_args = givaro_extra_compile_args),
+              extra_compile_args = givaro_extra_compile_args + ['-DNDEBUG']),
 
     Extension('sage.rings.polynomial.plural',
               sources = ['sage/rings/polynomial/plural.pyx'],
@@ -1611,7 +1611,7 @@
               language="c++",
               include_dirs = [SAGE_INC +'singular'],
               depends = [singular_depends],
-              extra_compile_args = givaro_extra_compile_args),
+              extra_compile_args = givaro_extra_compile_args + ['-DNDEBUG']),
 
     Extension('sage.rings.polynomial.multi_polynomial_libsingular',
               sources = ['sage/rings/polynomial/multi_polynomial_libsingular.pyx'],
@@ -1619,7 +1619,7 @@
               language="c++",
               include_dirs = [SAGE_INC + 'singular'],
               depends = singular_depends,
-              extra_compile_args = givaro_extra_compile_args),
+              extra_compile_args = givaro_extra_compile_args + ['-DNDEBUG']),
 
     Extension('sage.rings.polynomial.multi_polynomial_ring_generic',
               sources = ['sage/rings/polynomial/multi_polynomial_ring_generic.pyx']), 
