diff -Naur src.orig/sage/doctest/control.py src/sage/doctest/control.py
--- src.orig/sage/doctest/control.py	2016-02-08 22:12:20.089294214 +1300
+++ src/sage/doctest/control.py	2016-02-08 22:13:49.935517660 +1300
@@ -541,11 +541,11 @@
             'sagenb'
         """
         opj = os.path.join
-        from sage.env import SAGE_SRC, SAGE_ROOT
+        from sage.env import SAGE_SRC, SAGE_DOC, SAGE_ROOT
         def all_files():
             from glob import glob
             self.files.append(opj(SAGE_SRC, 'sage'))
-            self.files.extend(glob(opj(SAGE_SRC, 'doc', '[a-z][a-z]')))
+            self.files.extend(glob(opj(SAGE_DOC, '[a-z][a-z]')))
             self.options.sagenb = True
         DOT_GIT= opj(SAGE_ROOT, '.git')
         have_git = os.path.exists(DOT_GIT)
