diff --git a/sage_setup/command/sage_install.py b/sage_setup/command/sage_install.py
index 726049b..0929818 100644
--- a/sage_setup/command/sage_install.py
+++ b/sage_setup/command/sage_install.py
@@ -11,10 +11,10 @@ class sage_install(install):
     def run(self):
         install.run(self)
         self.install_kernel_spec()
-        log.info('Cleaning up stale installed files....')
-        t = time.time()
-        self.clean_stale_files()
-        log.info('Finished cleaning, time: %.2f seconds.' % (time.time() - t))
+        #log.info('Cleaning up stale installed files....')
+        #t = time.time()
+        #self.clean_stale_files()
+        #log.info('Finished cleaning, time: %.2f seconds.' % (time.time() - t))
 
     def install_kernel_spec(self):
         """
diff --git a/sage_setup/docbuild/__init__.py b/sage_setup/docbuild/__init__.py
index 910e414..6119d29 100644
--- a/sage_setup/docbuild/__init__.py
+++ b/sage_setup/docbuild/__init__.py
@@ -60,7 +60,7 @@ from sage.misc.cachefunc import cached_method
 from sage.misc.misc import sage_makedirs
 from sage.env import SAGE_DOC_SRC, SAGE_DOC, SAGE_SRC, DOT_SAGE
 
-from .build_options import (LANGUAGES, SPHINXOPTS, PAPER, OMIT,
+from sage_setup.docbuild.build_options import (LANGUAGES, SPHINXOPTS, PAPER, OMIT,
      PAPEROPTS, ALLSPHINXOPTS, NUM_THREADS, WEBSITESPHINXOPTS,
      INCREMENTAL_BUILD, ABORT_ON_ERROR)
 
diff --git a/sage_setup/docbuild/__main__.py b/sage_setup/docbuild/__main__.py
index 031df43..e68b438 100644
--- a/sage_setup/docbuild/__main__.py
+++ b/sage_setup/docbuild/__main__.py
@@ -1,2 +1,2 @@
-from . import main
+from sage_setup.docbuild import main
 main()
diff --git a/setup.py b/setup.py
index 01dd6c9..c1b2fe4 100755
--- a/setup.py
+++ b/setup.py
@@ -72,7 +72,7 @@ log.warn('distributions = {0}'.format(distributions))
 
 from sage_setup.find import find_python_sources
 python_packages, python_modules, cython_modules = find_python_sources(
-    SAGE_SRC, ['sage', 'sage_setup'], distributions=distributions)
+    SAGE_SRC, ['sage'], distributions=distributions)
 
 log.debug('python_packages = {0}'.format(python_packages))
 
