diff -Naur src.orig/sage/repl/rich_output/display_manager.py src/sage/repl/rich_output/display_manager.py
--- src.orig/sage/repl/rich_output/display_manager.py	2017-04-07 12:28:47.336595531 +1200
+++ src/sage/repl/rich_output/display_manager.py	2017-04-07 12:30:15.045264054 +1200
@@ -743,12 +743,10 @@
             offline threejs graphics
         """
         if online:
-            from sage.misc.package import installed_packages
-            version = installed_packages()['threejs']
             return """
-<script src="https://cdn.rawgit.com/mrdoob/three.js/{0}/build/three.min.js"></script>
-<script src="https://cdn.rawgit.com/mrdoob/three.js/{0}/examples/js/controls/OrbitControls.js"></script>
-            """.format(version)
+<script src="https://cdn.rawgit.com/mrdoob/three.js/r80/build/three.min.js"></script>
+<script src="https://cdn.rawgit.com/mrdoob/three.js/r80/examples/js/controls/OrbitControls.js"></script>
+            """
         try:
             return self._backend.threejs_offline_scripts()
         except AttributeError:
