--- spkg-install.orig	2009-11-26 22:37:46.529185716 +0000
+++ spkg-install	2009-11-27 12:43:29.722755949 +0000
@@ -77,41 +77,6 @@
 CFLAGS="-I$SAGE_LOCAL/include -L$SAGE_LOCAL/lib/ "$CFLAGS; export CFLAGS
 LDFLAGS="-L$SAGE_LOCAL/lib/ "$LDFLAGS; export LDFLAGS   
 
-# do not build recommended packages for now, for speed.
-
-if [ `uname` = "Darwin" ]; then
-     echo "Configuring R for OSX"
-    ./configure --prefix="$SAGE_LOCAL" --with-recommended-packages=no --enable-R-shlib --with-x=$XSUPPORT --with-readline="$SAGE_LOCAL" $OSXFW
-else
-     echo "Configuring R with ATLAS"
-    ./configure --prefix="$SAGE_LOCAL" --with-recommended-packages=no --enable-R-shlib --with-x=$XSUPPORT --with-readline="$SAGE_LOCAL" --with-blas="-L$SAGE_LOCAL/lib -lf77blas -latlas" --with-lapack="-L$SAGE_LOCAL/lib -llapack -lcblas" $SUN_FLAGS
-fi
-
-if [ $? -ne 0 ]; then
-     echo "Configuring R with fallback options"
-    ./configure --prefix="$SAGE_LOCAL" --with-recommended-packages=no --enable-R-shlib --with-x=no --with-readline="$SAGE_LOCAL" $OSXFW $SUN_FLAGS
-fi
-
-if [ $? -ne 0 ]; then
-    echo "Error configuring R."
-    exit 1
-fi
-
-$MAKE R
-if [ $? -ne 0 ]; then
-    echo "Error building R."
-    exit 1
-fi
-
-#parallel make install is broken
-export MAKE=make
-$MAKE vignettes #needed for help system
-$MAKE install
-if [ $? -ne 0 ]; then
-    echo "Error installing R."
-    exit 1
-fi
-
 #$MAKE install-libR
 #if [ $? -ne 0 ]; then
 #    echo "Error installing libR."
@@ -120,7 +85,6 @@
 
 # Fix hardcoding problems.
 cd "$CUR"/patches
-python fix_hardcode
 
 echo "Now install rpy"
 
