diff --git a/sage/libs/gap/saved_workspace.py b/sage/libs/gap/saved_workspace.py
index ad5adec..1dcc3c2 100644
--- a/sage/libs/gap/saved_workspace.py
+++ b/sage/libs/gap/saved_workspace.py
@@ -8,7 +8,6 @@ workspaces.
 
 import os
 import glob
-from sage.env import GAP_ROOT_DIR
 from sage.interfaces.gap_workspace import gap_workspace_file
 
 
@@ -31,8 +30,7 @@ def timestamp():
     """
     libgap_dir = os.path.dirname(__file__)
     libgap_files = glob.glob(os.path.join(libgap_dir, '*'))
-    gap_packages = glob.glob(os.path.join(GAP_ROOT_DIR, 'pkg', '*'))
-    files = libgap_files + gap_packages
+    files = libgap_files
     if len(files) == 0:
         print('Unable to find LibGAP files.')
         return float('inf')
diff --git a/sage/libs/gap/util.pyx b/sage/libs/gap/util.pyx
index 2649755..db45bf7 100644
--- a/sage/libs/gap/util.pyx
+++ b/sage/libs/gap/util.pyx
@@ -183,9 +183,6 @@ def gap_root():
         sage: gap_root()   # random output
         '/home/vbraun/opt/sage-5.3.rc0/local/gap/latest'
     """
-    if os.path.exists(sage.env.GAP_ROOT_DIR):
-        return sage.env.GAP_ROOT_DIR
-
     # Attempt to figure out the appropriate GAP_ROOT by reading the
     # local/bin/gap shell script; this is an ugly hack that exists for
     # historical reasons; the best approach to setting where Sage looks for
