--- a/sage/misc/sage_ostools.py	2013-03-28 19:19:40.000000000 +0000
+++ b/sage/misc/sage_ostools.py	2013-04-14 09:47:19.288092468 +0000
@@ -24,7 +24,7 @@
         True
         sage: have_program('there_is_not_a_program_with_this_name')
         False
-        sage: have_program('sage', path=SAGE_ROOT)
+        sage: have_program('sage')
         True
         sage: have_program('ls', path=SAGE_ROOT)
         False
