diff --git a/sage/plot/plot.py b/sage/plot/plot.py
index 5301a05..cef7876 100644
--- a/sage/plot/plot.py
+++ b/sage/plot/plot.py
@@ -1744,11 +1744,6 @@ def plot(funcs, *args, **kwds):
         sage: plot(2*x+1,(x,0,5),ticks=[[0,1,e,pi,sqrt(20)],2],tick_formatter="latex") # not tested (broken with matplotlib 3.6)
         Graphics object consisting of 1 graphics primitive
 
-    .. PLOT::
-
-        g = plot(2*x+1,(x,0,5),ticks=[[0,1,e,pi,sqrt(20)],2],tick_formatter="latex")
-        sphinx_plot(g)
-
     This is particularly useful when setting custom ticks in multiples of `pi`.
 
     ::
