--- sage/databases/cremona.py.orig	2013-02-19 10:33:56.000000000 +0000
+++ sage/databases/cremona.py	2013-03-18 16:51:58.950097433 +0000
@@ -634,19 +634,19 @@
 
             sage: it = CremonaDatabase().__iter__()
             sage: it.next().label()
-            '11a1'
+            '1001a1'
             sage: it.next().label()
-            '11a2'
+            '1001b1'
             sage: it.next().label()
-            '11a3'
+            '1001b2'
             sage: it.next().label()
-            '14a1'
+            '1001b3'
             sage: skip = [it.next() for _ in range(100)]
             sage: it.next().label()
-            '45a3'
+            '1014g2'
         """
         for c in self.__connection__.cursor().execute('SELECT curve FROM ' \
-            + 't_curve'):
+            + 't_curve ORDER BY curve'):
             yield self.elliptic_curve(c[0])
 
     def __getitem__(self, N):
