diff --git a/src/sage/repl/interpreter.py b/src/sage/repl/interpreter.py
index 5c0b980..f86818a 100644
--- a/src/sage/repl/interpreter.py
+++ b/src/sage/repl/interpreter.py
@@ -755,7 +755,8 @@ class SageTerminalApp(TerminalIPythonApp):
             sage: from sage.misc.temporary_file import tmp_dir
             sage: from sage.repl.interpreter import SageTerminalApp
             sage: d = tmp_dir()
-            sage: IPYTHONDIR = os.environ['IPYTHONDIR']
+            sage: from IPython.utils.path import get_ipython_dir
+            sage: IPYTHONDIR = get_ipython_dir()
             sage: os.environ['IPYTHONDIR'] = d
             sage: SageTerminalApp().load_config_file()
             sage: os.environ['IPYTHONDIR'] = IPYTHONDIR
diff --git a/src/sage/repl/ipython_kernel/install.py b/src/sage/repl/ipython_kernel/install.py
index e3c4cda..318eec7 100644
--- a/src/sage/repl/ipython_kernel/install.py
+++ b/src/sage/repl/ipython_kernel/install.py
@@ -12,7 +12,7 @@ import copy
 
 from IPython.kernel.kernelspec import (
     get_kernel_spec, install_kernel_spec, NoSuchKernel)
-    
+from IPython.utils.path import get_ipython_dir
 
 from sage.env import (
     SAGE_ROOT, SAGE_DOC, SAGE_LOCAL, SAGE_EXTCODE,
@@ -35,7 +35,7 @@ class SageKernelSpec(object):
             'Sage 6.6.beta2'
         """
         self._display_name = 'Sage {0}'.format(SAGE_VERSION)
-        self._ipython_dir = os.environ['IPYTHONDIR']
+        self._ipython_dir = get_ipython_dir()
         self._mkdirs()
 
     def _mkdirs(self):
@@ -113,9 +113,10 @@ class SageKernelSpec(object):
         EXAMPLES::
 
             sage: from sage.repl.ipython_kernel.install import SageKernelSpec
+            sage: from IPython.utils.path import get_ipython_dir
             sage: spec = SageKernelSpec()
             sage: spec.use_local_mathjax()
-            sage: ipython_dir = os.environ['IPYTHONDIR']
+            sage: ipython_dir = get_ipython_dir()
             sage: mathjax = os.path.join(ipython_dir, 'nbextensions', 'mathjax')
             sage: os.path.exists(mathjax)
             True
