diff -Naur src.orig/sage/all.py src/sage/all.py
--- src.orig/sage/all.py	2016-05-04 13:30:24.713199987 +1200
+++ src/sage/all.py	2016-05-04 13:33:30.648569877 +1200
@@ -291,33 +291,7 @@
 
 
 # Write a file indicating that Sage was started up successfully.
-# This is called by the sage-starts script.
-def _write_started_file():
-    """
-    Write a ``sage-started.txt`` file if it does not exist.  The
-    contents of this file do not matter, only its existence.
-
-    The non-existence of this file will be used as a trigger to run
-    ``sage-starts`` during the Sage build.
-
-    TESTS:
-
-    Check that the file exists when Sage is running::
-
-        sage: started_file = os.path.join(SAGE_LOCAL, 'etc', 'sage-started.txt')
-        sage: os.path.isfile(started_file)
-        True
-    """
-    started_file = os.path.join(SAGE_LOCAL, 'etc', 'sage-started.txt')
-
-    # Current time with a resolution of 1 second
-    import datetime
-    t = datetime.datetime.now().replace(microsecond=0)
-
-    O = open(started_file, 'w')
-    O.write("Sage {} was started at {}\n".format(sage.version.version, t))
-    O.close()
-
+# Don't do that on Gentoo.
 
 # Set a new random number seed as the very last thing
 # (so that printing initial_seed() and using that seed
