diff --git a/sage/misc/sagedoc.py b/sage/misc/sagedoc.py
index 644341f36b..84306be417 100644
--- a/sage/misc/sagedoc.py
+++ b/sage/misc/sagedoc.py
@@ -502,7 +502,7 @@ def process_dollars(s):
 # Sage trac ticket shortcuts. For example, :trac:`7549` .
 pythonversion = sys.version.split(' ')[0]
 extlinks = {
-    'python': ('https://docs.python.org/release/'+pythonversion+'/%s', ''),
+    'python': ('https://docs.python.org/release/'+pythonversion+'/%s', None),
     'trac': ('https://trac.sagemath.org/%s', 'trac ticket #%s'),
     'wikipedia': ('https://en.wikipedia.org/wiki/%s', 'Wikipedia article %s'),
     'arxiv': ('https://arxiv.org/abs/%s', 'arXiv %s'),
