diff --git a/sage/calculus/calculus.py b/sage/calculus/calculus.py
index 80e4a5d..c445be0 100644
--- a/sage/calculus/calculus.py
+++ b/sage/calculus/calculus.py
@@ -1628,8 +1628,11 @@ def laplace(ex, t, s, algorithm='maxima'):
 
         sage: laplace(dirac_delta(t), t, s)
         1
-        sage: laplace(dirac_delta(t), t, s, algorithm='sympy')
-        (-heaviside(0) + 1, -oo, True)
+        sage: F, a, cond = laplace(dirac_delta(t), t, s, algorithm='sympy')
+        sage: a, cond
+        (-oo, True)
+        sage: F       # random - sympy <1.9 includes undefined heaviside(0) in answer
+        1
         sage: laplace(dirac_delta(t), t, s, algorithm='giac')
         1
 
diff --git a/sage/calculus/test_sympy.py b/sage/calculus/test_sympy.py
index 86f9366..5383f58 100644
--- a/sage/calculus/test_sympy.py
+++ b/sage/calculus/test_sympy.py
@@ -193,7 +193,7 @@ This was fixed in Sympy, see :trac:`14437`::
     sage: u = Function('u')
     sage: n = Symbol('n', integer=True)
     sage: f = u(n+2) - u(n+1) + u(n)/4
-    sage: rsolve(f,u(n))
-    2**(-n)*(C0 + C1*n)
+    sage: 2**n * rsolve(f,u(n))
+    C1*n + C0
 
 """
diff --git a/sage/categories/sets_cat.py b/sage/categories/sets_cat.py
index 7e8cdc9..2b29f83 100644
--- a/sage/categories/sets_cat.py
+++ b/sage/categories/sets_cat.py
@@ -1710,7 +1710,7 @@ Please use, e.g., S.algebra(QQ, category=Semigroups())""".format(self))
                 sage: list(sF)
                 [1, 2, 3]
                 sage: from sympy import FiniteSet
-                sage: FiniteSet.fromiter(sF)
+                sage: FiniteSet.fromiter(sF)  # random - this output format is sympy >= 1.9
                 FiniteSet(1, 2, 3)
 
                 sage: RR._sympy_().is_finite_set
diff --git a/sage/sets/real_set.py b/sage/sets/real_set.py
index 0c9d276..1e8dfa6 100644
--- a/sage/sets/real_set.py
+++ b/sage/sets/real_set.py
@@ -452,8 +452,10 @@ class InternalRealInterval(UniqueRepresentation, Parent):
 
             sage: RealSet.open_closed(0, 1)[0]._sympy_()
             Interval.Lopen(0, 1)
-            sage: RealSet.point(0)[0]._sympy_()
-            FiniteSet(0)
+            sage: RealSet.point(0)[0]._sympy_()  # random - this output format is sympy >= 1.9
+            {0}
+            sage: type(_)
+            <class 'sympy.sets.sets.FiniteSet'>
             sage: RealSet.open(0,1)[0]._sympy_()
             Interval.open(0, 1)
             sage: RealSet.open(-oo,1)[0]._sympy_()
@@ -2454,10 +2456,10 @@ class RealSet(UniqueRepresentation, Parent, Set_base,
 
             sage: RealSet()._sympy_()
             EmptySet
-            sage: RealSet.point(5)._sympy_()
-            FiniteSet(5)
-            sage: (RealSet.point(1).union(RealSet.point(2)))._sympy_()
-            FiniteSet(1, 2)
+            sage: RealSet.point(5)._sympy_()  # random - this output format is sympy >= 1.9
+            {5}
+            sage: (RealSet.point(1).union(RealSet.point(2)))._sympy_()  # random
+            {1, 2}
             sage: (RealSet(1, 2).union(RealSet.closed(3, 4)))._sympy_()
             Union(Interval.open(1, 2), Interval(3, 4))
             sage: RealSet(-oo, oo)._sympy_()
diff --git a/sage/symbolic/ring.pyx b/sage/symbolic/ring.pyx
index ffe8d74..5a84fe0 100644
--- a/sage/symbolic/ring.pyx
+++ b/sage/symbolic/ring.pyx
@@ -1271,6 +1271,8 @@ cdef class UnderscoreSageMorphism(Morphism):
         import sage.categories.homset
         from sage.sets.pythonclass import Set_PythonType
         Morphism.__init__(self, sage.categories.homset.Hom(Set_PythonType(t), R))
+        from sage.interfaces.sympy import sympy_init
+        sympy_init()
 
     cpdef Element _call_(self, a):
         """
diff --git a/sage/tests/books/computational-mathematics-with-sagemath/recequadiff_doctest.py b/sage/tests/books/computational-mathematics-with-sagemath/recequadiff_doctest.py
index 97ac97b..1062f4f 100644
--- a/sage/tests/books/computational-mathematics-with-sagemath/recequadiff_doctest.py
+++ b/sage/tests/books/computational-mathematics-with-sagemath/recequadiff_doctest.py
@@ -374,8 +374,8 @@ Sage example in ./recequadiff.tex, line 1700::
 Sage example in ./recequadiff.tex, line 1707::
 
   sage: from sympy import rsolve
-  sage: rsolve(f, u(n), {u(0):-1,u(1):1})
-  3 - 4*2**(-n)
+  sage: rsolve(f, u(n), {u(0):-1,u(1):1}) == 3 - 4*2**(-n)
+  True
 
 Sage example in ./recequadiff.tex, line 1798::
