diff --git a/sage/misc/cython.py b/sage/misc/cython.py
index 8277842..822d115 100644
--- a/sage/misc/cython.py
+++ b/sage/misc/cython.py
@@ -529,6 +529,10 @@ def cython_import(filename, **kwds):
     try:
         sys.path.append(build_dir)
         return builtins.__import__(name)
+    except ModuleNotFoundError:
+        import importlib
+        importlib.invalidate_caches()
+        return builtins.__import__(name)
     finally:
         sys.path = oldpath
 
