diff --git a/bin/sage-notebook b/bin/sage-notebook
index 38b680da5d..83109e627c 100755
--- a/bin/sage-notebook
+++ b/bin/sage-notebook
@@ -10,7 +10,6 @@ import textwrap
 logging.basicConfig()
 logger = logging.getLogger()
 
-from sage.env import SAGE_ROOT
 from sage.misc.banner import banner
 
 
diff --git a/sage/doctest/control.py b/sage/doctest/control.py
index 96143a66f4..cf433e2b09 100644
--- a/sage/doctest/control.py
+++ b/sage/doctest/control.py
@@ -692,9 +692,13 @@ class DocTestController(SageObject):
             'sagenb'
         """
         opj = os.path.join
-        from sage.env import SAGE_SRC, SAGE_DOC_SRC, SAGE_ROOT
-        DOT_GIT = opj(SAGE_ROOT, '.git')
-        have_git = os.path.exists(DOT_GIT)
+        from sage.env import SAGE_SRC, SAGE_DOC_SRC, SAGE_ROOT, SAGE_ROOT_GIT
+        # SAGE_ROOT_GIT can be None on distributions which typically
+        # only have the SAGE_LOCAL install tree but not SAGE_ROOT
+        if SAGE_ROOT_GIT is not None:
+            have_git = os.path.isdir(SAGE_ROOT_GIT)
+        else:
+            have_git = False
 
         def all_files():
             self.files.append(opj(SAGE_SRC, 'sage'))
@@ -714,7 +718,7 @@ class DocTestController(SageObject):
             self.log("Doctesting files changed since last git commit")
             import subprocess
             change = subprocess.check_output(["git",
-                                              "--git-dir=" + DOT_GIT,
+                                              "--git-dir=" + SAGE_ROOT_GIT,
                                               "--work-tree=" + SAGE_ROOT,
                                               "status",
                                               "--porcelain"])
@@ -1199,13 +1203,14 @@ class DocTestController(SageObject):
         else:
             self.test_safe_directory()
             self.create_run_id()
-            from sage.env import SAGE_ROOT
-            DOT_GIT = os.path.join(SAGE_ROOT, '.git')
-            if os.path.isdir(DOT_GIT):
+            from sage.env import SAGE_ROOT_GIT
+            # SAGE_ROOT_GIT can be None on distributions which typically
+            # only have the SAGE_LOCAL install tree but not SAGE_ROOT
+            if (SAGE_ROOT_GIT is not None) and os.path.isdir(SAGE_ROOT_GIT):
                 import subprocess
                 try:
                     branch = subprocess.check_output(["git",
-                                                      "--git-dir=" + DOT_GIT,
+                                                      "--git-dir=" + SAGE_ROOT_GIT,
                                                       "rev-parse",
                                                       "--abbrev-ref",
                                                       "HEAD"])
diff --git a/sage/env.py b/sage/env.py
index 985fcdfb07..62c76f2300 100644
--- a/sage/env.py
+++ b/sage/env.py
@@ -162,6 +162,7 @@ var('SAGE_ROOT')           # no fallback for SAGE_ROOT
 var('SAGE_SRC',            join(SAGE_ROOT, 'src'), SAGE_LIB)
 var('SAGE_DOC_SRC',        join(SAGE_SRC, 'doc'))
 var('SAGE_PKGS',           join(SAGE_ROOT, 'build', 'pkgs'))
+var('SAGE_ROOT_GIT',       join(SAGE_ROOT, '.git'))
 
 var('DOT_SAGE',            join(os.environ.get('HOME'), '.sage'))
 var('SAGE_STARTUP_FILE',   join(DOT_SAGE, 'init.sage'))
diff --git a/sage/misc/citation.pyx b/sage/misc/citation.pyx
index dd59e70a87..3145a599f3 100644
--- a/sage/misc/citation.pyx
+++ b/sage/misc/citation.pyx
@@ -5,7 +5,7 @@ Dependency usage tracking for citations
 from __future__ import absolute_import
 
 from sage.misc.all import tmp_filename
-from sage.env import SAGE_ROOT
+from sage.env import SAGE_LOCAL
 
 systems = {}
 systems['PARI'] = ['cypari2', 'sage.interfaces.gp']
@@ -112,7 +112,7 @@ def get_systems(cmd):
     stats = pstats.Stats(filename)
 
     #Strings is a list of method names and modules which get run
-    strings = [a[0].replace(SAGE_ROOT, "") + " " + a[2]
+    strings = [a[0].replace(SAGE_LOCAL, "") + " " + a[2]
                for a in stats.stats]
 
     #Remove trivial functions
diff --git a/sage/misc/inline_fortran.py b/sage/misc/inline_fortran.py
index fe2565cb76..092b5041e2 100644
--- a/sage/misc/inline_fortran.py
+++ b/sage/misc/inline_fortran.py
@@ -141,12 +141,12 @@ class InlineFortran:
 
         TESTS::
 
-            sage: os.chdir(SAGE_ROOT)
+            sage: os.chdir(DOT_SAGE)
             sage: fortran.eval("SYNTAX ERROR !@#$")
             Traceback (most recent call last):
             ...
             RuntimeError: failed to compile Fortran code:...
-            sage: os.getcwd() == SAGE_ROOT
+            sage: os.getcwd() == DOT_SAGE
             True
         """
         if globals is None:
diff --git a/sage/misc/misc.py b/sage/misc/misc.py
index 8860669888..2d6857fce1 100644
--- a/sage/misc/misc.py
+++ b/sage/misc/misc.py
@@ -74,10 +74,9 @@ def sage_makedirs(dir):
         sage: sage_makedirs(DOT_SAGE) # no output
 
     The following fails because we are trying to create a directory in
-    place of an ordinary file (the main Sage executable)::
+    place of an ordinary file (the python executable)::
 
-        sage: sage_executable = os.path.join(SAGE_ROOT, 'sage')
-        sage: sage_makedirs(sage_executable)
+        sage: sage_makedirs(sys.executable)
         Traceback (most recent call last):
         ...
         OSError: ...
diff --git a/sage/misc/package.py b/sage/misc/package.py
index 689e5a23b9..8501acb53f 100644
--- a/sage/misc/package.py
+++ b/sage/misc/package.py
@@ -43,7 +43,7 @@ Functions
 #*****************************************************************************
 from __future__ import print_function
 
-from sage.env import SAGE_ROOT, SAGE_PKGS
+from sage.env import SAGE_PKGS
 
 import json
 import os
