--- deps	2009-10-27 18:41:31.586711209 +0000
+++ deps-use-system-mercurial	2009-10-28 13:04:26.405881473 +0000
@@ -224,9 +224,8 @@
 $(INST)/$(SQLITE): $(INST)/$(TERMCAP) $(INST)/$(READLINE)
 	$(SAGE_SPKG) $(SQLITE) 2>&1
 
-$(INST)/$(MERCURIAL): $(INST)/$(PYTHON)
-	$(SAGE_SPKG) $(MERCURIAL) 2>&1
-	$(MAKEREL)
+$(INST)/$(MERCURIAL):
+	@echo "not building $(MERCURIAL) - using system default"
 
 $(INST)/$(SETUPTOOLS): $(INST)/$(PYTHON)
 	$(SAGE_SPKG) $(SETUPTOOLS) 2>&1
