--- deps	2009-10-27 18:41:31.586711209 +0000
+++ use-gap-from-portage	2009-11-29 12:19:26.540696474 +0000
@@ -155,8 +155,8 @@
 	$(SAGE_SPKG) $(SYMMETRICA) 2>&1
 
 # gap requires SAGE so that gap_reset_workspace works.
-$(INST)/$(GAP): $(BASE)  $(INST)/$(MPIR) $(INST)/$(TERMCAP) $(INST)/$(READLINE) $(INST)/$(SAGE)
-	$(SAGE_SPKG) $(GAP) 2>&1
+$(INST)/$(GAP):
+	@echo "using gap from portage"
 
 $(INST)/$(IPYTHON): $(INST)/$(PYTHON)
 	$(SAGE_SPKG) $(IPYTHON) 2>&1
