--- deps	2009-10-27 18:41:31.586711209 +0000
+++ use-sqlite-from-portage	2009-11-03 10:34:18.410151683 +0000
@@ -221,8 +221,8 @@
 $(INST)/$(SQLALCHEMY): $(INST)/$(PYTHON) $(INST)/$(SETUPTOOLS)
 	$(SAGE_SPKG) $(SQLALCHEMY) 2>&1
 
-$(INST)/$(SQLITE): $(INST)/$(TERMCAP) $(INST)/$(READLINE)
-	$(SAGE_SPKG) $(SQLITE) 2>&1
+$(INST)/$(SQLITE):
+	@echo "using sqlite from portage"
 
 $(INST)/$(MERCURIAL): $(INST)/$(PYTHON)
 	$(SAGE_SPKG) $(MERCURIAL) 2>&1
