--- deps	2009-10-27 18:41:31.586711209 +0000
+++ deps-use-system-scons	2009-10-27 18:24:37.292711228 +0000
@@ -199,8 +199,8 @@
 $(INST)/$(PYTHON_GNUTLS):  $(INST)/$(GNUTLS) $(INST)/$(PYTHON)
 	$(SAGE_SPKG) $(PYTHON_GNUTLS) 2>&1
 
-$(INST)/$(SCONS):  $(INST)/$(PYTHON)
-	$(SAGE_SPKG) $(SCONS) 2>&1
+$(INST)/$(SCONS):
+	@echo "not building $(SCONS) - using system default"
 
 #$(INST)/$(OPENSSL): $(BASE)
 #	$(SAGE_SPKG) $(OPENSSL) 2>&1
