--- deps	2009-10-27 18:41:31.586711209 +0000
+++ use-ecm-from-portage	2009-11-19 16:48:57.539294360 +0000
@@ -279,8 +279,8 @@
 $(INST)/$(TACHYON): $(BASE) $(INST)/$(LIBPNG)
 	$(SAGE_SPKG) $(TACHYON) 2>&1
 
-$(INST)/$(ECM): $(BASE) $(INST)/$(MPIR)
-	$(SAGE_SPKG) $(ECM) 2>&1
+$(INST)/$(ECM):
+	@echo "using ecm from portage"
 
 $(INST)/$(RATPOINTS): $(BASE) $(INST)/$(MPIR)
 	$(SAGE_SPKG) $(RATPOINTS) 2>&1
