--- deps	2009-10-27 18:41:31.586711209 +0000
+++ use-znpoly-from-portage	2009-11-07 21:05:09.245718634 +0000
@@ -335,8 +335,8 @@
 $(INST)/$(LIBM4RI): $(BASE)
 	$(SAGE_SPKG) $(LIBM4RI) 2>&1
 
-$(INST)/$(ZNPOLY): $(INST)/$(MPIR)
-	$(SAGE_SPKG) $(ZNPOLY) 2>&1
+$(INST)/$(ZNPOLY):
+	@echo "using zn_poly from portage"
 
 # All components that are linked against extensions need to be listed here
 $(INST)/$(SAGE):  $(INST)/$(SAGE_SCRIPTS) \
