--- deps	2009-10-27 18:41:31.586711209 +0000
+++ use-givaro-from-portage	2009-11-07 20:58:44.480718687 +0000
@@ -124,8 +124,8 @@
 $(INST)/$(MPFI): $(BASE) $(INST)/$(MPIR) $(INST)/$(MPFR)
 	$(SAGE_SPKG) $(MPFI) 2>&1
 
-$(INST)/$(GIVARO): $(BASE) $(INST)/$(MPIR)
-	$(SAGE_SPKG) $(GIVARO) 2>&1
+$(INST)/$(GIVARO):
+	@echo "using givaro from portage"
 
 $(INST)/$(LINBOX): $(BASE) $(INST)/$(MPIR) $(INST)/$(NTL) $(INST)/$(GIVARO) $(INST)/$(GSL) $(INST)/$(ATLAS)
 	$(SAGE_SPKG) $(LINBOX) 2>&1
