--- deps	2009-10-27 18:41:31.586711209 +0000
+++ use-genus2reduction-from-portage	2009-11-23 12:45:52.475283928 +0000
@@ -136,8 +136,8 @@
 $(INST)/$(ECLIB): $(BASE) $(INST)/$(MPIR) $(INST)/$(PARI) $(INST)/$(NTL)
 	$(SAGE_SPKG) $(ECLIB) 2>&1
 
-$(INST)/$(G2RED): $(BASE) $(INST)/$(PARI)
-	$(SAGE_SPKG) $(G2RED) 2>&1
+$(INST)/$(G2RED):
+	@echo "using genus2reduction from portage"
 
 $(INST)/$(PALP): $(BASE)
 	$(SAGE_SPKG) $(PALP) 2>&1
