--- deps	2009-10-27 18:41:31.586711209 +0000
+++ deps-use-system-jinja2	2009-10-27 17:59:16.732711403 +0000
@@ -404,8 +404,8 @@
 $(INST)/$(JINJA): $(INST)/$(DOCUTILS) $(INST)/$(SETUPTOOLS)
 	$(SAGE_SPKG) $(JINJA) 2>&1
 
-$(INST)/$(JINJA2): $(INST)/$(DOCUTILS) $(INST)/$(SETUPTOOLS)
-	$(SAGE_SPKG) $(JINJA2) 2>&1
+$(INST)/$(JINJA2):
+	@echo "not building $(JINJA2) - using system default"
 
 $(INST)/$(LAPACK): $(INST)/$(FORTRAN)
 	$(SAGE_SPKG) $(INST)/$(LAPACK) 2>&1
