--- deps	2009-10-27 18:41:31.586711209 +0000
+++ use-atlas-from-portage	2009-10-31 10:12:48.695286710 +0000
@@ -49,8 +49,8 @@
 # Building normal packages
 ########################################
 
-$(INST)/$(ATLAS): $(BASE) $(INST)/$(LAPACK) $(INST)/$(PYTHON)
-	$(SAGE_SPKG) $(ATLAS) 2>&1
+$(INST)/$(ATLAS):
+	@echo "using atlas from portage"
 
 $(INST)/$(BOEHM_GC): $(BASE)
 	$(SAGE_SPKG) $(BOEHM_GC) 2>&1
