--- deps	2009-10-27 18:41:31.586711209 +0000
+++ deps-use-system-jinja	2009-10-27 17:56:56.439710391 +0000
@@ -401,8 +401,8 @@
 $(INST)/$(PIL): $(INST)/$(PYTHON)
 	$(SAGE_SPKG) $(PIL) 2>&1
 
-$(INST)/$(JINJA): $(INST)/$(DOCUTILS) $(INST)/$(SETUPTOOLS)
-	$(SAGE_SPKG) $(JINJA) 2>&1
+$(INST)/$(JINJA):
+	@echo "not building $(JINJA) - using system default"
 
 $(INST)/$(JINJA2): $(INST)/$(DOCUTILS) $(INST)/$(SETUPTOOLS)
 	$(SAGE_SPKG) $(JINJA2) 2>&1
