--- deps	2009-10-27 18:41:31.586711209 +0000
+++ use-mpfi-from-portage	2009-11-07 18:19:39.948720909 +0000
@@ -121,8 +121,8 @@
 $(INST)/$(MPFR): $(BASE) $(INST)/$(MPIR)
 	$(SAGE_SPKG) $(MPFR) 2>&1
 
-$(INST)/$(MPFI): $(BASE) $(INST)/$(MPIR) $(INST)/$(MPFR)
-	$(SAGE_SPKG) $(MPFI) 2>&1
+$(INST)/$(MPFI):
+	@echo "using mpfi from portage"
 
 $(INST)/$(GIVARO): $(BASE) $(INST)/$(MPIR)
 	$(SAGE_SPKG) $(GIVARO) 2>&1
