--- deps	2009-10-27 18:41:31.586711209 +0000
+++ deps-use-system-blas	2009-10-28 12:29:56.002882156 +0000
@@ -410,8 +410,8 @@
 $(INST)/$(LAPACK): $(INST)/$(FORTRAN)
 	$(SAGE_SPKG) $(INST)/$(LAPACK) 2>&1
 
-$(INST)/$(BLAS): $(INST)/$(FORTRAN)
-	$(SAGE_SPKG) $(INST)/$(BLAS) 2>&1
+$(INST)/$(BLAS):
+	@echo "not building $(BLAS) - using system default"
 
 $(INST)/$(NUMPY): $(BASE) $(INST)/$(PYTHON) $(INST)/$(FORTRAN) $(INST)/$(F2C) $(INST)/$(LAPACK) $(INST)/$(BLAS) $(INST)/$(ATLAS)
 	$(SAGE_SPKG) $(NUMPY) 2>&1
