--- deps	2009-10-27 18:41:31.586711209 +0000
+++ deps-use-system-boost	2009-10-27 17:20:33.388711657 +0000
@@ -55,8 +55,8 @@
 $(INST)/$(BOEHM_GC): $(BASE)
 	$(SAGE_SPKG) $(BOEHM_GC) 2>&1
 
-$(INST)/$(BOOST_CROPPED): $(BASE)
-	$(SAGE_SPKG) $(BOOST_CROPPED) 2>&1
+$(INST)/$(BOOST_CROPPED):
+	@echo "not building $(BOOST_CROPPED) - using system default"
 
 $(INST)/$(CLIQUER): $(BASE) $(INST)/$(SCONS)
 	$(SAGE_SPKG) $(CLIQUER) 2>&1
