--- deps	2009-10-27 18:41:31.586711209 +0000
+++ deps-use-system-libpng	2009-10-27 18:06:22.605712203 +0000
@@ -261,8 +261,8 @@
 $(INST)/$(FREETYPE): $(BASE)
 	$(SAGE_SPKG) $(FREETYPE) 2>&1
 
-$(INST)/$(LIBPNG): $(BASE)
-	$(SAGE_SPKG) $(LIBPNG) 2>&1
+$(INST)/$(LIBPNG):
+	@echo "not building $(LIBPNG) - using system default"
 
 $(INST)/$(MATPLOTLIB): $(BASE) $(INST)/$(PYTHON) $(INST)/$(NUMPY) $(INST)/$(FREETYPE) $(INST)/$(LIBPNG) $(INST)/$(GDMODULE)
 	$(SAGE_SPKG) $(MATPLOTLIB) 2>&1
