--- deps	2009-10-27 18:41:31.586711209 +0000
+++ deps-use-system-maxima	2009-10-27 17:18:46.494710573 +0000
@@ -292,8 +292,8 @@
 $(INST)/$(ECL): $(BASE) $(INST)/$(MPIR) $(INST)/$(TERMCAP) $(INST)/$(READLINE) $(INST)/$(BOEHM_GC)
 	$(SAGE_SPKG) $(ECL) 2>&1
 
-$(INST)/$(MAXIMA): $(BASE) $(INST)/$(ECL)
-	$(SAGE_SPKG) $(MAXIMA) 2>&1
+$(INST)/$(MAXIMA):
+	@echo "not building $(MAXIMA) - using system default"
 
 $(INST)/$(ZODB): $(INST)/$(PYTHON)
 	$(SAGE_SPKG) $(ZODB) 2>&1
