--- deps	2009-10-27 18:41:31.586711209 +0000
+++ deps-use-system-sqlalchemy	2009-10-27 18:34:53.401721629 +0000
@@ -218,8 +218,8 @@
 $(INST)/$(SPHINX): $(INST)/$(DOCUTILS) $(INST)/$(JINJA2) $(INST)/$(PYGMENTS)
 	$(SAGE_SPKG) $(INST)/$(SPHINX) 2>&1
 
-$(INST)/$(SQLALCHEMY): $(INST)/$(PYTHON) $(INST)/$(SETUPTOOLS)
-	$(SAGE_SPKG) $(SQLALCHEMY) 2>&1
+$(INST)/$(SQLALCHEMY):
+	@echo "not building $(SQLALCHEMY) - using system default"
 
 $(INST)/$(SQLITE): $(INST)/$(TERMCAP) $(INST)/$(READLINE)
 	$(SAGE_SPKG) $(SQLITE) 2>&1
