--- deps	2009-10-27 18:41:31.586711209 +0000
+++ deps-use-system-readline	2009-10-27 17:19:47.085710518 +0000
@@ -64,8 +64,8 @@
 $(INST)/$(TERMCAP): $(BASE)
 	$(SAGE_SPKG) $(TERMCAP) 2>&1
 
-$(INST)/$(READLINE): $(BASE)
-	$(SAGE_SPKG) $(READLINE) 2>&1
+$(INST)/$(READLINE):
+	@echo "not building $(READLINE) - using system default"
 
 $(INST)/$(DOCUTILS): $(INST)/$(PYTHON)
 	$(SAGE_SPKG) $(DOCUTILS) 2>&1
