--- Makefile.am.orig	2009-04-18 18:25:06.000000000 +0000
+++ Makefile.am	2010-06-08 16:42:51.948855703 +0000
@@ -24,7 +24,9 @@
 # in front of the full list of input object files. Automake does not allow
 # modifying the ecm_OBJECTS variable directly.
 ecm_LDFLAGS = 
-ecm_LDFLAGS += $(ECM_FASTOBJ)
+libecm_la_LIBADD = @LIBGOMP@
+ecm_LDADD = $(LDADD)
+ecm_LDADD += $(ECM_FASTOBJ)
 
 tune_CPPFLAGS = -DTUNE $(AM_CPPFLAGS)
 # tune_LDADD =
@@ -112,10 +114,5 @@
 ## (cf section "Man pages" in the automake manual)
 dist_man_MANS = ecm.1
 
-man_MANS =
-if MAKE_MANPAGE
-  man_MANS += ecm.1
-endif
-
 ecm.1: ecm.xml
 	xsltproc -o ecm.1 $(XSLDIR)/manpages/docbook.xsl ecm.xml
