--- setup.py.orig	2011-03-30 09:02:11.000000000 -0500
+++ setup.py	2011-03-30 09:03:32.000000000 -0500
@@ -29,7 +29,8 @@
       url         = 'http://www.sagemath.org',
       install_requires = ['twisted>=8.2'],
       test_suite = 'sagenb.testing.run_tests.all_tests',
-      packages    = ['sagenb',
+      packages    = ['flask_version',
+                     'sagenb',
                      'sagenb.interfaces',
                      'sagenb.misc',                                 
                      'sagenb.notebook',
