diff --git a/src/graphobject.c b/src/graphobject.c
index 1bbe590..e40013b 100644
--- a/src/graphobject.c
+++ b/src/graphobject.c
@@ -4804,7 +4804,7 @@ PyObject *igraphmodule_Graph_personalized_pagerank(igraphmodule_GraphObject *sel
   void *opts;
   int retval;
 
-  if (!PyArg_ParseTupleAndKeywords(args, kwds, "|OOdOOOO!Oid", kwlist, &vobj,
+  if (!PyArg_ParseTupleAndKeywords(args, kwds, "|OOdOOOO!Olf", kwlist, &vobj,
                                    &directed, &damping, &robj,
 				   &rvsobj, &wobj,
                                    &igraphmodule_ARPACKOptionsType,
