--- SConstruct.orig	2011-02-15 23:01:45.000000000 +0000
+++ SConstruct	2011-02-18 10:51:47.892324003 +0000
@@ -1314,6 +1323,11 @@
     env.Alias('install', ipboribin)
     env.AlwaysBuild(guibin)   
     env.Alias('install', guibin)
+
+    # we dump the flags for reuse by Sage
+    if not path.exists(InstPath()):
+        Execute(Mkdir(InstPath())) 
+    opts.Save(InstPath("flags.conf"), env)
     
 env.Alias('prepare-devel', devellibs + readabledevellibs)
 env.Alias('prepare-install', [pyroot, DocPath()])
