--- libpolybori/include/polybori/iterators/CTermStack.h.orig	2012-03-16 22:41:09.000000000 +0100
+++ libpolybori/include/polybori/iterators/CTermStack.h	2012-05-28 08:26:59.065981115 +0200
@@ -854,7 +854,7 @@
     PBORI_ASSERT(m_zero.isValid());
     PBORI_ASSERT(m_zero.isEmpty());
 
-    push(m_zero);
+    base::push(m_zero);
   }
 
 private:
--- libpolybori/include/polybori/orderings/CBlockOrderingFacade.h.orig	2012-03-16 22:41:09.000000000 +0100
+++ libpolybori/include/polybori/orderings/CBlockOrderingFacade.h	2012-05-28 08:28:37.024979695 +0200
@@ -91,7 +91,7 @@
     CacheManager<order_lead_tag> cache_mgr(poly.ring());
     typename base_type::descending_property descending;
 
-    return monom(  dd_block_degree_lead(cache_mgr, blockDegCache, 
+    return base_type::monom(  dd_block_degree_lead(cache_mgr, blockDegCache, 
                                         poly.navigation(), m_indices.begin(),
                                         set_type(poly.ring()), descending) );
   }
