--- SConstruct.orig     2011-12-02 11:24:27.000000000 +0100
+++ SConstruct  2011-12-02 11:25:42.000000000 +0100
@@ -633,6 +633,13 @@
         store_libs = env["LIBS"]
         env.Append(LIBS=gdlibs[1:])
         if conf.CheckLib(gdlibs[0]):
-            env.Append(CPPDEFINES=["HAVE_GD"])
-            GD_LIBS = gdlibs
+            env["LIBS"] = store_libs
+            env.Append(LIBS=gdlibs)
+            if conf.CheckFunc("Testing_PNGs",
+                              "#include<gd.h>\n#define Testing_PNGs() gdImagePng(NULL,NULL) "):
+                env.Append(CPPDEFINES=["HAVE_GD"])
+                GD_LIBS = gdlibs
+            else:
+                print "libgd is available, but could not generate png file -",
+                print "dependencies in GD_LIBS missing? - Skipping gd-based optional features." 
         env["LIBS"] = store_libs
@@ -803,7 +803,8 @@
 gb_shared = shared_object(gb_src)#env.SharedObject(gb_src)
 shared_resources += gb_shared

-libgbShared = slib('groebner', list(shared_resources))
+libgbShared = slib('groebner', list(shared_resources),
+                   LIBS = libpbShared + env['LIBS'] + GD_LIBS)


 env.Depends(libgbShared, libpbShared + pb_symlinks)
