--- SConstruct.orig	2011-01-14 13:21:09.000000000 +1300
+++ SConstruct	2011-01-14 16:45:41.000000000 +1300
@@ -164,7 +164,7 @@
     else:
         print "Flags", flag, "not in default environment!"        
     
-opts.Add('LINKFLAGS', "Linker flags", defaultenv['LINKFLAGS'] + ['-s'])
+#opts.Add('LINKFLAGS', "Linker flags", defaultenv['LINKFLAGS'] + ['-s'])
 opts.Add('LIBS', 'custom libraries needed for build', [], converter = Split)
 
 opts.Add('PREFIX', 'installation prefix directory', '/usr/local')
