--- src/doc/common/custom-sphinx-build.py.orig	2014-05-06 10:56:35.000000000 +0000
+++ src/doc/common/custom-sphinx-build.py	2014-05-26 12:08:19.607191250 +0000
@@ -28,7 +28,8 @@
     re.compile('^looking for now-outdated files... none found'),
     re.compile('^building \[.*\]: targets for 0 source files that are out of date'),
     re.compile('^loading pickled environment... done'),
-    re.compile('^loading cross citations... done \([0-9]* citations\).')
+    re.compile('^loading cross citations... done \([0-9]* citations\).'),
+    re.compile('WARNING: favicon file \'favicon.ico\' does not exist')
     )
 
 # replacements: pairs of regular expressions and their replacements,
