diff -Naur doc.orig/common/build_options.py doc/common/build_options.py
--- doc.orig/common/build_options.py	2015-03-04 13:27:11.455186635 +1300
+++ doc/common/build_options.py	2015-03-04 13:31:58.535867721 +1300
@@ -4,7 +4,20 @@
 
 import os, re
 SAGE_DOC = os.environ['SAGE_DOC']
-LANGUAGES = [d for d in os.listdir(SAGE_DOC) if re.match('^[a-z][a-z]$', d)]
+LANGUAGELIST = [d for d in os.listdir(SAGE_DOC) if re.match('^[a-z][a-z]$', d)]
+try:
+    DESIRED_LANGUAGES = (os.environ['LANGUAGES']).split()
+except KeyError:
+    DESIRED_LANGUAGES = LANGUAGELIST
+LANGUAGES = []
+for d in DESIRED_LANGUAGES:
+    if d in LANGUAGELIST:
+	LANGUAGES.append(d)
+if not LANGUAGES:
+    LANGUAGES=LANGUAGELIST
+
+print("building documentation for the following languages: %s"%LANGUAGES)
+
 SPHINXOPTS = ""
 PAPER = ""
 OMIT = ["introspect"]  # docs/dirs to omit when listing and building 'all'
