diff --git a/src/doc/bootstrap b/src/doc/bootstrap
index 69002a79c1..5ab0ef3fe2 100755
--- a/src/doc/bootstrap
+++ b/src/doc/bootstrap
@@ -176,6 +176,8 @@ for PKG_BASE in $(./sage --package list --has-file SPKG.rst | sort); do
 
             (echo ".. _spkg_$PKG_BASE:" && echo && OUTPUT_RST=1 sage-spkg-info $PKG_BASE) > "$OUTPUT_DIR"/$PKG_BASE.rst
             echo >> "$OUTPUT_INDEX" "   $PKG_BASE"
+            # for sphinx 4.4 we need to replace all direct links by some "extlink" (ticket 33272)
+            sed -e "s|https://trac.sagemath.org/ticket/\([0-9]*\)|:trac:\`ticket/\1\`|g" -i "$OUTPUT_DIR"/$PKG_BASE.rst
 done
 
 # #30064: Create the directory first
