diff --git a/src/bipart.cc b/src/bipart.cc
index b2ddf89..cc4f346 100644
--- a/src/bipart.cc
+++ b/src/bipart.cc
@@ -29,7 +29,7 @@
 #include <vector>
 
 #include "libsemigroups/semigroups.h"
-#include "src/compiled.h"
+#include "gap/compiled.h"
 
 using libsemigroups::Element;
 using libsemigroups::glob_reporter;
diff --git a/src/bipart.h b/src/bipart.h
index 8c09b75..fa48932 100644
--- a/src/bipart.h
+++ b/src/bipart.h
@@ -22,7 +22,7 @@
 #include "libsemigroups/elements.h"
 #include "pkg.h"
 #include "semigroups-debug.h"
-#include "src/compiled.h"
+#include "gap/compiled.h"
 
 using libsemigroups::Bipartition;
 using libsemigroups::Blocks;
diff --git a/src/congpairs.h b/src/congpairs.h
index ce50bd6..5eb5fb9 100644
--- a/src/congpairs.h
+++ b/src/congpairs.h
@@ -31,7 +31,7 @@
 
 #include <cstddef>
 
-#include "src/compiled.h"
+#include "gap/compiled.h"
 
 // GAP level functions
 
diff --git a/src/converter.h b/src/converter.h
index b8f29f7..54f5276 100644
--- a/src/converter.h
+++ b/src/converter.h
@@ -34,7 +34,7 @@
 #include <algorithm>
 #include <vector>
 
-#include "src/compiled.h"
+#include "gap/compiled.h"
 
 #include "pkg.h"
 
diff --git a/src/fropin.h b/src/fropin.h
index 1181694..739faa5 100644
--- a/src/fropin.h
+++ b/src/fropin.h
@@ -20,7 +20,7 @@
 #define SEMIGROUPS_SRC_FROPIN_H_
 
 #include "pkg.h"
-#include "src/compiled.h"
+#include "gap/compiled.h"
 
 #define ELM_PLIST2(plist, i, j) ELM_PLIST(ELM_PLIST(plist, i), j)
 
diff --git a/src/pkg.h b/src/pkg.h
index 715e12e..cfcca45 100644
--- a/src/pkg.h
+++ b/src/pkg.h
@@ -46,7 +46,7 @@
 #include "rnams.h"
 #include "semigroups-debug.h"
 
-#include "src/compiled.h"
+#include "gap/compiled.h"
 
 #if !defined(GAP_KERNEL_MAJOR_VERSION) || GAP_KERNEL_MAJOR_VERSION < 3
 // compatibility with GAP <= 4.9
diff --git a/src/rnams.h b/src/rnams.h
index 834c0c4..7756fe5 100644
--- a/src/rnams.h
+++ b/src/rnams.h
@@ -21,7 +21,7 @@
 #ifndef SEMIGROUPS_SRC_RNAMS_H_
 #define SEMIGROUPS_SRC_RNAMS_H_
 
-#include "src/compiled.h"  // GAP headers
+#include "gap/compiled.h"  // GAP headers
 
 static Int RNam_batch_size        = 0;
 static Int RNam_cong              = 0;
diff --git a/src/semigrp.cc b/src/semigrp.cc
index fd9c6fe..2331206 100644
--- a/src/semigrp.cc
+++ b/src/semigrp.cc
@@ -27,7 +27,7 @@
 #include "converter.h"
 #include "fropin.h"
 #include "pkg.h"
-#include "src/compiled.h"
+#include "gap/compiled.h"
 
 using libsemigroups::cayley_graph_t;
 using libsemigroups::Integers;
diff --git a/src/semigrp.h b/src/semigrp.h
index 1b99acc..e479b28 100644
--- a/src/semigrp.h
+++ b/src/semigrp.h
@@ -25,7 +25,7 @@
 #include "libsemigroups/semigroups.h"
 #include "pkg.h"
 #include "rnams.h"
-#include "src/compiled.h"  // GAP headers
+#include "gap/compiled.h"  // GAP headers
 
 using libsemigroups::Semigroup;
 
diff --git a/src/uf.cc b/src/uf.cc
index ae663c7..64462c4 100644
--- a/src/uf.cc
+++ b/src/uf.cc
@@ -24,7 +24,7 @@
 
 #include "pkg.h"
 #include "semigroups-debug.h"
-#include "src/compiled.h"
+#include "gap/compiled.h"
 
 #include "libsemigroups/uf.h"
 
diff --git a/src/uf.h b/src/uf.h
index 8a194a0..ac78195 100644
--- a/src/uf.h
+++ b/src/uf.h
@@ -24,7 +24,7 @@
 #define SEMIGROUPS_SRC_UF_H_
 
 #include "pkg.h"
-#include "src/compiled.h"
+#include "gap/compiled.h"
 
 // GAP level functions
 
