diff --git a/Makefile.am b/Makefile.am
index 48a01be..d00e90c 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -6,9 +6,6 @@
 #
 ACLOCAL_AMFLAGS = -I m4
 
-SUBDIRS = src/libsemigroups
-AM_CPPFLAGS = -I$(srcdir)/src/libsemigroups
-
 BINARCHDIR = bin/$(GAPARCH)
 GAPINSTALLLIB = $(abs_top_srcdir)/$(BINARCHDIR)/semigroups.so
 
@@ -24,15 +21,12 @@ semigroups_la_CPPFLAGS = $(GAP_CPPFLAGS) -DCONFIG_H
 # Note that the latter is only for GAP 4.4.12
 semigroups_la_LDFLAGS = -module -avoid-version
 
-semigroups_la_LIBADD = src/libsemigroups/libsemigroups.la
-
 if SYS_IS_CYGWIN
 semigroups_la_LDFLAGS += -no-undefined -version-info 0:0:0 -Wl,$(GAPROOT)/bin/$(GAPARCH)/gap.dll
 endif
 
 all-local: semigroups.la
-	$(mkdir_p) $(top_srcdir)/$(BINARCHDIR) $(top_srcdir)/bin/lib
-	cp -RL src/libsemigroups/.libs/* $(top_srcdir)/bin/lib/
+	$(mkdir_p) $(top_srcdir)/$(BINARCHDIR)
 if SYS_IS_CYGWIN
 	cp .libs/semigroups.dll $(GAPINSTALLLIB)
 # Cygwin will only look in this directory for dlls
diff --git a/configure.ac b/configure.ac
index fe5e40f..5d782b6 100644
--- a/configure.ac
+++ b/configure.ac
@@ -14,7 +14,6 @@ AC_CONFIG_SRCDIR([src/pkg.cc])
 AC_CONFIG_HEADER([src/_pkgconfig.h:cnf/pkgconfig.h.in])
 AC_CONFIG_MACRO_DIR([m4])
 AC_CONFIG_AUX_DIR([cnf])
-AC_CONFIG_SUBDIRS([src/libsemigroups])
 
 dnl ##
 dnl ## Get canonical host info
diff --git a/src/bipart.cc b/src/bipart.cc
index 8e7164a..263bb69 100644
--- a/src/bipart.cc
+++ b/src/bipart.cc
@@ -28,8 +28,8 @@
 #include <utility>
 #include <vector>
 
-#include "libsemigroups/src/semigroups.h"
-#include "src/compiled.h"
+#include "libsemigroups/semigroups.h"
+#include "gap/compiled.h"
 
 using libsemigroups::Element;
 using libsemigroups::glob_reporter;
diff --git a/src/bipart.h b/src/bipart.h
index f25912d..fa48932 100644
--- a/src/bipart.h
+++ b/src/bipart.h
@@ -19,10 +19,10 @@
 #ifndef SEMIGROUPS_SRC_BIPART_H_
 #define SEMIGROUPS_SRC_BIPART_H_
 
-#include "libsemigroups/src/elements.h"
+#include "libsemigroups/elements.h"
 #include "pkg.h"
 #include "semigroups-debug.h"
-#include "src/compiled.h"
+#include "gap/compiled.h"
 
 using libsemigroups::Bipartition;
 using libsemigroups::Blocks;
diff --git a/src/congpairs.cc b/src/congpairs.cc
index eb059aa..50b4d97 100644
--- a/src/congpairs.cc
+++ b/src/congpairs.cc
@@ -27,7 +27,7 @@
 #include "rnams.h"
 #include "semigrp.h"
 
-#include "libsemigroups/src/cong.h"
+#include "libsemigroups/cong.h"
 
 using libsemigroups::Congruence;
 using libsemigroups::Partition;
diff --git a/src/congpairs.h b/src/congpairs.h
index ce50bd6..5eb5fb9 100644
--- a/src/congpairs.h
+++ b/src/congpairs.h
@@ -31,7 +31,7 @@
 
 #include <cstddef>
 
-#include "src/compiled.h"
+#include "gap/compiled.h"
 
 // GAP level functions
 
diff --git a/src/converter.h b/src/converter.h
index 9018ab2..14eb1c4 100644
--- a/src/converter.h
+++ b/src/converter.h
@@ -34,13 +34,13 @@
 #include <algorithm>
 #include <vector>
 
-#include "src/compiled.h"
+#include "gap/compiled.h"
 
 #include "pkg.h"
 
 #include "semigroups-debug.h"
 
-#include "libsemigroups/src/elements.h"
+#include "libsemigroups/elements.h"
 
 using libsemigroups::Bipartition;
 using libsemigroups::BooleanMat;
diff --git a/src/fropin.cc b/src/fropin.cc
index 96e1d25..7fae76a 100644
--- a/src/fropin.cc
+++ b/src/fropin.cc
@@ -21,7 +21,7 @@
 #include <algorithm>
 #include <iostream>
 
-#include "libsemigroups/src/report.h"
+#include "libsemigroups/report.h"
 #include "rnams.h"
 #include "semigroups-debug.h"
 #include "semigrp.h"
diff --git a/src/fropin.h b/src/fropin.h
index 1181694..739faa5 100644
--- a/src/fropin.h
+++ b/src/fropin.h
@@ -20,7 +20,7 @@
 #define SEMIGROUPS_SRC_FROPIN_H_
 
 #include "pkg.h"
-#include "src/compiled.h"
+#include "gap/compiled.h"
 
 #define ELM_PLIST2(plist, i, j) ELM_PLIST(ELM_PLIST(plist, i), j)
 
diff --git a/src/pkg.cc b/src/pkg.cc
index 8270ea7..c8d62aa 100644
--- a/src/pkg.cc
+++ b/src/pkg.cc
@@ -32,9 +32,9 @@
 #include "semigrp.h"
 #include "uf.h"
 
-#include "libsemigroups/src/cong.h"
-#include "libsemigroups/src/semigroups.h"
-#include "libsemigroups/src/uf.h"
+#include "libsemigroups/cong.h"
+#include "libsemigroups/semigroups.h"
+#include "libsemigroups/uf.h"
 
 using libsemigroups::Congruence;
 using libsemigroups::UF;
diff --git a/src/pkg.h b/src/pkg.h
index 64973c1..056d7b4 100644
--- a/src/pkg.h
+++ b/src/pkg.h
@@ -46,7 +46,7 @@
 #include "rnams.h"
 #include "semigroups-debug.h"
 
-#include "src/compiled.h"
+#include "gap/compiled.h"
 
 // The following typedefs are used in the Semigroups package kernel module code
 // to increase the readability of the code.
diff --git a/src/rnams.h b/src/rnams.h
index 834c0c4..7756fe5 100644
--- a/src/rnams.h
+++ b/src/rnams.h
@@ -21,7 +21,7 @@
 #ifndef SEMIGROUPS_SRC_RNAMS_H_
 #define SEMIGROUPS_SRC_RNAMS_H_
 
-#include "src/compiled.h"  // GAP headers
+#include "gap/compiled.h"  // GAP headers
 
 static Int RNam_batch_size        = 0;
 static Int RNam_cong              = 0;
diff --git a/src/semigrp.cc b/src/semigrp.cc
index 176efd6..7c1a384 100644
--- a/src/semigrp.cc
+++ b/src/semigrp.cc
@@ -27,7 +27,7 @@
 #include "converter.h"
 #include "fropin.h"
 #include "pkg.h"
-#include "src/compiled.h"
+#include "gap/compiled.h"
 
 using libsemigroups::cayley_graph_t;
 using libsemigroups::Integers;
diff --git a/src/semigrp.h b/src/semigrp.h
index 627f934..e479b28 100644
--- a/src/semigrp.h
+++ b/src/semigrp.h
@@ -22,10 +22,10 @@
 #define SEMIGROUPS_SRC_SEMIGRP_H_
 
 #include "converter.h"
-#include "libsemigroups/src/semigroups.h"
+#include "libsemigroups/semigroups.h"
 #include "pkg.h"
 #include "rnams.h"
-#include "src/compiled.h"  // GAP headers
+#include "gap/compiled.h"  // GAP headers
 
 using libsemigroups::Semigroup;
 
diff --git a/src/uf.cc b/src/uf.cc
index 3620a10..999da58 100644
--- a/src/uf.cc
+++ b/src/uf.cc
@@ -24,9 +24,9 @@
 
 #include "pkg.h"
 #include "semigroups-debug.h"
-#include "src/compiled.h"
+#include "gap/compiled.h"
 
-#include "libsemigroups/src/uf.h"
+#include "libsemigroups/uf.h"
 
 using libsemigroups::UF;
 
diff --git a/src/uf.h b/src/uf.h
index 8a194a0..ac78195 100644
--- a/src/uf.h
+++ b/src/uf.h
@@ -24,7 +24,7 @@
 #define SEMIGROUPS_SRC_UF_H_
 
 #include "pkg.h"
-#include "src/compiled.h"
+#include "gap/compiled.h"
 
 // GAP level functions
 
