--- pexpect.py.orig	2010-03-14 07:12:13.613916842 +0000
+++ pexpect.py	2010-03-14 07:52:37.914541069 +0000
@@ -339,9 +339,12 @@
         If the child file descriptor was opened outside of this class
         (passed to the constructor) then this does not close it.
         """
-        if self.closed:
-            return
-        self.close()
+        try:
+            if self.closed:
+                return
+            self.close()
+        except:
+            pass
 
     def __str__(self):
         """This returns the current state of the pexpect object as a string.
@@ -1127,7 +1130,7 @@
     """
     # Special case where filename already contains a path.
     if os.path.dirname(filename) != '':
-        if os.access (filename, os.X_OK):
+        if os.access (filename, os.X_OK) and not os.path.isdir(f):
             return filename
 
     if not os.environ.has_key('PATH') or os.environ['PATH'] == '':
@@ -1142,7 +1145,7 @@
 
     for path in pathlist:
         f = os.path.join(path, filename)
-        if os.access(f, os.X_OK):
+        if os.access(f, os.X_OK) and not os.path.isdir(f):
             return f
     return None
 
