diff --git a/src/setup.py b/src/setup.py
index e398807..673575b 100644
--- a/setup.py
+++ b/setup.py
@@ -92,7 +92,7 @@ setup(
   url = "https://users.fmi.uni-jena.de/cohomology/documentation/",
   description = "Modular Cohomology Rings of Finite Groups",
   packages = find_packages(),
-  package_data = {'pGroupCohomology': ['*.pxd']},
+  package_data = {'pGroupCohomology': ['*.pxd', '*.pyx']},
   py_modules = ["pGroupCohomology.auxiliaries", "pGroupCohomology.barcode", "pGroupCohomology.factory", "pGroupCohomology.isomorphism_test"],
   ext_modules = cythonize(ext_mods, compiler_directives={'embedsignature': True,
                                                          'language_level': PY_MAJOR_VERSION}, build_dir=os.path.join("build","c_files-{}.{}".format(version_info.major, version_info.minor))),
