diff --git a/Makefile.in b/Makefile.in
index d5f2b98..61c0b87 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -26,7 +26,7 @@ leonconv: desauto
 	cd ./src; $(MAKE) CC="$(CC)" CPPFLAGS="$(CPPFLAGS)" CFLAGS="$(CFLAGS)" LDFLAGS="$(LDFLAGS)";
 
 desauto: 
-	cd $(SRCDIR); autoreconf --install --force || true ; ./configure; $(MAKE) CC="$(CC)" CPPFLAGS="$(CPPFLAGS)" CFLAGS="$(CFLAGS)" LDFLAGS="$(LDFLAGS)";
+	cd $(SRCDIR); $(MAKE) CC="$(CC)" CPPFLAGS="$(CPPFLAGS)" CFLAGS="$(CFLAGS)" LDFLAGS="$(LDFLAGS)";
 
 minimum-weight.o: $(CJSRCDIR)/minimum-weight.c $(CJSRCDIR)/minimum-weight-gf2.h $(CJSRCDIR)/minimum-weight-gf3.h $(CJSRCDIR)/popcount.h $(CJSRCDIR)/config.h $(CJSRCDIR)/types.h
 	$(CC) -c -O3 -Wall -I $(CJSRCDIR) $(CJSRCDIR)/minimum-weight.c
diff --git a/src/leon/configure.ac b/src/leon/configure.ac
index 8f9b6da..73abc3b 100644
--- a/src/leon/configure.ac
+++ b/src/leon/configure.ac
@@ -1,6 +1,5 @@
 AC_INIT(leon,1.0)
 AC_CONFIG_SRCDIR(src/group.h)
-AM_INIT_AUTOMAKE
 AC_CONFIG_HEADERS([src/leon_config.h])
 AC_CHECK_SIZEOF(int)
 
