--- Makefile.orig	2010-07-01 01:37:01.000000000 +1200
+++ Makefile	2010-12-20 11:56:41.000000000 +1300
@@ -23,15 +23,18 @@
 
 PRIME_SIZE = 7
 
-CC = gcc
+CC ?= gcc
 RM = rm -f
-INSTALL = cp
+INSTALL ?= cp
 
-INSTALL_DIR = /usr/local
+INSTALL_DIR ?= /usr/local
+LIB_DIR ?= lib
 
-CCFLAGS1 = -Wall -O2 -fomit-frame-pointer -DRATPOINTS_MAX_BITS_IN_PRIME=${PRIME_SIZE} -DUSE_SSE
+CFLAGS ?= -Wall -O2 -fomit-frame-pointer -DUSE_SSE
+LDFLAGS ?= 
+CCFLAGS1 = $(CFLAGS) -DRATPOINTS_MAX_BITS_IN_PRIME=${PRIME_SIZE}
 # for gcc on Apple, may have to add '-fnested-functions' to CCFLAGS1
-CCFLAGS2 = -lgmp -lgcc -lc -lm
+CCFLAGS2 = $(LDFLAGS) -lgmp -lgcc -lc -lm
 CCFLAGS3 = -L. -lratpoints
 CCFLAGS = 
 
@@ -47,24 +50,27 @@
             gen_find_points_h gen_init_sieve_h \
             rptest.out sift-debug.o find_points-debug.o main.o
 
-TARGETFILES = ratpoints libratpoints.a rptest ratpoints-debug
+TARGETFILES = ratpoints libratpoints.dylib rptest ratpoints-debug
 
-all: ratpoints libratpoints.a
+all: ratpoints libratpoints.dylib
 
 test: rptest testbase
-	time ./rptest > rptest.out
+	time LD_LIBRARY_PATH=. ./rptest > rptest.out
 #	diff -q testbase rptest.out
 	cmp -s testbase rptest.out || echo "Test failed!"
 
 install-bin: ratpoints
+	mkdir -p ${INSTALL_DIR}/bin
 	${INSTALL} ratpoints ${INSTALL_DIR}/bin/
 	chmod 755 ${INSTALL_DIR}/bin/ratpoints
 
-install-lib: ratpoints.h libratpoints.a
+install-lib: ratpoints.h libratpoints.dylib
+	mkdir -p ${INSTALL_DIR}/include
 	${INSTALL} ratpoints.h ${INSTALL_DIR}/include/
 	chmod 644 ${INSTALL_DIR}/include/ratpoints.h
-	${INSTALL} libratpoints.a ${INSTALL_DIR}/lib/
-	chmod 644 ${INSTALL_DIR}/lib/libratpoints.a
+	mkdir -p ${INSTALL_DIR}/${LIB_DIR}
+	${INSTALL} libratpoints.dylib ${INSTALL_DIR}/${LIB_DIR}
+	chmod 644 ${INSTALL_DIR}/${LIB_DIR}/libratpoints.dylib
 
 install: install-bin install-lib
 
@@ -83,10 +89,10 @@
 
 debug: ratpoints-debug
 
-libratpoints.a: sift.o init.o sturm.o find_points.o
-	ar rs libratpoints.a sift.o init.o sturm.o find_points.o
+libratpoints.dylib: sift.o init.o sturm.o find_points.o
+	$(CC) -dynamiclib $(LDFLAGS) -lgmp -lm -o libratpoints.dylib -install_name $(EPREFIX)/usr/$(LIB_DIR)/libratpoints.dylib sift.o init.o sturm.o find_points.o
 
-ratpoints: libratpoints.a main.c ratpoints.h
+ratpoints: libratpoints.dylib main.c ratpoints.h
 	${CC} main.c -o ratpoints ${CCFLAGS1} ${CCFLAGS2} ${CCFLAGS3} ${CCFLAGS}
 
 main.o: main.c ratpoints.h
@@ -97,33 +103,33 @@
               -o ratpoints-debug ${CCFLAGS1} ${CCFLAGS2} ${CCFLAGS}
 
 sift.o: sift.c ratpoints.h rp-private.h
-	${CC} sift.c -c -o sift.o ${CCFLAGS1} -funroll-loops ${CCFLAGS}
+	${CC} -fPIC sift.c -c -o sift.o ${CCFLAGS1} -funroll-loops ${CCFLAGS}
 
 sift-debug.o: sift.c ratpoints.h rp-private.h
-	${CC} sift.c -c -o sift-debug.o ${CCFLAGS1} -funroll-loops -DDEBUG ${CCFLAGS}
+	${CC} -fPIC sift.c -c -o sift-debug.o ${CCFLAGS1} -funroll-loops -DDEBUG ${CCFLAGS}
 
 sift.s: sift.c ratpoints.h rp-private.h
-	${CC} sift.c -S -o sift.s ${CCFLAGS1} -funroll-loops ${CCFLAGS}
+	${CC} -fPIC sift.c -S -o sift.s ${CCFLAGS1} -funroll-loops ${CCFLAGS}
 
 sift.i: sift.c ratpoints.h rp-private.h
-	${CC} sift.c -E -o sift.i ${CCFLAGS1} -funroll-loops ${CCFLAGS}
+	${CC} -fPIC sift.c -E -o sift.i ${CCFLAGS1} -funroll-loops ${CCFLAGS}
 
 init.o: init.c ratpoints.h rp-private.h init_sieve.h
-	${CC} init.c -c -o init.o ${CCFLAGS1} -funroll-loops -O3 ${CCFLAGS}
+	${CC} -fPIC init.c -c -o init.o ${CCFLAGS1} -funroll-loops -O3 ${CCFLAGS}
 
 init.s: init.c ratpoints.h rp-private.h init_sieve.h
-	${CC} init.c -S -o init.s ${CCFLAGS1} -funroll-loops -O3 ${CCFLAGS}
+	${CC} -fPIC init.c -S -o init.s ${CCFLAGS1} -funroll-loops -O3 ${CCFLAGS}
 
 sturm.o: sturm.c ratpoints.h rp-private.h
-	${CC} sturm.c -c -o sturm.o ${CCFLAGS1} ${CCFLAGS}
+	${CC} -fPIC sturm.c -c -o sturm.o ${CCFLAGS1} ${CCFLAGS}
 
 find_points.o: find_points.c ratpoints.h rp-private.h primes.h find_points.h
-	${CC} find_points.c -c -o find_points.o ${CCFLAGS1} ${CCFLAGS}
+	${CC} -fPIC find_points.c -c -o find_points.o ${CCFLAGS1} ${CCFLAGS}
 
 find_points-debug.o: find_points.c ratpoints.h rp-private.h primes.h find_points.h
-	${CC} find_points.c -c -o find_points-debug.o ${CCFLAGS1} -DDEBUG ${CCFLAGS}
+	${CC} -fPIC find_points.c -c -o find_points-debug.o ${CCFLAGS1} -DDEBUG ${CCFLAGS}
 
-rptest: libratpoints.a rptest.c ratpoints.h testdata.h
+rptest: libratpoints.so rptest.c ratpoints.h testdata.h
 	${CC} rptest.c -o rptest ${CCFLAGS1} ${CCFLAGS2} ${CCFLAGS3} ${CCFLAGS}
 
 gen_init_sieve_h: gen_init_sieve_h.c ratpoints.h rp-private.h primes.h
