Index: factory/GNUmakefile.in
===================================================================
--- factory/GNUmakefile.in.orig
+++ factory/GNUmakefile.in
@@ -490,6 +490,7 @@ mem:		libcfmem.a
 #
 # - explicit targets.
 #
+$(basefactorysrc:.cc=.o) $(basefactorysrc:.cc=.og):      factory.h
 $(libfactory):	$(factoryobj)
 		$(AR) $(ARFLAGS) $@ $^
 		$(RANLIB) $@
Index: Singular/Makefile.in
===================================================================
--- Singular/Makefile.in.orig
+++ Singular/Makefile.in
@@ -377,7 +377,7 @@ misc_ip.o emacs.o tesths.o: version.h
 claptmpl.o: claptmpl.cc mod2.h
 	${CXX} ${CXXFLAGS}  ${CPPFLAGS} ${DEFS} -c $<
 
-iparith.inc: claptmpl.o iparith.cc \
+iparith.inc: ${OBJS} claptmpl.o iparith.cc \
                           ipconv.cc tok.h mpsr_Tok.cc grammar.h mod2.h
 	${CXX} ${CXXFLAGS} ${CPPFLAGS} ${CXXTEMPLFLAGS} ${DEFS} -DGENTABLE \
 	     -o gentable1 claptmpl.o iparith.cc tesths.cc mpsr_Tok.cc \
@@ -385,7 +385,7 @@ iparith.inc: claptmpl.o iparith.cc \
 	./gentable1
 	/bin/rm -f gentable1 gentable1.exe
 
-mpsr_Tok.inc: claptmpl.o iparith.cc \
+mpsr_Tok.inc: ${OBJS} claptmpl.o iparith.cc \
                           ipconv.cc tok.h mpsr_Tok.cc grammar.h mod2.h
 	${CXX} ${CXXFLAGS} ${CPPFLAGS} ${CXXTEMPLFLAGS} ${DEFS} -DGENTABLE \
 	     -o gentable2 claptmpl.o iparith.cc tesths.cc mpsr_Tok.cc \
Index: Makefile.in
===================================================================
--- Makefile.in.orig
+++ Makefile.in
@@ -75,12 +75,12 @@ install-nolns:
 	echo "* do not forget to install Singular-@SINGULAR_VERSION@-share.tar.gz" 
 
 libsingular:
-	cd Singular; make libsingular
+	cd Singular; ${MAKE} libsingular
 
 install-libsingular: libsingular
-	(cd Singular && make install-libsingular);
-	(cd kernel && make install-libsingular);
-	(cd omalloc && make install-libsingular);
+	(cd Singular && ${MAKE} install-libsingular);
+	(cd kernel && ${MAKE} install-libsingular);
+	(cd omalloc && ${MAKE} install-libsingular);
 
 maintainer-install uninstall:
 	@ for SUBDIR in ${BUILD_SUBDIRS}; \
@@ -174,9 +174,9 @@ Tst-Long:  ${SINGULAR} Tst/Long/ok_l.lst
 test:
 	@echo "make Tst for the official Singular test suite"
 	@echo "this part here are unittest and still under development"
-	(cd Singular; make libsingular.a)
+	(cd Singular; ${MAKE} libsingular.a)
 	mkdir -p build
-	(cd build && cmake .. && make && make test)
+	(cd build && cmake .. && ${MAKE} && ${MAKE} test)
 
 ##############################################################
 # distribution targets
Index: libfac/Makefile.in
===================================================================
--- libfac/Makefile.in.orig
+++ libfac/Makefile.in
@@ -81,7 +81,8 @@ PROFOBJ = $(OPTFACTOROBJ1:%.o=%.op) $(OP
 
 .SUFFIXES:      .cc .h .o .op .og
 
-all: OPTOBJ ${TARGETNAME} libsingfac_g.a
+all: OPTOBJ
+	${MAKE} ${TARGETNAME} libsingfac_g.a
 
 OPTOBJ:
 	${MKINSTALLDIRS} OPTOBJ
