--- linbox/field/modular-float.h.orig	2011-12-05 15:46:04.486702473 +1300
+++ linbox/field/modular-float.h	2011-12-05 15:47:22.639635310 +1300
@@ -38,6 +38,9 @@
 	template< class Element >
 	class ModularRandIter;
 
+	template< class Field, class RandIter >
+	class NonzeroRandIter;
+
 	template <class Ring>
 	struct ClassifyRing; 
 	template <class Element>
@@ -67,6 +70,7 @@
 			       
 		typedef float Element;
 		typedef ModularRandIter<float> RandIter;
+		typedef NonzeroRandIter<Modular<float>, ModularRandIter<float> > NonZeroRandIter;
 
 		static ClassifyRing<Modular<float> >::categoryTag getCategory() {return ClassifyRing<Modular<float> >::categoryTag();}
 		
