--- doc/Makefile.am.orig	2009-11-22 10:24:29.497142925 +0000
+++ doc/Makefile.am	2009-11-22 10:25:34.447141574 +0000
@@ -13,16 +13,8 @@
 docs: 
 	doxygen 
 	cp tutorial.html linbox-html
-	cp install-dist.html linbox-html
-	cp install-dev.html linbox-html
-	cp ../INSTALL linbox-html
-	cp ../COPYING linbox-html
-	cp ../AUTHORS linbox-html
 
 install-doc :
-	mkdir $(docdir)
-	cp -r linbox-html $(docdir)/linbox-html
-	cp linbox.html $(docdir)/linbox.html
 
 EXTRA_DIST=\
 	linbox.html		\
