--- a/linbox/solutions/solve.h	2013-01-08 23:32:04.216963122 +1300
+++ b/linbox/solutions/solve.h	2013-01-08 23:32:56.613007720 +1300
@@ -311,7 +311,7 @@
 		      const RingCategories::ModularTag & tag,
 		      const Method::Dixon & m)
 	{
-		throw LinBoxFailure("You cannot do this");
+		throw LinBoxFailure(__func__, __FILE__, __LINE__,"You cannot do this");
 	}
 
 	/* Integer tag Specialization for Dixon method:
