--- interfaces/sage/Makefile.am.orig	2010-02-19 17:08:57.784372437 +0000
+++ interfaces/sage/Makefile.am	2010-02-24 19:52:21.512272781 +0000
@@ -10,7 +10,7 @@
 
 pkginclude_HEADERS = linbox-sage.h
 
-#liblinboxsage_la_LIBADD = -llinbox $(BLAS_LIBS)
+liblinboxsage_la_LIBADD = $(top_builddir)/linbox/liblinbox.la
 
 liblinboxsage_la_LDFLAGS =  $(GIVARO_LIBS) $(GMP_LIBS) $(NTL_LIBS) $(BLAS_LIBS) $(MAPLE_LIBS)  -version-info 0:0:0 #-Wl,-zmuldefs
 endif
\ No newline at end of file
