--- linbox/algorithms/blackbox-container-symmetric.h
+++ linbox/algorithms/blackbox-container-symmetric.h
@@ -47,7 +47,7 @@
 		{ init (u0, u0); }
 	BlackboxContainerSymmetric (const Blackbox *D, const Field &F, RandIter &g)
 		: BlackboxContainerBase<Field, _Blackbox> (D, F)
-		{ init (g); }
+		{ this->init (g); }
 
     protected:
 

--- linbox/matrix/matrix-domain.h
+++ linbox/matrix/matrix-domain.h
@@ -818,7 +818,7 @@
 	Vector1 &mulColSpecialized (Vector1 &w, const Matrix &A, const Vector2 &v,
 				    VectorCategories::DenseVectorTag,
 				    VectorCategories::DenseVectorTag) const
-	{ return mulColDense (_VD, w, A, v); } 
+	{ return this->mulColDense (_VD, w, A, v); } 
 	template <class Vector1, class Matrix, class Vector2>
 	Vector1 &mulColSpecialized (Vector1 &w, const Matrix &A, const Vector2 &v,
 				    VectorCategories::DenseVectorTag,
