diff --git a/src/sage/config.py.in b/src/sage/config.py.in
index 76f6e342a16..d1ff346cba6 100644
--- a/src/sage/config.py.in
+++ b/src/sage/config.py.in
@@ -11,7 +11,7 @@ VERSION = "@PACKAGE_VERSION@"
 SAGE_LOCAL = "@prefix@"
 
 # The path to the standalone maxima executable.
-MAXIMA = "@SAGE_MAXIMA@".replace("${prefix}", SAGE_LOCAL)
+MAXIMA = "maxima -l ecl"
 
 # Set this to the empty string if your ECL can load maxima without
 # further prodding.
@@ -57,6 +57,15 @@ rankwidth_enabled = @RANKWIDTH_ENABLED@
 sirocco_enabled = @SIROCCO_ENABLED@
 tdlib_enabled = @TDLIB_ENABLED@
 
+# Some Gentoo definitions
+MATHJAX_DIR  = "@GENTOO_PORTAGE_EPREFIX@/usr/share/mathjax"
+SAGE_SHARE   = "@GENTOO_PORTAGE_EPREFIX@/usr/share/sage"
+SAGE_DOC     = "@GENTOO_PORTAGE_EPREFIX@/usr/share/doc/@GENTOO_PORTAGE_PF@"
+JMOL_DIR     = "@GENTOO_PORTAGE_EPREFIX@/usr/share/sage-jmol-bin/lib"
+MTXLIB       = "@GENTOO_PORTAGE_EPREFIX@/usr/share/meataxe"
+PPLPY_DOCS   = "@GENTOO_PORTAGE_EPREFIX@/usr/share/doc/@PPLY_DOC_VERS@/html"
+LIE_INFO_DIR = "@GENTOO_PORTAGE_EPREFIX@/usr/share/lie"
+
 def is_editable_install() -> bool:
     """
     Check whether this is an editable install of Sage.
